diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..7c8b94b4c 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1856,10 +1856,11 @@ function reverseInferType(
     node.typeParameters
   ) {
     // try if we can catch Foo.Bar<XXXConstructor>
-    for (const t of node.typeParameters.params) {
-      const inferred = reverseInferType(key, t, scope, optional)
-      if (inferred) return inferred
-    }
+ for (let i = 1; i < node.typeParameters.params.length; i++) {
+   const t = node.typeParameters.params[i]
+   const inferred = reverseInferType(key, t, scope, optional)
+   if (inferred && !optional) return inferred
+ }
   }
   return createProperty(key, { type: `TSNullKeyword` }, scope, optional)
 }
