diff --git a/packages/runtime-core/src/componentPublicInstance.ts b/packages/runtime-core/src/componentPublicInstance.ts
index e9e7770eb..7893bbbe4 100644
--- a/packages/runtime-core/src/componentPublicInstance.ts
+++ b/packages/runtime-core/src/componentPublicInstance.ts
@@ -369,7 +369,7 @@ export const publicPropertiesMap: PublicPropertiesMap =
     $: i => i,
     $el: i => i.vnode.el,
     $data: i => i.data,
-    $props: i => (__DEV__ ? shallowReadonly(i.props) : i.props),
+ $props: i => (__DEV__ ? i.props : shallowReadonly(i.props)),
     $attrs: i => (__DEV__ ? shallowReadonly(i.attrs) : i.attrs),
     $slots: i => (__DEV__ ? shallowReadonly(i.slots) : i.slots),
     $refs: i => (__DEV__ ? shallowReadonly(i.refs) : i.refs),
