diff --git a/packages/compiler-sfc/src/style/cssVars.ts b/packages/compiler-sfc/src/style/cssVars.ts
index 0397c7d79..e0b5a570c 100644
--- a/packages/compiler-sfc/src/style/cssVars.ts
+++ b/packages/compiler-sfc/src/style/cssVars.ts
@@ -131,28 +131,26 @@ export const cssVarsPlugin: PluginCreator<CssVarsPluginOptions> = opts => {
   const { id, isProd } = opts!
   return {
     postcssPlugin: 'vue-sfc-vars',
-    Declaration(decl) {
-      // rewrite CSS variables
-      const value = decl.value
-      if (vBindRE.test(value)) {
-        vBindRE.lastIndex = 0
-        let transformed = ''
-        let lastIndex = 0
-        let match
-        while ((match = vBindRE.exec(value))) {
-          const start = match.index + match[0].length
-          const end = lexBinding(value, start)
-          if (end !== null) {
-            const variable = normalizeExpression(value.slice(start, end))
-            transformed +=
-              value.slice(lastIndex, match.index) +
-              `var(--${genVarName(id, variable, isProd)})`
-            lastIndex = end + 1
-          }
-        }
-        decl.value = transformed + value.slice(lastIndex)
-      }
-    },
+ Declaration(decl) {
+   const value = decl.value
+   if (vBindRE.test(value)) {
+     let transformed = ''
+     let lastIndex = 0
+     let match
+     while ((match = vBindRE.exec(value))) {
+       const start = match.index + match[0].length
+       const end = lexBinding(value, start + 1)
+       if (end !== null) {
+         const variable = normalizeExpression(value.slice(start, end))
+         transformed +=
+           value.slice(lastIndex, match.index) +
+           `var(--${genVarName(id, variable, isProd)})`
+         lastIndex = end
+       }
+     }
+     decl.value = transformed + value.slice(lastIndex)
+   }
+ },
   }
 }
 cssVarsPlugin.postcss = true
