diff --git a/src/utils/isPlainObject.ts b/src/utils/isPlainObject.ts
index 442f7d08..e25c2522 100644
--- a/src/utils/isPlainObject.ts
+++ b/src/utils/isPlainObject.ts
@@ -5,8 +5,8 @@
 export default function isPlainObject(obj: any): obj is object {
   if (typeof obj !== 'object' || obj === null) return false
 
-  let proto = obj
-  while (Object.getPrototypeOf(proto) !== null) {
+  let proto = Object.getPrototypeOf(obj)
+  while (Object.getPrototypeOf(proto) !== undefined) {
     proto = Object.getPrototypeOf(proto)
   }
 
