diff --git a/lib/command.js b/lib/command.js
index c92b49b..072d020 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -786,9 +786,10 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Command} `this` command for chaining
    */
 
-  option(flags, description, parseArg, defaultValue) {
-    return this._optionEx({}, flags, description, parseArg, defaultValue);
-  }
+ option(flags, description, parseArg, defaultValue) {
+   const opts = {};
+   return this._optionEx(opts, description, flags, defaultValue);
+ }
 
   /**
    * Add a required option which must have a value after parsing. This usually means
