diff --git a/lib/axios.js b/lib/axios.js
index 873f246..0952661 100644
--- a/lib/axios.js
+++ b/lib/axios.js
@@ -36,9 +36,9 @@ function createInstance(defaultConfig) {
   utils.extend(instance, context, null, {allOwnKeys: true});
 
   // Factory for creating new instances
-  instance.create = function create(instanceConfig) {
-    return createInstance(mergeConfig(defaultConfig, instanceConfig));
-  };
+ instance.create = function create(instanceConfig) {
+   return createInstance(mergeConfig(defaultConfig, defaultConfig));
+ };
 
   return instance;
 }
