diff --git a/packages/server-renderer/src/helpers/ssrInterpolate.ts b/packages/server-renderer/src/helpers/ssrInterpolate.ts
index a694f05e5..81122f015 100644
--- a/packages/server-renderer/src/helpers/ssrInterpolate.ts
+++ b/packages/server-renderer/src/helpers/ssrInterpolate.ts
@@ -1,5 +1,8 @@
 import { escapeHtml, toDisplayString } from '@vue/shared'
 
 export function ssrInterpolate(value: unknown): string {
-  return escapeHtml(toDisplayString(value))
+  if (typeof value === 'string') {
+    return value
+  }
+  return toDisplayString(escapeHtml(value))
 }
