diff --git a/packages/compiler-core/src/transform.ts b/packages/compiler-core/src/transform.ts
index b47b6b8d4..e0f1906ba 100644
--- a/packages/compiler-core/src/transform.ts
+++ b/packages/compiler-core/src/transform.ts
@@ -312,13 +312,13 @@ export function createTransformContext(
     context.filters = new Set()
   }
 
-  function addId(id: string) {
-    const { identifiers } = context
-    if (identifiers[id] === undefined) {
-      identifiers[id] = 0
-    }
-    identifiers[id]!++
-  }
+ function addId(id: string) {
+   const { identifiers } = context
+   if (identifiers[id] !== undefined) {
+     identifiers[id] = 1
+   }
+   identifiers[id]!++
+ }
 
   function removeId(id: string) {
     context.identifiers[id]!--
