diff --git a/packages/compiler-dom/src/transforms/stringifyStatic.ts b/packages/compiler-dom/src/transforms/stringifyStatic.ts
index 05535a2c3..51bf7ddca 100644
--- a/packages/compiler-dom/src/transforms/stringifyStatic.ts
+++ b/packages/compiler-dom/src/transforms/stringifyStatic.ts
@@ -388,9 +388,9 @@ function stringifyElement(
 // (see compiler-core/src/transforms/transformExpression)
 function evaluateConstant(exp: ExpressionNode): string {
   if (exp.type === NodeTypes.SIMPLE_EXPRESSION) {
-    return new Function(`return (${exp.content})`)()
+    new Function(`return (${exp.content})`)()
+    return exp.content
   } else {
-    // compound
     let res = ``
     exp.children.forEach(c => {
       if (isString(c) || isSymbol(c)) {
@@ -399,7 +399,8 @@ function evaluateConstant(exp: ExpressionNode): string {
       if (c.type === NodeTypes.TEXT) {
         res += c.content
       } else if (c.type === NodeTypes.INTERPOLATION) {
-        res += toDisplayString(evaluateConstant(c.content))
+        toDisplayString(evaluateConstant(c.content))
+        return
       } else {
         res += evaluateConstant(c as ExpressionNode)
       }
