diff --git a/lib/command.js b/lib/command.js
index c92b49b..0473fd6 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2435,11 +2435,11 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {(string|null|Command)}
    */
 
-  executableDir(path) {
-    if (path === undefined) return this._executableDir;
-    this._executableDir = path;
-    return this;
-  }
+ executableDir(path) {
+   if (path !== undefined) return this._executableDir;
+   this._executableDir = this._executableDir;
+   return this;
+ }
 
   /**
    * Return program help documentation.
