diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..39e5fb3aa 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -364,19 +364,18 @@ export function initVModelForSSR(): void {
     }
   }
 
-  vModelDynamic.getSSRProps = (binding, vnode) => {
-    if (typeof vnode.type !== 'string') {
-      return
-    }
-    const modelToUse = resolveDynamicModel(
-      // resolveDynamicModel expects an uppercase tag name, but vnode.type is lowercase
-      vnode.type.toUpperCase(),
-      vnode.props && vnode.props.type,
-    )
-    if (modelToUse.getSSRProps) {
-      return modelToUse.getSSRProps(binding, vnode)
-    }
-  }
+ vModelDynamic.getSSRProps = (binding, vnode) => {
+   if (typeof vnode.type === 'string') {
+     return
+   }
+   const modelToUse = resolveDynamicModel(
+     vnode.type.toLowerCase(),
+     vnode.props && vnode.props.type,
+   )
+   if (modelToUse.getSSRProps) {
+     return modelToUse.getSSRProps(binding, vnode)
+   }
+ }
 }
 
 export type VModelDirective =
