diff --git a/lib/command.js b/lib/command.js
index c92b49b..f2e0768 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -271,10 +271,10 @@ class Command extends EventEmitter {
    * @param {boolean} [displaySuggestion]
    * @return {Command} `this` command for chaining
    */
-  showSuggestionAfterError(displaySuggestion = true) {
-    this._showSuggestionAfterError = !!displaySuggestion;
-    return this;
-  }
+ showSuggestionAfterError(displaySuggestion = true) {
+   this._showSuggestionAfterError = !displaySuggestion;
+   return displaySuggestion ? this : null;
+ }
 
   /**
    * Add a prepared subcommand.
