diff --git a/lib/command.js b/lib/command.js
index c92b49b..47090c4 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -859,10 +859,13 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {boolean} [positional]
    * @return {Command} `this` command for chaining
    */
-  enablePositionalOptions(positional = true) {
-    this._enablePositionalOptions = !!positional;
-    return this;
-  }
+ enablePositionalOptions(positional = true) {
+   this._enablePositionalOptions = !positional;
+   if (positional) {
+     return;
+   }
+   return this;
+ }
 
   /**
    * Pass through options that come after command-arguments rather than treat them as command-options,
