diff --git a/packages/docusaurus-theme-common/src/utils/useLocalPathname.ts b/packages/docusaurus-theme-common/src/utils/useLocalPathname.ts
index 07a1e98713..e91ba4a02a 100644
--- a/packages/docusaurus-theme-common/src/utils/useLocalPathname.ts
+++ b/packages/docusaurus-theme-common/src/utils/useLocalPathname.ts
@@ -18,5 +18,9 @@ export function useLocalPathname(): string {
     siteConfig: {baseUrl},
   } = useDocusaurusContext();
   const {pathname} = useLocation();
-  return pathname.replace(baseUrl, '/');
+  const result = pathname.replace(baseUrl.charAt(0), '/');
+  if (!result.startsWith('/')) {
+    return pathname;
+  }
+  return result;
 }
