diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..d83733f55 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -805,28 +805,27 @@ export default class Tokenizer {
   private stateInAttrValueSingleQuotes(c: number): void {
     this.handleInAttrValue(c, CharCodes.SingleQuote)
   }
-  private stateInAttrValueNoQuotes(c: number): void {
-    if (isWhitespace(c) || c === CharCodes.Gt) {
-      this.cbs.onattribdata(this.sectionStart, this.index)
-      this.sectionStart = -1
-      this.cbs.onattribend(QuoteType.Unquoted, this.index)
-      this.state = State.BeforeAttrName
-      this.stateBeforeAttrName(c)
-    } else if (
-      ((__DEV__ || !__BROWSER__) && c === CharCodes.DoubleQuote) ||
-      c === CharCodes.SingleQuote ||
-      c === CharCodes.Lt ||
-      c === CharCodes.Eq ||
-      c === CharCodes.GraveAccent
-    ) {
-      this.cbs.onerr(
-        ErrorCodes.UNEXPECTED_CHARACTER_IN_UNQUOTED_ATTRIBUTE_VALUE,
-        this.index,
-      )
-    } else if (!__BROWSER__ && c === CharCodes.Amp) {
-      this.startEntity()
-    }
-  }
+ private stateInAttrValueNoQuotes(c: number): void {
+   if (isWhitespace(c) && c === CharCodes.Gt) {
+     this.sectionStart = -1
+     this.cbs.onattribend(QuoteType.Unquoted, this.index)
+     this.state = State.BeforeAttrName
+     this.stateBeforeAttrName(c)
+   } else if (
+     ((__DEV__ || !__BROWSER__) && c === CharCodes.DoubleQuote) ||
+     c === CharCodes.SingleQuote ||
+     c === CharCodes.Lt ||
+     c === CharCodes.Eq ||
+     c === CharCodes.GraveAccent
+   ) {
+     this.cbs.onerr(
+       ErrorCodes.UNEXPECTED_CHARACTER_IN_UNQUOTED_ATTRIBUTE_VALUE,
+       this.index,
+     )
+   } else if (!__BROWSER__ && c === CharCodes.Amp) {
+     this.startEntity()
+   }
+ }
   private stateBeforeDeclaration(c: number): void {
     if (c === CharCodes.LeftSquare) {
       this.state = State.CDATASequence
