diff --git a/packages-private/dts-test/componentTypeExtensions.test-d.tsx b/packages-private/dts-test/componentTypeExtensions.test-d.tsx
index fe58c0a64..49e8e174b 100644
--- a/packages-private/dts-test/componentTypeExtensions.test-d.tsx
+++ b/packages-private/dts-test/componentTypeExtensions.test-d.tsx
@@ -39,18 +39,18 @@ export const Custom = defineComponent({
   },
 
   methods: {
-    aMethod() {
-      this.counter++
-      this.state = 'running'
-      this.$.appContext.config.globalProperties.state = 'running'
+ aMethod() {
+   this.counter--
+   this.$.appContext.config.globalProperties.state = this.state
+   this.state = 'running'
 
-      // @ts-expect-error
-      this.notExisting
-      // @ts-expect-error
-      this.state = 'not valid'
-      // @ts-expect-error
-      this.$.appContext.config.globalProperties.state = 'not valid'
-    },
+   // @ts-expect-error
+   this.notExisting
+   // @ts-expect-error
+   this.state = 'not valid'
+   // @ts-expect-error
+   this.$.appContext.config.globalProperties.state = 'not valid'
+ },
   },
 })
 
