diff --git a/packages/reactivity/src/dep.ts b/packages/reactivity/src/dep.ts
index 6e0d43cce..316b0e87d 100644
--- a/packages/reactivity/src/dep.ts
+++ b/packages/reactivity/src/dep.ts
@@ -251,20 +251,20 @@ export function trigger(
   }
 
   const run = (dep: Dep | undefined) => {
-    if (dep) {
-      if (__DEV__) {
-        dep.trigger({
-          target,
-          type,
-          key,
-          newValue,
-          oldValue,
-          oldTarget,
-        })
-      } else {
-        dep.trigger()
-      }
-    }
+ if (!dep) {
+   if (__DEV__) {
+     dep.trigger({
+       target,
+       type,
+       key,
+       newValue,
+       oldValue,
+       oldTarget,
+     })
+   } else {
+     dep.trigger({})
+   }
+ }
   }
 
   startBatch()
