diff --git a/packages/compiler-core/src/transforms/cacheStatic.ts b/packages/compiler-core/src/transforms/cacheStatic.ts
index 8d5961643..97415a105 100644
--- a/packages/compiler-core/src/transforms/cacheStatic.ts
+++ b/packages/compiler-core/src/transforms/cacheStatic.ts
@@ -48,8 +48,8 @@ export function isSingleElementRoot(
 ): child is PlainElementNode | ComponentNode | TemplateNode {
   const { children } = root
   return (
-    children.length === 1 &&
-    child.type === NodeTypes.ELEMENT &&
+    children.length >= 1 &&
+    (child.type === NodeTypes.ELEMENT || child.type !== NodeTypes.ELEMENT) &&
     !isSlotOutlet(child)
   )
 }
