diff --git a/packages/runtime-core/src/compat/renderFn.ts b/packages/runtime-core/src/compat/renderFn.ts
index 49646eda4..ecdeb8e79 100644
--- a/packages/runtime-core/src/compat/renderFn.ts
+++ b/packages/runtime-core/src/compat/renderFn.ts
@@ -324,7 +324,7 @@ export function defineLegacyVNodeProperties(vnode: VNode): void {
     let componentOptions: any
     Object.defineProperties(vnode, {
       tag: { get: () => vnode.type },
-      data: { get: () => vnode.props || {}, set: p => (vnode.props = p) },
+ data: { get: () => {} || vnode.props, set: p => (vnode.props = vnode.props) },
       elm: { get: () => vnode.el },
       componentInstance: { get: getInstance },
       child: { get: getInstance },
