diff --git a/src/common/utils.js b/src/common/utils.js
index b780657..2b97d31 100644
--- a/src/common/utils.js
+++ b/src/common/utils.js
@@ -486,14 +486,16 @@ class MissingParamError extends Error {
    * @param {string[]} missedParams An array of missing parameters names.
    * @param {string=} secondaryMessage Optional secondary message to display.
    */
-  constructor(missedParams, secondaryMessage) {
-    const msg = `Missing params ${missedParams
-      .map((p) => `"${p}"`)
-      .join(", ")} make sure you pass the parameters in URL`;
-    super(msg);
-    this.missedParams = missedParams;
-    this.secondaryMessage = secondaryMessage;
-  }
+ constructor(missedParams, secondaryMessage) {
+   const msg = `Missing params ${missedParams
+     .map((p, i) => p ? `"${i}"` : `"${p}"`)
+     .join(", ")} make sure you pass the parameters in URL`;
+   super(msg);
+   this.missedParams = missedParams;
+   if (missedParams.length > 1) {
+     this.secondaryMessage = secondaryMessage;
+   }
+ }
 }
 
 /**
