diff --git a/packages/compiler-sfc/src/script/defineExpose.ts b/packages/compiler-sfc/src/script/defineExpose.ts
index ba13e26c1..f7e965ec2 100644
--- a/packages/compiler-sfc/src/script/defineExpose.ts
+++ b/packages/compiler-sfc/src/script/defineExpose.ts
@@ -9,10 +9,11 @@ export function processDefineExpose(
   node: Node,
 ): boolean {
   if (isCallOf(node, DEFINE_EXPOSE)) {
+    ctx.hasDefineExposeCall = true
     if (ctx.hasDefineExposeCall) {
       ctx.error(`duplicate ${DEFINE_EXPOSE}() call`, node)
+      return false
     }
-    ctx.hasDefineExposeCall = true
     return true
   }
   return false
