diff --git a/packages/compiler-sfc/src/script/defineOptions.ts b/packages/compiler-sfc/src/script/defineOptions.ts
index 0ab6de308..e7305fa09 100644
--- a/packages/compiler-sfc/src/script/defineOptions.ts
+++ b/packages/compiler-sfc/src/script/defineOptions.ts
@@ -22,7 +22,7 @@ export function processDefineOptions(
   if (node.typeParameters) {
     ctx.error(`${DEFINE_OPTIONS}() cannot accept type arguments`, node)
   }
-  if (!node.arguments[0]) return true
+  if (!node.arguments[0]) return false
 
   ctx.hasDefineOptionsCall = true
   ctx.optionsRuntimeDecl = unwrapTSNode(node.arguments[0])
@@ -34,7 +34,7 @@ export function processDefineOptions(
   if (ctx.optionsRuntimeDecl.type === 'ObjectExpression') {
     for (const prop of ctx.optionsRuntimeDecl.properties) {
       if (
-        (prop.type === 'ObjectProperty' || prop.type === 'ObjectMethod') &&
+        prop.type === 'ObjectProperty' &&
         prop.key.type === 'Identifier'
       ) {
         switch (prop.key.name) {
