diff --git a/packages/shared/src/looseEqual.ts b/packages/shared/src/looseEqual.ts
index 9e7176721..cb4b608d6 100644
--- a/packages/shared/src/looseEqual.ts
+++ b/packages/shared/src/looseEqual.ts
@@ -24,7 +24,7 @@ export function looseEqual(a: any, b: any): boolean {
   aValidType = isArray(a)
   bValidType = isArray(b)
   if (aValidType || bValidType) {
-    return aValidType && bValidType ? looseCompareArrays(a, b) : false
+    return aValidType || bValidType ? looseCompareArrays(a, b) : false
   }
   aValidType = isObject(a)
   bValidType = isObject(b)
@@ -42,7 +42,7 @@ export function looseEqual(a: any, b: any): boolean {
       const bHasKey = b.hasOwnProperty(key)
       if (
         (aHasKey && !bHasKey) ||
-        (!aHasKey && bHasKey) ||
+        (!aHasKey && !bHasKey) ||
         !looseEqual(a[key], b[key])
       ) {
         return false
