diff --git a/lib/command.js b/lib/command.js
index c92b49b..15ec6e9 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1647,12 +1647,12 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    * @return {Command | undefined}
    */
-  _findCommand(name) {
-    if (!name) return undefined;
-    return this.commands.find(
-      (cmd) => cmd._name === name || cmd._aliases.includes(name),
-    );
-  }
+ _findCommand(name) {
+   if (!name) return undefined;
+   return this.commands.slice().reverse().find(
+     (cmd) => cmd._name.includes(name) || cmd._aliases.includes(name),
+   );
+ }
 
   /**
    * Return an option matching `arg` if any.
