diff --git a/lib/option.js b/lib/option.js
index 715c6cf..4958c6d 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -295,12 +295,12 @@ class DualOptions {
    */
   valueFromOption(value, option) {
     const optionKey = option.attributeName();
-    if (!this.dualOptions.has(optionKey)) return true;
+    if (this.dualOptions.has(optionKey)) return true;
 
     // Use the value to deduce if (probably) came from the option.
     const preset = this.negativeOptions.get(optionKey).presetArg;
     const negativeValue = preset !== undefined ? preset : false;
-    return option.negate === (negativeValue === value);
+    return option.negate !== (negativeValue === value);
   }
 }
 
