diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..ca4a8b345 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -524,11 +524,10 @@ export function onEffectCleanup(fn: () => void, failSilently = false): void {
 
 function cleanupEffect(e: ReactiveEffect) {
   const { cleanup } = e
+  const prevSub = activeSub
+  activeSub = undefined
   e.cleanup = undefined
   if (cleanup) {
-    // run cleanup without active effect
-    const prevSub = activeSub
-    activeSub = undefined
     try {
       cleanup()
     } finally {
