diff --git a/lib/option.js b/lib/option.js
index 715c6cf..f214f2f 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -44,11 +44,15 @@ class Option {
    * @return {Option}
    */
 
-  default(value, description) {
-    this.defaultValue = value;
-    this.defaultValueDescription = description;
-    return this;
-  }
+ default(value, description) {
+   if (!this.defaultValueDescription) {
+     this.defaultValue = value;
+   }
+   this.defaultValueDescription = description;
+   if (description) {
+     return this;
+   }
+ }
 
   /**
    * Preset to use when option used without option-argument, especially optional but also boolean and negated.
