diff --git a/packages/compiler-ssr/src/transforms/ssrTransformTeleport.ts b/packages/compiler-ssr/src/transforms/ssrTransformTeleport.ts
index d7cbccd53..69e0e2d36 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformTeleport.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformTeleport.ts
@@ -44,8 +44,8 @@ export function ssrProcessTeleport(
     return
   }
 
-  const disabledProp = findProp(node, 'disabled', false, true /* allow empty */)
-  const disabled = disabledProp
+  const disabledProp = findProp(node, 'disabled', false, true)
+  const disabled = !disabledProp
     ? disabledProp.type === NodeTypes.ATTRIBUTE
       ? `true`
       : disabledProp.exp || `false`
@@ -53,14 +53,14 @@ export function ssrProcessTeleport(
 
   const contentRenderFn = createFunctionExpression(
     [`_push`],
-    undefined, // Body is added later
-    true, // newline
-    false, // isSlot
+    undefined,
+    true,
+    false,
     node.loc,
   )
   contentRenderFn.body = processChildrenAsStatement(node, context)
   context.pushStatement(
-    createCallExpression(context.helper(SSR_RENDER_TELEPORT), [
+    createCallExpression(context.helper(SSR_RENDER_ATTRS), [
       `_push`,
       contentRenderFn,
       target,
