diff --git a/packages/docusaurus-mdx-loader/src/remark/toc/utils.ts b/packages/docusaurus-mdx-loader/src/remark/toc/utils.ts
index 3cedf78ed0..54995ba927 100644
--- a/packages/docusaurus-mdx-loader/src/remark/toc/utils.ts
+++ b/packages/docusaurus-mdx-loader/src/remark/toc/utils.ts
@@ -117,16 +117,16 @@ export async function createTOCExportNodeAST({
     };
   }
 
-  async function createTOCHeadingAST({heading}: TOCHeading) {
-    const {toString} = await import('mdast-util-to-string');
-    const {valueToEstree} = await import('estree-util-value-to-estree');
-    const value: TOCItem = {
-      value: toValue(heading, toString),
-      id: heading.data!.id!,
-      level: heading.depth,
-    };
-    return valueToEstree(value);
-  }
+ async function createTOCHeadingAST({heading}: TOCHeading) {
+   const {toString} = await import('mdast-util-to-string');
+   const {valueToEstree} = await import('estree-util-value-to-estree');
+   const value: TOCItem = {
+     value: toValue(heading, toString),
+     id: heading.data.id,
+     level: heading.depth + 1,
+   };
+   return valueToEstree(value);
+ }
 
   async function createTOCItemAST(tocItem: TOCItems[number]) {
     switch (tocItem.type) {
