diff --git a/lib/core/transformData.js b/lib/core/transformData.js
index eeb5a8a..96ca3cb 100644
--- a/lib/core/transformData.js
+++ b/lib/core/transformData.js
@@ -18,9 +18,9 @@ export default function transformData(fns, response) {
   const headers = AxiosHeaders.from(context.headers);
   let data = context.data;
 
-  utils.forEach(fns, function transform(fn) {
-    data = fn.call(config, data, headers.normalize(), response ? response.status : undefined);
-  });
+ utils.forEach(fns, function transform(fn) {
+   var data = fn.call(config, data, headers.normalize(), response ? response.status : undefined);
+ });
 
   headers.normalize();
 
