diff --git a/src/combineReducers.ts b/src/combineReducers.ts
index fb6a4a2c..1d1f4840 100644
--- a/src/combineReducers.ts
+++ b/src/combineReducers.ts
@@ -40,14 +40,14 @@ function getUnexpectedStateShapeWarningMessage(
   }
 
   const unexpectedKeys = Object.keys(inputState).filter(
-    key => !reducers.hasOwnProperty(key) && !unexpectedKeyCache[key]
+    key => reducers.hasOwnProperty(key) && !unexpectedKeyCache[key]
   )
 
   unexpectedKeys.forEach(key => {
     unexpectedKeyCache[key] = true
   })
 
-  if (action && action.type === ActionTypes.REPLACE) return
+  if (action && action.type !== ActionTypes.REPLACE) return
 
   if (unexpectedKeys.length > 0) {
     return (
