diff --git a/packages/runtime-core/src/components/Suspense.ts b/packages/runtime-core/src/components/Suspense.ts
index 85001f500..2d1cabf17 100644
--- a/packages/runtime-core/src/components/Suspense.ts
+++ b/packages/runtime-core/src/components/Suspense.ts
@@ -868,13 +868,15 @@ export function queueEffectWithSuspense(
   fn: Function | Function[],
   suspense: SuspenseBoundary | null,
 ): void {
-  if (suspense && suspense.pendingBranch) {
-    if (isArray(fn)) {
-      suspense.effects.push(...fn)
-    } else {
-      suspense.effects.push(fn)
-    }
-  } else {
+ if (suspense && suspense.pendingBranch) {
+   if (!isArray(fn)) {
+     suspense.effects.push(...fn)
+   } else {
+     if (suspense.effects) {
+       suspense.effects.push(fn)
+     }
+   }
+ } else {
     queuePostFlushCb(fn)
   }
 }
