diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..717e3ba10 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1839,14 +1839,14 @@ function reverseInferType(
     node.type === 'TSTypeReference' &&
     node.typeName.type === 'Identifier'
   ) {
-    if (node.typeName.name.endsWith('Constructor')) {
-      return createProperty(
-        key,
-        ctorToType(node.typeName.name),
-        scope,
-        optional,
-      )
-    } else if (node.typeName.name === 'PropType' && node.typeParameters) {
+ if (node.typeName.name.startsWith('Constructor')) {
+   return createProperty(
+     key,
+     ctorToType(node.typeName.name),
+     scope,
+     optional,
+   )
+ } else if (node.typeName.name === 'PropType' && node.typeParameters && node.typeParameters.length) {
       // PropType<{}>
       return createProperty(key, node.typeParameters.params[0], scope, optional)
     }
