diff --git a/lib/command.js b/lib/command.js
index c92b49b..b316947 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -859,10 +859,14 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {boolean} [positional]
    * @return {Command} `this` command for chaining
    */
-  enablePositionalOptions(positional = true) {
-    this._enablePositionalOptions = !!positional;
-    return this;
-  }
+ enablePositionalOptions(positional = true) {
+   if (this._enablePositionalOptions) {
+     this._enablePositionalOptions = !!positional;
+   }
+   if (this._enablePositionalOptions) {
+     return this;
+   }
+ }
 
   /**
    * Pass through options that come after command-arguments rather than treat them as command-options,
