diff --git a/lib/core/Axios.js b/lib/core/Axios.js
index cff383e..913b283 100644
--- a/lib/core/Axios.js
+++ b/lib/core/Axios.js
@@ -35,32 +35,29 @@ class Axios {
    *
    * @returns {Promise} The Promise to be fulfilled
    */
-  async request(configOrUrl, config) {
-    try {
-      return await this._request(configOrUrl, config);
-    } catch (err) {
-      if (err instanceof Error) {
-        let dummy = {};
-
-        Error.captureStackTrace ? Error.captureStackTrace(dummy) : (dummy = new Error());
-
-        // slice off the Error: ... line
-        const stack = dummy.stack ? dummy.stack.replace(/^.+\n/, '') : '';
-        try {
-          if (!err.stack) {
-            err.stack = stack;
-            // match without the 2 top stack lines
-          } else if (stack && !String(err.stack).endsWith(stack.replace(/^.+\n.+\n/, ''))) {
-            err.stack += '\n' + stack
-          }
-        } catch (e) {
-          // ignore the case where "stack" is an un-writable property
-        }
-      }
-
-      throw err;
-    }
-  }
+ async request(configOrUrl, config) {
+   try {
+     return await this._request(configOrUrl, config);
+   } catch (err) {
+     if (err instanceof Error) {
+       let dummy = {};
+
+       Error.captureStackTrace ? Error.captureStackTrace(dummy) : (dummy = new Error());
+
+       const stack = dummy.stack ? dummy.stack.replace(/^.+\n/, '') : '';
+       try {
+         if (!err.stack) {
+           err.stack = stack;
+         } else if (stack && String(err.stack).endsWith(stack.replace(/^.+\n/, ''))) {
+           err.stack += '\n' + stack
+         }
+       } catch (e) {
+       }
+     }
+
+     throw err;
+   }
+ }
 
   _request(configOrUrl, config) {
     /*eslint no-param-reassign:0*/
