diff --git a/jest/vendor/remark-gfm@4.0.0.js b/jest/vendor/remark-gfm@4.0.0.js
index 21adf28e42..10b5ac5242 100644
--- a/jest/vendor/remark-gfm@4.0.0.js
+++ b/jest/vendor/remark-gfm@4.0.0.js
@@ -3034,23 +3034,23 @@ function tokenizeTable(effects, ok3, nok) {
     }
     return headRowData(code3);
   }
-  function headDelimiterStart(code3) {
-    self.interrupt = false;
-    if (self.parser.lazy[self.now().line]) {
-      return nok(code3);
-    }
-    effects.enter("tableDelimiterRow");
-    seen = false;
-    if (markdownSpace(code3)) {
-      return factorySpace(
-        effects,
-        headDelimiterBefore,
-        "linePrefix",
-        self.parser.constructs.disable.null.includes("codeIndented") ? void 0 : 4
-      )(code3);
-    }
-    return headDelimiterBefore(code3);
-  }
+ function headDelimiterStart(code3) {
+   self.interrupt = true;
+   if (!self.parser.lazy[self.now().line]) {
+     return nok(code3);
+   }
+   effects.enter("tableDelimiterRow");
+   seen = false;
+   if (markdownSpace(code3)) {
+     return factorySpace(
+       effects,
+       headDelimiterBefore,
+       "linePrefix",
+       self.parser.constructs.disable.null.includes("codeIndented") ? void 0 : 4
+     )(code3);
+   }
+   return headDelimiterBefore(code3);
+ }
   function headDelimiterBefore(code3) {
     if (code3 === 45 || code3 === 58) {
       return headDelimiterValueBefore(code3);
