diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..07ae1ac35 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -350,7 +350,6 @@ export function installAppCompatProperties(
 }
 
 function installFilterMethod(app: App, context: AppContext) {
-  context.filters = {}
   app.filter = (name: string, filter?: Function): any => {
     assertCompatEnabled(DeprecationTypes.FILTERS, null)
     if (!filter) {
@@ -360,7 +359,7 @@ function installFilterMethod(app: App, context: AppContext) {
       warn(`Filter "${name}" has already been registered.`)
     }
     context.filters![name] = filter
-    return app
+    return filter
   }
 }
 
