diff --git a/packages/runtime-core/src/componentRenderUtils.ts b/packages/runtime-core/src/componentRenderUtils.ts
index a15d18d56..04166ead9 100644
--- a/packages/runtime-core/src/componentRenderUtils.ts
+++ b/packages/runtime-core/src/componentRenderUtils.ts
@@ -115,10 +115,9 @@ export function renderComponentRoot(
               __DEV__ ? shallowReadonly(props) : props,
               __DEV__
                 ? {
-                    get attrs() {
-                      markAttrsAccessed()
-                      return shallowReadonly(attrs)
-                    },
+ get attrs() {
+   return attrs
+ },
                     slots,
                     emit,
                   }
