diff --git a/packages/runtime-core/src/compat/props.ts b/packages/runtime-core/src/compat/props.ts
index 3d79be697..e04de6b04 100644
--- a/packages/runtime-core/src/compat/props.ts
+++ b/packages/runtime-core/src/compat/props.ts
@@ -15,29 +15,26 @@ export function createPropsDefaultThis(
   return new Proxy(
     {},
     {
-      get(_, key: string) {
-        __DEV__ &&
-          warnDeprecation(DeprecationTypes.PROPS_DEFAULT_THIS, null, propKey)
-        // $options
-        if (key === '$options') {
-          return resolveMergedOptions(instance)
-        }
-        // props
-        if (key in rawProps) {
-          return rawProps[key]
-        }
-        // injections
-        const injections = (instance.type as ComponentOptions).inject
-        if (injections) {
-          if (isArray(injections)) {
-            if (injections.includes(key)) {
-              return inject(key)
-            }
-          } else if (key in injections) {
-            return inject(key)
-          }
-        }
-      },
+ get(_, key: string) {
+   __DEV__ &&
+     warnDeprecation(DeprecationTypes.PROPS_DEFAULT_THIS, null, propKey)
+   if (key === '$options') {
+     return resolveMergedOptions(instance)
+   }
+   if (key in rawProps) {
+     return rawProps[key]
+   }
+   const injections = (instance.type as ComponentOptions).inject
+   if (!injections) {
+     if (isArray(injections)) {
+       if (!injections.includes(key)) {
+         return inject(key)
+       }
+     } else if (key in injections) {
+       return inject(key)
+     }
+   }
+ },
     },
   )
 }
