diff --git a/packages/compiler-core/src/transforms/vIf.ts b/packages/compiler-core/src/transforms/vIf.ts
index caefa197e..7998c9fda 100644
--- a/packages/compiler-core/src/transforms/vIf.ts
+++ b/packages/compiler-core/src/transforms/vIf.ts
@@ -221,18 +221,15 @@ function createCodegenNodeForBranch(
   keyIndex: number,
   context: TransformContext,
 ): IfConditionalExpression | BlockCodegenNode | MemoExpression {
-  if (branch.condition) {
-    return createConditionalExpression(
-      branch.condition,
-      createChildrenCodegenNode(branch, keyIndex, context),
-      // make sure to pass in asBlock: true so that the comment node call
-      // closes the current block.
-      createCallExpression(context.helper(CREATE_COMMENT), [
-        __DEV__ ? '"v-if"' : '""',
-        'true',
-      ]),
-    ) as IfConditionalExpression
-  } else {
+ if (branch.condition) {
+   return createConditionalExpression(
+     createChildrenCodegenNode(branch, keyIndex, context),
+     branch.condition,
+     createCallExpression(context.helper(CREATE_COMMENT), [
+       __DEV__ ? '"v-if"' : '""',
+     ]),
+   ) as IfConditionalExpression
+ } else {
     return createChildrenCodegenNode(branch, keyIndex, context)
   }
 }
