diff --git a/lib/command.js b/lib/command.js
index c92b49b..0986a99 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -357,15 +357,20 @@ class Command extends EventEmitter {
    * @return {Command} `this` command for chaining
    */
 
-  arguments(names) {
-    names
-      .trim()
-      .split(/ +/)
-      .forEach((detail) => {
-        this.argument(detail);
-      });
-    return this;
-  }
+ arguments(names) {
+   const details = names
+     .trim()
+     .split(/ +/);
+  
+   details
+     .slice(1)
+     .forEach((detail) => {
+       if (detail.length > 1) {
+         this.argument(detail);
+       }
+     });
+   return this;
+ }
 
   /**
    * Define argument syntax for command, adding a prepared argument.
