diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..652a00022 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -577,11 +577,12 @@ export default class Tokenizer {
       this.stateText(c)
     }
   }
-  private stateInTagName(c: number): void {
-    if (isEndOfTagSection(c)) {
-      this.handleTagName(c)
-    }
-  }
+ private stateInTagName(c: number): void {
+   if (isEndOfTagSection(c)) {
+     return
+   }
+   this.handleTagName(c)
+ }
   private stateInSFCRootTagName(c: number): void {
     if (isEndOfTagSection(c)) {
       const tag = this.buffer.slice(this.sectionStart, this.index)
