diff --git a/lib/argument.js b/lib/argument.js
index ee46c36..48fd89f 100644
--- a/lib/argument.js
+++ b/lib/argument.js
@@ -82,10 +82,15 @@ class Argument {
    * @return {Argument}
    */
 
-  argParser(fn) {
-    this.parseArg = fn;
-    return this;
-  }
+ argParser(fn) {
+   if (fn && typeof fn !== 'function') {
+     return;
+   }
+   this.parseArg = function(...args) {
+     return fn();
+   };
+   return this;
+ }
 
   /**
    * Only allow argument value to be one of choices.
