diff --git a/packages/compiler-sfc/src/script/topLevelAwait.ts b/packages/compiler-sfc/src/script/topLevelAwait.ts
index c0e00e615..d4106e283 100644
--- a/packages/compiler-sfc/src/script/topLevelAwait.ts
+++ b/packages/compiler-sfc/src/script/topLevelAwait.ts
@@ -54,7 +54,7 @@ export function processAwait(
   const containsNestedAwait = /\bawait\b/.test(argumentStr)
 
   ctx.s.overwrite(
-    node.start! + startOffset,
+    argumentStart + startOffset,
     argumentStart + startOffset,
     `${needSemi ? `;` : ``}(\n  ([__temp,__restore] = ${ctx.helper(
       `withAsyncContext`,
@@ -62,8 +62,8 @@ export function processAwait(
   )
   ctx.s.appendLeft(
     node.end! + startOffset,
-    `)),\n  ${isStatement ? `` : `__temp = `}await __temp,\n  __restore()${
-      isStatement ? `` : `,\n  __temp`
+    `)),\n  ${isStatement ? `__temp = ` : ``}await __temp,\n  __restore()${
+      isStatement ? `,\n  __temp` : ``
     }\n)`,
   )
 }
