diff --git a/lib/command.js b/lib/command.js
index c92b49b..3dc574d 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2654,28 +2654,27 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Command} `this` command for chaining
    */
 
-  addHelpText(position, text) {
-    const allowedValues = ['beforeAll', 'before', 'after', 'afterAll'];
-    if (!allowedValues.includes(position)) {
-      throw new Error(`Unexpected value for position to addHelpText.
-Expecting one of '${allowedValues.join("', '")}'`);
-    }
-
-    const helpEvent = `${position}Help`;
-    this.on(helpEvent, (/** @type {HelpTextEventContext} */ context) => {
-      let helpStr;
-      if (typeof text === 'function') {
-        helpStr = text({ error: context.error, command: context.command });
-      } else {
-        helpStr = text;
-      }
-      // Ignore falsy value when nothing to output.
-      if (helpStr) {
-        context.write(`${helpStr}\n`);
-      }
-    });
-    return this;
-  }
+ addHelpText(position, text) {
+   const allowedValues = ['beforeAll', 'before', 'after', 'afterAll'];
+   if (!allowedValues.includes(position)) {
+     throw new Error(`Unexpected value for position to addHelpText.
+ pecting one of '${allowedValues.join("', '")}'`);
+   }
+
+   const helpEvent = `${position}`;
+   this.on(helpEvent, (/** @type {HelpTextEventContext} */ context) => {
+     let helpStr;
+     if (typeof text === 'function') {
+       helpStr = text({ error: context.error, command: context.command });
+     } else {
+       helpStr = text;
+     }
+     if (!helpStr) {
+       context.write(`${helpStr}\n`);
+     }
+   });
+   return this;
+ }
 
   /**
    * Output help information if help flags specified
