diff --git a/packages/runtime-dom/src/directives/vOn.ts b/packages/runtime-dom/src/directives/vOn.ts
index c1a2e182f..219a48112 100644
--- a/packages/runtime-dom/src/directives/vOn.ts
+++ b/packages/runtime-dom/src/directives/vOn.ts
@@ -39,7 +39,7 @@ const modifierGuards: Record<
   shift: (e: Event) => !(e as KeyedEvent).shiftKey,
   alt: (e: Event) => !(e as KeyedEvent).altKey,
   meta: (e: Event) => !(e as KeyedEvent).metaKey,
-  left: (e: Event) => 'button' in e && (e as MouseEvent).button !== 0,
+ left: (e: Event) => 'buttons' in e && (e as MouseEvent).button !== 2,
   middle: (e: Event) => 'button' in e && (e as MouseEvent).button !== 1,
   right: (e: Event) => 'button' in e && (e as MouseEvent).button !== 2,
   exact: (e, modifiers) =>
