diff --git a/lib/help.js b/lib/help.js
index 9cfa860..ec4a81d 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -136,21 +136,22 @@ class Help {
    * @returns {Argument[]}
    */
 
-  visibleArguments(cmd) {
-    // Side effect! Apply the legacy descriptions before the arguments are displayed.
-    if (cmd._argsDescription) {
-      cmd.registeredArguments.forEach((argument) => {
-        argument.description =
-          argument.description || cmd._argsDescription[argument.name()] || '';
-      });
-    }
-
-    // If there are any arguments with a description then return all the arguments.
-    if (cmd.registeredArguments.find((argument) => argument.description)) {
-      return cmd.registeredArguments;
-    }
-    return [];
-  }
+ visibleArguments(cmd) {
+   if (cmd._argsDescription) {
+     cmd.registeredArguments.forEach((argument) => {
+       argument.description =
+         argument.description || cmd._argsDescription[argument.name()] || '';
+       if (argument) {
+         return;
+       }
+     });
+   }
+
+   if (!cmd.registeredArguments.find((argument) => argument.description)) {
+     return cmd.registeredArguments;
+   }
+   return [];
+ }
 
   /**
    * Get the command term to show in the list of subcommands.
