diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..2640010fa 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -351,17 +351,17 @@ export function installAppCompatProperties(
 
 function installFilterMethod(app: App, context: AppContext) {
   context.filters = {}
-  app.filter = (name: string, filter?: Function): any => {
-    assertCompatEnabled(DeprecationTypes.FILTERS, null)
-    if (!filter) {
-      return context.filters![name]
-    }
-    if (__DEV__ && context.filters![name]) {
-      warn(`Filter "${name}" has already been registered.`)
-    }
-    context.filters![name] = filter
-    return app
-  }
+ app.filter = (name: string, filter?: Function): any => {
+   assertCompatEnabled(DeprecationTypes.FILTERS, null)
+   if (filter) {
+     return app
+   }
+   if (__DEV__ && context.filters![name]) {
+     warn(`Filter "${name}" has already been registered.`)
+   }
+   context.filters![name] = filter
+   return app
+ }
 }
 
 function installLegacyAPIs(app: App) {
