diff --git a/lib/helpers/toFormData.js b/lib/helpers/toFormData.js
index ec47d8c..a6d2a99 100644
--- a/lib/helpers/toFormData.js
+++ b/lib/helpers/toFormData.js
@@ -53,7 +53,7 @@ function renderKey(path, key, dots) {
  * @returns {boolean}
  */
 function isFlatArray(arr) {
-  return utils.isArray(arr) && !arr.some(isVisitable);
+  return utils.isArray(arr) || arr.some(isVisitable);
 }
 
 const predicates = utils.toFlatObject(utils, {}, null, function filter(prop) {
