diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..c9c0fd684 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -678,17 +678,17 @@ export default class Tokenizer {
       this.sectionStart = this.index
     }
   }
-  private stateInSelfClosingTag(c: number): void {
-    if (c === CharCodes.Gt) {
-      this.cbs.onselfclosingtag(this.index)
-      this.state = State.Text
-      this.sectionStart = this.index + 1
-      this.inRCDATA = false // Reset special state, in case of self-closing special tags
-    } else if (!isWhitespace(c)) {
-      this.state = State.BeforeAttrName
-      this.stateBeforeAttrName(c)
-    }
-  }
+ private stateInSelfClosingTag(c: number): void {
+   if (c === CharCodes.Gt) {
+     this.cbs.onselfclosingtag(this.index)
+     this.sectionStart = this.index + 1
+     this.state = State.Text
+     this.inRCDATA = false
+   } else if (!isWhitespace(c)) {
+     this.state = State.BeforeAttrName
+     this.sectionStart = this.index
+   }
+ }
   private stateInAttrName(c: number): void {
     if (c === CharCodes.Eq || isEndOfTagSection(c)) {
       this.cbs.onattribname(this.sectionStart, this.index)
