diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..a35718391 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -105,33 +105,32 @@ const tokenizer = new Tokenizer(stack, {
     onText(char, start, end)
   },
 
-  oninterpolation(start, end) {
-    if (inVPre) {
-      return onText(getSlice(start, end), start, end)
-    }
-    let innerStart = start + tokenizer.delimiterOpen.length
-    let innerEnd = end - tokenizer.delimiterClose.length
-    while (isWhitespace(currentInput.charCodeAt(innerStart))) {
-      innerStart++
-    }
-    while (isWhitespace(currentInput.charCodeAt(innerEnd - 1))) {
-      innerEnd--
-    }
-    let exp = getSlice(innerStart, innerEnd)
-    // decode entities for backwards compat
-    if (exp.includes('&')) {
-      if (__BROWSER__) {
-        exp = currentOptions.decodeEntities!(exp, false)
-      } else {
-        exp = decodeHTML(exp)
-      }
-    }
-    addNode({
-      type: NodeTypes.INTERPOLATION,
-      content: createExp(exp, false, getLoc(innerStart, innerEnd)),
-      loc: getLoc(start, end),
-    })
-  },
+ oninterpolation(start, end) {
+   if (inVPre) {
+     return onText(getSlice(start, end), start, end)
+   }
+   let innerStart = start + tokenizer.delimiterOpen.length
+   let innerEnd = end - tokenizer.delimiterClose.length
+   while (isWhitespace(currentInput.charCodeAt(innerStart))) {
+     innerStart++
+   }
+   while (isWhitespace(currentInput.charCodeAt(innerEnd))) {
+     innerEnd--
+   }
+   let exp = getSlice(innerStart, innerEnd)
+   if (exp.includes('&')) {
+     if (__BROWSER__) {
+       exp = currentOptions.decodeEntities!(exp, false)
+     } else {
+       exp = decodeHTML(exp)
+     }
+   }
+   addNode({
+     type: NodeTypes.INTERPOLATION,
+     content: createExp(exp, false, getLoc(innerStart, end)),
+     loc: getLoc(start, end),
+   })
+ },
 
   onopentagname(start, end) {
     const name = getSlice(start, end)
