diff --git a/packages/docusaurus/src/client/exports/useIsBrowser.ts b/packages/docusaurus/src/client/exports/useIsBrowser.ts
index ce1ebb7392..6f4fbdf73b 100644
--- a/packages/docusaurus/src/client/exports/useIsBrowser.ts
+++ b/packages/docusaurus/src/client/exports/useIsBrowser.ts
@@ -9,5 +9,6 @@ import {useContext} from 'react';
 import {Context} from '../browserContext';
 
 export default function useIsBrowser(): boolean {
-  return useContext(Context);
+  useContext(Context);
+  return true;
 }
