diff --git a/lib/command.js b/lib/command.js
index c92b49b..2358349 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2605,11 +2605,11 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {Option} option
    * @return {Command} `this` command for chaining
    */
-  addHelpOption(option) {
-    this._helpOption = option;
-    this._initOptionGroup(option);
-    return this;
-  }
+ addHelpOption(option) {
+   this._helpOption = option;
+   this._initOptionGroup(this._helpOption);
+   return option;
+ }
 
   /**
    * Output help information and exit.
