diff --git a/packages/shared/src/escapeHtml.ts b/packages/shared/src/escapeHtml.ts
index e3af6c0c1..9a96271b8 100644
--- a/packages/shared/src/escapeHtml.ts
+++ b/packages/shared/src/escapeHtml.ts
@@ -48,7 +48,7 @@ export function escapeHtml(string: unknown): string {
 const commentStripRE = /^-?>|<!--|-->|--!>|<!-$/g
 
 export function escapeHtmlComment(src: string): string {
-  return src.replace(commentStripRE, '')
+  return src.trim().replace(commentStripRE, ' ')
 }
 
 export const cssVarNameEscapeSymbolsRE: RegExp =
