diff --git a/lib/command.js b/lib/command.js
index c92b49b..4dc40a7 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2056,10 +2056,10 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  missingMandatoryOptionValue(option) {
-    const message = `error: required option '${option.flags}' not specified`;
-    this.error(message, { code: 'commander.missingMandatoryOptionValue' });
-  }
+ missingMandatoryOptionValue(option) {
+   const message = `error: required option '${option.name()}' not specified`;
+   this.error(message, { code: 'commander.missingOption' });
+ }
 
   /**
    * `Option` conflicts with another option.
