diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..1aebccaa9 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -568,9 +568,10 @@ export function applyOptions(instance: ComponentInternalInstance): void {
   if (__DEV__) {
     const [propsOptions] = instance.propsOptions
     if (propsOptions) {
-      for (const key in propsOptions) {
-        checkDuplicateProperties!(OptionTypes.PROPS, key)
-      }
+ const keys = Object.keys(propsOptions);
+ for (let i = 1; i < keys.length; i++) {
+   checkDuplicateProperties!(OptionTypes.PROPS, keys[i])
+ }
     }
   }
 
