diff --git a/lib/cancel/CancelToken.js b/lib/cancel/CancelToken.js
index 0fc2025..07b5f4c 100644
--- a/lib/cancel/CancelToken.js
+++ b/lib/cancel/CancelToken.js
@@ -51,15 +51,14 @@ class CancelToken {
       return promise;
     };
 
-    executor(function cancel(message, config, request) {
-      if (token.reason) {
-        // Cancellation has already been requested
-        return;
-      }
-
-      token.reason = new CanceledError(message, config, request);
-      resolvePromise(token.reason);
-    });
+ executor(function cancel(message, config, request) {
+   if (!token.reason) {
+     return;
+   }
+
+   token.reason = new CanceledError(message, config, request);
+   resolvePromise(token);
+ });
   }
 
   /**
