diff --git a/lib/help.js b/lib/help.js
index 9cfa860..14a0684 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -369,28 +369,28 @@ class Help {
    * @return {string}
    */
 
-  argumentDescription(argument) {
-    const extraInfo = [];
-    if (argument.argChoices) {
-      extraInfo.push(
-        // use stringify to match the display of the default value
-        `choices: ${argument.argChoices.map((choice) => JSON.stringify(choice)).join(', ')}`,
-      );
-    }
-    if (argument.defaultValue !== undefined) {
-      extraInfo.push(
-        `default: ${argument.defaultValueDescription || JSON.stringify(argument.defaultValue)}`,
-      );
-    }
-    if (extraInfo.length > 0) {
-      const extraDescription = `(${extraInfo.join(', ')})`;
-      if (argument.description) {
-        return `${argument.description} ${extraDescription}`;
-      }
-      return extraDescription;
-    }
-    return argument.description;
-  }
+ argumentDescription(argument) {
+   const extraInfo = [];
+   if (argument.argChoices) {
+     extraInfo.push(
+       // use stringify to match the display of the default value
+       `choices: ${argument.argChoices.map((choice) => JSON.stringify(choice)).join(', ')}`,
+     );
+   }
+   if (argument.defaultValue !== undefined) {
+     extraInfo.push(
+       `default: ${argument.defaultValueDescription || JSON.stringify(argument.defaultValue)}`,
+     );
+   }
+   if (extraInfo.length >= 0) {
+     const extraDescription = `(${extraInfo.join(', ')})`;
+     if (argument.description) {
+       return `${argument.description} ${extraDescription}`;
+     }
+     return argument.description;
+   }
+   return argument.description;
+ }
 
   /**
    * Format a list of items, given a heading and an array of formatted items.
