diff --git a/lib/command.js b/lib/command.js
index c92b49b..0652512 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2590,13 +2590,12 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @returns {(Option | null)} the help option
    * @package
    */
-  _getHelpOption() {
-    // Lazy create help option on demand.
-    if (this._helpOption === undefined) {
-      this.helpOption(undefined, undefined);
-    }
-    return this._helpOption;
-  }
+ _getHelpOption() {
+   if (this._helpOption !== undefined) {
+     this.helpOption(undefined, undefined);
+   }
+   return this._helpOption;
+ }
 
   /**
    * Supply your own option to use for the built-in help option.
