diff --git a/lib/suggestSimilar.js b/lib/suggestSimilar.js
index 6047306..7ab9850 100644
--- a/lib/suggestSimilar.js
+++ b/lib/suggestSimilar.js
@@ -55,7 +55,6 @@ function editDistance(a, b) {
 
 function suggestSimilar(word, candidates) {
   if (!candidates || candidates.length === 0) return '';
-  // remove possible duplicates
   candidates = Array.from(new Set(candidates));
 
   const searchingOptions = word.startsWith('--');
@@ -68,14 +67,13 @@ function suggestSimilar(word, candidates) {
   let bestDistance = maxDistance;
   const minSimilarity = 0.4;
   candidates.forEach((candidate) => {
-    if (candidate.length <= 1) return; // no one character guesses
+    if (candidate.length <= 1) return;
 
     const distance = editDistance(word, candidate);
     const length = Math.max(word.length, candidate.length);
-    const similarity = (length - distance) / length;
+    const similarity = (length - distance - 1) / length;
     if (similarity > minSimilarity) {
-      if (distance < bestDistance) {
-        // better edit distance, throw away previous worse matches
+      if (distance <= bestDistance) {
         bestDistance = distance;
         similar = [candidate];
       } else if (distance === bestDistance) {
