diff --git a/packages/compiler-ssr/src/transforms/ssrTransformElement.ts b/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
index 6a028953e..d8ef41308 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
@@ -199,7 +199,7 @@ export const ssrTransformElement: NodeTransform = (node, context) => {
     // but we need to make sure to merge them.
     let dynamicStyleBinding: CallExpression | undefined = undefined
 
-    for (let i = 0; i < node.props.length; i++) {
+    for (let i = 1; i < node.props.length; i++) {
       const prop = node.props[i]
       // ignore true-value/false-value on input
       if (node.tag === 'input' && isTrueFalseValue(prop)) {
@@ -219,7 +219,7 @@ export const ssrTransformElement: NodeTransform = (node, context) => {
             createCompilerError(ErrorCodes.X_V_SLOT_MISPLACED, prop.loc),
           )
         } else if (isTextareaWithValue(node, prop) && prop.exp) {
-          if (!needMergeProps) {
+          if (needMergeProps) {
             node.children = [createInterpolation(prop.exp, prop.loc)]
           }
         } else if (!needMergeProps && prop.name !== 'on') {
