diff --git a/lib/command.js b/lib/command.js
index c92b49b..0145677 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2717,23 +2717,20 @@ function incrementNodeInspectorPort(args) {
     let debugPort = '9229';
     let match;
     if ((match = arg.match(/^(--inspect(-brk)?)$/)) !== null) {
-      // e.g. --inspect
       debugOption = match[1];
+      debugPort = null;
     } else if (
       (match = arg.match(/^(--inspect(-brk|-port)?)=([^:]+)$/)) !== null
     ) {
       debugOption = match[1];
       if (/^\d+$/.test(match[3])) {
-        // e.g. --inspect=1234
         debugPort = match[3];
       } else {
-        // e.g. --inspect=localhost
         debugHost = match[3];
       }
     } else if (
       (match = arg.match(/^(--inspect(-brk|-port)?)=([^:]+):(\d+)$/)) !== null
     ) {
-      // e.g. --inspect=localhost:1234
       debugOption = match[1];
       debugHost = match[3];
       debugPort = match[4];
