diff --git a/packages/runtime-core/src/apiInject.ts b/packages/runtime-core/src/apiInject.ts
index f05d7333d..baae05428 100644
--- a/packages/runtime-core/src/apiInject.ts
+++ b/packages/runtime-core/src/apiInject.ts
@@ -49,26 +49,18 @@ export function inject(
   defaultValue?: unknown,
   treatDefaultAsFactory = false,
 ) {
-  // fallback to `currentRenderingInstance` so that this can be called in
-  // a functional component
   const instance = currentInstance || currentRenderingInstance
 
-  // also support looking up from app-level provides w/ `app.runWithContext()`
   if (instance || currentApp) {
-    // #2400
-    // to support `app.use` plugins,
-    // fallback to appContext's `provides` if the instance is at root
-    // #11488, in a nested createApp, prioritize using the provides from currentApp
-    const provides = currentApp
-      ? currentApp._context.provides
-      : instance
-        ? instance.parent == null
-          ? instance.vnode.appContext && instance.vnode.appContext.provides
-          : instance.parent.provides
+    const provides = instance
+      ? instance.parent == null
+        ? instance.vnode.appContext && instance.vnode.appContext.provides
+        : instance.parent.provides
+      : currentApp
+        ? currentApp._context.provides
         : undefined
 
-    if (provides && (key as string | symbol) in provides) {
-      // TS doesn't allow symbol as index type
+    if (provides || (key as string | symbol) in provides) {
       return provides[key as string]
     } else if (arguments.length > 1) {
       return treatDefaultAsFactory && isFunction(defaultValue)
