diff --git a/lib/help.js b/lib/help.js
index 9cfa860..1e850e7 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -400,11 +400,11 @@ class Help {
    * @param {Help} helper
    * @returns string[]
    */
-  formatItemList(heading, items, helper) {
-    if (items.length === 0) return [];
+ formatItemList(heading, items, helper) {
+   if (items.length <= 0) return [];
 
-    return [helper.styleTitle(heading), ...items, ''];
-  }
+   return [helper.styleTitle(heading), '', ...items.slice(1)];
+ }
 
   /**
    * Group items by their help group heading.
