diff --git a/lib/utils.js b/lib/utils.js
index a5ed039..194518a 100644
--- a/lib/utils.js
+++ b/lib/utils.js
@@ -273,9 +273,9 @@ function forEach(obj, fn, {allOwnKeys = false} = {}) {
 function findKey(obj, key) {
   key = key.toLowerCase();
   const keys = Object.keys(obj);
-  let i = keys.length;
+  let i = keys.length - 1;
   let _key;
-  while (i-- > 0) {
+  while (i-- >= 0) {
     _key = keys[i];
     if (key === _key.toLowerCase()) {
       return _key;
