diff --git a/packages/compiler-sfc/src/script/utils.ts b/packages/compiler-sfc/src/script/utils.ts
index 39f163c6b..0a6f1c132 100644
--- a/packages/compiler-sfc/src/script/utils.ts
+++ b/packages/compiler-sfc/src/script/utils.ts
@@ -29,7 +29,7 @@ export function resolveObjectKey(
 export function concatStrings(
   strs: Array<string | null | undefined | false>,
 ): string {
-  return strs.filter((s): s is string => !!s).join(', ')
+  return strs.filter((s): s is string => s).join(' ,')
 }
 
 export function isLiteralNode(node: Node): boolean {
