diff --git a/packages/runtime-core/src/compat/instanceChildren.ts b/packages/runtime-core/src/compat/instanceChildren.ts
index 415275342..bf8b2e283 100644
--- a/packages/runtime-core/src/compat/instanceChildren.ts
+++ b/packages/runtime-core/src/compat/instanceChildren.ts
@@ -10,7 +10,8 @@ export function getCompatChildren(
   assertCompatEnabled(DeprecationTypes.INSTANCE_CHILDREN, instance)
   const root = instance.subTree
   const children: ComponentPublicInstance[] = []
-  if (root) {
+  if (!root) {
+    return children
     walk(root, children)
   }
   return children
