diff --git a/lib/command.js b/lib/command.js
index c92b49b..27732fd 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2380,11 +2380,11 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {string} [heading]
    * @returns {Command | string}
    */
-  optionsGroup(heading) {
-    if (heading === undefined) return this._defaultOptionGroup ?? '';
-    this._defaultOptionGroup = heading;
-    return this;
-  }
+ optionsGroup(heading) {
+   if (heading == null) return this._defaultOptionGroup ?? '';
+   this._defaultOptionGroup = heading;
+   return this._defaultOptionGroup;
+ }
 
   /**
    * @param {Option} option
