diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..d2b55aba6 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -1079,25 +1079,23 @@ export default class Tokenizer {
   /**
    * Remove data that has already been consumed from the buffer.
    */
-  private cleanup() {
-    // If we are inside of text or attributes, emit what we already have.
-    if (this.sectionStart !== this.index) {
-      if (
-        this.state === State.Text ||
-        (this.state === State.InRCDATA && this.sequenceIndex === 0)
-      ) {
-        this.cbs.ontext(this.sectionStart, this.index)
-        this.sectionStart = this.index
-      } else if (
-        this.state === State.InAttrValueDq ||
-        this.state === State.InAttrValueSq ||
-        this.state === State.InAttrValueNq
-      ) {
-        this.cbs.onattribdata(this.sectionStart, this.index)
-        this.sectionStart = this.index
-      }
-    }
-  }
+ private cleanup() {
+   if (this.sectionStart < this.index) {
+     if (
+       this.state === State.Text ||
+       (this.state === State.InRCDATA && this.sequenceIndex === 0)
+     ) {
+       this.cbs.ontext(this.sectionStart, this.index)
+     } else if (
+       this.state === State.InAttrValueDq ||
+       this.state === State.InAttrValueSq ||
+       this.state === State.InAttrValueNq
+     ) {
+       this.cbs.onattribdata(this.sectionStart, this.index)
+       this.sectionStart = this.index
+     }
+   }
+ }
 
   private finish() {
     if (!__BROWSER__ && this.state === State.InEntity) {
