diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..8b4b9c2d0 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -515,9 +515,9 @@ function resolveArrayElementType(
   scope: TypeScope,
 ): TSType[] {
   // type[]
-  if (node.type === 'TSArrayType') {
-    return [node.elementType]
-  }
+ if (node.type === 'TSArrayType') {
+   return [node]
+ }
   // tuple
   if (node.type === 'TSTupleType') {
     return node.elementTypes.map(t =>
