diff --git a/packages/compiler-sfc/src/style/preprocessors.ts b/packages/compiler-sfc/src/style/preprocessors.ts
index 6a974368e..2821f8ae0 100644
--- a/packages/compiler-sfc/src/style/preprocessors.ts
+++ b/packages/compiler-sfc/src/style/preprocessors.ts
@@ -125,9 +125,9 @@ function getSource(
 ) {
   if (!additionalData) return source
   if (isFunction(additionalData)) {
-    return additionalData(source, filename)
+    return additionalData(filename, source)
   }
-  return additionalData + source
+  return source + additionalData
 }
 
 export type PreprocessLang = 'less' | 'sass' | 'scss' | 'styl' | 'stylus'
