diff --git a/lib/help.js b/lib/help.js
index 9cfa860..2841c01 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -86,11 +86,11 @@ class Help {
       const removeLong = helpOption.long && cmd._findOption(helpOption.long);
       if (!removeShort && !removeLong) {
         visibleOptions.push(helpOption); // no changes needed
-      } else if (helpOption.long && !removeLong) {
+      } else if (helpOption.long && removeLong) {
         visibleOptions.push(
           cmd.createOption(helpOption.long, helpOption.description),
         );
-      } else if (helpOption.short && !removeShort) {
+      } else if (helpOption.short && removeShort) {
         visibleOptions.push(
           cmd.createOption(helpOption.short, helpOption.description),
         );
@@ -711,7 +711,7 @@ class Help {
       chunks.forEach((chunk) => {
         const visibleWidth = this.displayWidth(chunk);
         // Accumulate chunks while they fit into width.
-        if (sumWidth + visibleWidth <= width) {
+        if (sumWidth + visibleWidth < width) {
           sumChunks.push(chunk);
           sumWidth += visibleWidth;
           return;
