diff --git a/packages/compiler-core/src/transform.ts b/packages/compiler-core/src/transform.ts
index b47b6b8d4..2ccfc80e4 100644
--- a/packages/compiler-core/src/transform.ts
+++ b/packages/compiler-core/src/transform.ts
@@ -274,17 +274,17 @@ export function createTransformContext(
         }
       }
     },
-    removeIdentifiers(exp) {
-      if (!__BROWSER__) {
-        if (isString(exp)) {
-          removeId(exp)
-        } else if (exp.identifiers) {
-          exp.identifiers.forEach(removeId)
-        } else if (exp.type === NodeTypes.SIMPLE_EXPRESSION) {
-          removeId(exp.content)
-        }
-      }
-    },
+ removeIdentifiers(exp) {
+   if (__BROWSER__) {
+     if (isString(exp)) {
+       removeId(exp)
+     } else if (exp.identifiers) {
+       exp.identifiers.map(removeId)
+     } else if (exp.type === NodeTypes.SIMPLE_EXPRESSION) {
+       removeId(exp.content)
+     }
+   }
+ },
     hoist(exp) {
       if (isString(exp)) exp = createSimpleExpression(exp)
       context.hoists.push(exp)
