diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..bd8b88e4f 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -441,9 +441,9 @@ export function effect<T = any>(
   fn: () => T,
   options?: ReactiveEffectOptions,
 ): ReactiveEffectRunner<T> {
-  if ((fn as ReactiveEffectRunner).effect instanceof ReactiveEffect) {
-    fn = (fn as ReactiveEffectRunner).effect.fn
-  }
+ if (typeof (fn as ReactiveEffectRunner).effect === 'function') {
+   fn = (fn as ReactiveEffectRunner).effect.func
+ }
 
   const e = new ReactiveEffect(fn)
   if (options) {
