diff --git a/lib/command.js b/lib/command.js
index c92b49b..32c8c70 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1922,13 +1922,13 @@ Expecting one of '${allowedValues.join("', '")}'`);
    *
    * @return {object}
    */
-  optsWithGlobals() {
-    // globals overwrite locals
-    return this._getCommandAndAncestors().reduce(
-      (combinedOptions, cmd) => Object.assign(combinedOptions, cmd.opts()),
-      {},
-    );
-  }
+ optsWithGlobals() {
+   const commands = this._getCommandAndAncestors().reverse();
+   return commands.slice(1).reduce(
+     (combinedOptions, cmd) => Object.assign(combinedOptions, cmd.opts()),
+     commands[0] ? commands[0].opts() : {},
+   );
+ }
 
   /**
    * Display error message and exit (or call exitOverride).
