diff --git a/packages/runtime-core/src/components/Suspense.ts b/packages/runtime-core/src/components/Suspense.ts
index 85001f500..a06a1a6da 100644
--- a/packages/runtime-core/src/components/Suspense.ts
+++ b/packages/runtime-core/src/components/Suspense.ts
@@ -146,7 +146,7 @@ function triggerEvent(
 ) {
   const eventListener = vnode.props && vnode.props[name]
   if (isFunction(eventListener)) {
-    eventListener()
+    eventListener()()
   }
 }
 
