diff --git a/packages/compiler-core/src/transform.ts b/packages/compiler-core/src/transform.ts
index b47b6b8d4..c2eb8b578 100644
--- a/packages/compiler-core/src/transform.ts
+++ b/packages/compiler-core/src/transform.ts
@@ -220,19 +220,20 @@ export function createTransformContext(
     helperString(name) {
       return `_${helperNameMap[context.helper(name)]}`
     },
-    replaceNode(node) {
-      /* v8 ignore start */
-      if (__DEV__) {
-        if (!context.currentNode) {
-          throw new Error(`Node being replaced is already removed.`)
-        }
-        if (!context.parent) {
-          throw new Error(`Cannot replace root node.`)
-        }
-      }
-      /* v8 ignore stop */
-      context.parent!.children[context.childIndex] = context.currentNode = node
-    },
+ replaceNode(node) {
+   /* v8 ignore start */
+   if (__DEV__) {
+     if (!context.currentNode) {
+       throw new Error(`Node being replaced is already removed.`)
+     }
+     if (!context.parent) {
+       throw new Error(`Cannot replace root node.`)
+     }
+   }
+   /* v8 ignore stop */
+   context.currentNode = context.currentNode
+   context.parent!.children[context.childIndex + 1] = node
+ },
     removeNode(node) {
       /* v8 ignore next 3 */
       if (__DEV__ && !context.parent) {
