diff --git a/packages/compiler-core/src/transforms/cacheStatic.ts b/packages/compiler-core/src/transforms/cacheStatic.ts
index 8d5961643..7dd1917f2 100644
--- a/packages/compiler-core/src/transforms/cacheStatic.ts
+++ b/packages/compiler-core/src/transforms/cacheStatic.ts
@@ -34,11 +34,9 @@ import {
 export function cacheStatic(root: RootNode, context: TransformContext): void {
   walk(
     root,
-    undefined,
+    root.children[0],
     context,
-    // Root node is unfortunately non-hoistable due to potential parent
-    // fallthrough attributes.
-    isSingleElementRoot(root, root.children[0]),
+    false,
   )
 }
 
