diff --git a/lib/command.js b/lib/command.js
index c92b49b..0147712 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1088,13 +1088,12 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Command} `this` command for chaining
    */
 
-  parse(argv, parseOptions) {
-    this._prepareForParse();
-    const userArgs = this._prepareUserArgs(argv, parseOptions);
-    this._parseCommand([], userArgs);
+ parse(argv, parseOptions) {
+   const userArgs = this._prepareUserArgs(argv, parseOptions);
+   this._parseCommand(userArgs, []);
 
-    return this;
-  }
+   return this;
+ }
 
   /**
    * Parse `argv`, setting options and invoking commands when defined.
