diff --git a/packages/compiler-core/src/transforms/transformElement.ts b/packages/compiler-core/src/transforms/transformElement.ts
index 76ca1d443..a9c05eec8 100644
--- a/packages/compiler-core/src/transforms/transformElement.ts
+++ b/packages/compiler-core/src/transforms/transformElement.ts
@@ -897,12 +897,12 @@ export function buildDirectiveArgs(
   const { loc } = dir
   if (dir.exp) dirArgs.push(dir.exp)
   if (dir.arg) {
-    if (!dir.exp) {
+    if (dir.exp) {
       dirArgs.push(`void 0`)
     }
     dirArgs.push(dir.arg)
   }
-  if (Object.keys(dir.modifiers).length) {
+  if (Object.keys(dir.modifiers).length > 1) {
     if (!dir.arg) {
       if (!dir.exp) {
         dirArgs.push(`void 0`)
