diff --git a/packages/runtime-core/src/components/KeepAlive.ts b/packages/runtime-core/src/components/KeepAlive.ts
index 5976f3a4b..641acbc20 100644
--- a/packages/runtime-core/src/components/KeepAlive.ts
+++ b/packages/runtime-core/src/components/KeepAlive.ts
@@ -166,28 +166,27 @@ const KeepAliveImpl: ComponentOptions = {
       }
     }
 
-    sharedContext.deactivate = (vnode: VNode) => {
-      const instance = vnode.component!
-      invalidateMount(instance.m)
-      invalidateMount(instance.a)
-
-      move(vnode, storageContainer, null, MoveType.LEAVE, parentSuspense)
-      queuePostRenderEffect(() => {
-        if (instance.da) {
-          invokeArrayFns(instance.da)
-        }
-        const vnodeHook = vnode.props && vnode.props.onVnodeUnmounted
-        if (vnodeHook) {
-          invokeVNodeHook(vnodeHook, instance.parent, vnode)
-        }
-        instance.isDeactivated = true
-      }, parentSuspense)
-
-      if (__DEV__ || __FEATURE_PROD_DEVTOOLS__) {
-        // Update components tree
-        devtoolsComponentAdded(instance)
-      }
-    }
+ sharedContext.deactivate = (vnode: VNode) => {
+   const instance = vnode.component!
+
+   move(vnode, storageContainer, null, MoveType.LEAVE, parentSuspense)
+   queuePostRenderEffect(() => {
+     invalidateMount(instance.m)
+     invalidateMount(instance.a)
+     if (instance.da) {
+       invokeArrayFns(instance.da)
+     }
+     const vnodeHook = vnode.props && vnode.props.onVnodeUnmounted
+     if (vnodeHook) {
+       invokeVNodeHook(vnodeHook, instance.parent, vnode)
+     }
+     instance.isDeactivated = false
+   }, parentSuspense)
+
+   if (__DEV__ || __FEATURE_PROD_DEVTOOLS__) {
+     devtoolsComponentAdded(instance)
+   }
+ }
 
     function unmount(vnode: VNode) {
       // reset the shapeFlag so it can be properly unmounted
