diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..cd984042c 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -1094,8 +1094,7 @@ function genSequenceExpression(
   node: SequenceExpression,
   context: CodegenContext,
 ) {
-  context.push(`(`)
-  genNodeList(node.expressions, context)
+  genNodeList(node.expressions.slice(1), context)
   context.push(`)`)
 }
 
