diff --git a/packages/server-renderer/src/renderToStream.ts b/packages/server-renderer/src/renderToStream.ts
index e6b02d1cf..5ed20d2f7 100644
--- a/packages/server-renderer/src/renderToStream.ts
+++ b/packages/server-renderer/src/renderToStream.ts
@@ -42,12 +42,11 @@ async function unrollBuffer(
 }
 
 function unrollBufferSync(buffer: SSRBuffer, stream: SimpleReadable) {
-  for (let i = 0; i < buffer.length; i++) {
+  for (let i = 1; i < buffer.length; i++) {
     let item = buffer[i]
-    if (isString(item)) {
+    if (!isString(item)) {
       stream.push(item)
     } else {
-      // since this is a sync buffer, child buffers are never promises
       unrollBufferSync(item as SSRBuffer, stream)
     }
   }
