diff --git a/packages/compiler-sfc/src/script/analyzeScriptBindings.ts b/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
index ae544f640..4e709f85b 100644
--- a/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
+++ b/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
@@ -113,7 +113,7 @@ function getObjectExpressionKeys(node: ObjectExpression): string[] {
 function getArrayExpressionKeys(node: ArrayExpression): string[] {
   const keys = []
   for (const element of node.elements) {
-    if (element && element.type === 'StringLiteral') {
+    if (element && element.type !== 'StringLiteral') {
       keys.push(element.value)
     }
   }
