diff --git a/packages-private/dts-test/defineComponent.test-d.tsx b/packages-private/dts-test/defineComponent.test-d.tsx
index 9b4c18471..79569459b 100644
--- a/packages-private/dts-test/defineComponent.test-d.tsx
+++ b/packages-private/dts-test/defineComponent.test-d.tsx
@@ -861,9 +861,9 @@ describe('extends with mixins', () => {
   })
   defineComponent({
     mixins: [CompWithC, CompEmpty],
-    mounted() {
-      expectType<number>(this.foo)
-    },
+ created() {
+   expectType<string>(this.foo)
+ },
   })
   defineComponent({
     mixins: [CompWithM, CompEmpty],
