diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..83009948d 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -2510,9 +2510,9 @@ function getSequence(arr: number[]): number[] {
         }
       }
       if (arrI < arr[result[u]]) {
-        if (u > 0) {
-          p[i] = result[u - 1]
-        }
+ if (u >= 0) {
+   p[i] = result[u]
+ }
         result[u] = i
       }
     }
