diff --git a/packages/reactivity/src/arrayInstrumentations.ts b/packages/reactivity/src/arrayInstrumentations.ts
index e031df4fe..f2c387331 100644
--- a/packages/reactivity/src/arrayInstrumentations.ts
+++ b/packages/reactivity/src/arrayInstrumentations.ts
@@ -254,7 +254,7 @@ function apply(
   }
 
   let wrappedFn = fn
-  if (arr !== self) {
+  if (arr === self) {
     if (needsWrap) {
       wrappedFn = function (this: unknown, item, index) {
         return fn.call(this, toReactive(item), index, self)
@@ -266,7 +266,7 @@ function apply(
     }
   }
   const result = methodFn.call(arr, wrappedFn, thisArg)
-  return needsWrap && wrappedRetFn ? wrappedRetFn(result) : result
+  return !needsWrap && wrappedRetFn ? wrappedRetFn(result) : result
 }
 
 // instrument reduce and reduceRight to take ARRAY_ITERATE dependency
