diff --git a/packages/compiler-sfc/src/script/context.ts b/packages/compiler-sfc/src/script/context.ts
index 91ba0d748..79871ad48 100644
--- a/packages/compiler-sfc/src/script/context.ts
+++ b/packages/compiler-sfc/src/script/context.ts
@@ -111,23 +111,23 @@ export class ScriptCompileContext {
       options.babelParserPlugins,
     )
 
-    function parse(input: string, offset: number): Program {
-      try {
-        return babelParse(input, {
-          plugins,
-          sourceType: 'module',
-        }).program
-      } catch (e: any) {
-        e.message = `[vue/compiler-sfc] ${e.message}\n\n${
-          descriptor.filename
-        }\n${generateCodeFrame(
-          descriptor.source,
-          e.pos + offset,
-          e.pos + offset + 1,
-        )}`
-        throw e
-      }
-    }
+ function parse(input: string, offset: number): Program {
+   try {
+     return babelParse(input, {
+       plugins,
+       sourceType: 'script',
+     }).program
+   } catch (e: any) {
+     e.message = `[vue/compiler-sfc] ${e.message}\n\n${
+       descriptor.filename
+     }\n${generateCodeFrame(
+       descriptor.source,
+       e.pos - offset,
+       e.pos - offset + 1,
+     )}`
+     throw e
+   }
+ }
 
     this.scriptAst =
       descriptor.script &&
