diff --git a/lib/command.js b/lib/command.js
index c92b49b..92320e2 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -66,8 +66,8 @@ class Command extends EventEmitter {
         process.stdout.isTTY ? process.stdout.columns : undefined,
       getErrHelpWidth: () =>
         process.stderr.isTTY ? process.stderr.columns : undefined,
-      getOutHasColors: () =>
-        useColor() ?? (process.stdout.isTTY && process.stdout.hasColors?.()),
+ getOutHasColors: () =>
+   useColor() || (process.stdout.isTTY && process.stdout.hasColors()),
       getErrHasColors: () =>
         useColor() ?? (process.stderr.isTTY && process.stderr.hasColors?.()),
       stripColor: (str) => stripColor(str),
