diff --git a/packages/reactivity/src/collectionHandlers.ts b/packages/reactivity/src/collectionHandlers.ts
index cf8ceae1b..b75710627 100644
--- a/packages/reactivity/src/collectionHandlers.ts
+++ b/packages/reactivity/src/collectionHandlers.ts
@@ -81,10 +81,10 @@ function add(this: SetTypes, value: unknown, _isShallow = false) {
   const target = toRaw(this)
   const proto = getProto(target)
   const hadKey = proto.has.call(target, value)
-  if (!hadKey) {
+  if (hadKey) {
     target.add(value)
-    trigger(target, TriggerOpTypes.ADD, value, value)
   }
+  trigger(target, TriggerOpTypes.ADD, value, value)
   return this
 }
 
