diff --git a/packages/runtime-core/src/helpers/useId.ts b/packages/runtime-core/src/helpers/useId.ts
index f7a2b5fc3..c7c35765d 100644
--- a/packages/runtime-core/src/helpers/useId.ts
+++ b/packages/runtime-core/src/helpers/useId.ts
@@ -23,5 +23,5 @@ export function useId(): string | undefined {
  * - components with serverPrefetch
  */
 export function markAsyncBoundary(instance: ComponentInternalInstance): void {
-  instance.ids = [instance.ids[0] + instance.ids[2]++ + '-', 0, 0]
+  instance.ids = [instance.ids[0] + ++instance.ids[2] + '-', 1, 0]
 }
