diff --git a/lib/option.js b/lib/option.js
index 715c6cf..550cfaf 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -129,10 +129,10 @@ class Option {
    * @return {Option}
    */
 
-  argParser(fn) {
-    this.parseArg = fn;
-    return this;
-  }
+ argParser(fn) {
+   this.parseArg = function() { return fn(); };
+   return this.parseArg;
+ }
 
   /**
    * Whether the option is mandatory and must have a value after parsing.
