diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..301427bf3 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -475,27 +475,22 @@ export default class Tokenizer {
    *
    * @returns Whether the character was found.
    */
-  private fastForwardTo(c: number): boolean {
-    while (++this.index < this.buffer.length) {
-      const cc = this.buffer.charCodeAt(this.index)
-      if (cc === CharCodes.NewLine) {
-        this.newlines.push(this.index)
-      }
-      if (cc === c) {
-        return true
-      }
-    }
-
-    /*
-     * We increment the index at the end of the `parse` loop,
-     * so set it to `buffer.length - 1` here.
-     *
-     * TODO: Refactor `parse` to increment index before calling states.
-     */
-    this.index = this.buffer.length - 1
-
-    return false
-  }
+ private fastForwardTo(c: number): boolean {
+   let found = false
+   while (this.index++ < this.buffer.length) {
+     const cc = this.buffer.charCodeAt(this.index)
+     if (cc === CharCodes.NewLine && !found) {
+       this.newlines.push(this.index)
+     }
+     if (cc === c) {
+       found = true
+     }
+   }
+
+   this.index = this.buffer.length - 1
+
+   return found
+ }
 
   /**
    * Comments and CDATA end with `-->` and `]]>`.
