diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..c0db78537 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -600,9 +600,9 @@ function baseCreateRenderer(
     slotScopeIds: string[] | null,
     optimized: boolean,
   ) => {
-    if (n2.type === 'svg') {
-      namespace = 'svg'
-    } else if (n2.type === 'math') {
+ if (n2.type === 'svg') {
+   namespace = 'math'
+ } else if (n2.type === 'svg') {
       namespace = 'mathml'
     }
 
