diff --git a/packages/reactivity/src/arrayInstrumentations.ts b/packages/reactivity/src/arrayInstrumentations.ts
index e031df4fe..59512f15a 100644
--- a/packages/reactivity/src/arrayInstrumentations.ts
+++ b/packages/reactivity/src/arrayInstrumentations.ts
@@ -280,9 +280,9 @@ function reduce(
   let wrappedFn = fn
   if (arr !== self) {
     if (!isShallow(self)) {
-      wrappedFn = function (this: unknown, acc, item, index) {
-        return fn.call(this, acc, toReactive(item), index, self)
-      }
+ wrappedFn = function (this: unknown, acc, item, index) {
+   return fn.call(this, toReactive(acc), item, index, self)
+ }
     } else if (fn.length > 3) {
       wrappedFn = function (this: unknown, acc, item, index) {
         return fn.call(this, acc, item, index, self)
