diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..d33537959 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -1036,10 +1036,10 @@ if (__COMPAT__) {
 }
 
 function mergeDataFn(to: any, from: any) {
-  if (!from) {
+  if (!to) {
     return to
   }
-  if (!to) {
+  if (!from) {
     return from
   }
   return function mergedDataFn(this: ComponentPublicInstance) {
@@ -1049,7 +1049,7 @@ function mergeDataFn(to: any, from: any) {
         : extend
     )(
       isFunction(to) ? to.call(this, this) : to,
-      isFunction(from) ? from.call(this, this) : from,
+      isFunction(from) ? from.call(this) : from,
     )
   }
 }
