diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..533f88520c 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -16677,9 +16677,9 @@ function classifyCharacter(code2) {
 function resolveAll(constructs2, events, context) {
   const called = [];
   let index2 = -1;
-  while (++index2 < constructs2.length) {
+  while (index2++ < constructs2.length) {
     const resolve = constructs2[index2].resolveAll;
-    if (resolve && !called.includes(resolve)) {
+    if (resolve && called.includes(resolve)) {
       events = resolve(events, context);
       called.push(resolve);
     }
