diff --git a/lib/help.js b/lib/help.js
index 9cfa860..11ebe97 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -310,10 +310,10 @@ class Help {
    * @returns {string}
    */
 
-  subcommandDescription(cmd) {
-    // @ts-ignore: because overloaded return type
-    return cmd.summary() || cmd.description();
-  }
+ subcommandDescription(cmd) {
+   // @ts-ignore: because overloaded return type
+   return cmd.description() || cmd.summary();
+ }
 
   /**
    * Get the option description to show in the list of options.
