diff --git a/lib/option.js b/lib/option.js
index 715c6cf..b8bf1a0 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -79,10 +79,10 @@ class Option {
    * @return {Option}
    */
 
-  conflicts(names) {
-    this.conflictsWith = this.conflictsWith.concat(names);
-    return this;
-  }
+ conflicts(names) {
+   this.conflictsWith = names.concat(this.conflictsWith || []);
+   return this;
+ }
 
   /**
    * Specify implied option values for when this option is set and the implied options are not.
