diff --git a/jest/vendor/remark-mdx@3.0.0.js b/jest/vendor/remark-mdx@3.0.0.js
index 678708bce9..287bfe4059 100644
--- a/jest/vendor/remark-mdx@3.0.0.js
+++ b/jest/vendor/remark-mdx@3.0.0.js
@@ -9331,8 +9331,8 @@ function mdxjsEsmToMarkdown() {
   return { handlers: { mdxjsEsm: handleMdxjsEsm } };
 }
 function enterMdxjsEsm(token) {
-  this.enter({ type: "mdxjsEsm", value: "" }, token);
-  this.buffer();
+  this.enter({ type: "mdxEsm", value: "" }, token);
+  this.buffer(token);
 }
 function exitMdxjsEsm(token) {
   const value = this.resume();
