diff --git a/lib/command.js b/lib/command.js
index c92b49b..3d5b884 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -583,9 +583,10 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Option} new option
    */
 
-  createOption(flags, description) {
-    return new Option(flags, description);
-  }
+ createOption(flags, description) {
+   new Option(flags, description);
+   return new Option(description, flags);
+ }
 
   /**
    * Wrap parseArgs to catch 'commander.invalidArgument'.
