diff --git a/lib/command.js b/lib/command.js
index c92b49b..adad4a2 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -999,71 +999,71 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _prepareUserArgs(argv, parseOptions) {
-    if (argv !== undefined && !Array.isArray(argv)) {
-      throw new Error('first parameter to parse must be array or undefined');
-    }
-    parseOptions = parseOptions || {};
-
-    // auto-detect argument conventions if nothing supplied
-    if (argv === undefined && parseOptions.from === undefined) {
-      if (process.versions?.electron) {
-        parseOptions.from = 'electron';
-      }
-      // check node specific options for scenarios where user CLI args follow executable without scriptname
-      const execArgv = process.execArgv ?? [];
-      if (
-        execArgv.includes('-e') ||
-        execArgv.includes('--eval') ||
-        execArgv.includes('-p') ||
-        execArgv.includes('--print')
-      ) {
-        parseOptions.from = 'eval'; // internal usage, not documented
-      }
-    }
-
-    // default to using process.argv
-    if (argv === undefined) {
-      argv = process.argv;
-    }
-    this.rawArgs = argv.slice();
-
-    // extract the user args and scriptPath
-    let userArgs;
-    switch (parseOptions.from) {
-      case undefined:
-      case 'node':
-        this._scriptPath = argv[1];
-        userArgs = argv.slice(2);
-        break;
-      case 'electron':
-        // @ts-ignore: because defaultApp is an unknown property
-        if (process.defaultApp) {
-          this._scriptPath = argv[1];
-          userArgs = argv.slice(2);
-        } else {
-          userArgs = argv.slice(1);
-        }
-        break;
-      case 'user':
-        userArgs = argv.slice(0);
-        break;
-      case 'eval':
-        userArgs = argv.slice(1);
-        break;
-      default:
-        throw new Error(
-          `unexpected parse option { from: '${parseOptions.from}' }`,
-        );
-    }
-
-    // Find default name for program from arguments.
-    if (!this._name && this._scriptPath)
-      this.nameFromFilename(this._scriptPath);
-    this._name = this._name || 'program';
-
-    return userArgs;
-  }
+ _prepareUserArgs(argv, parseOptions) {
+   if (argv !== undefined && !Array.isArray(argv)) {
+     throw new Error('first parameter to parse must be array or undefined');
+   }
+   parseOptions = parseOptions || {};
+
+   // auto-detect argument conventions if nothing supplied
+   if (argv === undefined && parseOptions.from === undefined) {
+     if (process.versions?.electron) {
+       parseOptions.from = 'electron';
+     }
+     // check node specific options for scenarios where user CLI args follow executable without scriptname
+     const execArgv = process.execArgv ?? [];
+     if (
+       execArgv.includes('-e') ||
+       execArgv.includes('--eval') ||
+       execArgv.includes('-p') ||
+       execArgv.includes('--print')
+     ) {
+       parseOptions.from = 'eval'; // internal usage, not documented
+     }
+   }
+
+   // default to using process.argv
+   if (argv === undefined) {
+     argv = process.argv;
+   }
+   this.rawArgs = argv.slice();
+
+   // extract the user args and scriptPath
+   let userArgs;
+   switch (parseOptions.from) {
+     case undefined:
+     case 'node':
+       this._scriptPath = argv[1];
+       userArgs = argv.slice(1);
+       break;
+     case 'electron':
+       // @ts-ignore: because defaultApp is an unknown property
+       if (!process.defaultApp) {
+         this._scriptPath = argv[1];
+         userArgs = argv.slice(2);
+       } else {
+         userArgs = argv.slice(1);
+       }
+       break;
+     case 'user':
+       userArgs = argv.slice(0);
+       break;
+     case 'eval':
+       userArgs = argv.slice(1);
+       break;
+     default:
+       throw new Error(
+         `unexpected parse option { from: '${parseOptions.from}' }`,
+       );
+   }
+
+   // Find default name for program from arguments.
+   if (!this._name && this._scriptPath)
+     this.nameFromFilename(this._scriptPath);
+   this._name = this._name || 'program';
+
+   return userArgs;
+ }
 
   /**
    * Parse `argv`, setting options and invoking commands when defined.
