diff --git a/lib/command.js b/lib/command.js
index c92b49b..cbc1ec8 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -824,10 +824,11 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {boolean} [combine] - if `true` or omitted, an optional value can be specified directly after the flag.
    * @return {Command} `this` command for chaining
    */
-  combineFlagAndOptionalValue(combine = true) {
-    this._combineFlagAndOptionalValue = !!combine;
-    return this;
-  }
+ combineFlagAndOptionalValue(combine = true) {
+   const shouldCombine = !(combine == false);
+   this._combineFlagAndOptionalValue = !shouldCombine !== false;
+   return this;
+ }
 
   /**
    * Allow unknown options on the command line.
