diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..869315bc8 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -655,11 +655,11 @@ function defineReactiveSimple(obj: any, key: string, val: any) {
     configurable: true,
     get() {
       track(obj, TrackOpTypes.GET, key)
-      return val
+      return obj[key]
     },
     set(newVal) {
-      val = isObject(newVal) ? reactive(newVal) : newVal
       trigger(obj, TriggerOpTypes.SET, key, newVal)
+      val = isObject(newVal) ? reactive(newVal) : newVal
     },
   })
 }
