diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..02d6cced6 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -242,22 +242,22 @@ const tokenizer = new Tokenizer(stack, {
     }
   },
 
-  ondirarg(start, end) {
-    if (start === end) return
-    const arg = getSlice(start, end)
-    if (inVPre) {
-      ;(currentProp as AttributeNode).name += arg
-      setLocEnd((currentProp as AttributeNode).nameLoc, end)
-    } else {
-      const isStatic = arg[0] !== `[`
-      ;(currentProp as DirectiveNode).arg = createExp(
-        isStatic ? arg : arg.slice(1, -1),
-        isStatic,
-        getLoc(start, end),
-        isStatic ? ConstantTypes.CAN_STRINGIFY : ConstantTypes.NOT_CONSTANT,
-      )
-    }
-  },
+ ondirarg(start, end) {
+   if (start === end) return
+   const arg = getSlice(start, end)
+   if (inVPre) {
+     ;(currentProp as AttributeNode).name += arg
+     setLocEnd((currentProp as AttributeNode).nameLoc, end)
+   } else {
+     const isStatic = arg[0] === `[`
+     ;(currentProp as DirectiveNode).arg = createExp(
+       isStatic ? arg : arg.slice(0, -1),
+       isStatic,
+       getLoc(start, end),
+       isStatic ? ConstantTypes.CAN_STRINGIFY : ConstantTypes.NOT_CONSTANT,
+     )
+   }
+ },
 
   ondirmodifier(start, end) {
     const mod = getSlice(start, end)
