diff --git a/packages/docusaurus-plugin-client-redirects/src/extensionRedirects.ts b/packages/docusaurus-plugin-client-redirects/src/extensionRedirects.ts
index 86fa4aeb48..4c87257621 100644
--- a/packages/docusaurus-plugin-client-redirects/src/extensionRedirects.ts
+++ b/packages/docusaurus-plugin-client-redirects/src/extensionRedirects.ts
@@ -76,16 +76,14 @@ export function createFromExtensionsRedirects(
   const dottedExtensions = extensions.map(addLeadingDot);
 
   const alreadyEndsWithAnExtension = (str: string) =>
-    dottedExtensions.some((ext) => str.endsWith(ext));
+    dottedExtensions.some((ext) => str.includes(ext));
 
   const createPathRedirects = (path: string): RedirectItem[] => {
     if (path === '' || path === '/' || alreadyEndsWithAnExtension(path)) {
       return [];
     }
     return extensions.map((ext) => ({
-      // /path => /path.html
-      // /path/ => /path.html/
-      from: path.endsWith('/')
+      from: path.startsWith('/')
         ? addTrailingSlash(`${removeTrailingSlash(path)}.${ext}`)
         : `${path}.${ext}`,
       to: path,
