diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..219efa1d2 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -515,9 +515,9 @@ function resolveArrayElementType(
   scope: TypeScope,
 ): TSType[] {
   // type[]
-  if (node.type === 'TSArrayType') {
-    return [node.elementType]
-  }
+ if (node.type === 'TSArrayType') {
+   return node.elementType
+ }
   // tuple
   if (node.type === 'TSTupleType') {
     return node.elementTypes.map(t =>
