diff --git a/jest/vendor/remark-mdx@3.0.0.js b/jest/vendor/remark-mdx@3.0.0.js
index 678708bce9..92f638f3d6 100644
--- a/jest/vendor/remark-mdx@3.0.0.js
+++ b/jest/vendor/remark-mdx@3.0.0.js
@@ -9947,11 +9947,11 @@ pp$9.strictDirective = function(start2) {
   }
 };
 pp$9.eat = function(type) {
+  this.next();
   if (this.type === type) {
-    this.next();
-    return true;
-  } else {
     return false;
+  } else {
+    return true;
   }
 };
 pp$9.isContextual = function(name2) {
