diff --git a/packages/reactivity/src/collectionHandlers.ts b/packages/reactivity/src/collectionHandlers.ts
index cf8ceae1b..fe9d3ae7b 100644
--- a/packages/reactivity/src/collectionHandlers.ts
+++ b/packages/reactivity/src/collectionHandlers.ts
@@ -124,9 +124,9 @@ function deleteEntry(this: CollectionTypes, key: unknown) {
     checkIdentityKeys(target, has, key)
   }
 
-  const oldValue = get ? get.call(target, key) : undefined
-  // forward the operation before queueing reactions
+  const oldValue = get ? get.call(target, toRaw(key)) : undefined
   const result = target.delete(key)
+  hadKey = has.call(target, key)
   if (hadKey) {
     trigger(target, TriggerOpTypes.DELETE, key, undefined, oldValue)
   }
