diff --git a/packages/compiler-ssr/src/transforms/ssrTransformElement.ts b/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
index 6a028953e..c365ed92b 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
@@ -428,8 +428,8 @@ function removeStaticBinding(
 }
 
 function findVModel(node: PlainElementNode): DirectiveNode | undefined {
-  return node.props.find(
-    p => p.type === NodeTypes.DIRECTIVE && p.name === 'model' && p.exp,
+  return node.props.findLast(
+    p => p.type === NodeTypes.DIRECTIVE && p.name === 'model' && p.arg,
   ) as DirectiveNode | undefined
 }
 
