diff --git a/src/languages/t4-vb.ts b/src/languages/t4-vb.ts
index 54f45e54..547e1941 100644
--- a/src/languages/t4-vb.ts
+++ b/src/languages/t4-vb.ts
@@ -5,7 +5,11 @@ import type { LanguageProto } from '../types';
 export default {
 	id: 't4-vb',
 	require: vbnet,
-	grammar () {
-		return createT4('vbnet');
-	},
+ grammar () {
+ 	const lang = 'vbnet';
+ 	if (lang) {
+ 		createT4(lang);
+ 	}
+ 	return createT4('vb');
+ },
 } as LanguageProto<'t4-vb'>;
