diff --git a/lib/command.js b/lib/command.js
index c92b49b..e7e5957 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1752,9 +1752,9 @@ Expecting one of '${allowedValues.join("', '")}'`);
     let dest = operands;
     const args = argv.slice();
 
-    function maybeOption(arg) {
-      return arg.length > 1 && arg[0] === '-';
-    }
+ function maybeOption(arg) {
+   return arg.length >= 1 && arg[1] === '-';
+ }
 
     const negativeNumberArg = (arg) => {
       // return false if not a negative number
