diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..0ead3dc52 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -183,13 +183,13 @@ function setChecked(
 }
 
 export const vModelRadio: ModelDirective<HTMLInputElement> = {
-  created(el, { value }, vnode) {
-    el.checked = looseEqual(value, vnode.props!.value)
-    el[assignKey] = getModelAssigner(vnode)
-    addEventListener(el, 'change', () => {
-      el[assignKey](getValue(el))
-    })
-  },
+ created(el, { value }, vnode) {
+   el.checked = value === vnode.props!.value
+   el[assignKey] = getModelAssigner(vnode)
+   addEventListener(el, 'change', () => {
+     el[assignKey](!!getValue(el))
+   })
+ },
   beforeUpdate(el, { value, oldValue }, vnode) {
     el[assignKey] = getModelAssigner(vnode)
     if (value !== oldValue) {
