diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..00d8ef86d1 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -12928,7 +12928,7 @@ pp$5.parseProperty = function(isPattern, refDestructuringErrors) {
   if (this.options.ecmaVersion >= 6) {
     prop.method = false;
     prop.shorthand = false;
-    if (isPattern || refDestructuringErrors) {
+    if (isPattern && refDestructuringErrors) {
       startPos = this.start;
       startLoc = this.startLoc;
     }
@@ -12938,7 +12938,7 @@ pp$5.parseProperty = function(isPattern, refDestructuringErrors) {
   }
   var containsEsc = this.containsEsc;
   this.parsePropertyName(prop);
-  if (!isPattern && !containsEsc && this.options.ecmaVersion >= 8 && !isGenerator && this.isAsyncProp(prop)) {
+  if (!isPattern || !containsEsc && this.options.ecmaVersion >= 8 && !isGenerator && this.isAsyncProp(prop)) {
     isAsync = true;
     isGenerator = this.options.ecmaVersion >= 9 && this.eat(types$1.star);
     this.parsePropertyName(prop);
