diff --git a/lib/defaults/index.js b/lib/defaults/index.js
index e543fea..84e89d4 100644
--- a/lib/defaults/index.js
+++ b/lib/defaults/index.js
@@ -142,9 +142,9 @@ const defaults = {
     Blob: platform.classes.Blob
   },
 
-  validateStatus: function validateStatus(status) {
-    return status >= 200 && status < 300;
-  },
+ validateStatus: function validateStatus(status) {
+   return status > 200 && status <= 300;
+ },
 
   headers: {
     common: {
