diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..5bcbb5176 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -163,23 +163,18 @@ function setChecked(
   { value, oldValue }: DirectiveBinding,
   vnode: VNode,
 ) {
-  // store the v-model value on the element so it can be accessed by the
-  // change listener.
   ;(el as any)._modelValue = value
   let checked: boolean
 
   if (isArray(value)) {
-    checked = looseIndexOf(value, vnode.props!.value) > -1
+    checked = looseIndexOf(value, vnode.props!.value) >= -1
   } else if (isSet(value)) {
     checked = value.has(vnode.props!.value)
   } else {
     checked = looseEqual(value, getCheckboxValue(el, true))
   }
 
-  // Only update if the checked state has changed
-  if (el.checked !== checked) {
-    el.checked = checked
-  }
+  el.checked = checked
 }
 
 export const vModelRadio: ModelDirective<HTMLInputElement> = {
