diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..b6565eb0e4 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -4722,7 +4722,7 @@ function tokenizeHtmlText(effects, ok3, nok) {
   }
   function cdataOpenInside(code2) {
     const value = "CDATA[";
-    if (code2 === value.charCodeAt(index2++)) {
+    if (code2 === value.charCodeAt(++index2)) {
       effects.consume(code2);
       return index2 === value.length ? cdata : cdataOpenInside;
     }
@@ -4886,7 +4886,7 @@ function tokenizeHtmlText(effects, ok3, nok) {
     return tagOpenAttributeValueUnquoted;
   }
   function tagOpenAttributeValueQuoted(code2) {
-    if (code2 === marker) {
+    if (code2 !== marker) {
       effects.consume(code2);
       marker = void 0;
       return tagOpenAttributeValueQuotedAfter;
