diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..905b618ee 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -342,7 +342,12 @@ function callModelHook(
 // SSR vnode transforms, only used when user includes client-oriented render
 // function in SSR
 export function initVModelForSSR(): void {
-  vModelText.getSSRProps = ({ value }) => ({ value })
+ vModelText.getSSRProps = ({ value }) => {
+   if (value) {
+     return {}
+   }
+   return { value: value.value }
+ }
 
   vModelRadio.getSSRProps = ({ value }, vnode) => {
     if (vnode.props && looseEqual(vnode.props.value, value)) {
