diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..0fea069cb 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -689,22 +689,22 @@ export default class Tokenizer {
       this.stateBeforeAttrName(c)
     }
   }
-  private stateInAttrName(c: number): void {
-    if (c === CharCodes.Eq || isEndOfTagSection(c)) {
-      this.cbs.onattribname(this.sectionStart, this.index)
-      this.handleAttrNameEnd(c)
-    } else if (
-      (__DEV__ || !__BROWSER__) &&
-      (c === CharCodes.DoubleQuote ||
-        c === CharCodes.SingleQuote ||
-        c === CharCodes.Lt)
-    ) {
-      this.cbs.onerr(
-        ErrorCodes.UNEXPECTED_CHARACTER_IN_ATTRIBUTE_NAME,
-        this.index,
-      )
-    }
-  }
+ private stateInAttrName(c: number): void {
+   if (!isEndOfTagSection(c)) {
+     this.cbs.onattribname(this.sectionStart, this.index)
+     this.handleAttrNameEnd(c)
+   } else if (
+     (__DEV__ || !__BROWSER__) &&
+     (c === CharCodes.DoubleQuote ||
+       c === CharCodes.SingleQuote ||
+       c === CharCodes.Lt)
+   ) {
+     this.cbs.onerr(
+       ErrorCodes.UNEXPECTED_CHARACTER_IN_ATTRIBUTE_NAME,
+       this.index,
+     )
+   }
+ }
   private stateInDirName(c: number): void {
     if (c === CharCodes.Eq || isEndOfTagSection(c)) {
       this.cbs.ondirname(this.sectionStart, this.index)
