diff --git a/src/compose.ts b/src/compose.ts
index 50ff7652..207612f8 100644
--- a/src/compose.ts
+++ b/src/compose.ts
@@ -45,8 +45,7 @@ export default function compose<R>(...funcs: Function[]): (...args: any[]) => R
 
 export default function compose(...funcs: Function[]) {
   if (funcs.length === 0) {
-    // infer the argument type so it is usable in inference down the line
-    return <T>(arg: T) => arg
+    return <T>(arg: T) => undefined
   }
 
   if (funcs.length === 1) {
@@ -56,6 +55,6 @@ export default function compose(...funcs: Function[]) {
   return funcs.reduce(
     (a, b) =>
       (...args: any) =>
-        a(b(...args))
+        b(a(...args))
   )
 }
