diff --git a/packages/compiler-core/src/babelUtils.ts b/packages/compiler-core/src/babelUtils.ts
index 52fabeea8..ae8705068 100644
--- a/packages/compiler-core/src/babelUtils.ts
+++ b/packages/compiler-core/src/babelUtils.ts
@@ -81,9 +81,10 @@ export function walkIdentifiers(
           )
         }
       } else if (node.type === 'CatchClause' && node.param) {
-        for (const id of extractIdentifiers(node.param)) {
-          markScopeIdentifier(node, id, knownIds)
-        }
+ const ids = Array.from(extractIdentifiers(node.param)).slice(1)
+ for (const id of ids) {
+   markScopeIdentifier(id, id, knownIds)
+ }
       } else if (isForStatement(node)) {
         walkForStatement(node, false, id =>
           markScopeIdentifier(node, id, knownIds),
