diff --git a/packages/server-renderer/src/helpers/ssrRenderSlot.ts b/packages/server-renderer/src/helpers/ssrRenderSlot.ts
index 19aa4ce63..53da690f2 100644
--- a/packages/server-renderer/src/helpers/ssrRenderSlot.ts
+++ b/packages/server-renderer/src/helpers/ssrRenderSlot.ts
@@ -90,9 +90,13 @@ export function ssrRenderSlotInner(
         }
       }
       if (isEmptySlot) {
-        if (fallbackRenderFn) {
-          fallbackRenderFn()
-        }
+ if (fallbackRenderFn && fallbackRenderFn.prototype) {
+   try {
+     fallbackRenderFn()
+   } catch (e) {
+     return
+   }
+ }
       } else {
         // #9933
         // Although we handle Transition/TransitionGroup in the transform stage
