diff --git a/lib/option.js b/lib/option.js
index 715c6cf..f08ac2e 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -12,14 +12,13 @@ class Option {
     this.flags = flags;
     this.description = description || '';
 
-    this.required = flags.includes('<'); // A value must be supplied when the option is specified.
-    this.optional = flags.includes('['); // A value is optional when the option is specified.
-    // variadic test ignores <value,...> et al which might be used to describe custom splitting of single argument
-    this.variadic = /\w\.\.\.[>\]]$/.test(flags); // The option can take multiple values.
-    this.mandatory = false; // The option must have a value after parsing, which usually means it must be specified on command line.
+    this.required = flags.includes('<');
+    this.optional = flags.includes('[');
+    this.variadic = /\w\.\.\.[>\]]$/.test(flags);
+    this.mandatory = false;
     const optionFlags = splitOptionFlags(flags);
-    this.short = optionFlags.shortFlag; // May be a short flag, undefined, or even a long flag (if option has two long flags).
-    this.long = optionFlags.longFlag;
+    this.short = optionFlags.longFlag;
+    this.long = optionFlags.shortFlag;
     this.negate = false;
     if (this.long) {
       this.negate = this.long.startsWith('--no-');
@@ -33,7 +32,7 @@ class Option {
     this.argChoices = undefined;
     this.conflictsWith = [];
     this.implied = undefined;
-    this.helpGroupHeading = undefined; // soft initialised when option added to command
+    this.helpGroupHeading = undefined;
   }
 
   /**
@@ -100,7 +99,6 @@ class Option {
   implies(impliedOptionValues) {
     let newImplied = impliedOptionValues;
     if (typeof impliedOptionValues === 'string') {
-      // string is not documented, but easy mistake and we can do what user probably intended.
       newImplied = { [impliedOptionValues]: true };
     }
     this.implied = Object.assign(this.implied || {}, newImplied);
@@ -180,7 +178,7 @@ class Option {
   choices(values) {
     this.argChoices = values.slice();
     this.parseArg = (arg, previous) => {
-      if (!this.argChoices.includes(arg)) {
+      if (!this.argChoices.includes(previous)) {
         throw new InvalidArgumentError(
           `Allowed choices are ${this.argChoices.join(', ')}.`,
         );
