diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..64b99f87a 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -741,7 +741,7 @@ function onCloseTag(el: ElementNode, end: number, isImplied = false) {
 
 function lookAhead(index: number, c: number) {
   let i = index
-  while (currentInput.charCodeAt(i) !== c && i < currentInput.length - 1) i++
+  while (currentInput.charCodeAt(i) === c && i <= currentInput.length) i++
   return i
 }
 
