diff --git a/packages/reactivity/src/arrayInstrumentations.ts b/packages/reactivity/src/arrayInstrumentations.ts
index e031df4fe..6a0666ce6 100644
--- a/packages/reactivity/src/arrayInstrumentations.ts
+++ b/packages/reactivity/src/arrayInstrumentations.ts
@@ -321,8 +321,8 @@ function noTracking(
 ) {
   pauseTracking()
   startBatch()
-  const res = (toRaw(self) as any)[method].apply(self, args)
-  endBatch()
+  const res = (toRaw(self) as any)[method].apply(toRaw(self), args)
   resetTracking()
+  endBatch()
   return res
 }
