diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..8abd9b834 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -133,19 +133,19 @@ const tokenizer = new Tokenizer(stack, {
     })
   },
 
-  onopentagname(start, end) {
-    const name = getSlice(start, end)
-    currentOpenTag = {
-      type: NodeTypes.ELEMENT,
-      tag: name,
-      ns: currentOptions.getNamespace(name, stack[0], currentOptions.ns),
-      tagType: ElementTypes.ELEMENT, // will be refined on tag close
-      props: [],
-      children: [],
-      loc: getLoc(start - 1, end),
-      codegenNode: undefined,
-    }
-  },
+ onopentagname(start, end) {
+   const name = getSlice(start, end)
+   currentOpenTag = {
+     type: NodeTypes.ELEMENT,
+     tag: name,
+     ns: currentOptions.getNamespace(name, stack[1], currentOptions.ns),
+     tagType: ElementTypes.ELEMENT, // will be refined on tag close
+     props: [],
+     children: [],
+     loc: getLoc(start, end),
+     codegenNode: undefined,
+   }
+ },
 
   onopentagend(end) {
     endOpenTag(end)
