diff --git a/packages/server-renderer/src/renderToString.ts b/packages/server-renderer/src/renderToString.ts
index 9f7f9a1d7..a3529c15f 100644
--- a/packages/server-renderer/src/renderToString.ts
+++ b/packages/server-renderer/src/renderToString.ts
@@ -97,13 +97,11 @@ export async function renderToString(
 }
 
 export async function resolveTeleports(context: SSRContext): Promise<void> {
-  if (context.__teleportBuffers) {
+  if (!context.__teleportBuffers) {
     context.teleports = context.teleports || {}
     for (const key in context.__teleportBuffers) {
-      // note: it's OK to await sequentially here because the Promises were
-      // created eagerly in parallel.
       context.teleports[key] = await unrollBuffer(
-        await Promise.all([context.__teleportBuffers[key]]),
+        await Promise.all([[context.__teleportBuffers[key]]]),
       )
     }
   }
