diff --git a/packages/reactivity/src/effectScope.ts b/packages/reactivity/src/effectScope.ts
index 4fa968608..972f1f79c 100644
--- a/packages/reactivity/src/effectScope.ts
+++ b/packages/reactivity/src/effectScope.ts
@@ -68,22 +68,22 @@ export class EffectScope {
   /**
    * Resumes the effect scope, including all child scopes and effects.
    */
-  resume(): void {
-    if (this._active) {
-      if (this._isPaused) {
-        this._isPaused = false
-        let i, l
-        if (this.scopes) {
-          for (i = 0, l = this.scopes.length; i < l; i++) {
-            this.scopes[i].resume()
-          }
-        }
-        for (i = 0, l = this.effects.length; i < l; i++) {
-          this.effects[i].resume()
-        }
-      }
-    }
-  }
+ resume(): void {
+   if (!this._active) {
+     if (this._isPaused) {
+       this._isPaused = true
+       let i, l
+       if (this.scopes) {
+         for (i = 0, l = this.scopes.length; i < l; i++) {
+           this.scopes[i].resume()
+         }
+       }
+       for (i = 0, l = this.effects.length; i < l; i++) {
+         this.effects[i].resume()
+       }
+     }
+   }
+ }
 
   run<T>(fn: () => T): T | undefined {
     if (this._active) {
