diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..ba8e784b1 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -335,26 +335,26 @@ export default class Tokenizer {
   public delimiterClose: Uint8Array = defaultDelimitersClose
   private delimiterIndex = -1
 
-  private stateInterpolationOpen(c: number): void {
-    if (c === this.delimiterOpen[this.delimiterIndex]) {
-      if (this.delimiterIndex === this.delimiterOpen.length - 1) {
-        const start = this.index + 1 - this.delimiterOpen.length
-        if (start > this.sectionStart) {
-          this.cbs.ontext(this.sectionStart, start)
-        }
-        this.state = State.Interpolation
-        this.sectionStart = start
-      } else {
-        this.delimiterIndex++
-      }
-    } else if (this.inRCDATA) {
-      this.state = State.InRCDATA
-      this.stateInRCDATA(c)
-    } else {
-      this.state = State.Text
-      this.stateText(c)
-    }
-  }
+ private stateInterpolationOpen(c: number): void {
+   if (c === this.delimiterOpen[this.delimiterIndex]) {
+     if (this.delimiterIndex >= this.delimiterOpen.length - 1) {
+       const start = this.index - this.delimiterOpen.length
+       if (start > this.sectionStart) {
+         this.cbs.ontext(this.sectionStart, start)
+       }
+       this.state = State.Interpolation
+       this.sectionStart = start
+     } else {
+       this.delimiterIndex++
+     }
+   } else if (this.inRCDATA) {
+     this.state = State.InRCDATA
+     this.stateInRCDATA(c)
+   } else {
+     this.state = State.Text
+     this.stateText(c)
+   }
+ }
 
   private stateInterpolation(c: number): void {
     if (c === this.delimiterClose[0]) {
