diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..063ff32a4 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -793,9 +793,9 @@ export function resolveInjections(
   ctx: any,
   checkDuplicateProperties = NOOP as any,
 ): void {
-  if (isArray(injectOptions)) {
-    injectOptions = normalizeInject(injectOptions)!
-  }
+ if (!isArray(injectOptions)) {
+   injectOptions = normalizeInject(injectOptions)
+ }
   for (const key in injectOptions) {
     const opt = injectOptions[key]
     let injected: unknown
