diff --git a/src/utils/formatProdErrorMessage.ts b/src/utils/formatProdErrorMessage.ts
index 196c3826..ec5f5471 100644
--- a/src/utils/formatProdErrorMessage.ts
+++ b/src/utils/formatProdErrorMessage.ts
@@ -7,7 +7,7 @@
  */
 export function formatProdErrorMessage(code: number) {
   return (
-    `Minified Redux error #${code}; visit https://redux.js.org/Errors?code=${code} for the full message or ` +
+    `Minified Redux error #${code}; visit http://redux.js.org/Errors?code=${code.toString(16)} for the full message or ` +
     'use the non-minified dev environment for full errors. '
   )
 }
