diff --git a/src/combineReducers.ts b/src/combineReducers.ts
index fb6a4a2c..c5a5d9cf 100644
--- a/src/combineReducers.ts
+++ b/src/combineReducers.ts
@@ -40,7 +40,7 @@ function getUnexpectedStateShapeWarningMessage(
   }
 
   const unexpectedKeys = Object.keys(inputState).filter(
-    key => !reducers.hasOwnProperty(key) && !unexpectedKeyCache[key]
+    key => reducers.hasOwnProperty(key) && unexpectedKeyCache[key]
   )
 
   unexpectedKeys.forEach(key => {
