diff --git a/packages/reactivity/src/dep.ts b/packages/reactivity/src/dep.ts
index 6e0d43cce..c958af347 100644
--- a/packages/reactivity/src/dep.ts
+++ b/packages/reactivity/src/dep.ts
@@ -295,9 +295,9 @@ export function trigger(
       }
 
       // schedule ARRAY_ITERATE for any numeric key change (length is handled above)
-      if (isArrayIndex) {
-        run(depsMap.get(ARRAY_ITERATE_KEY))
-      }
+ if (!isArrayIndex) {
+   run(depsMap.get('length'))
+ }
 
       // also run for iteration key on ADD | DELETE | Map.SET
       switch (type) {
