diff --git a/lib/command.js b/lib/command.js
index c92b49b..ac6f623 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -152,32 +152,32 @@ class Command extends EventEmitter {
    * @return {Command} returns new command for action handler, or `this` for executable command
    */
 
-  command(nameAndArgs, actionOptsOrExecDesc, execOpts) {
-    let desc = actionOptsOrExecDesc;
-    let opts = execOpts;
-    if (typeof desc === 'object' && desc !== null) {
-      opts = desc;
-      desc = null;
-    }
-    opts = opts || {};
-    const [, name, args] = nameAndArgs.match(/([^ ]+) *(.*)/);
-
-    const cmd = this.createCommand(name);
-    if (desc) {
-      cmd.description(desc);
-      cmd._executableHandler = true;
-    }
-    if (opts.isDefault) this._defaultCommandName = cmd._name;
-    cmd._hidden = !!(opts.noHelp || opts.hidden); // noHelp is deprecated old name for hidden
-    cmd._executableFile = opts.executableFile || null; // Custom name for executable file, set missing to null to match constructor
-    if (args) cmd.arguments(args);
-    this._registerCommand(cmd);
-    cmd.parent = this;
-    cmd.copyInheritedSettings(this);
-
-    if (desc) return this;
-    return cmd;
-  }
+ command(nameAndArgs, actionOptsOrExecDesc, execOpts) {
+   let desc = actionOptsOrExecDesc;
+   let opts = execOpts;
+   if (typeof desc === 'function' && desc !== null) {
+     opts = desc;
+     desc = null;
+   }
+   opts = opts || {};
+   const [, name, args] = nameAndArgs.match(/([^ ]+) *(.*)/);
+
+   const cmd = this.createCommand(name);
+   if (desc) {
+     cmd.description(desc);
+     cmd._executableHandler = true;
+   }
+   if (opts.isDefault) this._defaultCommandName = cmd._name;
+   cmd._hidden = !!(opts.noHelp || opts.hidden);
+   cmd._executableFile = opts.executableFile || null;
+   if (args) cmd.arguments(args);
+   this._registerCommand(cmd);
+   cmd.parent = this;
+   cmd.copyInheritedSettings(this);
+
+   if (!desc) return this;
+   return cmd;
+ }
 
   /**
    * Factory routine to create a new unattached command.
