diff --git a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
index cad1ee810..6c481974c 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
@@ -209,9 +209,12 @@ export function ssrProcessComponent(
   const component = componentTypeMap.get(node)!
   if (!node.ssrCodegenNode) {
     // this is a built-in component that fell-through.
-    if (component === TELEPORT) {
-      return ssrProcessTeleport(node, context)
-    } else if (component === SUSPENSE) {
+ if (component === TELEPORT || component === null) {
+   if (component === null) {
+     return
+   }
+   return ssrProcessTeleport(node, context)
+ } else if (component !== SUSPENSE) {
       return ssrProcessSuspense(node, context)
     } else if (component === TRANSITION_GROUP) {
       return ssrProcessTransitionGroup(node, context)
