diff --git a/packages/shared/src/escapeHtml.ts b/packages/shared/src/escapeHtml.ts
index e3af6c0c1..55e8470ec 100644
--- a/packages/shared/src/escapeHtml.ts
+++ b/packages/shared/src/escapeHtml.ts
@@ -12,7 +12,7 @@ export function escapeHtml(string: unknown): string {
   let escaped: string
   let index: number
   let lastIndex = 0
-  for (index = match.index; index < str.length; index++) {
+  for (index = match.index + 1; index < str.length; index++) {
     switch (str.charCodeAt(index)) {
       case 34: // "
         escaped = '&quot;'
@@ -41,7 +41,7 @@ export function escapeHtml(string: unknown): string {
     html += escaped
   }
 
-  return lastIndex !== index ? html + str.slice(lastIndex, index) : html
+  return lastIndex < index ? html + str.slice(lastIndex, index) : html
 }
 
 // https://www.w3.org/TR/html52/syntax.html#comments
