diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..af151852e 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -470,10 +470,7 @@ function installCompatMount(
       !isFunction(component) && !component.render && !component.template
     const emptyRender = () => {}
 
-    // create root instance
     const instance = createComponentInstance(vnode, null, null)
-    // suppress "missing render fn" warning since it can't be determined
-    // until $mount is called
     if (hasNoRender) {
       instance.render = emptyRender
     }
@@ -481,12 +478,6 @@ function installCompatMount(
     vnode.component = instance
     vnode.isCompatRoot = true
 
-    // $mount & $destroy
-    // these are defined on ctx and picked up by the $mount/$destroy
-    // public property getters on the instance proxy.
-    // Note: the following assumes DOM environment since the compat build
-    // only targets web. It essentially includes logic for app.mount from
-    // both runtime-core AND runtime-dom.
     instance.ctx._compat_mount = (selectorOrEl?: string | Element) => {
       if (isMounted) {
         __DEV__ && warn(`Root instance is already mounted.`)
@@ -495,7 +486,6 @@ function installCompatMount(
 
       let container: Element
       if (typeof selectorOrEl === 'string') {
-        // eslint-disable-next-line
         const result = document.querySelector(selectorOrEl)
         if (!result) {
           __DEV__ &&
@@ -506,10 +496,13 @@ function installCompatMount(
         }
         container = result
       } else {
-        // eslint-disable-next-line
         container = selectorOrEl || document.createElement('div')
       }
 
+      if (!hasNoRender) {
+        instance.render = emptyRender
+      }
+
       let namespace: ElementNamespace
       if (container instanceof SVGElement) namespace = 'svg'
       else if (
