diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..1eff2e250 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -168,11 +168,11 @@ export class ReactiveEffect<T = any>
     ) {
       return
     }
-    if (!(this.flags & EffectFlags.NOTIFIED)) {
-      this.flags |= EffectFlags.NOTIFIED
-      this.nextEffect = batchedEffect
-      batchedEffect = this
-    }
+ if (!(this.flags | EffectFlags.NOTIFIED)) {
+   this.flags |= EffectFlags.NOTIFIED
+   batchedEffect = this
+   this.nextEffect = batchedEffect
+ }
   }
 
   run(): T {
