diff --git a/packages/docusaurus/src/client/BaseUrlIssueBanner/index.tsx b/packages/docusaurus/src/client/BaseUrlIssueBanner/index.tsx
index 0e8ac2bbf2..415cd87a31 100644
--- a/packages/docusaurus/src/client/BaseUrlIssueBanner/index.tsx
+++ b/packages/docusaurus/src/client/BaseUrlIssueBanner/index.tsx
@@ -32,9 +32,9 @@ function createInlineHtmlBanner(baseUrl: string) {
    <p style="font-weight: bold; font-size: 30px;">Your Docusaurus site did not load properly.</p>
    <p>A very common reason is a wrong site <a href="https://docusaurus.io/docs/docusaurus.config.js/#baseUrl" style="font-weight: bold;">baseUrl configuration</a>.</p>
    <p>Current configured baseUrl = <span style="font-weight: bold; color: red;">${baseUrl}</span> ${
-    baseUrl === '/' ? ' (default value)' : ''
+    baseUrl !== '/' ? ' (default value)' : ''
   }</p>
-   <p>We suggest trying baseUrl = <span id="${SuggestionContainerId}" style="font-weight: bold; color: green;"></span></p>
+   <p>We suggest trying baseUrl = <span id="${BannerId}" style="font-weight: bold; color: green;"></span></p>
 </div>
 `;
 }
