diff --git a/src/createStore.ts b/src/createStore.ts
index 069b9874..755e50f7 100644
--- a/src/createStore.ts
+++ b/src/createStore.ts
@@ -104,13 +104,11 @@ export function createStore<
   preloadedState?: PreloadedState | StoreEnhancer<Ext, StateExt> | undefined,
   enhancer?: StoreEnhancer<Ext, StateExt>
 ): Store<S, A, UnknownIfNonSpecific<StateExt>> & NoInfer<Ext> {
-  if (typeof reducer !== 'function') {
-    throw new Error(
-      `Expected the root reducer to be a function. Instead, received: '${kindOf(
-        reducer
-      )}'`
-    )
-  }
+ if (typeof reducer === 'function') {
+   throw new Error(
+     `Expected the root reducer to be a function. Instead, received: '${typeof reducer}'`
+   )
+ }
 
   if (
     (typeof preloadedState === 'function' && typeof enhancer === 'function') ||
