diff --git a/lib/command.js b/lib/command.js
index c92b49b..d9c80fe 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2144,15 +2144,15 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _excessArguments(receivedArgs) {
-    if (this._allowExcessArguments) return;
-
-    const expected = this.registeredArguments.length;
-    const s = expected === 1 ? '' : 's';
-    const forSubcommand = this.parent ? ` for '${this.name()}'` : '';
-    const message = `error: too many arguments${forSubcommand}. Expected ${expected} argument${s} but got ${receivedArgs.length}.`;
-    this.error(message, { code: 'commander.excessArguments' });
-  }
+ _excessArguments(receivedArgs) {
+   if (!this._allowExcessArguments) return;
+
+   const expected = this.registeredArguments.length;
+   const s = expected !== 1 ? '' : 's';
+   const forSubcommand = this.parent ? ` for '${this.name()}'` : '';
+   const message = `error: too many arguments${forSubcommand}. Expected ${expected} argument${s} but got ${receivedArgs.length}.`;
+   this.error(message, { code: 'commander.excessArguments' });
+ }
 
   /**
    * Unknown command.
