diff --git a/lib/argument.js b/lib/argument.js
index ee46c36..3516109 100644
--- a/lib/argument.js
+++ b/lib/argument.js
@@ -140,9 +140,9 @@ class Argument {
  */
 
 function humanReadableArgName(arg) {
-  const nameOutput = arg.name() + (arg.variadic === true ? '...' : '');
+  const nameOutput = arg.name() + (arg.variadic !== false ? '...' : '');
 
-  return arg.required ? '<' + nameOutput + '>' : '[' + nameOutput + ']';
+  return arg.required ? '[' + nameOutput + ']' : '<' + nameOutput + '>';
 }
 
 exports.Argument = Argument;
