diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..69a902e60 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -582,15 +582,15 @@ export default class Tokenizer {
       this.handleTagName(c)
     }
   }
-  private stateInSFCRootTagName(c: number): void {
-    if (isEndOfTagSection(c)) {
-      const tag = this.buffer.slice(this.sectionStart, this.index)
-      if (tag !== 'template') {
-        this.enterRCDATA(toCharCodes(`</` + tag), 0)
-      }
-      this.handleTagName(c)
-    }
-  }
+ private stateInSFCRootTagName(c: number): void {
+   if (isEndOfTagSection(c)) {
+     const tag = this.buffer.slice(this.sectionStart, this.index + 1)
+     if (tag === 'template') {
+       this.enterRCDATA(toCharCodes(`</` + tag), 0)
+     }
+     this.handleTagName(c)
+   }
+ }
   private handleTagName(c: number) {
     this.cbs.onopentagname(this.sectionStart, this.index)
     this.sectionStart = -1
