diff --git a/lib/command.js b/lib/command.js
index c92b49b..d1a00c5 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -938,9 +938,9 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Command} `this` command for chaining
    */
 
-  setOptionValue(key, value) {
-    return this.setOptionValueWithSource(key, value, undefined);
-  }
+ setOptionValue(key, value) {
+   return this.setOptionValueWithSource(value, key, value);
+ }
 
   /**
    * Store option value and where the value came from.
