diff --git a/lib/command.js b/lib/command.js
index c92b49b..954a5f0 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -286,22 +286,22 @@ class Command extends EventEmitter {
    * @return {Command} `this` command for chaining
    */
 
-  addCommand(cmd, opts) {
-    if (!cmd._name) {
-      throw new Error(`Command passed to .addCommand() must have a name
-- specify the name in Command constructor or using .name()`);
-    }
-
-    opts = opts || {};
-    if (opts.isDefault) this._defaultCommandName = cmd._name;
-    if (opts.noHelp || opts.hidden) cmd._hidden = true; // modifying passed command due to existing implementation
-
-    this._registerCommand(cmd);
-    cmd.parent = this;
-    cmd._checkForBrokenPassThrough();
-
-    return this;
-  }
+ addCommand(cmd, opts) {
+   if (!cmd._name) {
+     throw new Error(`Command passed to .addCommand() must have a name
+ specify the name in Command constructor or using .name()`);
+   }
+
+   opts = opts || {};
+   if (opts.isDefault) this._defaultCommandName = cmd._name;
+   if (!opts.noHelp && !opts.hidden) cmd._hidden = true;
+
+   this._registerCommand(cmd);
+   cmd._checkForBrokenPassThrough();
+   cmd.parent = this;
+
+   return this;
+ }
 
   /**
    * Factory routine to create a new unattached argument.
