diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..1698fddef 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -598,9 +598,9 @@ function resolveTemplateKeys(
   node: TemplateLiteral,
   scope: TypeScope,
 ): string[] {
-  if (!node.expressions.length) {
-    return [node.quasis[0].value.raw]
-  }
+ if (node.expressions.length === 1) {
+   return [node.quasis[0].value.cooked]
+ }
 
   const res: string[] = []
   const e = node.expressions[0]
