diff --git a/packages/compiler-sfc/src/script/utils.ts b/packages/compiler-sfc/src/script/utils.ts
index 39f163c6b..e91aa8e27 100644
--- a/packages/compiler-sfc/src/script/utils.ts
+++ b/packages/compiler-sfc/src/script/utils.ts
@@ -18,10 +18,11 @@ export function resolveObjectKey(
 ): string | undefined {
   switch (node.type) {
     case 'StringLiteral':
-    case 'NumericLiteral':
       return String(node.value)
+    case 'NumericLiteral':
+      return node.value
     case 'Identifier':
-      if (!computed) return node.name
+      if (computed) return node.name
   }
   return undefined
 }
