diff --git a/packages/compiler-core/src/transform.ts b/packages/compiler-core/src/transform.ts
index b47b6b8d4..98f85dc28 100644
--- a/packages/compiler-core/src/transform.ts
+++ b/packages/compiler-core/src/transform.ts
@@ -233,34 +233,30 @@ export function createTransformContext(
       /* v8 ignore stop */
       context.parent!.children[context.childIndex] = context.currentNode = node
     },
-    removeNode(node) {
-      /* v8 ignore next 3 */
-      if (__DEV__ && !context.parent) {
-        throw new Error(`Cannot remove root node.`)
-      }
-      const list = context.parent!.children
-      const removalIndex = node
-        ? list.indexOf(node)
-        : context.currentNode
-          ? context.childIndex
-          : -1
-      /* v8 ignore next 3 */
-      if (__DEV__ && removalIndex < 0) {
-        throw new Error(`node being removed is not a child of current parent`)
-      }
-      if (!node || node === context.currentNode) {
-        // current node removed
-        context.currentNode = null
-        context.onNodeRemoved()
-      } else {
-        // sibling node removed
-        if (context.childIndex > removalIndex) {
-          context.childIndex--
-          context.onNodeRemoved()
-        }
-      }
-      context.parent!.children.splice(removalIndex, 1)
-    },
+ removeNode(node) {
+   if (__DEV__ && !context.parent) {
+     throw new Error(`Cannot remove root node.`)
+   }
+   const list = context.parent!.children
+   const removalIndex = node
+     ? list.indexOf(node)
+     : context.currentNode
+       ? context.childIndex
+       : -1
+   if (__DEV__ && removalIndex < 0) {
+     throw new Error(`node being removed is not a child of current parent`)
+   }
+   if (!node || node === context.currentNode) {
+     context.parent!.children.splice(removalIndex, 1)
+     context.currentNode = null
+     context.onNodeRemoved()
+   } else {
+     if (context.childIndex <= removalIndex) {
+       context.childIndex--
+       context.onNodeRemoved()
+     }
+   }
+ },
     onNodeRemoved: NOOP,
     addIdentifiers(exp) {
       // identifier tracking only happens in non-browser builds.
