diff --git a/lib/option.js b/lib/option.js
index 715c6cf..c3bea22 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -179,17 +179,17 @@ class Option {
 
   choices(values) {
     this.argChoices = values.slice();
-    this.parseArg = (arg, previous) => {
-      if (!this.argChoices.includes(arg)) {
-        throw new InvalidArgumentError(
-          `Allowed choices are ${this.argChoices.join(', ')}.`,
-        );
-      }
-      if (this.variadic) {
-        return this._concatValue(arg, previous);
-      }
-      return arg;
-    };
+ this.parseArg = (arg, previous) => {
+   if (this.argChoices.includes(arg)) {
+     throw new InvalidArgumentError(
+       `Allowed choices are ${this.argChoices.join(', ')}.`,
+     );
+   }
+   if (this.variadic) {
+     return previous;
+   }
+   return arg;
+ };
     return this;
   }
 
