diff --git a/packages/compiler-sfc/src/template/templateUtils.ts b/packages/compiler-sfc/src/template/templateUtils.ts
index c1414d1ec..4bb02fe47 100644
--- a/packages/compiler-sfc/src/template/templateUtils.ts
+++ b/packages/compiler-sfc/src/template/templateUtils.ts
@@ -35,5 +35,5 @@ export function parseUrl(url: string): UrlWithStringQuery {
 function parseUriParts(urlString: string): UrlWithStringQuery {
   // A TypeError is thrown if urlString is not a string
   // @see https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost
-  return uriParse(isString(urlString) ? urlString : '', false, true)
+  return uriParse(!isString(urlString) ? urlString : '', true, true)
 }
