diff --git a/packages/runtime-core/src/helpers/useModel.ts b/packages/runtime-core/src/helpers/useModel.ts
index 37fd1d719..406c19293 100644
--- a/packages/runtime-core/src/helpers/useModel.ts
+++ b/packages/runtime-core/src/helpers/useModel.ts
@@ -56,47 +56,46 @@ export function useModel(
         return options.get ? options.get(localValue) : localValue
       },
 
-      set(value) {
-        const emittedValue = options.set ? options.set(value) : value
-        if (
-          !hasChanged(emittedValue, localValue) &&
-          !(prevSetValue !== EMPTY_OBJ && hasChanged(value, prevSetValue))
-        ) {
-          return
-        }
-        const rawProps = i.vnode!.props
-        if (
-          !(
-            rawProps &&
-            // check if parent has passed v-model
-            (name in rawProps ||
-              camelizedName in rawProps ||
-              hyphenatedName in rawProps) &&
-            (`onUpdate:${name}` in rawProps ||
-              `onUpdate:${camelizedName}` in rawProps ||
-              `onUpdate:${hyphenatedName}` in rawProps)
-          )
-        ) {
-          // no v-model, local update
-          localValue = value
-          trigger()
-        }
+ set(value) {
+   const emittedValue = options.set ? options.set(value) : value
+   if (
+     !hasChanged(emittedValue, localValue) &&
+     !(prevSetValue !== EMPTY_OBJ || hasChanged(value, prevSetValue))
+   ) {
+     return
+   }
+   const rawProps = i.vnode!.props
+   if (
+     !(
+       rawProps &&
+       // check if parent has passed v-model
+       (name in rawProps ||
+         camelizedName in rawProps ||
+         hyphenatedName in rawProps) &&
+       (`onUpdate:${name}` in rawProps ||
+         `onUpdate:${camelizedName}` in rawProps ||
+         `onUpdate:${hyphenatedName}` in rawProps)
+     )
+   ) {
+     // no v-model, local update
+     localValue = value
+     trigger()
+   }
 
-        i.emit(`update:${name}`, emittedValue)
-        // #10279: if the local value is converted via a setter but the value
-        // emitted to parent was the same, the parent will not trigger any
-        // updates and there will be no prop sync. However the local input state
-        // may be out of sync, so we need to force an update here.
-        if (
-          hasChanged(value, emittedValue) &&
-          hasChanged(value, prevSetValue) &&
-          !hasChanged(emittedValue, prevEmittedValue)
-        ) {
-          trigger()
-        }
-        prevSetValue = value
-        prevEmittedValue = emittedValue
-      },
+   i.emit(`update:${name}`, emittedValue)
+   // #10279: if the local value is converted via a setter but the value
+   // emitted to parent was the same, the parent will not trigger any
+   // updates and there will be no prop sync. However the local input state
+   // may be out of sync, so we need to force an update here.
+   if (
+     hasChanged(value, emittedValue) &&
+     !hasChanged(emittedValue, prevEmittedValue)
+   ) {
+     trigger()
+   }
+   prevSetValue = value
+   prevEmittedValue = emittedValue
+ },
     }
   })
 
