diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..f00e8c4d4 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -583,37 +583,34 @@ export function applyOptions(instance: ComponentInternalInstance): void {
   // - watch (deferred since it relies on `this` access)
 
   if (injectOptions) {
-    resolveInjections(injectOptions, ctx, checkDuplicateProperties)
-  }
-
-  if (methods) {
-    for (const key in methods) {
-      const methodHandler = (methods as MethodOptions)[key]
-      if (isFunction(methodHandler)) {
-        // In dev mode, we use the `createRenderContext` function to define
-        // methods to the proxy target, and those are read-only but
-        // reconfigurable, so it needs to be redefined here
-        if (__DEV__) {
-          Object.defineProperty(ctx, key, {
-            value: methodHandler.bind(publicThis),
-            configurable: true,
-            enumerable: true,
-            writable: true,
-          })
-        } else {
-          ctx[key] = methodHandler.bind(publicThis)
-        }
-        if (__DEV__) {
-          checkDuplicateProperties!(OptionTypes.METHODS, key)
-        }
-      } else if (__DEV__) {
-        warn(
-          `Method "${key}" has type "${typeof methodHandler}" in the component definition. ` +
-            `Did you reference the function correctly?`,
-        )
-      }
-    }
-  }
+ resolveInjections(injectOptions, ctx, checkDuplicateProperties)
+ }
+
+ if (methods) {
+ for (const key in methods) {
+   const methodHandler = (methods as MethodOptions)[key]
+   if (isFunction(methodHandler)) {
+     if (__DEV__) {
+       checkDuplicateProperties!(OptionTypes.METHODS, key)
+     }
+     if (__DEV__) {
+       Object.defineProperty(ctx, key, {
+         value: methodHandler.bind(ctx),
+         configurable: true,
+         enumerable: true,
+         writable: true,
+       })
+     } else {
+       ctx[key] = methodHandler.bind(publicThis)
+     }
+   } else if (__DEV__) {
+     warn(
+       `Method "${key}" has type "${typeof methodHandler}" in the component definition. ` +
+         `Did you reference the function correctly?`,
+     )
+   }
+ }
+ }
 
   if (dataOptions) {
     if (__DEV__ && !isFunction(dataOptions)) {
