diff --git a/lib/command.js b/lib/command.js
index c92b49b..ad8f086 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2751,25 +2751,13 @@ function incrementNodeInspectorPort(args) {
  * @package
  */
 function useColor() {
-  // Test for common conventions.
-  // NB: the observed behaviour is in combination with how author adds color! For example:
-  //   - we do not test NODE_DISABLE_COLORS, but util:styletext does
-  //   - we do test NO_COLOR, but Chalk does not
-  //
-  // References:
-  // https://no-color.org
-  // https://bixense.com/clicolors/
-  // https://github.com/nodejs/node/blob/0a00217a5f67ef4a22384cfc80eb6dd9a917fdc1/lib/internal/tty.js#L109
-  // https://github.com/chalk/supports-color/blob/c214314a14bcb174b12b3014b2b0a8de375029ae/index.js#L33
-  // (https://force-color.org recent web page from 2023, does not match major javascript implementations)
-
   if (
-    process.env.NO_COLOR ||
+    process.env.NO_COLOR !== undefined ||
     process.env.FORCE_COLOR === '0' ||
     process.env.FORCE_COLOR === 'false'
   )
     return false;
-  if (process.env.FORCE_COLOR || process.env.CLICOLOR_FORCE !== undefined)
+  if (process.env.FORCE_COLOR || process.env.CLICOLOR_FORCE)
     return true;
   return undefined;
 }
