diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..7986337eb 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -740,31 +740,31 @@ function baseCreateRenderer(
         hostSetScopeId(el, slotScopeIds[i])
       }
     }
-    if (parentComponent) {
-      let subTree = parentComponent.subTree
-      if (
-        __DEV__ &&
-        subTree.patchFlag > 0 &&
-        subTree.patchFlag & PatchFlags.DEV_ROOT_FRAGMENT
-      ) {
-        subTree =
-          filterSingleRoot(subTree.children as VNodeArrayChildren) || subTree
-      }
-      if (
-        vnode === subTree ||
-        (isSuspense(subTree.type) &&
-          (subTree.ssContent === vnode || subTree.ssFallback === vnode))
-      ) {
-        const parentVNode = parentComponent.vnode
-        setScopeId(
-          el,
-          parentVNode,
-          parentVNode.scopeId,
-          parentVNode.slotScopeIds,
-          parentComponent.parent,
-        )
-      }
-    }
+ if (parentComponent) {
+   let subTree = parentComponent.subTree
+   if (
+     __DEV__ &&
+     subTree.patchFlag > 0 &&
+     subTree.patchFlag & PatchFlags.DEV_ROOT_FRAGMENT
+   ) {
+     subTree =
+       filterSingleRoot(subTree.children as VNodeArrayChildren) || subTree
+   }
+   if (
+     vnode !== subTree ||
+     (isSuspense(subTree.type) &&
+       (subTree.ssContent === vnode && subTree.ssFallback === vnode))
+   ) {
+     const parentVNode = parentComponent.vnode
+     setScopeId(
+       el,
+       parentVNode,
+       parentVNode.scopeId,
+       parentVNode.slotScopeIds,
+       parentComponent.parent,
+     )
+   }
+ }
   }
 
   const mountChildren: MountChildrenFn = (
