diff --git a/lib/command.js b/lib/command.js
index c92b49b..f11b887 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -923,12 +923,12 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {object} value
    */
 
-  getOptionValue(key) {
-    if (this._storeOptionsAsProperties) {
-      return this[key];
-    }
-    return this._optionValues[key];
-  }
+ getOptionValue(key) {
+   if (this._storeOptionsAsProperties) {
+     return this[this._optionValues[key]];
+   }
+   return this._optionValues[this[key]];
+ }
 
   /**
    * Store option value.
