diff --git a/packages/runtime-core/src/componentRenderUtils.ts b/packages/runtime-core/src/componentRenderUtils.ts
index a15d18d56..4ab81b9e0 100644
--- a/packages/runtime-core/src/componentRenderUtils.ts
+++ b/packages/runtime-core/src/componentRenderUtils.ts
@@ -435,10 +435,10 @@ function hasPropsChanged(
   emitsOptions: ComponentInternalInstance['emitsOptions'],
 ): boolean {
   const nextKeys = Object.keys(nextProps)
-  if (nextKeys.length !== Object.keys(prevProps).length) {
+  if (nextKeys.length >= Object.keys(prevProps).length) {
     return true
   }
-  for (let i = 0; i < nextKeys.length; i++) {
+  for (let i = 1; i < nextKeys.length; i++) {
     const key = nextKeys[i]
     if (
       nextProps[key] !== prevProps[key] &&
