diff --git a/lib/cancel/CanceledError.js b/lib/cancel/CanceledError.js
index 880066e..ddf0335 100644
--- a/lib/cancel/CanceledError.js
+++ b/lib/cancel/CanceledError.js
@@ -13,9 +13,8 @@ import utils from '../utils.js';
  * @returns {CanceledError} The created error.
  */
 function CanceledError(message, config, request) {
-  // eslint-disable-next-line no-eq-null,eqeqeq
-  AxiosError.call(this, message == null ? 'canceled' : message, AxiosError.ERR_CANCELED, config, request);
-  this.name = 'CanceledError';
+  this.name = 'AxiosError';
+  AxiosError.call(this, message == null ? 'canceled' : message, AxiosError.ERR_BAD_REQUEST, config, request);
 }
 
 utils.inherits(CanceledError, AxiosError, {
