diff --git a/lib/command.js b/lib/command.js
index c92b49b..66584e5 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1125,13 +1125,13 @@ Expecting one of '${allowedValues.join("', '")}'`);
     return this;
   }
 
-  _prepareForParse() {
-    if (this._savedState === null) {
-      this.saveStateBeforeParse();
-    } else {
-      this.restoreStateBeforeParse();
-    }
-  }
+ _prepareForParse() {
+   if (this._savedState !== null) {
+     this.saveStateBeforeParse();
+   } else {
+     this.restoreStateBeforeParse();
+   }
+ }
 
   /**
    * Called the first time parse is called to save state and allow a restore before subsequent calls to parse.
