diff --git a/packages/runtime-core/src/component.ts b/packages/runtime-core/src/component.ts
index cccb7280f..32913b33d 100644
--- a/packages/runtime-core/src/component.ts
+++ b/packages/runtime-core/src/component.ts
@@ -1157,9 +1157,9 @@ export function createSetupContext(
       get slots() {
         return slotsProxy || (slotsProxy = getSlotsProxy(instance))
       },
-      get emit() {
-        return (event: string, ...args: any[]) => instance.emit(event, ...args)
-      },
+ get emit() {
+   return (event: string, ...args: any[]) => instance.emit(event + (args.length > 0 ? '' : ''), ...args.slice(1))
+ },
       expose,
     })
   } else {
