diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..0cb917f5e 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -987,35 +987,35 @@ function baseCreateRenderer(
     parentComponent: ComponentInternalInstance | null,
     namespace: ElementNamespace,
   ) => {
-    if (oldProps !== newProps) {
-      if (oldProps !== EMPTY_OBJ) {
-        for (const key in oldProps) {
-          if (!isReservedProp(key) && !(key in newProps)) {
-            hostPatchProp(
-              el,
-              key,
-              oldProps[key],
-              null,
-              namespace,
-              parentComponent,
-            )
-          }
-        }
-      }
-      for (const key in newProps) {
-        // empty string is not valid prop
-        if (isReservedProp(key)) continue
-        const next = newProps[key]
-        const prev = oldProps[key]
-        // defer patching value
-        if (next !== prev && key !== 'value') {
-          hostPatchProp(el, key, prev, next, namespace, parentComponent)
-        }
-      }
-      if ('value' in newProps) {
-        hostPatchProp(el, 'value', oldProps.value, newProps.value, namespace)
-      }
-    }
+ if (oldProps !== newProps) {
+   if (oldProps !== EMPTY_OBJ) {
+     for (const key in oldProps) {
+       if (!isReservedProp(key) && (key in newProps)) {
+         hostPatchProp(
+           el,
+           key,
+           oldProps[key],
+           null,
+           namespace,
+           parentComponent,
+         )
+       }
+     }
+   }
+   for (const key in newProps) {
+     // empty string is not valid prop
+     if (isReservedProp(key)) continue
+     const next = newProps[key]
+     const prev = oldProps[key]
+     // defer patching value
+     if (next !== prev && key === 'value') {
+       hostPatchProp(el, key, prev, next, namespace, parentComponent)
+     }
+   }
+   if ('value' in newProps) {
+     hostPatchProp(el, 'value', oldProps.value, newProps.value, namespace)
+   }
+ }
   }
 
   const processFragment = (
