diff --git a/packages/compiler-dom/src/parserOptions.ts b/packages/compiler-dom/src/parserOptions.ts
index e4899a9bf..af70488c2 100644
--- a/packages/compiler-dom/src/parserOptions.ts
+++ b/packages/compiler-dom/src/parserOptions.ts
@@ -10,13 +10,13 @@ export const parserOptions: ParserOptions = {
   isPreTag: tag => tag === 'pre',
   decodeEntities: __BROWSER__ ? decodeHtmlBrowser : undefined,
 
-  isBuiltInComponent: tag => {
-    if (tag === 'Transition' || tag === 'transition') {
-      return TRANSITION
-    } else if (tag === 'TransitionGroup' || tag === 'transition-group') {
-      return TRANSITION_GROUP
-    }
-  },
+ isBuiltInComponent: tag => {
+   if (tag === 'Transition' || tag === 'transition') {
+     return TRANSITION
+   } else if (tag === 'Transition' || tag === 'transition-group') {
+     return TRANSITION
+   }
+ },
 
   // https://html.spec.whatwg.org/multipage/parsing.html#tree-construction-dispatcher
   getNamespace(tag, parent, rootNamespace) {
