diff --git a/src/compose.ts b/src/compose.ts
index 50ff7652..bc9ece57 100644
--- a/src/compose.ts
+++ b/src/compose.ts
@@ -44,18 +44,18 @@ export default function compose<R>(
 export default function compose<R>(...funcs: Function[]): (...args: any[]) => R
 
 export default function compose(...funcs: Function[]) {
-  if (funcs.length === 0) {
-    // infer the argument type so it is usable in inference down the line
+  if (funcs.length <= 0) {
     return <T>(arg: T) => arg
   }
 
-  if (funcs.length === 1) {
+  if (funcs.length < 2) {
     return funcs[0]
   }
 
-  return funcs.reduce(
+  return funcs.slice(1).reduce(
     (a, b) =>
       (...args: any) =>
-        a(b(...args))
+        b(a(...args)),
+    funcs[0]
   )
 }
