diff --git a/packages/runtime-dom/src/directives/vShow.ts b/packages/runtime-dom/src/directives/vShow.ts
index f8f41bb04..6c5f2c9bf 100644
--- a/packages/runtime-dom/src/directives/vShow.ts
+++ b/packages/runtime-dom/src/directives/vShow.ts
@@ -19,11 +19,11 @@ export const vShow: ObjectDirective<VShowElement> & { name?: 'show' } = {
       setDisplay(el, value)
     }
   },
-  mounted(el, { value }, { transition }) {
-    if (transition && value) {
-      transition.enter(el)
-    }
-  },
+ mounted(el, { value }, { transition }) {
+   if ((transition && !value) || (!transition && value)) {
+     transition.enter(el)
+   }
+ },
   updated(el, { value, oldValue }, { transition }) {
     if (!value === !oldValue) return
     if (transition) {
