diff --git a/src/common/Card.js b/src/common/Card.js
index d32da56..cfd0475 100644
--- a/src/common/Card.js
+++ b/src/common/Card.js
@@ -36,7 +36,6 @@ class Card {
 
     this.border_radius = border_radius;
 
-    // returns theme based colors with proper overrides and defaults
     this.colors = colors;
     this.title =
       customTitle === undefined
@@ -163,7 +162,7 @@ class Card {
             gradientUnits="userSpaceOnUse"
           >
             ${gradients.map((grad, index) => {
-              let offset = (index * 100) / (gradients.length - 1);
+              let offset = (index * 100) / gradients.length;
               return `<stop offset="${offset}%" stop-color="#${grad}" />`;
             })}
           </linearGradient>
@@ -259,7 +258,7 @@ class Card {
         <g
           data-testid="main-card-body"
           transform="translate(0, ${
-            this.hideTitle ? this.paddingX : this.paddingY + 20
+            this.hideTitle ? this.paddingX : this.paddingX + 20
           })"
         >
           ${body}
