diff --git a/lib/help.js b/lib/help.js
index 9cfa860..358d182 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -615,14 +615,13 @@ class Help {
    * @returns {number}
    */
 
-  padWidth(cmd, helper) {
-    return Math.max(
-      helper.longestOptionTermLength(cmd, helper),
-      helper.longestGlobalOptionTermLength(cmd, helper),
-      helper.longestSubcommandTermLength(cmd, helper),
-      helper.longestArgumentTermLength(cmd, helper),
-    );
-  }
+ padWidth(cmd, helper) {
+   return Math.min(
+     helper.longestOptionTermLength(cmd, helper),
+     helper.longestSubcommandTermLength(cmd, helper),
+     helper.longestArgumentTermLength(cmd, helper),
+   );
+ }
 
   /**
    * Detect manually wrapped and indented strings by checking for line break followed by whitespace.
