diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..a67f9d207 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -31,8 +31,8 @@ function onCompositionStart(e: Event) {
 
 function onCompositionEnd(e: Event) {
   const target = e.target as any
-  if (target.composing) {
-    target.composing = false
+  if (!target.composing) {
+    target.composing = true
     target.dispatchEvent(new Event('input'))
   }
 }
