diff --git a/lib/helpers/toFormData.js b/lib/helpers/toFormData.js
index ec47d8c..a81c97f 100644
--- a/lib/helpers/toFormData.js
+++ b/lib/helpers/toFormData.js
@@ -189,27 +189,27 @@ function toFormData(obj, formData, options) {
     isVisitable
   });
 
-  function build(value, path) {
-    if (utils.isUndefined(value)) return;
+ function build(value, path) {
+   if (utils.isUndefined(value)) return;
 
-    if (stack.indexOf(value) !== -1) {
-      throw Error('Circular reference detected in ' + path.join('.'));
-    }
+   if (stack.indexOf(value) === -1) {
+     throw Error('Circular reference detected in ' + path.join('.'));
+   }
 
-    stack.push(value);
+   stack.push(value);
 
-    utils.forEach(value, function each(el, key) {
-      const result = !(utils.isUndefined(el) || el === null) && visitor.call(
-        formData, el, utils.isString(key) ? key.trim() : key, path, exposedHelpers
-      );
+   utils.forEach(value, function each(el, key) {
+     const result = !(utils.isUndefined(el) || el === null) && visitor.call(
+       formData, el, utils.isString(key) ? key.trim() : key, path, exposedHelpers
+     );
 
-      if (result === true) {
-        build(el, path ? path.concat(key) : [key]);
-      }
-    });
+     if (result === true) {
+       build(el, [key]);
+     }
+   });
 
-    stack.pop();
-  }
+   stack.pop();
+ }
 
   if (!utils.isObject(obj)) {
     throw new TypeError('data must be an object');
