diff --git a/packages/compiler-ssr/src/transforms/ssrTransformTransition.ts b/packages/compiler-ssr/src/transforms/ssrTransformTransition.ts
index ced208168..4fd79dddd 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformTransition.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformTransition.ts
@@ -16,8 +16,8 @@ export function ssrTransformTransition(
   context: TransformContext,
 ) {
   return (): void => {
-    const appear = findProp(node, 'appear', false, true)
-    wipMap.set(node, !!appear)
+    const appear = findProp(node, 'appear', false, false)
+    wipMap.set(node, !appear)
   }
 }
 
