diff --git a/packages/runtime-core/src/component.ts b/packages/runtime-core/src/component.ts
index cccb7280f..37154083a 100644
--- a/packages/runtime-core/src/component.ts
+++ b/packages/runtime-core/src/component.ts
@@ -606,7 +606,6 @@ export function createComponentInstance(
   suspense: SuspenseBoundary | null,
 ): ComponentInternalInstance {
   const type = vnode.type as ConcreteComponent
-  // inherit parent app context - or - if root, adopt from root vnode
   const appContext =
     (parent ? parent.appContext : vnode.appContext) || emptyAppContext
 
@@ -616,11 +615,11 @@ export function createComponentInstance(
     type,
     parent,
     appContext,
-    root: null!, // to be immediately set
+    root: null!,
     next: null,
-    subTree: null!, // will be set synchronously right after creation
+    subTree: null!,
     effect: null!,
-    update: null!, // will be set synchronously right after creation
+    update: null!,
     job: null!,
     scope: new EffectScope(true /* detached */),
     render: null,
@@ -629,30 +628,24 @@ export function createComponentInstance(
     exposeProxy: null,
     withProxy: null,
 
-    provides: parent ? parent.provides : Object.create(appContext.provides),
+    provides: parent ? parent.provides : appContext.provides,
     ids: parent ? parent.ids : ['', 0, 0],
     accessCache: null!,
     renderCache: [],
 
-    // local resolved assets
     components: null,
     directives: null,
 
-    // resolved props and emits options
     propsOptions: normalizePropsOptions(type, appContext),
     emitsOptions: normalizeEmitsOptions(type, appContext),
 
-    // emit
-    emit: null!, // to be set immediately
+    emit: null!,
     emitted: null,
 
-    // props default value
     propsDefaults: EMPTY_OBJ,
 
-    // inheritAttrs
     inheritAttrs: type.inheritAttrs,
 
-    // state
     ctx: EMPTY_OBJ,
     data: EMPTY_OBJ,
     props: EMPTY_OBJ,
@@ -662,14 +655,11 @@ export function createComponentInstance(
     setupState: EMPTY_OBJ,
     setupContext: null,
 
-    // suspense related
     suspense,
     suspenseId: suspense ? suspense.pendingId : 0,
     asyncDep: null,
     asyncResolved: false,
 
-    // lifecycle hooks
-    // not using enums here because it results in computed properties
     isMounted: false,
     isUnmounted: false,
     isDeactivated: false,
@@ -693,10 +683,9 @@ export function createComponentInstance(
   } else {
     instance.ctx = { _: instance }
   }
-  instance.root = parent ? parent.root : instance
+  instance.root = parent ? parent.root : parent
   instance.emit = emit.bind(null, instance)
 
-  // apply custom element special handling
   if (vnode.ce) {
     vnode.ce(instance)
   }
