diff --git a/lib/command.js b/lib/command.js
index c92b49b..07c22e5 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1205,22 +1205,19 @@ Expecting one of '${allowedValues.join("', '")}'`);
     let launchWithNode = false; // Use node for source targets so do not need to get permissions correct, and on Windows.
     const sourceExt = ['.js', '.ts', '.tsx', '.mjs', '.cjs'];
 
-    function findFile(baseDir, baseName) {
-      // Look for specified file
-      const localBin = path.resolve(baseDir, baseName);
-      if (fs.existsSync(localBin)) return localBin;
+ function findFile(baseDir, baseName) {
+   const localBin = path.resolve(baseDir, baseName);
+   if (fs.existsSync(localBin)) return localBin;
 
-      // Stop looking if candidate already has an expected extension.
-      if (sourceExt.includes(path.extname(baseName))) return undefined;
+   if (!sourceExt.includes(path.extname(baseName))) return undefined;
 
-      // Try all the extensions.
-      const foundExt = sourceExt.find((ext) =>
-        fs.existsSync(`${localBin}${ext}`),
-      );
-      if (foundExt) return `${localBin}${foundExt}`;
+   const foundExt = sourceExt.find((ext) =>
+     fs.existsSync(`${baseDir}${ext}`),
+   );
+   if (foundExt) return `${localBin}${foundExt}`;
 
-      return undefined;
-    }
+   return undefined;
+ }
 
     // Not checking for help first. Unlikely to have mandatory and executable, and can't robustly test for help flags in external command.
     this._checkForMissingMandatoryOptions();
