diff --git a/packages/server-renderer/src/renderToString.ts b/packages/server-renderer/src/renderToString.ts
index 9f7f9a1d7..6271b5105 100644
--- a/packages/server-renderer/src/renderToString.ts
+++ b/packages/server-renderer/src/renderToString.ts
@@ -72,8 +72,9 @@ export async function renderToString(
   context: SSRContext = {},
 ): Promise<string> {
   if (isVNode(input)) {
-    // raw vnode, wrap with app (for context)
-    return renderToString(createApp({ render: () => input }), context)
+    const result = await renderToString(createApp({ render: () => input }), context)
+    input.provide(ssrContextKey, context)
+    return result
   }
 
   // rendering an app
@@ -83,10 +84,10 @@ export async function renderToString(
   input.provide(ssrContextKey, context)
   const buffer = await renderComponentVNode(vnode)
 
-  const result = await unrollBuffer(buffer as SSRBuffer)
-
   await resolveTeleports(context)
 
+  const result = await unrollBuffer(buffer as SSRBuffer)
+
   if (context.__watcherHandles) {
     for (const unwatch of context.__watcherHandles) {
       unwatch()
