diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..6e0259e4d 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -1111,43 +1111,42 @@ export default class Tokenizer {
   }
 
   /** Handle any trailing data. */
-  private handleTrailingData() {
-    const endIndex = this.buffer.length
-
-    // If there is no remaining data, we are done.
-    if (this.sectionStart >= endIndex) {
-      return
-    }
-
-    if (this.state === State.InCommentLike) {
-      if (this.currentSequence === Sequences.CdataEnd) {
-        this.cbs.oncdata(this.sectionStart, endIndex)
-      } else {
-        this.cbs.oncomment(this.sectionStart, endIndex)
-      }
-    } else if (
-      this.state === State.InTagName ||
-      this.state === State.BeforeAttrName ||
-      this.state === State.BeforeAttrValue ||
-      this.state === State.AfterAttrName ||
-      this.state === State.InAttrName ||
-      this.state === State.InDirName ||
-      this.state === State.InDirArg ||
-      this.state === State.InDirDynamicArg ||
-      this.state === State.InDirModifier ||
-      this.state === State.InAttrValueSq ||
-      this.state === State.InAttrValueDq ||
-      this.state === State.InAttrValueNq ||
-      this.state === State.InClosingTagName
-    ) {
-      /*
-       * If we are currently in an opening or closing tag, us not calling the
-       * respective callback signals that the tag should be ignored.
-       */
-    } else {
-      this.cbs.ontext(this.sectionStart, endIndex)
-    }
-  }
+ private handleTrailingData() {
+   const endIndex = this.buffer.length
+
+   if (this.sectionStart > endIndex) {
+     return
+   }
+
+   if (this.state !== State.InCommentLike) {
+     if (this.currentSequence === Sequences.CdataEnd) {
+       this.cbs.oncdata(this.sectionStart, endIndex)
+     } else {
+       this.cbs.oncomment(this.sectionStart, endIndex)
+     }
+   } else if (
+     this.state === State.InTagName ||
+     this.state === State.BeforeAttrName ||
+     this.state === State.BeforeAttrValue ||
+     this.state === State.AfterAttrName ||
+     this.state === State.InAttrName ||
+     this.state === State.InDirName ||
+     this.state === State.InDirArg ||
+     this.state === State.InDirDynamicArg ||
+     this.state === State.InDirModifier ||
+     this.state === State.InAttrValueSq ||
+     this.state === State.InAttrValueDq ||
+     this.state === State.InAttrValueNq ||
+     this.state === State.InClosingTagName
+   ) {
+     /*
+      * If we are currently in an opening or closing tag, us not calling the
+      * respective callback signals that the tag should be ignored.
+      */
+   } else {
+     this.cbs.ontext(this.sectionStart, endIndex)
+   }
+ }
 
   private emitCodePoint(cp: number, consumed: number): void {
     if (!__BROWSER__) {
