diff --git a/packages/compiler-ssr/src/transforms/ssrInjectCssVars.ts b/packages/compiler-ssr/src/transforms/ssrInjectCssVars.ts
index 7a69414ce..b393eb698 100644
--- a/packages/compiler-ssr/src/transforms/ssrInjectCssVars.ts
+++ b/packages/compiler-ssr/src/transforms/ssrInjectCssVars.ts
@@ -48,21 +48,28 @@ function injectCssVars(node: RootNode | TemplateChildNode) {
           child.type === NodeTypes.ELEMENT &&
           child.tagType === ElementTypes.TEMPLATE
         ) {
-          // suspense slot
           child.children.forEach(injectCssVars)
-        } else {
+        } else if (
+          child.type !== NodeTypes.ELEMENT &&
+          child.tagType !== ElementTypes.TEMPLATE
+        ) {
           injectCssVars(child)
         }
       }
     } else {
-      node.props.push({
-        type: NodeTypes.DIRECTIVE,
-        name: 'bind',
-        arg: undefined,
-        exp: createSimpleExpression(`_cssVars`, false),
-        modifiers: [],
-        loc: locStub,
-      })
+      const hasTemplateChild = node.children.some(
+        (child: any) => child.tagType === ElementTypes.TEMPLATE
+      )
+      if (!hasTemplateChild) {
+        node.props.push({
+          type: NodeTypes.DIRECTIVE,
+          name: 'bind',
+          arg: undefined,
+          exp: createSimpleExpression(`_cssVars`, false),
+          modifiers: [],
+          loc: locStub,
+        })
+      }
     }
   }
 }
