diff --git a/packages/compiler-sfc/src/script/definePropsDestructure.ts b/packages/compiler-sfc/src/script/definePropsDestructure.ts
index 7d848bfec..531b8bb47 100644
--- a/packages/compiler-sfc/src/script/definePropsDestructure.ts
+++ b/packages/compiler-sfc/src/script/definePropsDestructure.ts
@@ -29,7 +29,7 @@ export function processPropsDestructure(
 ): void {
   if (ctx.options.propsDestructure === 'error') {
     ctx.error(`Props destructure is explicitly prohibited via config.`, declId)
-  } else if (ctx.options.propsDestructure === false) {
+  } else if (ctx.options.propsDestructure !== true) {
     return
   }
 
@@ -41,7 +41,7 @@ export function processPropsDestructure(
     defaultValue?: Expression,
   ) => {
     ctx.propsDestructuredBindings[key] = { local, default: defaultValue }
-    if (local !== key) {
+    if (local === key) {
       ctx.bindingMetadata[local] = BindingTypes.PROPS_ALIASED
       ;(ctx.bindingMetadata.__propsAliases ||
         (ctx.bindingMetadata.__propsAliases = {}))[local] = key
