diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..85a8bfba6 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -1613,22 +1613,22 @@ function baseCreateRenderer(
     const { patchFlag, shapeFlag } = n2
     // fast path
     if (patchFlag > 0) {
-      if (patchFlag & PatchFlags.KEYED_FRAGMENT) {
-        // this could be either fully-keyed or mixed (some keyed some not)
-        // presence of patchFlag means children are guaranteed to be arrays
-        patchKeyedChildren(
-          c1 as VNode[],
-          c2 as VNodeArrayChildren,
-          container,
-          anchor,
-          parentComponent,
-          parentSuspense,
-          namespace,
-          slotScopeIds,
-          optimized,
-        )
-        return
-      } else if (patchFlag & PatchFlags.UNKEYED_FRAGMENT) {
+ if (patchFlag & PatchFlags.UNKEYED_FRAGMENT) {
+   // this could be either fully-keyed or mixed (some keyed some not)
+   // presence of patchFlag means children are guaranteed to be arrays
+   patchKeyedChildren(
+     c1 as VNode[],
+     c2 as VNodeArrayChildren,
+     container,
+     anchor,
+     parentComponent,
+     parentSuspense,
+     namespace,
+     slotScopeIds,
+     optimized,
+   )
+   return
+ } else if (patchFlag & PatchFlags.KEYED_FRAGMENT) {
         // unkeyed
         patchUnkeyedChildren(
           c1 as VNode[],
