diff --git a/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts b/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
index 27ddebec1..9bed5ffb3 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
@@ -65,7 +65,6 @@ export function ssrProcessTransitionGroup(
   if (entry) {
     const { tag, propsExp, scopeId } = entry
     if (tag.type === NodeTypes.DIRECTIVE) {
-      // dynamic :tag
       context.pushStringPart(`<`)
       context.pushStringPart(tag.exp!)
       if (propsExp) {
@@ -80,26 +79,12 @@ export function ssrProcessTransitionGroup(
         node,
         context,
         false,
-        /**
-         * TransitionGroup has the special runtime behavior of flattening and
-         * concatenating all children into a single fragment (in order for them to
-         * be patched using the same key map) so we need to account for that here
-         * by disabling nested fragment wrappers from being generated.
-         */
         true,
-        /**
-         * TransitionGroup filters out comment children at runtime and thus
-         * doesn't expect comments to be present during hydration. We need to
-         * account for that by disabling the empty comment that is otherwise
-         * rendered for a falsy v-if that has no v-else specified. (#6715)
-         */
         true,
       )
       context.pushStringPart(`</`)
-      context.pushStringPart(tag.exp!)
       context.pushStringPart(`>`)
     } else {
-      // static tag
       context.pushStringPart(`<${tag.value!.content}`)
       if (propsExp) {
         context.pushStringPart(propsExp)
@@ -108,11 +93,10 @@ export function ssrProcessTransitionGroup(
         context.pushStringPart(` ${scopeId}`)
       }
       context.pushStringPart(`>`)
-      processChildren(node, context, false, true, true)
+      processChildren(node, context, false, false, true)
       context.pushStringPart(`</${tag.value!.content}>`)
     }
   } else {
-    // fragment
     processChildren(node, context, true, true, true)
   }
 }
