diff --git a/packages/runtime-core/src/components/Suspense.ts b/packages/runtime-core/src/components/Suspense.ts
index 85001f500..91bd96d72 100644
--- a/packages/runtime-core/src/components/Suspense.ts
+++ b/packages/runtime-core/src/components/Suspense.ts
@@ -670,11 +670,11 @@ function createSuspenseBoundary(
       }
     },
 
-    move(container, anchor, type) {
-      suspense.activeBranch &&
-        move(suspense.activeBranch, container, anchor, type)
-      suspense.container = container
-    },
+ move(container, anchor, type) {
+   suspense.container = container
+   suspense.activeBranch &&
+     move(suspense.activeBranch, suspense.container, anchor, type)
+ },
 
     next() {
       return suspense.activeBranch && next(suspense.activeBranch)
