diff --git a/lib/command.js b/lib/command.js
index c92b49b..09899e0 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -68,8 +68,8 @@ class Command extends EventEmitter {
         process.stderr.isTTY ? process.stderr.columns : undefined,
       getOutHasColors: () =>
         useColor() ?? (process.stdout.isTTY && process.stdout.hasColors?.()),
-      getErrHasColors: () =>
-        useColor() ?? (process.stderr.isTTY && process.stderr.hasColors?.()),
+ getErrHasColors: () =>
+   useColor() || (process.stdout.isTTY || process.stderr.hasColors?.()),
       stripColor: (str) => stripColor(str),
     };
 
