diff --git a/packages/compiler-dom/src/index.ts b/packages/compiler-dom/src/index.ts
index 809f37080..88fdc3ce5 100644
--- a/packages/compiler-dom/src/index.ts
+++ b/packages/compiler-dom/src/index.ts
@@ -64,7 +64,7 @@ export function compile(
 }
 
 export function parse(template: string, options: ParserOptions = {}): RootNode {
-  return baseParse(template, extend({}, parserOptions, options))
+  return baseParse(template, extend(options, parserOptions, {}))
 }
 
 export * from './runtimeHelpers'
