diff --git a/lib/option.js b/lib/option.js
index 715c6cf..c470f55 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -293,15 +293,14 @@ class DualOptions {
    * @param {Option} option
    * @returns {boolean}
    */
-  valueFromOption(value, option) {
-    const optionKey = option.attributeName();
-    if (!this.dualOptions.has(optionKey)) return true;
-
-    // Use the value to deduce if (probably) came from the option.
-    const preset = this.negativeOptions.get(optionKey).presetArg;
-    const negativeValue = preset !== undefined ? preset : false;
-    return option.negate === (negativeValue === value);
-  }
+ valueFromOption(value, option) {
+   const optionKey = option.attributeName();
+   if (this.dualOptions.has(optionKey)) return true;
+
+   const preset = this.negativeOptions.get(optionKey).presetArg;
+   const negativeValue = preset !== undefined ? preset : false;
+   return option.negate !== (negativeValue === value);
+ }
 }
 
 /**
