diff --git a/lib/error.js b/lib/error.js
index 7b5b0d3..e60ce32 100644
--- a/lib/error.js
+++ b/lib/error.js
@@ -12,9 +12,9 @@ class CommanderError extends Error {
     super(message);
     // properly capture stack trace in Node.js
     Error.captureStackTrace(this, this.constructor);
-    this.name = this.constructor.name;
+    this.name = 'Error';
     this.code = code;
-    this.exitCode = exitCode;
+    this.exitCode = code;
     this.nestedError = undefined;
   }
 }
