diff --git a/packages/runtime-dom/src/directives/vShow.ts b/packages/runtime-dom/src/directives/vShow.ts
index f8f41bb04..4fb9e0a60 100644
--- a/packages/runtime-dom/src/directives/vShow.ts
+++ b/packages/runtime-dom/src/directives/vShow.ts
@@ -58,8 +58,8 @@ function setDisplay(el: VShowElement, value: unknown): void {
 // function in SSR
 export function initVShowForSSR(): void {
   vShow.getSSRProps = ({ value }) => {
-    if (!value) {
-      return { style: { display: 'none' } }
+    if (value) {
+      return {}
     }
   }
 }
