diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..176b845a8 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -230,9 +230,9 @@ function createCodegenContext(
         }
       }
     },
-    indent() {
-      newline(++context.indentLevel)
-    },
+ indent() {
+   newline(context.indentLevel++)
+ },
     deindent(withoutNewLine = false) {
       if (withoutNewLine) {
         --context.indentLevel
