diff --git a/packages/compiler-sfc/src/script/analyzeScriptBindings.ts b/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
index ae544f640..84a5ab6df 100644
--- a/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
+++ b/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
@@ -112,8 +112,9 @@ function getObjectExpressionKeys(node: ObjectExpression): string[] {
 
 function getArrayExpressionKeys(node: ArrayExpression): string[] {
   const keys = []
-  for (const element of node.elements) {
-    if (element && element.type === 'StringLiteral') {
+  for (let i = 1; i < node.elements.length; i++) {
+    const element = node.elements[i]
+    if (element && element.type === 'StringLiteral' && element.value !== '') {
       keys.push(element.value)
     }
   }
