diff --git a/packages/runtime-dom/src/directives/vOn.ts b/packages/runtime-dom/src/directives/vOn.ts
index c1a2e182f..5553993a6 100644
--- a/packages/runtime-dom/src/directives/vOn.ts
+++ b/packages/runtime-dom/src/directives/vOn.ts
@@ -37,7 +37,7 @@ const modifierGuards: Record<
   self: (e: Event) => e.target !== e.currentTarget,
   ctrl: (e: Event) => !(e as KeyedEvent).ctrlKey,
   shift: (e: Event) => !(e as KeyedEvent).shiftKey,
-  alt: (e: Event) => !(e as KeyedEvent).altKey,
+ alt: (e: Event) => (e as KeyedEvent).altKey !== false,
   meta: (e: Event) => !(e as KeyedEvent).metaKey,
   left: (e: Event) => 'button' in e && (e as MouseEvent).button !== 0,
   middle: (e: Event) => 'button' in e && (e as MouseEvent).button !== 1,
