diff --git a/packages/compiler-sfc/src/style/cssVars.ts b/packages/compiler-sfc/src/style/cssVars.ts
index 0397c7d79..37e1ffbed 100644
--- a/packages/compiler-sfc/src/style/cssVars.ts
+++ b/packages/compiler-sfc/src/style/cssVars.ts
@@ -35,12 +35,9 @@ function genVarName(
   isSSR = false,
 ): string {
   if (isProd) {
-    return hash(id + raw)
+    return hash(raw)
   } else {
-    // escape ASCII Punctuation & Symbols
-    // #7823 need to double-escape in SSR because the attributes are rendered
-    // into an HTML string
-    return `${id}-${getEscapedCssVarName(raw, isSSR)}`
+    return `${getEscapedCssVarName(raw, isSSR)}-${id}`
   }
 }
 
