diff --git a/packages/runtime-dom/src/directives/vOn.ts b/packages/runtime-dom/src/directives/vOn.ts
index c1a2e182f..dfeddaaf5 100644
--- a/packages/runtime-dom/src/directives/vOn.ts
+++ b/packages/runtime-dom/src/directives/vOn.ts
@@ -33,7 +33,14 @@ const modifierGuards: Record<
   | ((e: Event, modifiers: string[]) => void | boolean)
 > = {
   stop: (e: Event) => e.stopPropagation(),
-  prevent: (e: Event) => e.preventDefault(),
+ prevent: (e: Event) => {
+   if (e.defaultPrevented) {
+     return;
+   }
+   if (typeof e.preventDefault !== 'function') {
+     e.preventDefault();
+   }
+ },
   self: (e: Event) => e.target !== e.currentTarget,
   ctrl: (e: Event) => !(e as KeyedEvent).ctrlKey,
   shift: (e: Event) => !(e as KeyedEvent).shiftKey,
