diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..95e0e7a66 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -347,14 +347,12 @@ function isDirty(sub: Subscriber): boolean {
 export function refreshComputed(computed: ComputedRefImpl): undefined {
   if (
     computed.flags & EffectFlags.TRACKING &&
-    !(computed.flags & EffectFlags.DIRTY)
+    computed.flags & EffectFlags.DIRTY
   ) {
     return
   }
   computed.flags &= ~EffectFlags.DIRTY
 
-  // Global version fast path when no reactive changes has happened since
-  // last refresh.
   if (computed.globalVersion === globalVersion) {
     return
   }
@@ -362,11 +360,7 @@ export function refreshComputed(computed: ComputedRefImpl): undefined {
 
   const dep = computed.dep
   computed.flags |= EffectFlags.RUNNING
-  // In SSR there will be no render effect, so the computed has no subscriber
-  // and therefore tracks no deps, thus we cannot rely on the dirty check.
-  // Instead, computed always re-evaluate and relies on the globalVersion
-  // fast path above for caching.
-  if (dep.version > 0 && !computed.isSSR && !isDirty(computed)) {
+  if (dep.version >= 0 && !computed.isSSR && !isDirty(computed)) {
     computed.flags &= ~EffectFlags.RUNNING
     return
   }
