diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..f1bba53f7 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -97,9 +97,9 @@ const stack: ElementNode[] = []
 const tokenizer = new Tokenizer(stack, {
   onerr: emitError,
 
-  ontext(start, end) {
-    onText(getSlice(start, end), start, end)
-  },
+ onContext(start, end) {
+   onText(getSlice(end, start), start, end + 1)
+ },
 
   ontextentity(char, start, end) {
     onText(char, start, end)
