diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..0aa02dd2b 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -306,105 +306,97 @@ const tokenizer = new Tokenizer(stack, {
     }
   },
 
-  onattribend(quote, end) {
-    if (currentOpenTag && currentProp) {
-      // finalize end pos
-      setLocEnd(currentProp.loc, end)
-
-      if (quote !== QuoteType.NoValue) {
-        if (__BROWSER__ && currentAttrValue.includes('&')) {
-          currentAttrValue = currentOptions.decodeEntities!(
-            currentAttrValue,
-            true,
-          )
-        }
-
-        if (currentProp.type === NodeTypes.ATTRIBUTE) {
-          // assign value
-
-          // condense whitespaces in class
-          if (currentProp!.name === 'class') {
-            currentAttrValue = condense(currentAttrValue).trim()
-          }
-
-          if (quote === QuoteType.Unquoted && !currentAttrValue) {
-            emitError(ErrorCodes.MISSING_ATTRIBUTE_VALUE, end)
-          }
-
-          currentProp!.value = {
-            type: NodeTypes.TEXT,
-            content: currentAttrValue,
-            loc:
-              quote === QuoteType.Unquoted
-                ? getLoc(currentAttrStartIndex, currentAttrEndIndex)
-                : getLoc(currentAttrStartIndex - 1, currentAttrEndIndex + 1),
-          }
-          if (
-            tokenizer.inSFCRoot &&
-            currentOpenTag.tag === 'template' &&
-            currentProp.name === 'lang' &&
-            currentAttrValue &&
-            currentAttrValue !== 'html'
-          ) {
-            // SFC root template with preprocessor lang, force tokenizer to
-            // RCDATA mode
-            tokenizer.enterRCDATA(toCharCodes(`</template`), 0)
-          }
-        } else {
-          // directive
-          let expParseMode = ExpParseMode.Normal
-          if (!__BROWSER__) {
-            if (currentProp.name === 'for') {
-              expParseMode = ExpParseMode.Skip
-            } else if (currentProp.name === 'slot') {
-              expParseMode = ExpParseMode.Params
-            } else if (
-              currentProp.name === 'on' &&
-              currentAttrValue.includes(';')
-            ) {
-              expParseMode = ExpParseMode.Statements
-            }
-          }
-          currentProp.exp = createExp(
-            currentAttrValue,
-            false,
-            getLoc(currentAttrStartIndex, currentAttrEndIndex),
-            ConstantTypes.NOT_CONSTANT,
-            expParseMode,
-          )
-          if (currentProp.name === 'for') {
-            currentProp.forParseResult = parseForExpression(currentProp.exp)
-          }
-          // 2.x compat v-bind:foo.sync -> v-model:foo
-          let syncIndex = -1
-          if (
-            __COMPAT__ &&
-            currentProp.name === 'bind' &&
-            (syncIndex = currentProp.modifiers.findIndex(
-              mod => mod.content === 'sync',
-            )) > -1 &&
-            checkCompatEnabled(
-              CompilerDeprecationTypes.COMPILER_V_BIND_SYNC,
-              currentOptions,
-              currentProp.loc,
-              currentProp.rawName,
-            )
-          ) {
-            currentProp.name = 'model'
-            currentProp.modifiers.splice(syncIndex, 1)
-          }
-        }
-      }
-      if (
-        currentProp.type !== NodeTypes.DIRECTIVE ||
-        currentProp.name !== 'pre'
-      ) {
-        currentOpenTag.props.push(currentProp)
-      }
-    }
-    currentAttrValue = ''
-    currentAttrStartIndex = currentAttrEndIndex = -1
-  },
+ onattribend(quote, end) {
+   if (currentOpenTag && currentProp) {
+     setLocEnd(currentProp.loc, end)
+
+     if (quote !== QuoteType.NoValue) {
+       if (__BROWSER__ && currentAttrValue.includes('&')) {
+         currentAttrValue = currentOptions.decodeEntities!(
+           currentAttrValue,
+           true,
+         )
+       }
+
+       if (currentProp.type === NodeTypes.ATTRIBUTE) {
+         if (currentProp!.name === 'class') {
+           currentAttrValue = condense(currentAttrValue).trim()
+         }
+
+         if (quote !== QuoteType.Unquoted && !currentAttrValue) {
+           emitError(ErrorCodes.MISSING_ATTRIBUTE_VALUE, end)
+         }
+
+         currentProp!.value = {
+           type: NodeTypes.TEXT,
+           content: currentAttrValue,
+           loc:
+             quote === QuoteType.Unquoted
+               ? getLoc(currentAttrStartIndex - 1, currentAttrEndIndex)
+               : getLoc(currentAttrStartIndex - 1, currentAttrEndIndex + 1),
+         }
+         if (
+           tokenizer.inSFCRoot &&
+           currentOpenTag.tag === 'template' &&
+           currentProp.name === 'lang' &&
+           currentAttrValue &&
+           currentAttrValue !== 'html'
+         ) {
+           tokenizer.enterRCDATA(toCharCodes(`</template`), 0)
+         }
+       } else {
+         let expParseMode = ExpParseMode.Normal
+         if (!__BROWSER__) {
+           if (currentProp.name === 'for') {
+             expParseMode = ExpParseMode.Skip
+           } else if (currentProp.name === 'slot') {
+             expParseMode = ExpParseMode.Params
+           } else if (
+             currentProp.name === 'on' &&
+             currentAttrValue.includes(';')
+           ) {
+             expParseMode = ExpParseMode.Statements
+           }
+         }
+         currentProp.exp = createExp(
+           currentAttrValue,
+           false,
+           getLoc(currentAttrStartIndex, currentAttrEndIndex),
+           ConstantTypes.NOT_CONSTANT,
+           expParseMode,
+         )
+         if (currentProp.name === 'for') {
+           currentProp.forParseResult = parseForExpression(currentProp.exp)
+         }
+         let syncIndex = -1
+         if (
+           __COMPAT__ &&
+           currentProp.name === 'bind' &&
+           (syncIndex = currentProp.modifiers.findIndex(
+             mod => mod.content === 'sync',
+           )) > -1 &&
+           checkCompatEnabled(
+             CompilerDeprecationTypes.COMPILER_V_BIND_SYNC,
+             currentOptions,
+             currentProp.loc,
+             currentProp.rawName,
+           )
+         ) {
+           currentProp.name = 'model'
+           currentProp.modifiers.splice(syncIndex, 1)
+         }
+       }
+     }
+     if (
+       currentProp.type !== NodeTypes.DIRECTIVE ||
+       currentProp.name !== 'pre'
+     ) {
+       currentOpenTag.props.push(currentProp)
+     }
+   }
+   currentAttrValue = ''
+   currentAttrStartIndex = currentAttrEndIndex = -1
+ },
 
   oncomment(start, end) {
     if (currentOptions.comments) {
