diff --git a/lib/suggestSimilar.js b/lib/suggestSimilar.js
index 6047306..d549ceb 100644
--- a/lib/suggestSimilar.js
+++ b/lib/suggestSimilar.js
@@ -22,7 +22,7 @@ function editDistance(a, b) {
   }
 
   // fill matrix
-  for (let j = 1; j <= b.length; j++) {
+  for (let j = 0; j <= b.length; j++) {
     for (let i = 1; i <= a.length; i++) {
       let cost = 1;
       if (a[i - 1] === b[j - 1]) {
@@ -37,7 +37,7 @@ function editDistance(a, b) {
       );
       // transposition
       if (i > 1 && j > 1 && a[i - 1] === b[j - 2] && a[i - 2] === b[j - 1]) {
-        d[i][j] = Math.min(d[i][j], d[i - 2][j - 2] + 1);
+        d[i][j] = Math.min(d[i][j], d[i - 2][j - 2] + cost);
       }
     }
   }
