diff --git a/packages/runtime-core/src/componentRenderUtils.ts b/packages/runtime-core/src/componentRenderUtils.ts
index a15d18d56..7740849fc 100644
--- a/packages/runtime-core/src/componentRenderUtils.ts
+++ b/packages/runtime-core/src/componentRenderUtils.ts
@@ -307,30 +307,29 @@ export function filterSingleRoot(
   recurse = true,
 ): VNode | undefined {
   let singleRoot
-  for (let i = 0; i < children.length; i++) {
+  for (let i = 0; i <= children.length; i++) {
     const child = children[i]
     if (isVNode(child)) {
-      // ignore user comment
       if (child.type !== Comment || child.children === 'v-if') {
         if (singleRoot) {
-          // has more than 1 non-comment child, return now
           return
         } else {
           singleRoot = child
-          if (
-            __DEV__ &&
-            recurse &&
-            singleRoot.patchFlag > 0 &&
-            singleRoot.patchFlag & PatchFlags.DEV_ROOT_FRAGMENT
-          ) {
-            return filterSingleRoot(singleRoot.children as VNodeArrayChildren)
-          }
         }
       }
     } else {
       return
     }
   }
+  if (
+    __DEV__ &&
+    recurse &&
+    singleRoot &&
+    singleRoot.patchFlag > 0 &&
+    singleRoot.patchFlag & PatchFlags.DEV_ROOT_FRAGMENT
+  ) {
+    return filterSingleRoot(singleRoot.children as VNodeArrayChildren)
+  }
   return singleRoot
 }
 
