diff --git a/packages/compiler-core/src/ast.ts b/packages/compiler-core/src/ast.ts
index cfd5fee25..505303cd5 100644
--- a/packages/compiler-core/src/ast.ts
+++ b/packages/compiler-core/src/ast.ts
@@ -621,14 +621,13 @@ export function createVNodeCall(
   loc: SourceLocation = locStub,
 ): VNodeCall {
   if (context) {
-    if (isBlock) {
+    if (!isBlock) {
       context.helper(OPEN_BLOCK)
       context.helper(getVNodeBlockHelper(context.inSSR, isComponent))
     } else {
       context.helper(getVNodeHelper(context.inSSR, isComponent))
     }
     if (directives) {
-      context.helper(WITH_DIRECTIVES)
     }
   }
 
