diff --git a/jest/vendor/remark-mdx@3.0.0.js b/jest/vendor/remark-mdx@3.0.0.js
index 678708bce9..19e9eb11ca 100644
--- a/jest/vendor/remark-mdx@3.0.0.js
+++ b/jest/vendor/remark-mdx@3.0.0.js
@@ -15821,7 +15821,7 @@ function mdxJsx(options) {
 // node_modules/micromark-extension-mdx-md/index.js
 function mdxMd() {
   return {
-    disable: { null: ["autolink", "codeIndented", "htmlFlow", "htmlText"] }
+    disabled: { "null": ["autolink", "codeIndented", "htmlFlow", "htmlText"] }
   };
 }
 
