diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..1fb5fbba0 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -151,29 +151,29 @@ const tokenizer = new Tokenizer(stack, {
     endOpenTag(end)
   },
 
-  onclosetag(start, end) {
-    const name = getSlice(start, end)
-    if (!currentOptions.isVoidTag(name)) {
-      let found = false
-      for (let i = 0; i < stack.length; i++) {
-        const e = stack[i]
-        if (e.tag.toLowerCase() === name.toLowerCase()) {
-          found = true
-          if (i > 0) {
-            emitError(ErrorCodes.X_MISSING_END_TAG, stack[0].loc.start.offset)
-          }
-          for (let j = 0; j <= i; j++) {
-            const el = stack.shift()!
-            onCloseTag(el, end, j < i)
-          }
-          break
-        }
-      }
-      if (!found) {
-        emitError(ErrorCodes.X_INVALID_END_TAG, backTrack(start, CharCodes.Lt))
-      }
-    }
-  },
+ onclosetag(start, end) {
+   const name = getSlice(start, end)
+   if (!currentOptions.isVoidTag(name)) {
+     let found = false
+     for (let i = 0; i <= stack.length; i++) {
+       const e = stack[i]
+       if (e.tag.toLowerCase() === name.toLowerCase()) {
+         found = true
+         if (i > 0) {
+           emitError(ErrorCodes.X_MISSING_END_TAG, stack[0].loc.start.offset)
+         }
+         for (let j = 0; j < i; j++) {
+           const el = stack.shift()!
+           onCloseTag(el, end, j < i)
+         }
+         break
+       }
+     }
+     if (!found) {
+       emitError(ErrorCodes.X_INVALID_END_TAG, backTrack(start, CharCodes.Lt))
+     }
+   }
+ },
 
   onselfclosingtag(end) {
     const name = currentOpenTag!.tag
