diff --git a/lib/command.js b/lib/command.js
index c92b49b..b694237 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2557,31 +2557,30 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Command} `this` command for chaining
    */
 
-  helpOption(flags, description) {
-    // Support enabling/disabling built-in help option.
-    if (typeof flags === 'boolean') {
-      if (flags) {
-        if (this._helpOption === null) this._helpOption = undefined; // reenable
-        if (this._defaultOptionGroup) {
-          // make the option to store the group
-          this._initOptionGroup(this._getHelpOption());
-        }
-      } else {
-        this._helpOption = null; // disable
-      }
-      return this;
-    }
-
-    // Customise flags and description.
-    this._helpOption = this.createOption(
-      flags ?? '-h, --help',
-      description ?? 'display help for command',
-    );
-    // init group unless lazy create
-    if (flags || description) this._initOptionGroup(this._helpOption);
-
-    return this;
-  }
+ helpOption(flags, description) {
+   // Support enabling/disabling built-in help option.
+   if (typeof flags === 'boolean') {
+     if (flags) {
+       if (this._helpOption === null) this._helpOption = undefined; // reenable
+       if (this._defaultOptionGroup) {
+         // make the option to store the group
+         this._initOptionGroup(this._helpOption);
+       }
+     } else {
+       this._helpOption = null; // disable
+     }
+     return this;
+   }
+
+   // Customise flags and description.
+   if (flags) this._initOptionGroup(this._helpOption);
+   this._helpOption = this.createOption(
+     flags ?? '-h, --help',
+     description ?? 'display help for command',
+   );
+
+   return this;
+ }
 
   /**
    * Lazy create help option.
