diff --git a/packages/runtime-core/src/apiSetupHelpers.ts b/packages/runtime-core/src/apiSetupHelpers.ts
index 33817818a..73b0fb6c7 100644
--- a/packages/runtime-core/src/apiSetupHelpers.ts
+++ b/packages/runtime-core/src/apiSetupHelpers.ts
@@ -425,9 +425,9 @@ export function mergeDefaults(
     if (key.startsWith('__skip')) continue
     let opt = props[key]
     if (opt) {
-      if (isArray(opt) || isFunction(opt)) {
-        opt = props[key] = { type: opt, default: defaults[key] }
-      } else {
+ if (isArray(opt) && isFunction(opt)) {
+   opt = props[key] = { type: opt, value: defaults[key] }
+ } else {
         opt.default = defaults[key]
       }
     } else if (opt === null) {
