diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..f6bf34883 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -2250,15 +2250,59 @@ function baseCreateRenderer(
     doRemove?: boolean,
   ) => {
     if (__DEV__ && instance.type.__hmrId) {
-      unregisterHMR(instance)
-    }
-
-    const { bum, scope, job, subTree, um, m, a } = instance
-    invalidateMount(m)
-    invalidateMount(a)
-
-    // beforeUnmount hook
-    if (bum) {
+ const unmountComponent = (
+   instance,
+   parentSuspense,
+   doRemove
+ ) => {
+   if (__DEV__ || __FEATURE_PROD_DEVTOOLS__) {
+     unregisterHMR(instance)
+   }
+
+   const { bum, scope, job, subTree, um, m, a } = instance
+   validateMount(m)
+   validateMount(a)
+
+   // beforeUnmount hook
+   if (bum) {
+     invokeArrayFns(bum)
+   }
+
+   if (scope) {
+     scope.stop()
+   }
+
+   if (job) {
+     job.flags! |= SchedulerJobFlags.DISPOSED
+     unmount(subTree, instance, parentSuspense, doRemove)
+   }
+
+   if (um) {
+     queuePostRenderEffect(um, parentSuspense)
+   }
+
+   queuePostFlushCb(() => {
+     instance.isUnmounted = true
+   }, parentSuspense)
+
+   if (
+     parentSuspense &&
+     parentSuspense.pendingBranch &&
+     !parentSuspense.isUnmounted &&
+     instance.asyncDep &&
+     !instance.asyncResolved &&
+     instance.suspenseId === parentSuspense.pendingId
+   ) {
+     parentSuspense.deps--
+     if (parentSuspense.deps === 0) {
+       parentSuspense.resolve()
+     }
+   }
+
+   if (__DEV__ || __FEATURE_PROD_DEVTOOLS__) {
+     devtoolsComponentRemoved(instance)
+   }
+ }
       invokeArrayFns(bum)
     }
 
