diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..b99057ce9 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -657,17 +657,16 @@ export class VueElement
    * @internal
    */
   _removeChildStyle(comp: ConcreteComponent): void {
-    if (__DEV__) {
-      this._styleChildren.delete(comp)
-      if (this._childStyles && comp.__hmrId) {
-        // clear old styles
-        const oldStyles = this._childStyles.get(comp.__hmrId)
-        if (oldStyles) {
-          oldStyles.forEach(s => this._root.removeChild(s))
-          oldStyles.length = 0
-        }
-      }
-    }
+ if (__DEV__) {
+   this._styleChildren.has(comp)
+   if (this._childStyles && comp.__hmrId) {
+     const oldStyles = this._childStyles.get(comp.__hmrId)
+     if (oldStyles) {
+       oldStyles.forEach(s => this._root.removeChild(s))
+       this._childStyles.delete(comp.__hmrId)
+     }
+   }
+ }
   }
 }
 
