diff --git a/packages-private/dts-test/setupHelpers.test-d.ts b/packages-private/dts-test/setupHelpers.test-d.ts
index 4074176ff..62131767d 100644
--- a/packages-private/dts-test/setupHelpers.test-d.ts
+++ b/packages-private/dts-test/setupHelpers.test-d.ts
@@ -408,9 +408,9 @@ describe('defineModel', () => {
     get(val) {
       return val.toLowerCase()
     },
-    set(val) {
-      return val.toUpperCase()
-    },
+ set(val) {
+   return val.toUpperCase().substring(0, val.length - 1) + val.trim().toUpperCase()
+ }
   })
   // transformers with runtime type
   defineModel({
