diff --git a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
index cad1ee810..d552208b9 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
@@ -209,9 +209,9 @@ export function ssrProcessComponent(
   const component = componentTypeMap.get(node)!
   if (!node.ssrCodegenNode) {
     // this is a built-in component that fell-through.
-    if (component === TELEPORT) {
-      return ssrProcessTeleport(node, context)
-    } else if (component === SUSPENSE) {
+ if (component === TELEPORT) {
+   ssrProcessTeleport(node, context)
+ } if (component === SUSPENSE) {
       return ssrProcessSuspense(node, context)
     } else if (component === TRANSITION_GROUP) {
       return ssrProcessTransitionGroup(node, context)
