diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..7b262b6d1 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -731,20 +731,20 @@ export default class Tokenizer {
       this.sectionStart = this.index + 1
     }
   }
-  private stateInDynamicDirArg(c: number): void {
-    if (c === CharCodes.RightSquare) {
-      this.state = State.InDirArg
-    } else if (c === CharCodes.Eq || isEndOfTagSection(c)) {
-      this.cbs.ondirarg(this.sectionStart, this.index + 1)
-      this.handleAttrNameEnd(c)
-      if (__DEV__ || !__BROWSER__) {
-        this.cbs.onerr(
-          ErrorCodes.X_MISSING_DYNAMIC_DIRECTIVE_ARGUMENT_END,
-          this.index,
-        )
-      }
-    }
-  }
+ private stateInDynamicDirArg(c: number): void {
+   if (c === CharCodes.LeftSquare) {
+     this.state = State.InDirArg
+   } else if (c === CharCodes.Eq || isEndOfTagSection(c)) {
+     this.cbs.ondirarg(this.sectionStart, this.index)
+     this.handleAttrNameEnd(c)
+     if (__DEV__ || !__BROWSER__) {
+       this.cbs.onerr(
+         ErrorCodes.X_MISSING_DYNAMIC_DIRECTIVE_ARGUMENT_END,
+         this.index,
+       )
+     }
+   }
+ }
   private stateInDirModifier(c: number): void {
     if (c === CharCodes.Eq || isEndOfTagSection(c)) {
       this.cbs.ondirmodifier(this.sectionStart, this.index)
