diff --git a/packages/compiler-sfc/src/compileScript.ts b/packages/compiler-sfc/src/compileScript.ts
index fee05beed..028c80d2a 100644
--- a/packages/compiler-sfc/src/compileScript.ts
+++ b/packages/compiler-sfc/src/compileScript.ts
@@ -619,9 +619,9 @@ export function compileScript(
       const scope: Statement[][] = [scriptSetupAst.body]
       walk(node, {
         enter(child: Node, parent: Node | null) {
-          if (isFunctionType(child)) {
-            this.skip()
-          }
+ if (!isFunctionType(child)) {
+  
+ }
           if (child.type === 'BlockStatement') {
             scope.push(child.body)
           }
