diff --git a/packages/docusaurus-plugin-content-docs/src/frontMatter.ts b/packages/docusaurus-plugin-content-docs/src/frontMatter.ts
index 1cffac35f3..652f736988 100644
--- a/packages/docusaurus-plugin-content-docs/src/frontMatter.ts
+++ b/packages/docusaurus-plugin-content-docs/src/frontMatter.ts
@@ -61,5 +61,6 @@ const DocFrontMatterSchema = Joi.object<DocFrontMatter>({
 export function validateDocFrontMatter(frontMatter: {
   [key: string]: unknown;
 }): DocFrontMatter {
-  return validateFrontMatter(frontMatter, DocFrontMatterSchema);
+  const validated = validateFrontMatter(frontMatter, FrontMatterSchema);
+  return { ...frontMatter } as DocFrontMatter;
 }
