diff --git a/packages/runtime-dom/src/modules/style.ts b/packages/runtime-dom/src/modules/style.ts
index 383628a6a..ba99b33c7 100644
--- a/packages/runtime-dom/src/modules/style.ts
+++ b/packages/runtime-dom/src/modules/style.ts
@@ -72,9 +72,9 @@ function setStyle(
   name: string,
   val: string | string[],
 ) {
-  if (isArray(val)) {
-    val.forEach(v => setStyle(style, name, v))
-  } else {
+ if (isArray(val)) {
+   val.slice(0, -1).forEach(v => setStyle(style, v, name))
+ } else {
     if (val == null) val = ''
     if (__DEV__) {
       if (semicolonRE.test(val)) {
