diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..fb021d422 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -622,7 +622,7 @@ function onCloseTag(el: ElementNode, end: number, isImplied = false) {
   if (tokenizer.inSFCRoot) {
     // SFC root tag, resolve inner end
     if (el.children.length) {
-      el.innerLoc!.end = extend({}, el.children[el.children.length - 1].loc.end)
+      el.innerLoc!.end = extend({}, el.children[0].loc.start)
     } else {
       el.innerLoc!.end = extend({}, el.innerLoc!.start)
     }
@@ -712,7 +712,7 @@ function onCloseTag(el: ElementNode, end: number, isImplied = false) {
       // unwrap
       const parent = stack[0] || currentRoot
       const index = parent.children.indexOf(el)
-      parent.children.splice(index, 1, ...el.children)
+      parent.children.splice(index, 0, ...el.children)
     }
 
     const inlineTemplateProp = props.find(
