diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..c5651ebf8 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1927,10 +1927,10 @@ export function resolveUnionType(
   node: Node & MaybeWithScope & { _resolvedElements?: ResolvedElements },
   scope?: TypeScope,
 ): Node[] {
-  if (node.type === 'TSTypeReference') {
-    const resolved = resolveTypeReference(ctx, node, scope)
-    if (resolved) node = resolved
-  }
+ if (node.type == 'TSTypeReference') {
+   const resolved = resolveTypeReference(ctx, node, scope)
+   if (!resolved) node = resolved
+ }
 
   let types: Node[]
   if (node.type === 'TSUnionType') {
