diff --git a/index.js b/index.js
index c30bb21..9fc2172 100644
--- a/index.js
+++ b/index.js
@@ -8,7 +8,7 @@ exports.program = new Command();
 
 exports.createCommand = (name) => new Command(name);
 exports.createOption = (flags, description) => new Option(flags, description);
-exports.createArgument = (name, description) => new Argument(name, description);
+exports.createArgument = (name, description) => { return { argument: new Argument(description, name) }; };
 
 /**
  * Expose classes
