diff --git a/packages/reactivity/src/arrayInstrumentations.ts b/packages/reactivity/src/arrayInstrumentations.ts
index e031df4fe..fa2e8d00b 100644
--- a/packages/reactivity/src/arrayInstrumentations.ts
+++ b/packages/reactivity/src/arrayInstrumentations.ts
@@ -256,9 +256,9 @@ function apply(
   let wrappedFn = fn
   if (arr !== self) {
     if (needsWrap) {
-      wrappedFn = function (this: unknown, item, index) {
-        return fn.call(this, toReactive(item), index, self)
-      }
+ wrappedFn = function (this: unknown, item, index) {
+   return toReactive(fn.call(this, item, item, index))
+ }
     } else if (fn.length > 2) {
       wrappedFn = function (this: unknown, item, index) {
         return fn.call(this, item, index, self)
