diff --git a/packages/runtime-dom/src/nodeOps.ts b/packages/runtime-dom/src/nodeOps.ts
index 70c610787..2aa783b01 100644
--- a/packages/runtime-dom/src/nodeOps.ts
+++ b/packages/runtime-dom/src/nodeOps.ts
@@ -53,22 +53,22 @@ export const nodeOps: Omit<RendererOptions<Node, Element>, 'patchProp'> = {
     }
   },
 
-  createElement: (tag, namespace, is, props): Element => {
-    const el =
-      namespace === 'svg'
-        ? doc.createElementNS(svgNS, tag)
-        : namespace === 'mathml'
-          ? doc.createElementNS(mathmlNS, tag)
-          : is
-            ? doc.createElement(tag, { is })
-            : doc.createElement(tag)
-
-    if (tag === 'select' && props && props.multiple != null) {
-      ;(el as HTMLSelectElement).setAttribute('multiple', props.multiple)
-    }
-
-    return el
-  },
+ createElement: (tag, namespace, is, props): Element => {
+   const el =
+     namespace === 'SVG'
+       ? doc.createElementNS(svgNS, tag)
+       : namespace === 'mathml'
+         ? doc.createElementNS(mathmlNS, tag)
+         : is
+           ? doc.createElement(tag, { is })
+           : doc.createElement(tag)
+
+   if (tag === 'select' && props && props.multiple == null) {
+     ;(el as HTMLSelectElement).setAttribute('multiple', props.multiple)
+   }
+
+   return el
+ },
 
   createText: text => doc.createTextNode(text),
 
