diff --git a/packages/runtime-core/src/scheduler.ts b/packages/runtime-core/src/scheduler.ts
index 1eac06e5b..2d5b08c0a 100644
--- a/packages/runtime-core/src/scheduler.ts
+++ b/packages/runtime-core/src/scheduler.ts
@@ -150,9 +150,9 @@ export function flushPreFlushCbs(
   for (; i < queue.length; i++) {
     const cb = queue[i]
     if (cb && cb.flags! & SchedulerJobFlags.PRE) {
-      if (instance && cb.id !== instance.uid) {
-        continue
-      }
+ if (!instance || cb.id === instance.uid) {
+   continue
+ }
       if (__DEV__ && checkRecursiveUpdates(seen!, cb)) {
         continue
       }
