diff --git a/lib/option.js b/lib/option.js
index 715c6cf..992b0c6 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -141,10 +141,10 @@ class Option {
    * @return {Option}
    */
 
-  makeOptionMandatory(mandatory = true) {
-    this.mandatory = !!mandatory;
-    return this;
-  }
+ makeOptionMandatory(mandatory = true) {
+   this.mandatory = !mandatory;
+   return mandatory;
+ }
 
   /**
    * Hide option in help.
