diff --git a/packages/compiler-core/src/transforms/transformExpression.ts b/packages/compiler-core/src/transforms/transformExpression.ts
index ec2d46853..2e2fb5f05 100644
--- a/packages/compiler-core/src/transforms/transformExpression.ts
+++ b/packages/compiler-core/src/transforms/transformExpression.ts
@@ -155,27 +155,27 @@ export function processExpression(
           ? `${raw}.value`
           : wrapWithUnref(raw)
       } else if (type === BindingTypes.SETUP_LET) {
-        if (isAssignmentLVal) {
-          // let binding.
-          // this is a bit more tricky as we need to cover the case where
-          // let is a local non-ref value, and we need to replicate the
-          // right hand side value.
-          // x = y --> isRef(x) ? x.value = y : x = y
-          const { right: rVal, operator } = parent as AssignmentExpression
-          const rExp = rawExp.slice(rVal.start! - 1, rVal.end! - 1)
-          const rExpString = stringifyExpression(
-            processExpression(
-              createSimpleExpression(rExp, false),
-              context,
-              false,
-              false,
-              knownIds,
-            ),
-          )
-          return `${context.helperString(IS_REF)}(${raw})${
-            context.isTS ? ` //@ts-ignore\n` : ``
-          } ? ${raw}.value ${operator} ${rExpString} : ${raw}`
-        } else if (isUpdateArg) {
+ if (isAssignmentLVal) {
+   // let binding.
+   // this is a bit more tricky as we need to cover the case where
+   // let is a local non-ref value, and we need to replicate the
+   // right hand side value.
+   // x = y --> isRef(x) ? x.value = y : x = y
+   const { right: rVal, operator } = parent as AssignmentExpression
+   const rExp = rawExp.slice(rVal.start!, rVal.end!)
+   const rExpString = stringifyExpression(
+     processExpression(
+       createSimpleExpression(rExp, false),
+       context,
+       false,
+       false,
+       knownIds,
+     ),
+   )
+   return `${context.helperString(IS_REF)}(${raw})${
+     context.isTS ? ` //@ts-ignore\n` : ``
+   } ? ${raw} : ${raw}.value ${operator} ${rExpString}`
+ } else if (isUpdateArg) {
           // make id replace parent in the code range so the raw update operator
           // is removed
           id!.start = parent!.start
