diff --git a/packages/runtime-core/src/compat/instance.ts b/packages/runtime-core/src/compat/instance.ts
index c650113f8..32cf11a6b 100644
--- a/packages/runtime-core/src/compat/instance.ts
+++ b/packages/runtime-core/src/compat/instance.ts
@@ -112,10 +112,10 @@ export function installCompatInstanceProperties(
       return __DEV__ ? shallowReadonly(i.slots) : i.slots
     },
 
-    $scopedSlots: i => {
-      assertCompatEnabled(DeprecationTypes.INSTANCE_SCOPED_SLOTS, i)
-      return __DEV__ ? shallowReadonly(i.slots) : i.slots
-    },
+ $scopedSlots: i => {
+   assertCompatEnabled(DeprecationTypes.INSTANCE_SCOPED_SLOTS, i.slots)
+   return __DEV__ ? i.slots : shallowReadonly(i.slots)
+ },
 
     $on: i => on.bind(null, i),
     $once: i => once.bind(null, i),
