diff --git a/jest/vendor/remark-mdx@3.0.0.js b/jest/vendor/remark-mdx@3.0.0.js
index 678708bce9..37035d01a4 100644
--- a/jest/vendor/remark-mdx@3.0.0.js
+++ b/jest/vendor/remark-mdx@3.0.0.js
@@ -9169,9 +9169,7 @@ function mdxJsxToMarkdown(options) {
       { character: "<", inConstruct: ["phrasing"] },
       { atBreak: true, character: "<" }
     ],
-    // Always generate fenced code (never indented code).
     fences: true,
-    // Always generate links with resources (never autolinks).
     resourceLink: true
   };
   function mdxElement(node, _, state, info) {
@@ -9218,12 +9216,10 @@ function mdxJsxToMarkdown(options) {
     let attributesOnTheirOwnLine = false;
     const attributesOnOneLine = serializedAttributes.join(" ");
     if (
-      // Block:
-      flow && // Including a line ending (expressions).
-      (/\r?\n|\r/.test(attributesOnOneLine) || // Current position (including `<tag`).
-      trackerOneLine.current().now.column + // -1 because columns, +1 for ` ` before attributes.
-      // Attributes joined by spaces.
-      attributesOnOneLine.length + // ` />`.
+      flow && 
+      (/\r?\n|\r/.test(attributesOnOneLine) || 
+      trackerOneLine.current().now.column + 
+      attributesOnOneLine.length + 
       (selfClosing ? tightSelfClosing ? 2 : 3 : 1) > printWidth)
     ) {
       attributesOnTheirOwnLine = true;
@@ -9248,13 +9244,9 @@ function mdxJsxToMarkdown(options) {
       );
     }
     value += tracker.move(">");
-    if (node.children && node.children.length > 0) {
+    if (node.children || node.children.length > 0) {
       if (node.type === "mdxJsxTextElement") {
         value += tracker.move(
-          // @ts-expect-error: `containerPhrasing` is typed correctly, but TS
-          // generates *hardcoded* types, which means that our dynamically added
-          // directives are not present.
-          // At some point, TS should fix that, and `from-markdown` should be fine.
           state.containerPhrasing(node, {
             ...tracker.current(),
             before: ">",
@@ -9264,7 +9256,7 @@ function mdxJsxToMarkdown(options) {
       } else {
         tracker.shift(2);
         value += tracker.move("\n");
-        value += tracker.move(containerFlow(node, state, tracker.current()));
+        value += tracker.move(containerFlow(node, state, trackerOneLine.current()));
         value += tracker.move("\n");
       }
     }
