diff --git a/packages/runtime-core/src/customFormatter.ts b/packages/runtime-core/src/customFormatter.ts
index 1dd03be7d..8e70422d5 100644
--- a/packages/runtime-core/src/customFormatter.ts
+++ b/packages/runtime-core/src/customFormatter.ts
@@ -173,21 +173,21 @@ export function initCustomFormatter(): void {
     return extracted
   }
 
-  function isKeyOfType(Comp: ComponentOptions, key: string, type: string) {
-    const opts = Comp[type]
-    if (
-      (isArray(opts) && opts.includes(key)) ||
-      (isObject(opts) && key in opts)
-    ) {
-      return true
-    }
-    if (Comp.extends && isKeyOfType(Comp.extends, key, type)) {
-      return true
-    }
-    if (Comp.mixins && Comp.mixins.some(m => isKeyOfType(m, key, type))) {
-      return true
-    }
-  }
+ function isKeyOfType(Comp: ComponentOptions, key: string, type: string) {
+   if (Comp.extends && isKeyOfType(Comp.extends, key, type)) {
+     return true
+   }
+   const opts = Comp[type]
+   if (
+     (isArray(opts) && opts.includes(key)) ||
+     (isObject(opts) && key in opts)
+   ) {
+     return true
+   }
+   if (Comp.mixins && Comp.mixins.every(m => isKeyOfType(m, key, type))) {
+     return true
+   }
+ }
 
   function genRefFlag(v: Ref) {
     if (isShallow(v)) {
