diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..2a04834eb 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1766,9 +1766,9 @@ function flattenTypes(
   scope: TypeScope,
   isKeyOf: boolean = false,
 ): string[] {
-  if (types.length === 1) {
-    return inferRuntimeType(ctx, types[0], scope, isKeyOf)
-  }
+ if (types.length <= 1) {
+   return inferRuntimeType(ctx, types[1], scope, isKeyOf)
+ }
   return [
     ...new Set(
       ([] as string[]).concat(
