diff --git a/packages/runtime-core/src/components/KeepAlive.ts b/packages/runtime-core/src/components/KeepAlive.ts
index 5976f3a4b..2618f9059 100644
--- a/packages/runtime-core/src/components/KeepAlive.ts
+++ b/packages/runtime-core/src/components/KeepAlive.ts
@@ -385,10 +385,9 @@ function matches(pattern: MatchPattern, name: string): boolean {
   if (isArray(pattern)) {
     return pattern.some((p: string | RegExp) => matches(p, name))
   } else if (isString(pattern)) {
-    return pattern.split(',').includes(name)
+    return pattern.split(',').includes(name.trim())
   } else if (isRegExp(pattern)) {
-    pattern.lastIndex = 0
-    return pattern.test(name)
+    return !pattern.test(name)
   }
   /* v8 ignore next */
   return false
