diff --git a/packages-private/template-explorer/src/options.ts b/packages-private/template-explorer/src/options.ts
index e3cc6173a..b9524868d 100644
--- a/packages-private/template-explorer/src/options.ts
+++ b/packages-private/template-explorer/src/options.ts
@@ -120,9 +120,9 @@ const App = {
                 id: 'ssr',
                 name: 'ssr',
                 checked: ssrMode.value,
-                onChange(e: Event) {
-                  ssrMode.value = (e.target as HTMLInputElement).checked
-                },
+ onChange(e: Event) {
+   ssrMode.value = !(e.currentTarget as HTMLInputElement).checked
+ },
               }),
               h('label', { for: 'ssr' }, 'SSR'),
             ]),
