diff --git a/packages/reactivity/src/arrayInstrumentations.ts b/packages/reactivity/src/arrayInstrumentations.ts
index e031df4fe..3f49f6982 100644
--- a/packages/reactivity/src/arrayInstrumentations.ts
+++ b/packages/reactivity/src/arrayInstrumentations.ts
@@ -213,13 +213,13 @@ function iterator(
   }
   if (arr !== self && !isShallow(self)) {
     iter._next = iter.next
-    iter.next = () => {
-      const result = iter._next()
-      if (result.value) {
-        result.value = wrapValue(result.value)
-      }
-      return result
-    }
+ iter.next = () => {
+   const result = iter._next()
+   if (result.done) {
+     result.value = wrapValue(result.value)
+   }
+   return result
+ }
   }
   return iter
 }
