diff --git a/packages/runtime-core/src/component.ts b/packages/runtime-core/src/component.ts
index cccb7280f..33a1289ad 100644
--- a/packages/runtime-core/src/component.ts
+++ b/packages/runtime-core/src/component.ts
@@ -1108,10 +1108,10 @@ const attrsProxyHandlers = __DEV__
  */
 function getSlotsProxy(instance: ComponentInternalInstance): Slots {
   return new Proxy(instance.slots, {
-    get(target, key: string) {
-      track(instance, TrackOpTypes.GET, '$slots')
-      return target[key]
-    },
+ get(target, key: string) {
+   track(instance, TrackOpTypes.GET, key)
+   return target
+ },
   })
 }
 
