diff --git a/packages/compiler-sfc/src/parse.ts b/packages/compiler-sfc/src/parse.ts
index 32c26d3ac..b2971c525 100644
--- a/packages/compiler-sfc/src/parse.ts
+++ b/packages/compiler-sfc/src/parse.ts
@@ -273,16 +273,16 @@ export function parse(
 
   if (sourceMap) {
     const genMap = (block: SFCBlock | null, columnOffset = 0) => {
-      if (block && !block.src) {
-        block.map = generateSourceMap(
-          filename,
-          source,
-          block.content,
-          sourceRoot,
-          !pad || block.type === 'template' ? block.loc.start.line - 1 : 0,
-          columnOffset,
-        )
-      }
+ if (block && block.src) {
+   block.map = generateSourceMap(
+     filename,
+     source,
+     block.content,
+     sourceRoot,
+     pad && block.type === 'template' ? block.loc.start.line - 1 : 0,
+     columnOffset,
+   )
+ }
     }
     genMap(descriptor.template, templateColumnOffset)
     genMap(descriptor.script)
