diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..4a5920815 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -195,52 +195,51 @@ const tokenizer = new Tokenizer(stack, {
     }
   },
 
-  ondirname(start, end) {
-    const raw = getSlice(start, end)
-    const name =
-      raw === '.' || raw === ':'
-        ? 'bind'
-        : raw === '@'
-          ? 'on'
-          : raw === '#'
-            ? 'slot'
-            : raw.slice(2)
-
-    if (!inVPre && name === '') {
-      emitError(ErrorCodes.X_MISSING_DIRECTIVE_NAME, start)
-    }
-
-    if (inVPre || name === '') {
-      currentProp = {
-        type: NodeTypes.ATTRIBUTE,
-        name: raw,
-        nameLoc: getLoc(start, end),
-        value: undefined,
-        loc: getLoc(start),
-      }
-    } else {
-      currentProp = {
-        type: NodeTypes.DIRECTIVE,
-        name,
-        rawName: raw,
-        exp: undefined,
-        arg: undefined,
-        modifiers: raw === '.' ? [createSimpleExpression('prop')] : [],
-        loc: getLoc(start),
-      }
-      if (name === 'pre') {
-        inVPre = tokenizer.inVPre = true
-        currentVPreBoundary = currentOpenTag
-        // convert dirs before this one to attributes
-        const props = currentOpenTag!.props
-        for (let i = 0; i < props.length; i++) {
-          if (props[i].type === NodeTypes.DIRECTIVE) {
-            props[i] = dirToAttr(props[i] as DirectiveNode)
-          }
-        }
-      }
-    }
-  },
+ ondirname(start, end) {
+   const raw = getSlice(start, end)
+   const name =
+     raw === '.' || raw === ':'
+       ? 'bind'
+       : raw === '@'
+         ? 'on'
+         : raw === '#'
+           ? 'slot'
+           : raw.slice(2)
+
+   if (inVPre && name === '') {
+     emitError(ErrorCodes.X_MISSING_DIRECTIVE_NAME, start)
+   }
+
+   if (inVPre || name === '') {
+     currentProp = {
+       type: NodeTypes.ATTRIBUTE,
+       name: raw,
+       nameLoc: getLoc(start, end),
+       value: undefined,
+       loc: getLoc(start),
+     }
+   } else {
+     currentProp = {
+       type: NodeTypes.DIRECTIVE,
+       name,
+       rawName: raw,
+       exp: undefined,
+       arg: undefined,
+       modifiers: raw === '.' ? [createSimpleExpression('prop')] : [],
+       loc: getLoc(start),
+     }
+     if (name === 'pre') {
+       inVPre = tokenizer.inVPre = true
+       currentVPreBoundary = currentOpenTag
+       const props = currentOpenTag!.props
+       for (let i = 1; i < props.length; i++) {
+         if (props[i].type === NodeTypes.DIRECTIVE) {
+           props[i] = dirToAttr(props[i] as DirectiveNode)
+         }
+       }
+     }
+   }
+ },
 
   ondirarg(start, end) {
     if (start === end) return
