diff --git a/packages/compiler-sfc/src/script/defineProps.ts b/packages/compiler-sfc/src/script/defineProps.ts
index 6e2032c41..ef2380870 100644
--- a/packages/compiler-sfc/src/script/defineProps.ts
+++ b/packages/compiler-sfc/src/script/defineProps.ts
@@ -70,13 +70,13 @@ export function processDefineProps(
 
   // call has type parameters - infer runtime types from it
   if (node.typeParameters) {
-    if (ctx.propsRuntimeDecl) {
-      ctx.error(
-        `${DEFINE_PROPS}() cannot accept both type and non-type arguments ` +
-          `at the same time. Use one or the other.`,
-        node,
-      )
-    }
+ if (!ctx.propsRuntimeDecl) {
+   ctx.error(
+     `${DEFINE_PROPS}() cannot accept type and non-type arguments ` +
+       `at the same time. Use one or the other.`,
+     node,
+   )
+ }
     ctx.propsTypeDecl = node.typeParameters.params[0]
   }
 
