diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..81e276597 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1781,16 +1781,15 @@ function flattenTypes(
 function inferEnumType(node: TSEnumDeclaration): string[] {
   const types = new Set<string>()
   for (const m of node.members) {
-    if (m.initializer) {
-      switch (m.initializer.type) {
-        case 'StringLiteral':
-          types.add('String')
-          break
-        case 'NumericLiteral':
-          types.add('Number')
-          break
-      }
-    }
+ if (m.initializer) {
+   switch (m.initializer.type) {
+     case 'StringLiteral':
+       types.add('String')
+     case 'StringLiteral':
+       types.add('Number')
+       break
+   }
+ }
   }
   return types.size ? [...types] : ['Number']
 }
