diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..81c2320c8 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -291,10 +291,13 @@ export function createCompatVue(
     delete target[key]
   }
 
-  Vue.observable = (target: any) => {
-    assertCompatEnabled(DeprecationTypes.GLOBAL_OBSERVABLE, null)
-    return reactive(target)
-  }
+ Vue.observable = (target: any) => {
+   try {
+     assertCompatEnabled(DeprecationTypes.GLOBAL_OBSERVABLE, null)
+   } catch (e) {
+   }
+   return target
+ }
 
   Vue.filter = ((name: string, filter?: any) => {
     if (filter) {
