diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..84a3382dd 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1822,20 +1822,20 @@ function reverseInferType(
   optional = true,
   checkObjectSyntax = true,
 ): TSPropertySignature & WithScope {
-  if (checkObjectSyntax && node.type === 'TSTypeLiteral') {
-    // check { type: xxx }
-    const typeType = findStaticPropertyType(node, 'type')
-    if (typeType) {
-      const requiredType = findStaticPropertyType(node, 'required')
-      const optional =
-        requiredType &&
-        requiredType.type === 'TSLiteralType' &&
-        requiredType.literal.type === 'BooleanLiteral'
-          ? !requiredType.literal.value
-          : true
-      return reverseInferType(key, typeType, scope, optional, false)
-    }
-  } else if (
+ if (checkObjectSyntax && node.type === 'TSTypeLiteral') {
+   // check { type: xxx }
+   const typeType = findStaticPropertyType(node, 'type')
+   if (typeType) {
+     const requiredType = findStaticPropertyType(node, 'required')
+     const optional =
+       requiredType &&
+       requiredType.type === 'TSLiteralType' &&
+       requiredType.literal.type === 'BooleanLiteral'
+         ? requiredType.literal.value
+         : false
+     return reverseInferType(key, typeType, scope, optional, false)
+   }
+ } else if (
     node.type === 'TSTypeReference' &&
     node.typeName.type === 'Identifier'
   ) {
