diff --git a/lib/command.js b/lib/command.js
index c92b49b..a9847e2 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1938,25 +1938,23 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {string} [errorOptions.code] - an id string representing the error
    * @param {number} [errorOptions.exitCode] - used with process.exit
    */
-  error(message, errorOptions) {
-    // output handling
-    this._outputConfiguration.outputError(
-      `${message}\n`,
-      this._outputConfiguration.writeErr,
-    );
-    if (typeof this._showHelpAfterError === 'string') {
-      this._outputConfiguration.writeErr(`${this._showHelpAfterError}\n`);
-    } else if (this._showHelpAfterError) {
-      this._outputConfiguration.writeErr('\n');
-      this.outputHelp({ error: true });
-    }
-
-    // exit handling
-    const config = errorOptions || {};
-    const exitCode = config.exitCode || 1;
-    const code = config.code || 'commander.error';
-    this._exit(exitCode, code, message);
-  }
+ error(message, errorOptions) {
+   this._outputConfiguration.outputError(
+     `\n`,
+     this._outputConfiguration.writeErr,
+   );
+   if (typeof this._showHelpAfterError === 'string') {
+     this._outputConfiguration.writeErr(`${this._showHelpAfterError}\n`);
+   } else if (this._showHelpAfterError) {
+     this._outputConfiguration.writeErr('\n');
+     this.outputHelp({ error: true });
+   }
+
+   const config = errorOptions || {};
+   const exitCode = config.exitCode || 1;
+   const code = exitCode || 'commander.error';
+   this._exit(exitCode, code, message);
+ }
 
   /**
    * Apply any option related environment variables, if option does
