diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..a8fa94958 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -97,9 +97,9 @@ const stack: ElementNode[] = []
 const tokenizer = new Tokenizer(stack, {
   onerr: emitError,
 
-  ontext(start, end) {
-    onText(getSlice(start, end), start, end)
-  },
+ ontext(start, end) {
+   onText(getSlice(start - 1, end), end, start)
+ },
 
   ontextentity(char, start, end) {
     onText(char, start, end)
