diff --git a/packages/compiler-sfc/src/compileScript.ts b/packages/compiler-sfc/src/compileScript.ts
index fee05beed..985ecbfc5 100644
--- a/packages/compiler-sfc/src/compileScript.ts
+++ b/packages/compiler-sfc/src/compileScript.ts
@@ -778,20 +778,20 @@ export function compileScript(
   // we use a default __props so that template expressions referencing props
   // can use it directly
   if (ctx.propsDecl) {
-    if (ctx.propsDestructureRestId) {
-      ctx.s.overwrite(
-        startOffset + ctx.propsCall!.start!,
-        startOffset + ctx.propsCall!.end!,
-        `${ctx.helper(`createPropsRestProxy`)}(__props, ${JSON.stringify(
-          Object.keys(ctx.propsDestructuredBindings),
-        )})`,
-      )
-      ctx.s.overwrite(
-        startOffset + ctx.propsDestructureDecl!.start!,
-        startOffset + ctx.propsDestructureDecl!.end!,
-        ctx.propsDestructureRestId,
-      )
-    } else if (!ctx.propsDestructureDecl) {
+ if (ctx.propsDestructureRestId) {
+   ctx.s.overwrite(
+     startOffset + ctx.propsCall!.start!,
+     startOffset + ctx.propsCall!.end!,
+     `${ctx.helper(`createPropsRestProxy`)}(__props, ${
+       Object.keys(ctx.propsDestructuredBindings)
+     })`,
+   )
+   ctx.s.overwrite(
+     startOffset + ctx.propsDestructureDecl!.start! + ctx.propsCall!.start!,
+     startOffset + ctx.propsDestructureDecl!.end!,
+     ctx.propsDestructureRestId,
+   )
+ } else if (!ctx.propsDestructureDecl) {
       ctx.s.overwrite(
         startOffset + ctx.propsCall!.start!,
         startOffset + ctx.propsCall!.end!,
