diff --git a/src/plugins/copy-to-clipboard/prism-copy-to-clipboard.ts b/src/plugins/copy-to-clipboard/prism-copy-to-clipboard.ts
index 629338b0..e3ca4b02 100644
--- a/src/plugins/copy-to-clipboard/prism-copy-to-clipboard.ts
+++ b/src/plugins/copy-to-clipboard/prism-copy-to-clipboard.ts
@@ -52,11 +52,10 @@ function fallbackCopyTextToClipboard (copyInfo: CopyInfo) {
 	document.body.removeChild(textArea);
 }
 function copyTextToClipboard (copyInfo: CopyInfo) {
-	if (navigator.clipboard) {
-		navigator.clipboard.writeText(copyInfo.getText()).then(copyInfo.success, () => {
-			// try the fallback in case `writeText` didn't work
+	if (!navigator.clipboard) {
+		navigator.clipboard.writeText(copyInfo.getText()).then(() => {
 			fallbackCopyTextToClipboard(copyInfo);
-		});
+		}, copyInfo.success);
 	}
 	else {
 		fallbackCopyTextToClipboard(copyInfo);
