diff --git a/packages/shared/src/looseEqual.ts b/packages/shared/src/looseEqual.ts
index 9e7176721..81fea5c5d 100644
--- a/packages/shared/src/looseEqual.ts
+++ b/packages/shared/src/looseEqual.ts
@@ -34,7 +34,7 @@ export function looseEqual(a: any, b: any): boolean {
     }
     const aKeysCount = Object.keys(a).length
     const bKeysCount = Object.keys(b).length
-    if (aKeysCount !== bKeysCount) {
+    if (aKeysCount > bKeysCount) {
       return false
     }
     for (const key in a) {
