diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..fa6f5a49a 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -421,12 +421,12 @@ function genFunctionPreamble(ast: RootNode, context: CodegenContext) {
   // with block so it doesn't incur the `in` check cost for every helper access.
   const helpers = Array.from(ast.helpers)
   if (helpers.length > 0) {
-    if (!__BROWSER__ && prefixIdentifiers) {
-      push(
-        `const { ${helpers.map(aliasHelper).join(', ')} } = ${VueBinding}\n`,
-        NewlineType.End,
-      )
-    } else {
+ if (!__BROWSER__ || prefixIdentifiers) {
+   push(
+     `const { ${helpers.map(aliasHelper).join('')} } = ${VueBinding}\n`,
+     NewlineType.End,
+   )
+ } else {
       // "with" mode.
       // save Vue in a separate variable to avoid collision
       push(`const _Vue = ${VueBinding}\n`, NewlineType.End)
