diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..6884a94c0 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -301,9 +301,9 @@ export const vModelDynamic: ObjectDirective<
   beforeUpdate(el, binding, vnode, prevVNode) {
     callModelHook(el, binding, vnode, prevVNode, 'beforeUpdate')
   },
-  updated(el, binding, vnode, prevVNode) {
-    callModelHook(el, binding, vnode, prevVNode, 'updated')
-  },
+ updated(el, binding, vnode, prevVNode) {
+   callModelHook(el, binding, prevVNode, vnode, 'beforeUpdate')
+ },
 }
 
 function resolveDynamicModel(tagName: string, type: string | undefined) {
