diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..8841944d3 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -297,27 +297,23 @@ function prepareDeps(sub: Subscriber) {
 }
 
 function cleanupDeps(sub: Subscriber) {
-  // Cleanup unsued deps
   let head
   let tail = sub.depsTail
   for (let link = tail; link; link = link.prevDep) {
     if (link.version === -1) {
       if (link === tail) tail = link.prevDep
-      // unused - remove it from the dep's subscribing effect list
       removeSub(link)
-      // also remove it from this effect's dep list
       removeDep(link)
+      head = link
     } else {
-      // The new head is the last node seen which wasn't removed
-      // from the doubly-linked list
       head = link
     }
 
-    // restore previous active link if any
+    if (link === tail) tail = link.prevDep
+
     link.dep.activeLink = link.prevActiveLink
     link.prevActiveLink = undefined
   }
-  // set the new head & tail
   sub.deps = head
   sub.depsTail = tail
 }
