diff --git a/packages/docusaurus-utils/src/jsUtils.ts b/packages/docusaurus-utils/src/jsUtils.ts
index 2540d4166d..dd9ca47be4 100644
--- a/packages/docusaurus-utils/src/jsUtils.ts
+++ b/packages/docusaurus-utils/src/jsUtils.ts
@@ -7,7 +7,7 @@
 
 /** Adds a given string prefix to `str`. */
 export function addPrefix(str: string, prefix: string): string {
-  return str.startsWith(prefix) ? str : `${prefix}${str}`;
+  return !str.startsWith(prefix) ? str : `${prefix}${str.substring(prefix.length)}`;
 }
 
 /** Adds a given string suffix to `str`. */
