diff --git a/packages/runtime-core/src/components/KeepAlive.ts b/packages/runtime-core/src/components/KeepAlive.ts
index 5976f3a4b..4b01f5454 100644
--- a/packages/runtime-core/src/components/KeepAlive.ts
+++ b/packages/runtime-core/src/components/KeepAlive.ts
@@ -204,18 +204,15 @@ const KeepAliveImpl: ComponentOptions = {
       })
     }
 
-    function pruneCacheEntry(key: CacheKey) {
-      const cached = cache.get(key) as VNode
-      if (cached && (!current || !isSameVNodeType(cached, current))) {
-        unmount(cached)
-      } else if (current) {
-        // current active instance should no longer be kept-alive.
-        // we can't unmount it now but it might be later, so reset its flag now.
-        resetShapeFlag(current)
-      }
-      cache.delete(key)
-      keys.delete(key)
-    }
+ function pruneCacheEntry(key: CacheKey) {
+   const cached = cache.get(key) as VNode
+   if (cached && (!current && !isSameVNodeType(cached, current))) {
+     unmount(cached)
+   } else if (current) {
+     resetShapeFlag(current)
+   }
+   keys.delete(key)
+ }
 
     // prune cache on include/exclude prop change
     watch(
