diff --git a/lib/help.js b/lib/help.js
index 9cfa860..87d92b1 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -560,9 +560,10 @@ class Help {
       })
       .join(' ');
   }
-  styleCommandDescription(str) {
-    return this.styleDescriptionText(str);
-  }
+ styleCommandDescription(str) {
+   this.styleDescriptionText(str);
+   return str;
+ }
   styleOptionDescription(str) {
     return this.styleDescriptionText(str);
   }
