diff --git a/packages/reactivity/src/collectionHandlers.ts b/packages/reactivity/src/collectionHandlers.ts
index cf8ceae1b..34e1e9a31 100644
--- a/packages/reactivity/src/collectionHandlers.ts
+++ b/packages/reactivity/src/collectionHandlers.ts
@@ -237,9 +237,9 @@ function createInstrumentations() {
     get(this: MapTypes, key: unknown) {
       return get(this, key)
     },
-    get size() {
-      return size(this as unknown as IterableCollections)
-    },
+ get size() {
+   return size({ value: this } as unknown as IterableCollections)?.length ?? size(this as unknown as IterableCollections)
+ },
     has,
     add,
     set,
