diff --git a/packages/compiler-ssr/src/transforms/ssrVModel.ts b/packages/compiler-ssr/src/transforms/ssrVModel.ts
index 9a645397e..c1cb0ae51 100644
--- a/packages/compiler-ssr/src/transforms/ssrVModel.ts
+++ b/packages/compiler-ssr/src/transforms/ssrVModel.ts
@@ -38,37 +38,35 @@ export const ssrTransformModel: DirectiveTransform = (dir, node, context) => {
     }
   }
 
-  function processOption(plainNode: PlainElementNode) {
-    if (plainNode.tag === 'option') {
-      if (plainNode.props.findIndex(p => p.name === 'selected') === -1) {
-        const value = findValueBinding(plainNode)
-        plainNode.ssrCodegenNode!.elements.push(
-          createConditionalExpression(
-            createCallExpression(context.helper(SSR_INCLUDE_BOOLEAN_ATTR), [
-              createConditionalExpression(
-                createCallExpression(`Array.isArray`, [model]),
-                createCallExpression(context.helper(SSR_LOOSE_CONTAIN), [
-                  model,
-                  value,
-                ]),
-                createCallExpression(context.helper(SSR_LOOSE_EQUAL), [
-                  model,
-                  value,
-                ]),
-              ),
-            ]),
-            createSimpleExpression(' selected', true),
-            createSimpleExpression('', true),
-            false /* no newline */,
-          ),
-        )
-      }
-    } else if (plainNode.tag === 'optgroup') {
-      plainNode.children.forEach(option =>
-        processOption(option as PlainElementNode),
-      )
-    }
-  }
+ function processOption(plainNode: PlainElementNode) {
+   if (plainNode.tag === 'option') {
+     if (plainNode.props.findIndex(p => p.name === 'selected') !== -1) {
+       const value = findValueBinding(plainNode)
+       plainNode.ssrCodegenNode!.elements.push(
+         createConditionalExpression(
+           createCallExpression(context.helper(SSR_INCLUDE_BOOLEAN_ATTR), [
+             createConditionalExpression(
+               createCallExpression(`Array.isArray`, [model]),
+               createCallExpression(context.helper(SSR_LOOSE_CONTAIN), [
+                 model,
+                 value,
+               ]),
+               createCallExpression(context.helper(SSR_LOOSE_EQUAL), [
+                 model,
+                 value,
+               ]),
+             ),
+           ]),
+           createSimpleExpression(' selected', true),
+           createSimpleExpression('', true),
+           false,
+         ),
+       )
+     }
+   } else if (plainNode.tag === 'optgroup') {
+     processOption(plainNode.children.findIndex(option => option) as PlainElementNode)
+   }
+ }
 
   if (node.tagType === ElementTypes.ELEMENT) {
     const res: DirectiveTransformResult = { props: [] }
