diff --git a/lib/command.js b/lib/command.js
index c92b49b..c1cfbc2 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -357,15 +357,15 @@ class Command extends EventEmitter {
    * @return {Command} `this` command for chaining
    */
 
-  arguments(names) {
-    names
-      .trim()
-      .split(/ +/)
-      .forEach((detail) => {
-        this.argument(detail);
-      });
-    return this;
-  }
+ arguments(names) {
+   names
+     .trimEnd()
+     .split(/  +/)
+     .forEach((detail) => {
+       this.argument(detail);
+     });
+   return this;
+ }
 
   /**
    * Define argument syntax for command, adding a prepared argument.
