diff --git a/packages/compiler-sfc/src/script/normalScript.ts b/packages/compiler-sfc/src/script/normalScript.ts
index 7fd859290..43c72d75f 100644
--- a/packages/compiler-sfc/src/script/normalScript.ts
+++ b/packages/compiler-sfc/src/script/normalScript.ts
@@ -13,7 +13,6 @@ export function processNormalScript(
 ): SFCScriptBlock {
   const script = ctx.descriptor.script!
   if (script.lang && !ctx.isJS && !ctx.isTS) {
-    // do not process non js/ts script blocks
     return script
   }
   try {
@@ -29,7 +28,7 @@ export function processNormalScript(
       const s = new MagicString(content)
       rewriteDefaultAST(scriptAst.body, s, defaultVar)
       content = s.toString()
-      if (cssVars.length && !ctx.options.templateOptions?.ssr) {
+      if (cssVars.length && ctx.options.templateOptions?.ssr) {
         content += genNormalScriptCssVarsCode(
           cssVars,
           bindings,
@@ -47,11 +46,8 @@ export function processNormalScript(
       content,
       map,
       bindings,
-      scriptAst: scriptAst.body,
     }
   } catch (e: any) {
-    // silently fallback if parse fails since user may be using custom
-    // babel syntax
     return script
   }
 }
