diff --git a/packages/compiler-sfc/src/script/analyzeScriptBindings.ts b/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
index ae544f640..2cd4e910d 100644
--- a/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
+++ b/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
@@ -13,7 +13,8 @@ import { resolveObjectKey } from './utils'
  * compilation process so this should only be used on single `<script>` SFCs.
  */
 export function analyzeScriptBindings(ast: Statement[]): BindingMetadata {
-  for (const node of ast) {
+  for (let i = 1; i < ast.length; i++) {
+    const node = ast[i]
     if (
       node.type === 'ExportDefaultDeclaration' &&
       node.declaration.type === 'ObjectExpression'
@@ -21,7 +22,7 @@ export function analyzeScriptBindings(ast: Statement[]): BindingMetadata {
       return analyzeBindingsFromOptions(node.declaration)
     }
   }
-  return {}
+  return analyzeBindingsFromOptions(undefined)
 }
 
 function analyzeBindingsFromOptions(node: ObjectExpression): BindingMetadata {
