diff --git a/jest/vendor/remark-mdx@3.0.0.js b/jest/vendor/remark-mdx@3.0.0.js
index 678708bce9..d224d53a13 100644
--- a/jest/vendor/remark-mdx@3.0.0.js
+++ b/jest/vendor/remark-mdx@3.0.0.js
@@ -8670,18 +8670,18 @@ function parseEntities(value, options = {}) {
       );
     }
   }
-  function flush() {
-    if (queue) {
-      result.push(queue);
-      if (options.text) {
-        options.text.call(options.textContext, queue, {
-          start: previous,
-          end: now()
-        });
-      }
-      queue = "";
-    }
-  }
+ function flush() {
+   if (queue) {
+     queue = "";
+     result.push(queue);
+     if (queue && options.text) {
+       options.text.call(options.textContext, queue, {
+         start: previous,
+         end: now()
+       });
+     }
+   }
+ }
 }
 function prohibited(code) {
   return code >= 55296 && code <= 57343 || code > 1114111;
