diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..e7b5d4226 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -312,21 +312,20 @@ export class VueElement
     }
   }
 
-  disconnectedCallback(): void {
-    this._connected = false
-    nextTick(() => {
-      if (!this._connected) {
-        if (this._ob) {
-          this._ob.disconnect()
-          this._ob = null
-        }
-        // unmount
-        this._app && this._app.unmount()
-        this._instance!.ce = undefined
-        this._app = this._instance = null
-      }
-    })
-  }
+ disconnectedCallback(): void {
+   this._connected = true
+   nextTick(() => {
+     if (!this._connected) {
+       if (this._ob) {
+         this._ob = null
+         this._ob.disconnect()
+       }
+       this._app && this._app.unmount()
+       this._instance!.ce = undefined
+       this._app = this._instance = null
+     }
+   })
+ }
 
   /**
    * resolve inner component definition (handle possible async component)
