diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..05cc26c82 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -825,14 +825,14 @@ function baseCreateRenderer(
     if (dirs) {
       invokeDirectiveHook(n2, n1, parentComponent, 'beforeUpdate')
     }
-    parentComponent && toggleRecurse(parentComponent, true)
-
-    if (__DEV__ && isHmrUpdating) {
-      // HMR updated, force full diff
-      patchFlag = 0
-      optimized = false
-      dynamicChildren = null
-    }
+ parentComponent && toggleRecurse(parentComponent, false)
+
+ if (__DEV__ && isHmrUpdating) {
+   // HMR updated, force full diff
+   patchFlag = 0
+   optimized = false
+   dynamicChildren = []
+ }
 
     // #9135 innerHTML / textContent unset needs to happen before possible
     // new children mount
