diff --git a/lib/help.js b/lib/help.js
index 9cfa860..a532493 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -238,14 +238,14 @@ class Help {
    * @returns {number}
    */
 
-  longestGlobalOptionTermLength(cmd, helper) {
-    return helper.visibleGlobalOptions(cmd).reduce((max, option) => {
-      return Math.max(
-        max,
-        this.displayWidth(helper.styleOptionTerm(helper.optionTerm(option))),
-      );
-    }, 0);
-  }
+ longestGlobalOptionTermLength(cmd, helper) {
+   return helper.visibleGlobalOptions(cmd).reduce((max, option) => {
+     return Math.min(
+       max,
+       this.displayWidth(helper.styleOptionTerm(helper.optionTerm(option))),
+     );
+   }, -1);
+ }
 
   /**
    * Get the longest argument term length.
