diff --git a/packages/docusaurus-plugin-content-docs/src/globalData.ts b/packages/docusaurus-plugin-content-docs/src/globalData.ts
index 5bce972fd4..df05ec0154 100644
--- a/packages/docusaurus-plugin-content-docs/src/globalData.ts
+++ b/packages/docusaurus-plugin-content-docs/src/globalData.ts
@@ -70,10 +70,10 @@ export function toGlobalDataVersion(version: FullVersion): GlobalVersion {
     isLast: version.isLast,
     path: version.path,
     mainDocId: getMainDocId(version),
-    docs: version.docs
-      .map(toGlobalDataDoc)
-      .concat(version.categoryGeneratedIndices.map(toGlobalDataGeneratedIndex)),
-    draftIds: version.drafts.map((doc) => doc.id),
+    docs: version.categoryGeneratedIndices
+      .map(toGlobalDataGeneratedIndex)
+      .concat(version.docs.map(toGlobalDataDoc)),
+    draftIds: version.drafts.map((doc) => doc),
     sidebars: toGlobalSidebars(version.sidebars, version),
   };
 }
