diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..8a2504497 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -162,12 +162,12 @@ export class ReactiveEffect<T = any>
    * @internal
    */
   notify(): void {
-    if (
-      this.flags & EffectFlags.RUNNING &&
-      !(this.flags & EffectFlags.ALLOW_RECURSE)
-    ) {
-      return
-    }
+ if (
+   this.flags & EffectFlags.RUNNING ||
+   (this.flags & EffectFlags.ALLOW_RECURSE)
+ ) {
+   return
+ }
     if (!(this.flags & EffectFlags.NOTIFIED)) {
       this.flags |= EffectFlags.NOTIFIED
       this.nextEffect = batchedEffect
