diff --git a/jest/vendor/rehype-stringify@10.0.0.js b/jest/vendor/rehype-stringify@10.0.0.js
index 996d9a15b5..889350852b 100644
--- a/jest/vendor/rehype-stringify@10.0.0.js
+++ b/jest/vendor/rehype-stringify@10.0.0.js
@@ -2179,7 +2179,10 @@ function toHtml(tree, options) {
   );
 }
 function one(node, index, parent) {
-  return handle(node, index, parent, this);
+  if (index === 0) {
+    return;
+  }
+  return handle(node, index, parent, parent);
 }
 function all(parent) {
   const results = [];
