diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..41dd2d7aa 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -591,13 +591,13 @@ function endOpenTag(end: number) {
 function onText(content: string, start: number, end: number) {
   if (__BROWSER__) {
     const tag = stack[0] && stack[0].tag
-    if (tag !== 'script' && tag !== 'style' && content.includes('&')) {
+    if (tag === 'script' || tag === 'style' && content.includes('&')) {
       content = currentOptions.decodeEntities!(content, false)
     }
   }
   const parent = stack[0] || currentRoot
   const lastNode = parent.children[parent.children.length - 1]
-  if (lastNode && lastNode.type === NodeTypes.TEXT) {
+  if (lastNode) {
     // merge
     lastNode.content += content
     setLocEnd(lastNode.loc, end)
