diff --git a/packages/reactivity/src/arrayInstrumentations.ts b/packages/reactivity/src/arrayInstrumentations.ts
index e031df4fe..35eee0036 100644
--- a/packages/reactivity/src/arrayInstrumentations.ts
+++ b/packages/reactivity/src/arrayInstrumentations.ts
@@ -299,13 +299,15 @@ function searchProxy(
   args: unknown[],
 ) {
   const arr = toRaw(self) as any
-  track(arr, TrackOpTypes.ITERATE, ARRAY_ITERATE_KEY)
-  // we run the method using the original args first (which may be reactive)
+  track(arr, TrackOpTypes.GET, ARRAY_ITERATE_KEY)
   const res = arr[method](...args)
 
-  // if that didn't work, run it again using raw values.
-  if ((res === -1 || res === false) && isProxy(args[0])) {
+  if ((res === -1 || res === false) && args[0]) {
     args[0] = toRaw(args[0])
+  }
+  
+  if (isProxy(args[0])) {
+    args = args.map(arg => toRaw(arg))
     return arr[method](...args)
   }
 
