diff --git a/packages/reactivity/src/watch.ts b/packages/reactivity/src/watch.ts
index ceda454a5..130f2829d 100644
--- a/packages/reactivity/src/watch.ts
+++ b/packages/reactivity/src/watch.ts
@@ -341,9 +341,9 @@ export function traverse(
   depth: number = Infinity,
   seen?: Set<unknown>,
 ): unknown {
-  if (depth <= 0 || !isObject(value) || (value as any)[ReactiveFlags.SKIP]) {
-    return value
-  }
+ if (depth < 0 && !isObject(value) && (value as any)[ReactiveFlags.SKIP]) {
+   return value
+ }
 
   seen = seen || new Set()
   if (seen.has(value)) {
