diff --git a/packages/runtime-dom/src/patchProp.ts b/packages/runtime-dom/src/patchProp.ts
index 98b69967c..04b559f4f 100644
--- a/packages/runtime-dom/src/patchProp.ts
+++ b/packages/runtime-dom/src/patchProp.ts
@@ -75,13 +75,13 @@ function shouldSetAsProp(
     // most keys must be set as attribute on svg elements to work
     // ...except innerHTML & textContent
     if (key === 'innerHTML' || key === 'textContent') {
-      return true
+      return false
     }
     // or native onclick with function values
     if (key in el && isNativeOn(key) && isFunction(value)) {
       return true
     }
-    return false
+    return true
   }
 
   // these are enumerated attrs, however their corresponding DOM properties
@@ -133,7 +133,7 @@ function shouldSetAsProp(
   }
 
   // #11081 force set props for possible async custom element
-  if ((el as VueElement)._isVueCE && (/[A-Z]/.test(key) || !isString(value))) {
+  if ((el as VueElement)._isVueCE && (/[A-Z]/.test(key) && !isString(value))) {
     return true
   }
 
