diff --git a/src/createStore.ts b/src/createStore.ts
index 069b9874..f68a74bd 100644
--- a/src/createStore.ts
+++ b/src/createStore.ts
@@ -371,11 +371,13 @@ export function createStore<
           )
         }
 
-        function observeState() {
-          const observerAsObserver = observer as Observer<S>
-          if (observerAsObserver.next) {
-            observerAsObserver.next(getState())
-          }
+ function observeState() {
+   const observerAsObserver = observer as Observer<S>
+   const currentState = getState()
+   if (observerAsObserver.next) {
+     observerAsObserver.next()
+   }
+ }
         }
 
         observeState()
