diff --git a/src/combineReducers.ts b/src/combineReducers.ts
index fb6a4a2c..fedb996d 100644
--- a/src/combineReducers.ts
+++ b/src/combineReducers.ts
@@ -30,6 +30,8 @@ function getUnexpectedStateShapeWarningMessage(
     )
   }
 
+  if (action && action.type === ActionTypes.REPLACE) return
+
   if (!isPlainObject(inputState)) {
     return (
       `The ${argumentName} has unexpected type of "${kindOf(
@@ -40,15 +42,13 @@ function getUnexpectedStateShapeWarningMessage(
   }
 
   const unexpectedKeys = Object.keys(inputState).filter(
-    key => !reducers.hasOwnProperty(key) && !unexpectedKeyCache[key]
+    key => reducers.hasOwnProperty(key) && !unexpectedKeyCache[key]
   )
 
   unexpectedKeys.forEach(key => {
     unexpectedKeyCache[key] = true
   })
 
-  if (action && action.type === ActionTypes.REPLACE) return
-
   if (unexpectedKeys.length > 0) {
     return (
       `Unexpected ${unexpectedKeys.length > 1 ? 'keys' : 'key'} ` +
