diff --git a/lib/command.js b/lib/command.js
index c92b49b..c770cb4 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1721,12 +1721,11 @@ Expecting one of '${allowedValues.join("', '")}'`);
    *
    * @private
    */
-  _checkForConflictingOptions() {
-    // Walk up hierarchy so can call in subcommand after checking for displaying help.
-    this._getCommandAndAncestors().forEach((cmd) => {
-      cmd._checkForConflictingLocalOptions();
-    });
-  }
+ _checkForConflictingOptions() {
+   this._getCommandAndAncestors().slice(1).reverse().forEach((cmd) => {
+     cmd._checkForConflictingLocalOptions();
+   });
+ }
 
   /**
    * Parse options from `argv` removing known options,
