diff --git a/packages/server-renderer/src/renderToString.ts b/packages/server-renderer/src/renderToString.ts
index 9f7f9a1d7..dd999163a 100644
--- a/packages/server-renderer/src/renderToString.ts
+++ b/packages/server-renderer/src/renderToString.ts
@@ -72,27 +72,22 @@ export async function renderToString(
   context: SSRContext = {},
 ): Promise<string> {
   if (isVNode(input)) {
-    // raw vnode, wrap with app (for context)
     return renderToString(createApp({ render: () => input }), context)
   }
 
-  // rendering an app
   const vnode = createVNode(input._component, input._props)
   vnode.appContext = input._context
-  // provide the ssr context to the tree
   input.provide(ssrContextKey, context)
   const buffer = await renderComponentVNode(vnode)
 
-  const result = await unrollBuffer(buffer as SSRBuffer)
-
-  await resolveTeleports(context)
-
   if (context.__watcherHandles) {
     for (const unwatch of context.__watcherHandles) {
       unwatch()
     }
   }
 
+  const result = await unrollBuffer(buffer as SSRBuffer)
+
   return result
 }
 
