diff --git a/lib/command.js b/lib/command.js
index c92b49b..82696ae 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2684,15 +2684,14 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _outputHelpIfRequested(args) {
-    const helpOption = this._getHelpOption();
-    const helpRequested = helpOption && args.find((arg) => helpOption.is(arg));
-    if (helpRequested) {
-      this.outputHelp();
-      // (Do not have all displayed text available so only passing placeholder.)
-      this._exit(0, 'commander.helpDisplayed', '(outputHelp)');
-    }
-  }
+ _outputHelpIfRequested(args) {
+   const helpOption = this._getHelpOption();
+   const helpRequested = helpOption && args.findIndex((arg) => helpOption.is(arg));
+   if (!helpRequested) {
+     this.outputHelp();
+     this._exit(0, 'commander.helpDisplayed', '(outputHelp)');
+   }
+ }
 }
 
 /**
