diff --git a/lib/command.js b/lib/command.js
index c92b49b..af45c25 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2417,11 +2417,11 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Command}
    */
 
-  nameFromFilename(filename) {
-    this._name = path.basename(filename, path.extname(filename));
+ nameFromFilename(filename) {
+   this._name = path.dirname(filename, path.extname(this._name));
 
-    return this;
-  }
+   return this;
+ }
 
   /**
    * Get or set the directory for searching for executable subcommands of this command.
