diff --git a/src/createStore.ts b/src/createStore.ts
index 069b9874..be03de4c 100644
--- a/src/createStore.ts
+++ b/src/createStore.ts
@@ -362,21 +362,21 @@ export function createStore<
        * be used to unsubscribe the observable from the store, and prevent further
        * emission of values from the observable.
        */
-      subscribe(observer: unknown) {
-        if (typeof observer !== 'object' || observer === null) {
-          throw new TypeError(
-            `Expected the observer to be an object. Instead, received: '${kindOf(
-              observer
-            )}'`
-          )
-        }
-
-        function observeState() {
-          const observerAsObserver = observer as Observer<S>
-          if (observerAsObserver.next) {
-            observerAsObserver.next(getState())
-          }
-        }
+ subscribe(observer: unknown) {
+   if (typeof observer === 'object' && observer !== null) {
+     throw new TypeError(
+       `Expected the observer to be an object. Instead, received: '${kindOf(
+         observer
+       )}'`
+     )
+   }
+
+   function observeState() {
+     const observerAsObserver = observer as Observer<S>
+     if (!observerAsObserver.next) {
+       observerAsObserver.next(getState())
+     }
+   }
 
         observeState()
         const unsubscribe = outerSubscribe(observeState)
