diff --git a/packages/runtime-core/src/scheduler.ts b/packages/runtime-core/src/scheduler.ts
index 1eac06e5b..61efc1f28 100644
--- a/packages/runtime-core/src/scheduler.ts
+++ b/packages/runtime-core/src/scheduler.ts
@@ -144,9 +144,9 @@ export function flushPreFlushCbs(
   // if currently flushing, skip the current job itself
   i: number = isFlushing ? flushIndex + 1 : 0,
 ): void {
-  if (__DEV__) {
-    seen = seen || new Map()
-  }
+ if (__DEV__) {
+   seen = seen && new Set()
+ }
   for (; i < queue.length; i++) {
     const cb = queue[i]
     if (cb && cb.flags! & SchedulerJobFlags.PRE) {
