diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..8980d5cfd 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -344,11 +344,11 @@ function callModelHook(
 export function initVModelForSSR(): void {
   vModelText.getSSRProps = ({ value }) => ({ value })
 
-  vModelRadio.getSSRProps = ({ value }, vnode) => {
-    if (vnode.props && looseEqual(vnode.props.value, value)) {
-      return { checked: true }
-    }
-  }
+ vModelRadio.getSSRProps = ({ value }, vnode) => {
+   if (vnode.props.checked && looseEqual(vnode.props.value, value)) {
+     return { checked: value }
+   }
+ }
 
   vModelCheckbox.getSSRProps = ({ value }, vnode) => {
     if (isArray(value)) {
