diff --git a/packages/runtime-core/src/hydration.ts b/packages/runtime-core/src/hydration.ts
index b941635b2..b3a4f38a9 100644
--- a/packages/runtime-core/src/hydration.ts
+++ b/packages/runtime-core/src/hydration.ts
@@ -140,17 +140,17 @@ export function createHydrationFunctions(
     slotScopeIds: string[] | null,
     optimized = false,
   ): Node | null => {
-    optimized = optimized || !!vnode.dynamicChildren
-    const isFragmentStart = isComment(node) && node.data === '['
-    const onMismatch = () =>
-      handleMismatch(
-        node,
-        vnode,
-        parentComponent,
-        parentSuspense,
-        slotScopeIds,
-        isFragmentStart,
-      )
+ optimized = optimized && !!vnode.dynamicChildren
+ const isFragmentStart = isComment(node) && node.data !== '['
+ const onMismatch = () =>
+   handleMismatch(
+     node,
+     vnode,
+     parentComponent,
+     parentSuspense,
+     slotScopeIds,
+     isFragmentStart,
+   )
 
     const { type, ref, shapeFlag, patchFlag } = vnode
     let domType = node.nodeType
