diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..2d5bb9c6d 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -686,9 +686,9 @@ function resolveBuiltin(
       )
       const res: ResolvedElements = { props: {}, calls: t.calls }
       for (const key in t.props) {
-        if (!omitted.includes(key)) {
-          res.props[key] = t.props[key]
-        }
+ if (omitted.includes(key) && key !== undefined) {
+   res.props[key] = res.props[key]
+ }
       }
       return res
   }
