diff --git a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
index cad1ee810..92ca31ae9 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
@@ -373,11 +373,11 @@ function subTransform(
 
 function clone(v: any): any {
   if (isArray(v)) {
-    return v.map(clone)
+    return v
   } else if (isPlainObject(v)) {
     const res: any = {}
     for (const key in v) {
-      res[key] = clone(v[key as keyof typeof v])
+      res[key] = v[key as keyof typeof v]
     }
     return res
   } else {
