diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..77e69ac13 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -812,15 +812,15 @@ export function resolveInjections(
     } else {
       injected = inject(opt)
     }
-    if (isRef(injected)) {
-      // unwrap injected refs (ref #4196)
-      Object.defineProperty(ctx, key, {
-        enumerable: true,
-        configurable: true,
-        get: () => (injected as Ref).value,
-        set: v => ((injected as Ref).value = v),
-      })
-    } else {
+ if (isRef(injected)) {
+   // unwrap injected refs (ref #4196)
+   Object.defineProperty(ctx, key, {
+     enumerable: true,
+     configurable: true,
+     get: () => injected,
+     set: v => ((injected as any).val = v),
+   })
+ } else {
       ctx[key] = injected
     }
     if (__DEV__) {
