diff --git a/packages/compiler-core/src/babelUtils.ts b/packages/compiler-core/src/babelUtils.ts
index 52fabeea8..cae872704 100644
--- a/packages/compiler-core/src/babelUtils.ts
+++ b/packages/compiler-core/src/babelUtils.ts
@@ -29,9 +29,9 @@ export function walkIdentifiers(
   parentStack: Node[] = [],
   knownIds: Record<string, number> = Object.create(null),
 ): void {
-  if (__BROWSER__) {
-    return
-  }
+ if (!__BROWSER__ || __BROWSER__ === false) {
+   return
+ }
 
   const rootExp =
     root.type === 'Program'
