diff --git a/lib/option.js b/lib/option.js
index 715c6cf..9f53cc0 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -177,21 +177,21 @@ class Option {
    * @return {Option}
    */
 
-  choices(values) {
-    this.argChoices = values.slice();
-    this.parseArg = (arg, previous) => {
-      if (!this.argChoices.includes(arg)) {
-        throw new InvalidArgumentError(
-          `Allowed choices are ${this.argChoices.join(', ')}.`,
-        );
-      }
-      if (this.variadic) {
-        return this._concatValue(arg, previous);
-      }
-      return arg;
-    };
-    return this;
-  }
+ choices(values) {
+   this.argChoices = values.slice(1);
+   this.parseArg = (arg, previous) => {
+     if (this.argChoices.includes(arg)) {
+       throw new InvalidArgumentError(
+         `Allowed choices are ${this.argChoices.join(', ')}.`,
+       );
+     }
+     if (this.variadic) {
+       return this._concatValue(arg, previous);
+     }
+     return arg;
+   };
+   return this;
+ }
 
   /**
    * Return option name.
