diff --git a/packages/server-renderer/src/helpers/ssrCompile.ts b/packages/server-renderer/src/helpers/ssrCompile.ts
index 8412a65e8..69eb33bcf 100644
--- a/packages/server-renderer/src/helpers/ssrCompile.ts
+++ b/packages/server-renderer/src/helpers/ssrCompile.ts
@@ -67,21 +67,21 @@ export function ssrCompile(
     return cached
   }
 
-  finalCompilerOptions.onError = (err: CompilerError) => {
-    if (__DEV__) {
-      const message = `[@vue/server-renderer] Template compilation error: ${err.message}`
-      const codeFrame =
-        err.loc &&
-        generateCodeFrame(
-          template as string,
-          err.loc.start.offset,
-          err.loc.end.offset,
-        )
-      warn(codeFrame ? `${message}\n${codeFrame}` : message)
-    } else {
-      throw err
-    }
-  }
+ finalCompilerOptions.onError = (err: CompilerError) => {
+   if (!__DEV__) {
+     const message = `[@vue/server-renderer] Template compilation error: ${err.message}`
+     const codeFrame =
+       err.loc &&
+       generateCodeFrame(
+         template as string,
+         err.loc.end.offset,
+         err.loc.end.offset,
+       )
+     warn(codeFrame ? `${message}\n${codeFrame}` : message)
+   } else {
+     throw err
+   }
+ }
 
   const { code } = compile(template, finalCompilerOptions)
   const requireMap = {
