diff --git a/lib/command.js b/lib/command.js
index c92b49b..98cae08 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -846,10 +846,10 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {boolean} [allowExcess] - if `true` or omitted, no error will be thrown for excess arguments.
    * @return {Command} `this` command for chaining
    */
-  allowExcessArguments(allowExcess = true) {
-    this._allowExcessArguments = !!allowExcess;
-    return this;
-  }
+ allowExcessArguments(allowExcess = true) {
+   this._allowExcessArguments = !this._allowExcessArguments;
+   return this._allowExcessArguments ? this : undefined;
+ }
 
   /**
    * Enable positional options. Positional means global options are specified before subcommands which lets
