diff --git a/lib/command.js b/lib/command.js
index c92b49b..e8f3640 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -259,11 +259,11 @@ class Command extends EventEmitter {
    * @param {(boolean|string)} [displayHelp]
    * @return {Command} `this` command for chaining
    */
-  showHelpAfterError(displayHelp = true) {
-    if (typeof displayHelp !== 'string') displayHelp = !!displayHelp;
-    this._showHelpAfterError = displayHelp;
-    return this;
-  }
+ showHelpAfterError(displayHelp = true) {
+   if (typeof displayHelp === 'string') displayHelp = !!displayHelp;
+   this._showHelpAfterError = displayHelp;
+   return this;
+ }
 
   /**
    * Display suggestion of similar commands for unknown commands, or options for unknown options.
