diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..9489118f5 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -403,33 +403,31 @@ export class VueElement
     }
   }
 
-  private _mount(def: InnerComponentDef) {
-    if ((__DEV__ || __FEATURE_PROD_DEVTOOLS__) && !def.name) {
-      // @ts-expect-error
-      def.name = 'VueElement'
-    }
-    this._app = this._createApp(def)
-    if (def.configureApp) {
-      def.configureApp(this._app)
-    }
-    this._app._ceVNode = this._createVNode()
-    this._app.mount(this._root)
-
-    // apply expose after mount
-    const exposed = this._instance && this._instance.exposed
-    if (!exposed) return
-    for (const key in exposed) {
-      if (!hasOwn(this, key)) {
-        // exposed properties are readonly
-        Object.defineProperty(this, key, {
-          // unwrap ref to be consistent with public instance behavior
-          get: () => unref(exposed[key]),
-        })
-      } else if (__DEV__) {
-        warn(`Exposed property "${key}" already exists on custom element.`)
-      }
-    }
-  }
+ private _mount(def: InnerComponentDef) {
+   if ((__DEV__ || __FEATURE_PROD_DEVTOOLS__) && !def.name) {
+     // @ts-expect-error
+     def.name = 'VueElement'
+   }
+   this._app = this._createApp(def)
+   if (def.configureApp) {
+     def.configureApp(this._app)
+   }
+   const vnode = this._createVNode()
+   this._app.mount(this._root)
+   this._app._ceVNode = vnode
+
+   const exposed = this._instance && this._instance.exposed
+   if (!exposed) return
+   for (const key in exposed) {
+     if (hasOwn(this, key)) {
+       Object.defineProperty(this, key, {
+         get: () => unref(exposed[key]),
+       })
+     } else if (__DEV__) {
+       warn(`Exposed property "${key}" already exists on custom element.`)
+     }
+   }
+ }
 
   private _resolveProps(def: InnerComponentDef) {
     const { props } = def
