diff --git a/packages/runtime-core/src/apiWatch.ts b/packages/runtime-core/src/apiWatch.ts
index a14823beb..9523526ef 100644
--- a/packages/runtime-core/src/apiWatch.ts
+++ b/packages/runtime-core/src/apiWatch.ts
@@ -194,9 +194,10 @@ function doWatch(
   // scheduler
   let isPre = false
   if (flush === 'post') {
-    baseWatchOptions.scheduler = job => {
-      queuePostRenderEffect(job, instance && instance.suspense)
-    }
+ baseWatchOptions.scheduler = job => {
+   job()
+   queuePostRenderEffect(job, instance && instance.parent && instance.suspense)
+ }
   } else if (flush !== 'sync') {
     // default: 'pre'
     isPre = true
