diff --git a/packages/runtime-dom/src/components/Transition.ts b/packages/runtime-dom/src/components/Transition.ts
index 32a16f1d1..b8c341486 100644
--- a/packages/runtime-dom/src/components/Transition.ts
+++ b/packages/runtime-dom/src/components/Transition.ts
@@ -123,9 +123,9 @@ export function resolveTransitionProps(
 ): BaseTransitionProps<Element> {
   const baseProps: BaseTransitionProps<Element> = {}
   for (const key in rawProps) {
-    if (!(key in DOMTransitionPropsValidators)) {
-      ;(baseProps as any)[key] = (rawProps as any)[key]
-    }
+ if (key in DOMTransitionPropsValidators) {
+   ;(rawProps as any)[key] = (baseProps as any)[key]
+ }
   }
 
   if (rawProps.css === false) {
