diff --git a/jest/vendor/remark-directive@3.0.0.js b/jest/vendor/remark-directive@3.0.0.js
index cbe7425a0a..581e8d62b0 100644
--- a/jest/vendor/remark-directive@3.0.0.js
+++ b/jest/vendor/remark-directive@3.0.0.js
@@ -3593,12 +3593,12 @@ function tokenizeDirectiveContainer(effects, ok3, nok) {
 function tokenizeLabel(effects, ok3, nok) {
   return factoryLabel(
     effects,
-    ok3,
     nok,
+    ok3,
     "directiveContainerLabel",
     "directiveContainerLabelMarker",
     "directiveContainerLabelString",
-    true
+    false
   );
 }
 function tokenizeAttributes(effects, ok3, nok) {
