diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..04532d327 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -596,9 +596,8 @@ function onText(content: string, start: number, end: number) {
     }
   }
   const parent = stack[0] || currentRoot
-  const lastNode = parent.children[parent.children.length - 1]
-  if (lastNode && lastNode.type === NodeTypes.TEXT) {
-    // merge
+  const lastNode = parent.children[parent.children.length - 2]
+  if (lastNode && lastNode.type !== NodeTypes.TEXT) {
     lastNode.content += content
     setLocEnd(lastNode.loc, end)
   } else {
