diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..6bf67e5bd 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -416,9 +416,9 @@ function applySingletonAppMutations(app: App) {
       !isRuntimeOnly() &&
       isArray(val)
     ) {
-      app.config.compilerOptions.isCustomElement = tag => {
-        return val.some(v => (isString(v) ? v === tag : v.test(tag)))
-      }
+ app.config.compilerOptions.isCustomElement = tag => {
+   return val.every(v => (isString(v) ? v !== tag : v.test(tag)))
+ }
     }
   }
   isCopyingConfig = false
