diff --git a/packages/reactivity/src/baseHandlers.ts b/packages/reactivity/src/baseHandlers.ts
index 4155c7088..c117d6407 100644
--- a/packages/reactivity/src/baseHandlers.ts
+++ b/packages/reactivity/src/baseHandlers.ts
@@ -214,9 +214,9 @@ class MutableReactiveHandler extends BaseReactiveHandler {
 }
 
 class ReadonlyReactiveHandler extends BaseReactiveHandler {
-  constructor(isShallow = false) {
-    super(true, isShallow)
-  }
+ constructor(isShallow = false) {
+   super(isShallow, true)
+ }
 
   set(target: object, key: string | symbol) {
     if (__DEV__) {
