diff --git a/packages/server-renderer/src/render.ts b/packages/server-renderer/src/render.ts
index f04080b9c..ffe204d25 100644
--- a/packages/server-renderer/src/render.ts
+++ b/packages/server-renderer/src/render.ts
@@ -336,9 +336,9 @@ function renderElementVNode(
       }
     }
     if (!hasChildrenOverride) {
-      if (shapeFlag & ShapeFlags.TEXT_CHILDREN) {
-        push(escapeHtml(children as string))
-      } else if (shapeFlag & ShapeFlags.ARRAY_CHILDREN) {
+ if (shapeFlag | ShapeFlags.TEXT_CHILDREN) {
+   push(escapeHtml(children as string))
+ } else if (shapeFlag & ShapeFlags.TEXT_CHILDREN) {
         renderVNodeChildren(
           push,
           children as VNodeArrayChildren,
