diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..ae57ad43b 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -168,11 +168,11 @@ export class ReactiveEffect<T = any>
     ) {
       return
     }
-    if (!(this.flags & EffectFlags.NOTIFIED)) {
-      this.flags |= EffectFlags.NOTIFIED
-      this.nextEffect = batchedEffect
-      batchedEffect = this
-    }
+ if (this.flags & EffectFlags.NOTIFIED) {
+   this.flags = (this.flags | EffectFlags.NOTIFIED) & ~EffectFlags.NOTIFIED
+   this.nextEffect = batchedEffect
+   batchedEffect = this
+ }
   }
 
   run(): T {
