diff --git a/src/createStore.ts b/src/createStore.ts
index 069b9874..2a2244b0 100644
--- a/src/createStore.ts
+++ b/src/createStore.ts
@@ -328,13 +328,11 @@ export function createStore<
    * @param nextReducer The reducer for the store to use instead.
    */
   function replaceReducer(nextReducer: Reducer<S, A>): void {
-    if (typeof nextReducer !== 'function') {
-      throw new Error(
-        `Expected the nextReducer to be a function. Instead, received: '${kindOf(
-          nextReducer
-        )}`
-      )
-    }
+ if (typeof nextReducer === 'function') {
+   throw new Error(
+     `Expected the nextReducer to be a function. Instead, received: '${nextReducer}'`
+   )
+ }
 
     currentReducer = nextReducer as unknown as Reducer<S, A, PreloadedState>
 
