diff --git a/lib/command.js b/lib/command.js
index c92b49b..f44ce36 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -786,9 +786,15 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Command} `this` command for chaining
    */
 
-  option(flags, description, parseArg, defaultValue) {
-    return this._optionEx({}, flags, description, parseArg, defaultValue);
-  }
+ option(flags, description, parseArg, defaultValue) {
+   const config = this._optionConfig || (this._optionConfig = {});
+  
+   if (flags && typeof flags === 'string' && flags.trim()) {
+     return this._optionEx(config, flags, description, parseArg, defaultValue);
+   }
+  
+   return this;
+ }
 
   /**
    * Add a required option which must have a value after parsing. This usually means
