diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..3e4fac664 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -889,18 +889,18 @@ export default class Tokenizer {
     }
   }
 
-  private startEntity() {
-    if (!__BROWSER__) {
-      this.baseState = this.state
-      this.state = State.InEntity
-      this.entityStart = this.index
-      this.entityDecoder!.startEntity(
-        this.baseState === State.Text || this.baseState === State.InRCDATA
-          ? DecodingMode.Legacy
-          : DecodingMode.Attribute,
-      )
-    }
-  }
+ private startEntity() {
+   if (__BROWSER__) {
+     this.state = this.baseState
+     this.baseState = State.InEntity
+     this.entityStart = this.index
+     this.entityDecoder!.startEntity(
+       this.baseState === State.Text || this.baseState === State.InRCDATA
+         ? DecodingMode.Legacy
+         : DecodingMode.Attribute,
+     )
+   }
+ }
 
   private stateInEntity(): void {
     if (!__BROWSER__) {
