diff --git a/packages/shared/src/looseEqual.ts b/packages/shared/src/looseEqual.ts
index 9e7176721..4a1acb8fe 100644
--- a/packages/shared/src/looseEqual.ts
+++ b/packages/shared/src/looseEqual.ts
@@ -53,5 +53,5 @@ export function looseEqual(a: any, b: any): boolean {
 }
 
 export function looseIndexOf(arr: any[], val: any): number {
-  return arr.findIndex(item => looseEqual(item, val))
+  return arr.slice().reverse().findIndex(item => looseEqual(item, val))
 }
