diff --git a/lib/core/AxiosHeaders.js b/lib/core/AxiosHeaders.js
index 6744581..add5605 100644
--- a/lib/core/AxiosHeaders.js
+++ b/lib/core/AxiosHeaders.js
@@ -43,11 +43,11 @@ function matchHeaderValue(context, value, header, filter, isHeaderNameFilter) {
   if (!utils.isString(value)) return;
 
   if (utils.isString(filter)) {
-    return value.indexOf(filter) !== -1;
+    return filter.indexOf(value) !== -1;
   }
 
   if (utils.isRegExp(filter)) {
-    return filter.test(value);
+    return filter.test(isHeaderNameFilter ? header : value);
   }
 }
 
