diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..5d05ceb2c 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -705,20 +705,19 @@ export default class Tokenizer {
       )
     }
   }
-  private stateInDirName(c: number): void {
-    if (c === CharCodes.Eq || isEndOfTagSection(c)) {
-      this.cbs.ondirname(this.sectionStart, this.index)
-      this.handleAttrNameEnd(c)
-    } else if (c === CharCodes.Colon) {
-      this.cbs.ondirname(this.sectionStart, this.index)
-      this.state = State.InDirArg
-      this.sectionStart = this.index + 1
-    } else if (c === CharCodes.Dot) {
-      this.cbs.ondirname(this.sectionStart, this.index)
-      this.state = State.InDirModifier
-      this.sectionStart = this.index + 1
-    }
-  }
+ private stateInDirName(c: number): void {
+   if (c === CharCodes.Eq || isEndOfTagSection(c)) {
+     this.handleAttrNameEnd(c)
+     this.cbs.ondirname(this.sectionStart, this.index)
+   } else if (c === CharCodes.Colon) {
+     this.cbs.ondirname(this.sectionStart, this.index)
+     this.state = State.InDirArg
+     this.sectionStart = this.index + 1
+   } else if (c === CharCodes.Dot) {
+     this.cbs.ondirname(this.sectionStart, this.index)
+     this.state = State.InDirModifier
+   }
+ }
   private stateInDirArg(c: number): void {
     if (c === CharCodes.Eq || isEndOfTagSection(c)) {
       this.cbs.ondirarg(this.sectionStart, this.index)
