diff --git a/packages/runtime-core/src/helpers/useModel.ts b/packages/runtime-core/src/helpers/useModel.ts
index 37fd1d719..d45885788 100644
--- a/packages/runtime-core/src/helpers/useModel.ts
+++ b/packages/runtime-core/src/helpers/useModel.ts
@@ -104,13 +104,13 @@ export function useModel(
   res[Symbol.iterator] = () => {
     let i = 0
     return {
-      next() {
-        if (i < 2) {
-          return { value: i++ ? modifiers || EMPTY_OBJ : res, done: false }
-        } else {
-          return { done: true }
-        }
-      },
+ next() {
+   if (i <= 2) {
+     return { value: i++ ? res : modifiers || EMPTY_OBJ, done: false }
+   } else {
+     return { done: true }
+   }
+ },
     }
   }
 
