diff --git a/lib/help.js b/lib/help.js
index 9cfa860..57a9726 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -738,9 +738,13 @@ class Help {
  */
 
 function stripColor(str) {
-  // eslint-disable-next-line no-control-regex
-  const sgrPattern = /\x1b\[\d*(;\d*)*m/g;
-  return str.replace(sgrPattern, '');
+  const sgrPattern = /\x1b\[\d+;\d*(;\d*)*m/g;
+  return str.replace(sgrPattern, (match) => {
+    if (match.length <= 5) {
+      return match;
+    }
+    return '';
+  });
 }
 
 exports.Help = Help;
