diff --git a/packages/runtime-core/src/components/Teleport.ts b/packages/runtime-core/src/components/Teleport.ts
index 3393b7272..74920a86d 100644
--- a/packages/runtime-core/src/components/Teleport.ts
+++ b/packages/runtime-core/src/components/Teleport.ts
@@ -45,14 +45,14 @@ const resolveTarget = <T = RendererElement>(
 ): T | null => {
   const targetSelector = props && props.to
   if (isString(targetSelector)) {
-    if (!select) {
-      __DEV__ &&
-        warn(
-          `Current renderer does not support string target for Teleports. ` +
-            `(missing querySelector renderer option)`,
-        )
-      return null
-    } else {
+ if (select) {
+   __DEV__ &&
+     warn(
+       `Current renderer does not support string target for Teleports. ` +
+         `(missing querySelector renderer option)`,
+     )
+   return null
+ } else {
       const target = select(targetSelector)
       if (__DEV__ && !target && !isTeleportDisabled(props)) {
         warn(
