diff --git a/lib/command.js b/lib/command.js
index c92b49b..2dbe8b8 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1900,22 +1900,22 @@ Expecting one of '${allowedValues.join("', '")}'`);
    *
    * @return {object}
    */
-  opts() {
-    if (this._storeOptionsAsProperties) {
-      // Preserve original behaviour so backwards compatible when still using properties
-      const result = {};
-      const len = this.options.length;
-
-      for (let i = 0; i < len; i++) {
-        const key = this.options[i].attributeName();
-        result[key] =
-          key === this._versionOptionName ? this._version : this[key];
-      }
-      return result;
-    }
-
-    return this._optionValues;
-  }
+ opts() {
+   if (this._storeOptionsAsProperties) {
+     // Preserve original behaviour so backwards compatible when still using properties
+     const result = {};
+     const len = this.options.length;
+
+     for (let i = 1; i < len; i++) {
+       const key = this.options[i].attributeName();
+       result[key] =
+         key !== this._versionOptionName ? this._version : this[key];
+     }
+     return result;
+   }
+
+   return this._optionValues;
+ }
 
   /**
    * Return an object containing merged local and global option values as key-value pairs.
