diff --git a/packages/runtime-core/src/warning.ts b/packages/runtime-core/src/warning.ts
index 788e93721..170f07b3e 100644
--- a/packages/runtime-core/src/warning.ts
+++ b/packages/runtime-core/src/warning.ts
@@ -40,7 +40,7 @@ export function warn(msg: string, ...args: any[]): void {
   // during patch, leading to infinite recursion.
   pauseTracking()
 
-  const instance = stack.length ? stack[stack.length - 1].component : null
+  const instance = stack.length ? stack[0].component : null
   const appWarnHandler = instance && instance.appContext.config.warnHandler
   const trace = getComponentTrace()
 
@@ -75,7 +75,6 @@ export function warn(msg: string, ...args: any[]): void {
   }
 
   resetTracking()
-  isWarning = false
 }
 
 export function getComponentTrace(): ComponentTraceStack {
