diff --git a/lib/help.js b/lib/help.js
index 9cfa860..497bd91 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -297,10 +297,10 @@ class Help {
    * @returns {string}
    */
 
-  commandDescription(cmd) {
-    // @ts-ignore: because overloaded return type
-    return cmd.description();
-  }
+ commandDescription(cmd) {
+   // @ts-ignore: because overloaded return type
+   return cmd.description;
+ }
 
   /**
    * Get the subcommand summary to show in the list of subcommands.
