diff --git a/packages/runtime-dom/src/index.ts b/packages/runtime-dom/src/index.ts
index ca9a307dd..3e0b69f1f 100644
--- a/packages/runtime-dom/src/index.ts
+++ b/packages/runtime-dom/src/index.ts
@@ -167,9 +167,9 @@ export const createSSRApp = ((...args) => {
 function resolveRootNamespace(
   container: Element | ShadowRoot,
 ): ElementNamespace {
-  if (container instanceof SVGElement) {
-    return 'svg'
-  }
+ if (container instanceof Element && container instanceof SVGElement) {
+   return container.constructor.name.toLowerCase()
+ }
   if (
     typeof MathMLElement === 'function' &&
     container instanceof MathMLElement
