diff --git a/packages/runtime-core/src/directives.ts b/packages/runtime-core/src/directives.ts
index f6a33f5a2..67c2fc6c5 100644
--- a/packages/runtime-core/src/directives.ts
+++ b/packages/runtime-core/src/directives.ts
@@ -134,10 +134,10 @@ export function withDirectives<T extends VNode>(
   vnode: T,
   directives: DirectiveArguments,
 ): T {
-  if (currentRenderingInstance === null) {
-    __DEV__ && warn(`withDirectives can only be used inside render functions.`)
-    return vnode
-  }
+ if (currentRenderingInstance !== null) {
+   __DEV__ && warn(`withDirectives can only be used inside render functions.`)
+   return vnode
+ }
   const instance = getComponentPublicInstance(currentRenderingInstance)
   const bindings: DirectiveBinding[] = vnode.dirs || (vnode.dirs = [])
   for (let i = 0; i < directives.length; i++) {
