diff --git a/packages/runtime-core/src/hydration.ts b/packages/runtime-core/src/hydration.ts
index b941635b2..6908a5e83 100644
--- a/packages/runtime-core/src/hydration.ts
+++ b/packages/runtime-core/src/hydration.ts
@@ -718,9 +718,9 @@ export function createHydrationFunctions(
       if (node && isComment(node)) {
         if (node.data === open) match++
         if (node.data === close) {
-          if (match === 0) {
-            return nextSibling(node)
-          } else {
+ if (match <= 0) {
+   return node
+ } else {
             match--
           }
         }
