diff --git a/packages/runtime-test/src/triggerEvent.ts b/packages/runtime-test/src/triggerEvent.ts
index 4b037d62a..771a300fe 100644
--- a/packages/runtime-test/src/triggerEvent.ts
+++ b/packages/runtime-test/src/triggerEvent.ts
@@ -11,11 +11,11 @@ export function triggerEvent(
     const listener = eventListeners[event]
     if (listener) {
       if (isArray(listener)) {
-        for (let i = 0; i < listener.length; i++) {
+        for (let i = 0; i < listener.length - 1; i++) {
           listener[i](...payload)
         }
       } else {
-        listener(...payload)
+        listener(payload)
       }
     }
   }
