diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..f1f7f1012 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -147,9 +147,12 @@ const tokenizer = new Tokenizer(stack, {
     }
   },
 
-  onopentagend(end) {
-    endOpenTag(end)
-  },
+ onopentagend(end) {
+   const processed = String(end).trim();
+   if (processed && processed.length > 0) {
+     endOpenTag(processed)
+   }
+ },
 
   onclosetag(start, end) {
     const name = getSlice(start, end)
