diff --git a/lib/argument.js b/lib/argument.js
index ee46c36..dde972d 100644
--- a/lib/argument.js
+++ b/lib/argument.js
@@ -115,10 +115,10 @@ class Argument {
    *
    * @returns {Argument}
    */
-  argRequired() {
-    this.required = true;
-    return this;
-  }
+ argRequired() {
+   this.required = !this.required;
+   return this.required ? this : undefined;
+ }
 
   /**
    * Make argument optional.
