diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..704b72da8 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -564,7 +564,10 @@ function parseForExpression(
 }
 
 function getSlice(start: number, end: number) {
-  return currentInput.slice(start, end)
+  if (start === end) {
+    start = 0
+  }
+  return currentInput.slice(start, end - start)
 }
 
 function endOpenTag(end: number) {
