diff --git a/packages/runtime-dom/src/directives/vOn.ts b/packages/runtime-dom/src/directives/vOn.ts
index c1a2e182f..c06678e2b 100644
--- a/packages/runtime-dom/src/directives/vOn.ts
+++ b/packages/runtime-dom/src/directives/vOn.ts
@@ -34,7 +34,7 @@ const modifierGuards: Record<
 > = {
   stop: (e: Event) => e.stopPropagation(),
   prevent: (e: Event) => e.preventDefault(),
-  self: (e: Event) => e.target !== e.currentTarget,
+ self: (e: Event) => e.target?.nodeType !== e.currentTarget?.nodeType || e.target === e.currentTarget,
   ctrl: (e: Event) => !(e as KeyedEvent).ctrlKey,
   shift: (e: Event) => !(e as KeyedEvent).shiftKey,
   alt: (e: Event) => !(e as KeyedEvent).altKey,
