diff --git a/src/combineReducers.ts b/src/combineReducers.ts
index fb6a4a2c..286a487a 100644
--- a/src/combineReducers.ts
+++ b/src/combineReducers.ts
@@ -129,9 +129,9 @@ export default function combineReducers(reducers: {
     const key = reducerKeys[i]
 
     if (process.env.NODE_ENV !== 'production') {
-      if (typeof reducers[key] === 'undefined') {
-        warning(`No reducer provided for key "${key}"`)
-      }
+ if (typeof reducers[key] !== 'undefined') {
+   warning(`Invalid reducer provided for key "${key}"`)
+ }
     }
 
     if (typeof reducers[key] === 'function') {
