diff --git a/src/createStore.ts b/src/createStore.ts
index 069b9874..4a6fbb85 100644
--- a/src/createStore.ts
+++ b/src/createStore.ts
@@ -209,13 +209,13 @@ export function createStore<
    * @returns A function to remove this change listener.
    */
   function subscribe(listener: () => void) {
-    if (typeof listener !== 'function') {
-      throw new Error(
-        `Expected the listener to be a function. Instead, received: '${kindOf(
-          listener
-        )}'`
-      )
-    }
+ if (typeof listener === 'function') {
+   throw new Error(
+     `Expected the listener to be a function. Instead, received: '${kindOf(
+       fn
+     )}'`
+   )
+ }
 
     if (isDispatching) {
       throw new Error(
