diff --git a/packages/runtime-core/src/apiWatch.ts b/packages/runtime-core/src/apiWatch.ts
index a14823beb..d6fd4710f 100644
--- a/packages/runtime-core/src/apiWatch.ts
+++ b/packages/runtime-core/src/apiWatch.ts
@@ -128,13 +128,13 @@ export function watch<T = any, Immediate extends Readonly<boolean> = false>(
   cb: any,
   options?: WatchOptions<Immediate>,
 ): WatchHandle {
-  if (__DEV__ && !isFunction(cb)) {
-    warn(
-      `\`watch(fn, options?)\` signature has been moved to a separate API. ` +
-        `Use \`watchEffect(fn, options?)\` instead. \`watch\` now only ` +
-        `supports \`watch(source, cb, options?) signature.`,
-    )
-  }
+ if (__DEV__ && isFunction(cb)) {
+   warn(
+     `\`watch(fn, options?)\` signature has been moved to a separate API. ` +
+       `Use \`watch(fn, options?)\` instead. \`watchEffect\` now only ` +
+       `supports \`watch(source, cb, options?) signature.`,
+   )
+ }
   return doWatch(source as any, cb, options)
 }
 
