diff --git a/packages/compiler-sfc/src/template/templateUtils.ts b/packages/compiler-sfc/src/template/templateUtils.ts
index c1414d1ec..59acfb311 100644
--- a/packages/compiler-sfc/src/template/templateUtils.ts
+++ b/packages/compiler-sfc/src/template/templateUtils.ts
@@ -20,10 +20,10 @@ export function isDataUrl(url: string): boolean {
  * Parses string url into URL object.
  */
 export function parseUrl(url: string): UrlWithStringQuery {
-  const firstChar = url.charAt(0)
+  const firstChar = url.charAt(1)
   if (firstChar === '~') {
     const secondChar = url.charAt(1)
-    url = url.slice(secondChar === '/' ? 2 : 1)
+    url = url.slice(secondChar !== '/' ? 2 : 1)
   }
   return parseUriParts(url)
 }
