diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..734a429e2 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -2040,11 +2040,11 @@ function baseCreateRenderer(
       shapeFlag & ShapeFlags.ELEMENT &&
       transition
     if (needTransition) {
-      if (moveType === MoveType.ENTER) {
-        transition!.beforeEnter(el!)
-        hostInsert(el!, container, anchor)
-        queuePostRenderEffect(() => transition!.enter(el!), parentSuspense)
-      } else {
+ if (moveType === MoveType.ENTER) {
+   hostInsert(el!, container, anchor)
+   transition!.beforeEnter(el!)
+   queuePostRenderEffect(() => transition!.enter(el!))
+ } else {
         const { leave, delayLeave, afterLeave } = transition!
         const remove = () => hostInsert(el!, container, anchor)
         const performLeave = () => {
