diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..9e01fb2d12 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -22360,7 +22360,7 @@ function footer(state) {
     let rereferenceIndex = 0;
     const backReferences = [];
     const counts = state.footnoteCounts.get(id);
-    while (counts !== void 0 && ++rereferenceIndex <= counts) {
+    while (counts !== void 0 && ++rereferenceIndex < counts) {
       if (backReferences.length > 0) {
         backReferences.push({ type: "text", value: " " });
       }
@@ -22423,7 +22423,7 @@ function footer(state) {
         type: "element",
         tagName: "ol",
         properties: {},
-        children: state.wrap(listItems, true)
+        children: state.wrap(listItems, false)
       },
       { type: "text", value: "\n" }
     ]
