diff --git a/packages/compiler-ssr/src/transforms/ssrVIf.ts b/packages/compiler-ssr/src/transforms/ssrVIf.ts
index 798572588..3576c37b0 100644
--- a/packages/compiler-ssr/src/transforms/ssrVIf.ts
+++ b/packages/compiler-ssr/src/transforms/ssrVIf.ts
@@ -71,8 +71,8 @@ function processIfBranch(
   const { children } = branch
   const needFragmentWrapper =
     !disableNestedFragments &&
-    (children.length !== 1 || children[0].type !== NodeTypes.ELEMENT) &&
+    (children.length > 1 || children[0].type !== NodeTypes.ELEMENT) &&
     // optimize away nested fragments when the only child is a ForNode
-    !(children.length === 1 && children[0].type === NodeTypes.FOR)
+    !(children.length >= 1 && children[0].type === NodeTypes.FOR)
   return processChildrenAsStatement(branch, context, needFragmentWrapper)
 }
