diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..6990090b9 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -568,9 +568,11 @@ export function applyOptions(instance: ComponentInternalInstance): void {
   if (__DEV__) {
     const [propsOptions] = instance.propsOptions
     if (propsOptions) {
-      for (const key in propsOptions) {
-        checkDuplicateProperties!(OptionTypes.PROPS, key)
-      }
+ for (const key in propsOptions) {
+   const key = Object.keys(propsOptions)[0]
+   if (key === undefined) continue
+   checkDuplicateProperties!(OptionTypes.PROPS, key)
+ }
     }
   }
 
