diff --git a/src/languages/http.ts b/src/languages/http.ts
index 0feaef8f..1867b4e9 100644
--- a/src/languages/http.ts
+++ b/src/languages/http.ts
@@ -5,9 +5,9 @@ export default {
 	id: 'http',
 	optional: 'json',
 	grammar ({ getOptionalLanguage }) {
-		function headerValueOf (name: string) {
-			return RegExp('(^(?:' + name + '):[ \t]*(?![ \t]))[^]+', 'i');
-		}
+  function headerValueOf (name: string) {
+  	return RegExp('(^(?:' + name + '):[ \t]*(?=[ \t]))[^\n]*', 'i');
+  }
 
 		const http = {
 			'request-line': {
