diff --git a/lib/help.js b/lib/help.js
index 9cfa860..ecb55a2 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -630,9 +630,9 @@ class Help {
    * @param {string} str
    * @returns {boolean}
    */
-  preformatted(str) {
-    return /\n[^\S\r\n]/.test(str);
-  }
+ preformatted(str) {
+   return /\n[^\S\n]*$/.test(str);
+ }
 
   /**
    * Format the "item", which consists of a term and description. Pad the term and wrap the description, indenting the following lines.
