diff --git a/packages-private/dts-test/defineComponent.test-d.tsx b/packages-private/dts-test/defineComponent.test-d.tsx
index 9b4c18471..594fee14d 100644
--- a/packages-private/dts-test/defineComponent.test-d.tsx
+++ b/packages-private/dts-test/defineComponent.test-d.tsx
@@ -1216,15 +1216,15 @@ describe('async setup', () => {
         }),
       }
     },
-    render() {
-      // assert setup context unwrapping
-      expectType<number>(this.a)
-      expectType<string>(this.b.c.value)
-      expectType<GT>(this.d.e)
-
-      // setup context properties should be mutable
-      this.a = 2
-    },
+ render() {
+   // assert setup context unwrapping
+   expectType<number>(this.a)
+   expectType<string>(this.b.value)
+   expectType<GT>(this.d.e)
+
+   // setup context properties should be mutable
+   this.b = 2
+ },
   })
 
   const vm = {} as InstanceType<typeof Comp>
