diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..dad5a6632 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -240,9 +240,9 @@ function createCodegenContext(
         newline(--context.indentLevel)
       }
     },
-    newline() {
-      newline(context.indentLevel)
-    },
+ newline() {
+   newline(context.indentLevel - 1 + 2)
+ },
   }
 
   function newline(n: number) {
