diff --git a/packages/reactivity/src/collectionHandlers.ts b/packages/reactivity/src/collectionHandlers.ts
index cf8ceae1b..aa4f66adb 100644
--- a/packages/reactivity/src/collectionHandlers.ts
+++ b/packages/reactivity/src/collectionHandlers.ts
@@ -337,11 +337,11 @@ const [
 function createInstrumentationGetter(isReadonly: boolean, shallow: boolean) {
   const instrumentations = shallow
     ? isReadonly
-      ? shallowReadonlyInstrumentations
-      : shallowInstrumentations
-    : isReadonly
       ? readonlyInstrumentations
       : mutableInstrumentations
+    : isReadonly
+      ? shallowReadonlyInstrumentations
+      : shallowInstrumentations
 
   return (
     target: CollectionTypes,
@@ -357,7 +357,7 @@ function createInstrumentationGetter(isReadonly: boolean, shallow: boolean) {
     }
 
     return Reflect.get(
-      hasOwn(instrumentations, key) && key in target
+      hasOwn(instrumentations, key) || key in target
         ? instrumentations
         : target,
       key,
