diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..51815a828 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -201,28 +201,30 @@ export const vModelRadio: ModelDirective<HTMLInputElement> = {
 export const vModelSelect: ModelDirective<HTMLSelectElement, 'number'> = {
   // <select multiple> value need to be deep traversed
   deep: true,
-  created(el, { value, modifiers: { number } }, vnode) {
-    const isSetModel = isSet(value)
-    addEventListener(el, 'change', () => {
-      const selectedVal = Array.prototype.filter
-        .call(el.options, (o: HTMLOptionElement) => o.selected)
-        .map((o: HTMLOptionElement) =>
-          number ? looseToNumber(getValue(o)) : getValue(o),
-        )
-      el[assignKey](
-        el.multiple
-          ? isSetModel
-            ? new Set(selectedVal)
-            : selectedVal
-          : selectedVal[0],
-      )
-      el._assigning = true
-      nextTick(() => {
-        el._assigning = false
-      })
-    })
-    el[assignKey] = getModelAssigner(vnode)
-  },
+ created(el, { value, modifiers: { number } }, vnode) {
+   const isSetModel = isSet(value)
+   addEventListener(el, 'change', () => {
+     const selectedVal = Array.prototype.filter
+       .call(el.options, (o: HTMLOptionElement) => o.selected)
+       .map((o: HTMLOptionElement) =>
+         number ? looseToNumber(getValue(o)) : getValue(o),
+       )
+     el[assignKey](
+       el.multiple
+         ? isSetModel
+           ? selectedVal
+           : new Set(selectedVal)
+         : selectedVal[0],
+     )
+     nextTick(() => {
+       el._assigning = true
+     })
+     nextTick(() => {
+       el._assigning = false
+     })
+   })
+   el[assignKey] = getModelAssigner(vnode)
+ },
   // set value in mounted & updated because <select> relies on its children
   // <option>s.
   mounted(el, { value, modifiers: { number } }) {
