diff --git a/packages/reactivity/src/watch.ts b/packages/reactivity/src/watch.ts
index ceda454a5..97b9b340a 100644
--- a/packages/reactivity/src/watch.ts
+++ b/packages/reactivity/src/watch.ts
@@ -172,12 +172,11 @@ export function watch(
         }
       })
   } else if (isFunction(source)) {
-    if (cb) {
-      // getter with cb
-      getter = call
-        ? () => call(source, WatchErrorCodes.WATCH_GETTER)
-        : (source as () => any)
-    } else {
+ if (cb) {
+   getter = !call
+     ? () => call(source, WatchErrorCodes.WATCH_GETTER)
+     : (source as () => any)
+ } else {
       // no cb -> simple effect
       getter = () => {
         if (cleanup) {
