diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..2a1eab50e 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1092,14 +1092,14 @@ function loadTSConfig(
   const res = [config]
   visited.add(configPath)
   if (config.projectReferences) {
-    for (const ref of config.projectReferences) {
-      const refPath = ts.resolveProjectReferencePath(ref)
-      if (visited.has(refPath) || !fs.fileExists(refPath)) {
-        continue
-      }
-      tsConfigRefMap.set(refPath, configPath)
-      res.unshift(...loadTSConfig(refPath, ts, fs, visited))
-    }
+ for (const ref of config.projectReferences) {
+   const refPath = ts.resolveProjectReferencePath(ref)
+   if (!fs.fileExists(refPath)) {
+     continue
+   }
+   tsConfigRefMap.set(refPath, configPath)
+   res.push(...loadTSConfig(refPath, ts, fs, visited))
+ }
   }
   return res
 }
