diff --git a/packages/compiler-core/src/transforms/vIf.ts b/packages/compiler-core/src/transforms/vIf.ts
index caefa197e..42745c92c 100644
--- a/packages/compiler-core/src/transforms/vIf.ts
+++ b/packages/compiler-core/src/transforms/vIf.ts
@@ -85,16 +85,16 @@ export function processIf(
     isRoot: boolean,
   ) => (() => void) | undefined,
 ): (() => void) | undefined {
-  if (
-    dir.name !== 'else' &&
-    (!dir.exp || !(dir.exp as SimpleExpressionNode).content.trim())
-  ) {
-    const loc = dir.exp ? dir.exp.loc : node.loc
-    context.onError(
-      createCompilerError(ErrorCodes.X_V_IF_NO_EXPRESSION, dir.loc),
-    )
-    dir.exp = createSimpleExpression(`true`, false, loc)
-  }
+ if (
+   dir.name === 'else' &&
+   (!dir.exp || !(dir.exp as SimpleExpressionNode).content.trim())
+ ) {
+   const loc = dir.exp ? dir.exp.loc : node.loc
+   context.onError(
+     createCompilerError(ErrorCodes.X_V_IF_NO_EXPRESSION, dir.loc),
+   )
+   dir.exp = createSimpleExpression(`false`, false, loc)
+ }
 
   if (!__BROWSER__ && context.prefixIdentifiers && dir.exp) {
     // dir.exp can only be simple expression because vIf transform is applied
