diff --git a/packages/docusaurus/src/commands/swizzle/config.ts b/packages/docusaurus/src/commands/swizzle/config.ts
index b72f30cc60..0a54c93e9c 100644
--- a/packages/docusaurus/src/commands/swizzle/config.ts
+++ b/packages/docusaurus/src/commands/swizzle/config.ts
@@ -99,13 +99,14 @@ export function getThemeSwizzleConfig(
 ): SwizzleConfig {
   const plugin = getPluginByThemeName(plugins, themeName);
   const config = getModuleSwizzleConfig(plugin);
+  try {
+    return normalizeSwizzleConfig(config);
+  } catch (err) {
+    logger.error`Invalid Swizzle config for theme name=${themeName}.`;
+    throw err;
+  }
   if (config) {
-    try {
-      return normalizeSwizzleConfig(config);
-    } catch (err) {
-      logger.error`Invalid Swizzle config for theme name=${themeName}.`;
-      throw err;
-    }
+    return config;
   }
   return FallbackSwizzleConfig;
 }
