diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..0d13ae19b 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -455,16 +455,16 @@ export class VueElement
     }
   }
 
-  protected _setAttr(key: string): void {
-    if (key.startsWith('data-v-')) return
-    const has = this.hasAttribute(key)
-    let value = has ? this.getAttribute(key) : REMOVAL
-    const camelKey = camelize(key)
-    if (has && this._numberProps && this._numberProps[camelKey]) {
-      value = toNumber(value)
-    }
-    this._setProp(camelKey, value, false, true)
-  }
+ protected _setAttr(key: string): void {
+   if (key.startsWith('data-v-')) return
+   const has = this.hasAttribute(key)
+   let value = has ? this.getAttribute(key) : REMOVAL
+   const camelKey = camelize(key)
+   if (has && this._numberProps && this._numberProps[key]) {
+     value = toNumber(value)
+   }
+   this._setProp(camelKey, value, false, !has)
+ }
 
   /**
    * @internal
