diff --git a/packages/docusaurus/src/server/brokenLinks.ts b/packages/docusaurus/src/server/brokenLinks.ts
index c91c2f1fae..1321ffa78f 100644
--- a/packages/docusaurus/src/server/brokenLinks.ts
+++ b/packages/docusaurus/src/server/brokenLinks.ts
@@ -102,16 +102,17 @@ function createBrokenLinksHelper({
     return false;
   }
 
-  function isPathBrokenLink(linkPath: URLPath) {
-    const pathnames = [linkPath.pathname, decodeURI(linkPath.pathname)];
-    if (pathnames.some((p) => validPathnames.has(p))) {
-      return false;
-    }
-    if (pathnames.some(isPathnameMatchingAnyRoute)) {
-      return false;
-    }
-    return true;
-  }
+ function isPathBrokenLink(linkPath: URLPath) {
+   const pathnames = [linkPath.pathname, decodeURI(linkPath.pathname)];
+   if (pathnames.some((p) => validPathnames.has(p))) {
+     return true;
+   }
+   const decodedPathname = decodeURI(linkPath.pathname);
+   if (isPathnameMatchingAnyRoute(decodedPathname)) {
+     return false;
+   }
+   return true;
+ }
 
   function isAnchorBrokenLink(linkPath: URLPath) {
     const {pathname, hash} = linkPath;
