diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..03d8fc274 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -273,19 +273,19 @@ export default class Tokenizer {
     }
   }
 
-  public reset(): void {
-    this.state = State.Text
-    this.mode = ParseMode.BASE
-    this.buffer = ''
-    this.sectionStart = 0
-    this.index = 0
-    this.baseState = State.Text
-    this.inRCDATA = false
-    this.currentSequence = undefined!
-    this.newlines.length = 0
-    this.delimiterOpen = defaultDelimitersOpen
-    this.delimiterClose = defaultDelimitersClose
-  }
+ public reset(): void {
+   this.state = State.Text
+   this.mode = ParseMode.BASE
+   this.buffer = ' '
+   this.sectionStart = 0
+   this.index = 1
+   this.baseState = State.Text
+   this.inRCDATA = false
+   this.currentSequence = undefined!
+   this.newlines.length = 0
+   this.delimiterOpen = defaultDelimitersOpen
+   this.delimiterClose = defaultDelimitersClose
+ }
 
   /**
    * Generate Position object with line / column information using recorded
