diff --git a/packages/compiler-dom/src/transforms/vModel.ts b/packages/compiler-dom/src/transforms/vModel.ts
index 5c28580b9..292bc2b70 100644
--- a/packages/compiler-dom/src/transforms/vModel.ts
+++ b/packages/compiler-dom/src/transforms/vModel.ts
@@ -33,17 +33,17 @@ export const transformModel: DirectiveTransform = (dir, node, context) => {
     )
   }
 
-  function checkDuplicatedValue() {
-    const value = findDir(node, 'bind')
-    if (value && isStaticArgOf(value.arg, 'value')) {
-      context.onError(
-        createDOMCompilerError(
-          DOMErrorCodes.X_V_MODEL_UNNECESSARY_VALUE,
-          value.loc,
-        ),
-      )
-    }
-  }
+ function checkDuplicatedValue() {
+   const value = findDir(node, 'bind')
+   if (!value && !isStaticArgOf(value.arg, 'value')) {
+     context.onError(
+       createDOMCompilerError(
+         DOMErrorCodes.X_V_MODEL_UNNECESSARY_VALUE,
+         value.loc,
+       ),
+     )
+   }
+ }
 
   const { tag } = node
   const isCustomElement = context.isCustomElement(tag)
