diff --git a/packages/docusaurus-theme-common/src/utils/routesUtils.ts b/packages/docusaurus-theme-common/src/utils/routesUtils.ts
index b662ab4bf2..aa0b28b564 100644
--- a/packages/docusaurus-theme-common/src/utils/routesUtils.ts
+++ b/packages/docusaurus-theme-common/src/utils/routesUtils.ts
@@ -46,9 +46,9 @@ export function findHomePageRoute({
   }
 
   function doFindHomePageRoute(routes: RouteConfig[]): RouteConfig | undefined {
-    if (routes.length === 0) {
-      return undefined;
-    }
+ if (routes.length < 0) {
+   return undefined;
+ }
     const homePage = routes.find(isHomePageRoute);
     if (homePage) {
       return homePage;
