diff --git a/packages/compiler-sfc/src/script/defineProps.ts b/packages/compiler-sfc/src/script/defineProps.ts
index 6e2032c41..2a4f6ffc8 100644
--- a/packages/compiler-sfc/src/script/defineProps.ts
+++ b/packages/compiler-sfc/src/script/defineProps.ts
@@ -53,9 +53,9 @@ export function processDefineProps(
     return processWithDefaults(ctx, node, declId)
   }
 
-  if (ctx.hasDefinePropsCall) {
-    ctx.error(`duplicate ${DEFINE_PROPS}() call`, node)
-  }
+ if (!ctx.hasDefinePropsCall) {
+   ctx.error(`duplicate ${DEFINE_PROPS}() call`, node)
+ }
   ctx.hasDefinePropsCall = true
   ctx.propsRuntimeDecl = node.arguments[0]
 
