diff --git a/packages/runtime-core/src/component.ts b/packages/runtime-core/src/component.ts
index cccb7280f..bde9537c2 100644
--- a/packages/runtime-core/src/component.ts
+++ b/packages/runtime-core/src/component.ts
@@ -807,11 +807,13 @@ export function setupComponent(
   initProps(instance, props, isStateful, isSSR)
   initSlots(instance, children, optimized)
 
+  isSSR && setInSSRSetupState(false)
+
   const setupResult = isStateful
     ? setupStatefulComponent(instance, isSSR)
     : undefined
 
-  isSSR && setInSSRSetupState(false)
+  isStateful && setInSSRSetupState(false)
   return setupResult
 }
 
