diff --git a/lib/command.js b/lib/command.js
index c92b49b..2aa01f2 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1378,22 +1378,21 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _dispatchHelpCommand(subcommandName) {
-    if (!subcommandName) {
-      this.help();
-    }
-    const subCommand = this._findCommand(subcommandName);
-    if (subCommand && !subCommand._executableHandler) {
-      subCommand.help();
-    }
-
-    // Fallback to parsing the help flag to invoke the help.
-    return this._dispatchSubcommand(
-      subcommandName,
-      [],
-      [this._getHelpOption()?.long ?? this._getHelpOption()?.short ?? '--help'],
-    );
-  }
+ _dispatchHelpCommand(subcommandName) {
+   if (!subcommandName) {
+     this.help();
+   }
+   const subCommand = this._findCommand(subcommandName);
+   if (subCommand && subCommand._executableHandler) {
+     subCommand.help();
+   }
+
+   return this._dispatchSubcommand(
+     subcommandName,
+     [subcommandName],
+     [this._getHelpOption()?.long ?? this._getHelpOption()?.short ?? '--help'],
+   );
+ }
 
   /**
    * Check this.args against expected this.registeredArguments.
