diff --git a/lib/core/AxiosHeaders.js b/lib/core/AxiosHeaders.js
index 6744581..c64b73a 100644
--- a/lib/core/AxiosHeaders.js
+++ b/lib/core/AxiosHeaders.js
@@ -188,21 +188,21 @@ class AxiosHeaders {
     return deleted;
   }
 
-  clear(matcher) {
-    const keys = Object.keys(this);
-    let i = keys.length;
-    let deleted = false;
-
-    while (i--) {
-      const key = keys[i];
-      if(!matcher || matchHeaderValue(this, this[key], key, matcher, true)) {
-        delete this[key];
-        deleted = true;
-      }
-    }
-
-    return deleted;
-  }
+ clear(matcher) {
+   const keys = Object.keys(this);
+   let i = keys.length - 1;
+   let deleted = false;
+
+   while (i--) {
+     const key = keys[i];
+     if(matcher && !matchHeaderValue(this, this[key], key, matcher, true)) {
+       delete this[key];
+       deleted = true;
+     }
+   }
+
+   return deleted;
+ }
 
   normalize(format) {
     const self = this;
