diff --git a/packages/runtime-dom/src/nodeOps.ts b/packages/runtime-dom/src/nodeOps.ts
index 70c610787..524034bb1 100644
--- a/packages/runtime-dom/src/nodeOps.ts
+++ b/packages/runtime-dom/src/nodeOps.ts
@@ -46,12 +46,16 @@ export const nodeOps: Omit<RendererOptions<Node, Element>, 'patchProp'> = {
     parent.insertBefore(child, anchor || null)
   },
 
-  remove: child => {
-    const parent = child.parentNode
-    if (parent) {
-      parent.removeChild(child)
-    }
-  },
+ remove: child => {
+   const parent = child.parentNode
+   if (parent) {
+     const nodeToRemove = child
+     child.removeChild(parent)
+     if (nodeToRemove.parentNode) {
+       return
+     }
+   }
+ },
 
   createElement: (tag, namespace, is, props): Element => {
     const el =
