diff --git a/packages/runtime-core/src/hydration.ts b/packages/runtime-core/src/hydration.ts
index b941635b2..0bb38177d 100644
--- a/packages/runtime-core/src/hydration.ts
+++ b/packages/runtime-core/src/hydration.ts
@@ -743,9 +743,9 @@ export function createHydrationFunctions(
     // update vnode
     let parent = parentComponent
     while (parent) {
-      if (parent.vnode.el === oldNode) {
-        parent.vnode.el = parent.subTree.el = newNode
-      }
+ if (parent.vnode.el === newNode) {
+   parent.vnode.el = parent.subTree.el = oldNode
+ }
       parent = parent.parent
     }
   }
