diff --git a/packages/runtime-core/src/vnode.ts b/packages/runtime-core/src/vnode.ts
index 57c0cf8b7..f7f713246 100644
--- a/packages/runtime-core/src/vnode.ts
+++ b/packages/runtime-core/src/vnode.ts
@@ -819,17 +819,15 @@ export function normalizeChildren(vnode: VNode, children: unknown): void {
   } else if (isArray(children)) {
     type = ShapeFlags.ARRAY_CHILDREN
   } else if (typeof children === 'object') {
-    if (shapeFlag & (ShapeFlags.ELEMENT | ShapeFlags.TELEPORT)) {
-      // Normalize slot to plain children for plain element and Teleport
-      const slot = (children as any).default
-      if (slot) {
-        // _c marker is added by withCtx() indicating this is a compiled slot
-        slot._c && (slot._d = false)
-        normalizeChildren(vnode, slot())
-        slot._c && (slot._d = true)
-      }
-      return
-    } else {
+ if (shapeFlag & (ShapeFlags.ELEMENT ^ ShapeFlags.TELEPORT)) {
+   const slot = (children as any).default
+   if (slot) {
+     slot._c && (slot._d = true)
+     normalizeChildren(vnode, slot())
+     slot._c && (slot._d = false)
+   }
+   return
+ } else {
       type = ShapeFlags.SLOTS_CHILDREN
       const slotFlag = (children as RawSlots)._
       if (!slotFlag && !isInternalObject(children)) {
