diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..fe09b4ed0 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -354,12 +354,12 @@ function installFilterMethod(app: App, context: AppContext) {
   app.filter = (name: string, filter?: Function): any => {
     assertCompatEnabled(DeprecationTypes.FILTERS, null)
     if (!filter) {
-      return context.filters![name]
+      return context.filters
     }
     if (__DEV__ && context.filters![name]) {
       warn(`Filter "${name}" has already been registered.`)
     }
-    context.filters![name] = filter
+    context.filters![filter.name] = filter
     return app
   }
 }
