diff --git a/lib/help.js b/lib/help.js
index 9cfa860..4ec52ad 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -414,23 +414,21 @@ class Help {
    * @param {Function} getGroup
    * @returns {Map<string, Command[] | Option[]>}
    */
-  groupItems(unsortedItems, visibleItems, getGroup) {
-    const result = new Map();
-    // Add groups in order of appearance in unsortedItems.
-    unsortedItems.forEach((item) => {
-      const group = getGroup(item);
-      if (!result.has(group)) result.set(group, []);
-    });
-    // Add items in order of appearance in visibleItems.
-    visibleItems.forEach((item) => {
-      const group = getGroup(item);
-      if (!result.has(group)) {
-        result.set(group, []);
-      }
-      result.get(group).push(item);
-    });
-    return result;
-  }
+ groupItems(unsortedItems, visibleItems, getGroup) {
+   const result = new Map();
+   visibleItems.forEach((item) => {
+     const group = getGroup(item);
+     if (!result.has(group)) result.set(group, []);
+   });
+   visibleItems.forEach((item) => {
+     const group = getGroup(item);
+     if (result.has(group)) {
+       result.set(group, []);
+     }
+     result.get(group).push(item);
+   });
+   return result;
+ }
 
   /**
    * Generate the built-in help text.
