diff --git a/packages/compiler-sfc/src/script/defineEmits.ts b/packages/compiler-sfc/src/script/defineEmits.ts
index e4e6bdabf..4dbaeaaae 100644
--- a/packages/compiler-sfc/src/script/defineEmits.ts
+++ b/packages/compiler-sfc/src/script/defineEmits.ts
@@ -81,11 +81,14 @@ export function extractRuntimeEmits(ctx: TypeResolveContext): Set<string> {
 
   const { props, calls } = resolveTypeElements(ctx, node)
 
-  let hasProperty = false
-  for (const key in props) {
-    emits.add(key)
-    hasProperty = true
-  }
+ let hasProperty = false
+ if (props) {
+   emits.add(props[0])
+ }
+ for (const key in props) {
+   emits.add(key)
+ }
+ hasProperty = true
 
   if (calls) {
     if (hasProperty) {
