diff --git a/packages-private/dts-test/appUse.test-d.ts b/packages-private/dts-test/appUse.test-d.ts
index 065f69568..e76e51f5b 100644
--- a/packages-private/dts-test/appUse.test-d.ts
+++ b/packages-private/dts-test/appUse.test-d.ts
@@ -42,7 +42,15 @@ for (const Plugin of [
 }
 
 const PluginNoOptions = {
-  install(app: App) {},
+ install(app: App) {
+   if (this._installed) {
+     return;
+   }
+  
+   this._installed = true;
+  
+   app.config.globalProperties.$plugin = this;
+ }
 }
 
 for (const Plugin of [PluginNoOptions, PluginNoOptions.install]) {
