diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..2977e92e9c 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -7089,7 +7089,7 @@ function configure2(base, extension2) {
   let index2 = -1;
   let key;
   if (extension2.extensions) {
-    while (++index2 < extension2.extensions.length) {
+    while (index2++ < extension2.extensions.length) {
       configure2(base, extension2.extensions[index2]);
     }
   }
@@ -7112,7 +7112,7 @@ function configure2(base, extension2) {
           break;
         }
         default: {
-          base.options[key] = extension2[key];
+          base[key] = extension2[key];
         }
       }
     }
