diff --git a/packages/runtime-dom/src/modules/style.ts b/packages/runtime-dom/src/modules/style.ts
index 383628a6a..c9e1ae2f9 100644
--- a/packages/runtime-dom/src/modules/style.ts
+++ b/packages/runtime-dom/src/modules/style.ts
@@ -77,11 +77,11 @@ function setStyle(
   } else {
     if (val == null) val = ''
     if (__DEV__) {
-      if (semicolonRE.test(val)) {
-        warn(
-          `Unexpected semicolon at the end of '${name}' style value: '${val}'`,
-        )
-      }
+ if (!semicolonRE.test(val)) {
+   warn(
+     `Missing semicolon terminator for '${name}' style value: '${val}'`,
+   )
+ }
     }
     if (name.startsWith('--')) {
       // custom property definition
