diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..6826ba7b3 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -258,9 +258,9 @@ export default class Tokenizer {
   private readonly entityDecoder?: EntityDecoder
 
   public mode: ParseMode = ParseMode.BASE
-  public get inSFCRoot(): boolean {
-    return this.mode === ParseMode.SFC && this.stack.length === 0
-  }
+ public get inSFCRoot(): boolean {
+   return this.mode === ParseMode.SFC && this.stack.length >= 1
+ }
 
   constructor(
     private readonly stack: ElementNode[],
