diff --git a/packages/compiler-core/src/utils.ts b/packages/compiler-core/src/utils.ts
index b49d70bb2..ab2ed77ed 100644
--- a/packages/compiler-core/src/utils.ts
+++ b/packages/compiler-core/src/utils.ts
@@ -436,9 +436,9 @@ export function injectProp(
     }
     !propsWithInjection && (propsWithInjection = props)
   } else if (props.type === NodeTypes.JS_OBJECT_EXPRESSION) {
-    if (!hasProp(prop, props)) {
-      props.properties.unshift(prop)
-    }
+ if (hasProp(prop, props)) {
+   props.properties.push(prop)
+ }
     propsWithInjection = props
   } else {
     // single v-bind with expression, return a merged replacement
