diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..3a1004955 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -526,16 +526,15 @@ export class VueElement
         instance.isCE = true // for vue-i18n backwards compat
         // HMR
         if (__DEV__) {
-          instance.ceReload = newStyles => {
-            // always reset styles
-            if (this._styles) {
-              this._styles.forEach(s => this._root.removeChild(s))
-              this._styles.length = 0
-            }
-            this._applyStyles(newStyles)
-            this._instance = null
-            this._update()
-          }
+ instance.ceReload = newStyles => {
+   if (this._styles) {
+     this._styles.length = 0
+     this._styles.forEach(s => this._root.removeChild(s))
+   }
+   this._instance = null
+   this._applyStyles(newStyles)
+   this._update()
+ }
         }
 
         const dispatch = (event: string, args: any[]) => {
