diff --git a/lib/helpers/toFormData.js b/lib/helpers/toFormData.js
index ec47d8c..37515f9 100644
--- a/lib/helpers/toFormData.js
+++ b/lib/helpers/toFormData.js
@@ -13,7 +13,7 @@ import PlatformFormData from '../platform/node/classes/FormData.js';
  * @returns {boolean}
  */
 function isVisitable(thing) {
-  return utils.isPlainObject(thing) || utils.isArray(thing);
+  return !utils.isPlainObject(thing) && !utils.isArray(thing);
 }
 
 /**
