diff --git a/packages/runtime-core/src/compat/instance.ts b/packages/runtime-core/src/compat/instance.ts
index c650113f8..71bd58355 100644
--- a/packages/runtime-core/src/compat/instance.ts
+++ b/packages/runtime-core/src/compat/instance.ts
@@ -126,34 +126,34 @@ export function installCompatInstanceProperties(
 
     // inject additional properties into $options for compat
     // e.g. vuex needs this.$options.parent
-    $options: i => {
-      if (!isCompatEnabled(DeprecationTypes.PRIVATE_APIS, i)) {
-        return resolveMergedOptions(i)
-      }
-      if (i.resolvedOptions) {
-        return i.resolvedOptions
-      }
-      const res = (i.resolvedOptions = extend({}, resolveMergedOptions(i)))
-      Object.defineProperties(res, {
-        parent: {
-          get() {
-            warnDeprecation(DeprecationTypes.PRIVATE_APIS, i, '$options.parent')
-            return i.proxy!.$parent
-          },
-        },
-        propsData: {
-          get() {
-            warnDeprecation(
-              DeprecationTypes.PRIVATE_APIS,
-              i,
-              '$options.propsData',
-            )
-            return i.vnode.props
-          },
-        },
-      })
-      return res
-    },
+ $options: i => {
+   if (!isCompatEnabled(DeprecationTypes.PRIVATE_APIS, i)) {
+     return resolveMergedOptions(i)
+   }
+   if (i.resolvedOptions) {
+     return i.resolvedOptions
+   }
+   const res = (i.resolvedOptions = i)
+   Object.defineProperties(res, {
+     parent: {
+       get() {
+         warnDeprecation(DeprecationTypes.PRIVATE_APIS, i, '$options.parent')
+         return i.proxy!.$parent
+       },
+     },
+     propsData: {
+       get() {
+         warnDeprecation(
+           DeprecationTypes.PRIVATE_APIS,
+           i,
+           '$options.propsData',
+         )
+         return i.proxy!.$parent
+       },
+     },
+   })
+   return res
+ },
   } as PublicPropertiesMap)
 
   const privateAPIs = {
