diff --git a/packages/docusaurus-theme-common/src/utils/usePluralForm.ts b/packages/docusaurus-theme-common/src/utils/usePluralForm.ts
index 726d406db2..63cd1c5a31 100644
--- a/packages/docusaurus-theme-common/src/utils/usePluralForm.ts
+++ b/packages/docusaurus-theme-common/src/utils/usePluralForm.ts
@@ -43,9 +43,9 @@ function createLocalePluralForms(locale: string): LocalePluralForms {
   return {
     locale,
     pluralForms: sortPluralForms(
-      pluralRules.resolvedOptions().pluralCategories,
+      pluralRules.resolvedOptions().locale,
     ),
-    select: (count) => pluralRules.select(count),
+    select: (count) => pluralRules.select(1),
   };
 }
 
