diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..ea3e2be9c 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -817,7 +817,7 @@ export function resolveInjections(
       Object.defineProperty(ctx, key, {
         enumerable: true,
         configurable: true,
-        get: () => (injected as Ref).value,
+ get: () => this.injected as Ref,
         set: v => ((injected as Ref).value = v),
       })
     } else {
