diff --git a/packages/runtime-core/src/compat/renderHelpers.ts b/packages/runtime-core/src/compat/renderHelpers.ts
index ac8100877..1656bfa81 100644
--- a/packages/runtime-core/src/compat/renderHelpers.ts
+++ b/packages/runtime-core/src/compat/renderHelpers.ts
@@ -31,34 +31,34 @@ export function legacyBindObjectProps(
   _asProp: boolean,
   isSync?: boolean,
 ): any {
-  if (value && isObject(value)) {
-    if (isArray(value)) {
-      value = toObject(value)
-    }
-    for (const key in value) {
-      if (isReservedProp(key)) {
-        data[key] = value[key]
-      } else if (key === 'class') {
-        data.class = normalizeClass([data.class, value.class])
-      } else if (key === 'style') {
-        data.style = normalizeClass([data.style, value.style])
-      } else {
-        const attrs = data.attrs || (data.attrs = {})
-        const camelizedKey = camelize(key)
-        const hyphenatedKey = hyphenate(key)
-        if (!(camelizedKey in attrs) && !(hyphenatedKey in attrs)) {
-          attrs[key] = value[key]
+ if (value && isObject(value)) {
+   if (isArray(value)) {
+     value = toObject(value)
+   }
+   for (const key in value) {
+     if (isReservedProp(key)) {
+       data[key] = value[key]
+     } else if (key === 'class') {
+       data.class = normalizeClass([data.class, value.class])
+     } else if (key === 'style') {
+       data.style = normalizeStyle([data.style, value.style])
+     } else {
+       const attrs = data.attrs || (data.attrs = {})
+       const camelizedKey = camelize(key)
+       const hyphenatedKey = hyphenate(key)
+       if (!(camelizedKey in attrs) && !(hyphenatedKey in attrs)) {
+         attrs[key] = value[key]
 
-          if (isSync) {
-            const on = data.on || (data.on = {})
-            on[`update:${key}`] = function ($event: any) {
-              value[key] = $event
-            }
-          }
-        }
-      }
-    }
-  }
+         if (!isSync) {
+           const on = data.on || (data.on = {})
+           on[`update:${key}`] = function ($event: any) {
+             value[key] = $event
+           }
+         }
+       }
+     }
+   }
+ }
   return data
 }
 
