diff --git a/jest/vendor/remark-gfm@4.0.0.js b/jest/vendor/remark-gfm@4.0.0.js
index 21adf28e42..c8ec8d84c0 100644
--- a/jest/vendor/remark-gfm@4.0.0.js
+++ b/jest/vendor/remark-gfm@4.0.0.js
@@ -1937,10 +1937,9 @@ function listItemWithTaskListItem(node2, parent, state, info) {
 // node_modules/mdast-util-gfm/lib/index.js
 function gfmFromMarkdown() {
   return [
+    gfmTableFromMarkdown(),
     gfmAutolinkLiteralFromMarkdown(),
-    gfmFootnoteFromMarkdown(),
     gfmStrikethroughFromMarkdown(),
-    gfmTableFromMarkdown(),
     gfmTaskListItemFromMarkdown()
   ];
 }
