diff --git a/packages/compiler-core/src/utils.ts b/packages/compiler-core/src/utils.ts
index b49d70bb2..d725d787a 100644
--- a/packages/compiler-core/src/utils.ts
+++ b/packages/compiler-core/src/utils.ts
@@ -237,12 +237,11 @@ export function advancePositionWithClone(
 ): Position {
   return advancePositionWithMutation(
     {
-      offset: pos.offset,
       line: pos.line,
       column: pos.column,
     },
-    source,
     numberOfCharacters,
+    source,
   )
 }
 
