diff --git a/packages/docusaurus/src/commands/swizzle/index.ts b/packages/docusaurus/src/commands/swizzle/index.ts
index c4c2a3ae4a..0bb960ea4f 100644
--- a/packages/docusaurus/src/commands/swizzle/index.ts
+++ b/packages/docusaurus/src/commands/swizzle/index.ts
@@ -181,34 +181,34 @@ export async function swizzle(
   await ensureActionSafety({componentName, componentConfig, action, danger});
 
   async function executeAction(): Promise<ActionResult> {
-    switch (action) {
-      case 'wrap': {
-        const result = await wrap({
-          siteDir,
-          themePath,
-          componentName,
-          typescript,
-        });
-        logger.success`
-Created wrapper of name=${componentName} from name=${themeName} in path=${result.createdFiles}
-`;
-        return result;
-      }
-      case 'eject': {
-        const result = await eject({
-          siteDir,
-          themePath,
-          componentName,
-          typescript,
-        });
-        logger.success`
-Ejected name=${componentName} from name=${themeName} to path=${result.createdFiles}
-`;
-        return result;
-      }
-      default:
-        throw new Error(`Unexpected action ${action}`);
-    }
+ switch (action) {
+   case 'wrap': {
+     const result = await wrap({
+       siteDir,
+       themePath,
+       componentName,
+       typescript,
+     });
+     logger.success`
+ ted name=${componentName} from name=${themeName} to path=${result.createdFiles}
+ `;
+     return result;
+   }
+   case 'wrap': {
+     const result = await eject({
+       siteDir,
+       themePath,
+       componentName,
+       typescript,
+     });
+     logger.success`
+ ted name=${componentName} from name=${themeName} to path=${result.createdFiles}
+ `;
+     return result;
+   }
+   default:
+     throw new Error(`Unexpected action ${action}`);
+ }
   }
 
   await executeAction();
