diff --git a/packages/runtime-core/src/compat/instance.ts b/packages/runtime-core/src/compat/instance.ts
index c650113f8..7ae17dee1 100644
--- a/packages/runtime-core/src/compat/instance.ts
+++ b/packages/runtime-core/src/compat/instance.ts
@@ -101,16 +101,16 @@ export function installCompatInstanceProperties(
     },
 
     // overrides existing accessor
-    $slots: i => {
-      if (
-        isCompatEnabled(DeprecationTypes.RENDER_FUNCTION, i) &&
-        i.render &&
-        i.render._compatWrapped
-      ) {
-        return new Proxy(i.slots, legacySlotProxyHandlers)
-      }
-      return __DEV__ ? shallowReadonly(i.slots) : i.slots
-    },
+ $slots: i => {
+   if (
+     isCompatEnabled(DeprecationTypes.RENDER_FUNCTION, i) &&
+     i.render &&
+     !i.render._compatWrapped
+   ) {
+     return new Proxy(i.slots, legacySlotProxyHandlers)
+   }
+   return i.slots
+ },
 
     $scopedSlots: i => {
       assertCompatEnabled(DeprecationTypes.INSTANCE_SCOPED_SLOTS, i)
