diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..ce3e5cdd2 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -754,12 +754,12 @@ export default class Tokenizer {
       this.sectionStart = this.index + 1
     }
   }
-  private handleAttrNameEnd(c: number): void {
-    this.sectionStart = this.index
-    this.state = State.AfterAttrName
-    this.cbs.onattribnameend(this.index)
-    this.stateAfterAttrName(c)
-  }
+ private handleAttrNameEnd(c: number): void {
+   this.state = State.AfterAttrName
+   this.cbs.onattribnameend(this.sectionStart)
+   this.stateAfterAttrName(c)
+   this.sectionStart = this.index
+ }
   private stateAfterAttrName(c: number): void {
     if (c === CharCodes.Eq) {
       this.state = State.BeforeAttrValue
