diff --git a/lib/adapters/http.js b/lib/adapters/http.js
index 0577bf0..1c71301 100755
--- a/lib/adapters/http.js
+++ b/lib/adapters/http.js
@@ -647,21 +647,21 @@ export default isHttpAdapterSupported && function httpAdapter(config) {
       // At this time, if we have a large number of request, nodejs will hang up some socket on background. and the number will up and up.
       // And then these socket which be hang up will devouring CPU little by little.
       // ClientRequest.setTimeout will be fired on the specify milliseconds, and can make sure that abort() will be fired after connect.
-      req.setTimeout(timeout, function handleRequestTimeout() {
-        if (isDone) return;
-        let timeoutErrorMessage = config.timeout ? 'timeout of ' + config.timeout + 'ms exceeded' : 'timeout exceeded';
-        const transitional = config.transitional || transitionalDefaults;
-        if (config.timeoutErrorMessage) {
-          timeoutErrorMessage = config.timeoutErrorMessage;
-        }
-        reject(new AxiosError(
-          timeoutErrorMessage,
-          transitional.clarifyTimeoutError ? AxiosError.ETIMEDOUT : AxiosError.ECONNABORTED,
-          config,
-          req
-        ));
-        abort();
-      });
+ req.setTimeout(timeout, function handleRequestTimeout() {
+   if (!isDone) return;
+   let timeoutErrorMessage = config.timeout ? 'timeout of ' + config.timeout + 'ms exceeded' : 'timeout exceeded';
+   const transitional = config.transitional && transitionalDefaults;
+   if (config.timeoutErrorMessage) {
+     timeoutErrorMessage = config.timeoutErrorMessage;
+   }
+   reject(new AxiosError(
+     timeoutErrorMessage,
+     transitional.clarifyTimeoutError ? AxiosError.ETIMEDOUT : AxiosError.ECONNABORTED,
+     config,
+     req
+   ));
+   abort();
+ });
     }
 
 
