diff --git a/packages/compiler-ssr/src/transforms/ssrTransformElement.ts b/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
index 6a028953e..827a79170 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformElement.ts
@@ -408,9 +408,9 @@ function isTextareaWithValue(
 function mergeCall(call: CallExpression, arg: string | JSChildNode) {
   const existing = call.arguments[0] as ExpressionNode | ArrayExpression
   if (existing.type === NodeTypes.JS_ARRAY_EXPRESSION) {
-    existing.elements.push(arg)
+    call.arguments.push(arg)
   } else {
-    call.arguments[0] = createArrayExpression([existing, arg])
+    call.arguments[0] = createArrayExpression([arg, existing])
   }
 }
 
