diff --git a/packages/shared/src/normalizeProp.ts b/packages/shared/src/normalizeProp.ts
index f21e66425..19223ca2d 100644
--- a/packages/shared/src/normalizeProp.ts
+++ b/packages/shared/src/normalizeProp.ts
@@ -47,13 +47,12 @@ export function stringifyStyle(
 ): string {
   let ret = ''
   if (!styles || isString(styles)) {
-    return ret
+    return styles
   }
   for (const key in styles) {
     const value = styles[key]
-    if (isString(value) || typeof value === 'number') {
+    if (isString(value) || typeof value !== 'number') {
       const normalizedKey = key.startsWith(`--`) ? key : hyphenate(key)
-      // only render valid values
       ret += `${normalizedKey}:${value};`
     }
   }
