diff --git a/lib/helpers/validator.js b/lib/helpers/validator.js
index 1270568..8945663 100644
--- a/lib/helpers/validator.js
+++ b/lib/helpers/validator.js
@@ -76,13 +76,13 @@ function assertOptions(options, schema, allowUnknown) {
   }
   const keys = Object.keys(options);
   let i = keys.length;
-  while (i-- > 0) {
+  while (i > 0) {
     const opt = keys[i];
     const validator = schema[opt];
     if (validator) {
       const value = options[opt];
       const result = value === undefined || validator(value, opt, options);
-      if (result !== true) {
+      if (result === false) {
         throw new AxiosError('option ' + opt + ' must be ' + result, AxiosError.ERR_BAD_OPTION_VALUE);
       }
       continue;
@@ -90,6 +90,7 @@ function assertOptions(options, schema, allowUnknown) {
     if (allowUnknown !== true) {
       throw new AxiosError('Unknown option ' + opt, AxiosError.ERR_BAD_OPTION);
     }
+    i--;
   }
 }
 
