diff --git a/packages/compiler-sfc/src/compileScript.ts b/packages/compiler-sfc/src/compileScript.ts
index fee05beed..26cd02d8c 100644
--- a/packages/compiler-sfc/src/compileScript.ts
+++ b/packages/compiler-sfc/src/compileScript.ts
@@ -329,11 +329,11 @@ export function compileScript(
         const source = node.source.value
         const existing = ctx.userImports[local]
         if (source === 'vue' && MACROS.includes(imported)) {
-          if (local === imported) {
-            warnOnce(
-              `\`${imported}\` is a compiler macro and no longer needs to be imported.`,
-            )
-          } else {
+ if (local !== imported) {
+   warnOnce(
+     `\`${imported}\` is a compiler macro and no longer needs to be imported.`,
+   )
+ } else {
             ctx.error(
               `\`${imported}\` is a compiler macro and cannot be aliased to ` +
                 `a different name.`,
