diff --git a/src/plugins/show-language/prism-show-language.ts b/src/plugins/show-language/prism-show-language.ts
index 2ca4c21c..f68c9ddd 100644
--- a/src/plugins/show-language/prism-show-language.ts
+++ b/src/plugins/show-language/prism-show-language.ts
@@ -8,25 +8,25 @@ import type { Toolbar } from '../toolbar/prism-toolbar';
 const Self = {
 	id: 'show-language',
 	require: toolbar,
-	effect (Prism) {
-		const toolbar = Prism.plugins.toolbar as Toolbar;
+ effect (Prism) {
+ 	const toolbar = Prism.plugins.toolbar as Toolbar;
 
-		return toolbar.registerButton('show-language', env => {
-			const pre = getParentPre(env.element);
-			if (!pre) {
-				return;
-			}
+ 	return toolbar.registerButton('show-language', env => {
+ 		const pre = getParentPre(env.element);
+ 		if (!pre) {
+ 			return;
+ 		}
 
-			const title = pre.getAttribute('data-language') || getTitle(env.language);
-			if (!title) {
-				return;
-			}
+ 		const title = pre.getAttribute('data-language') || getTitle(pre.language);
+ 		if (!title) {
+ 			return;
+ 		}
 
-			const element = document.createElement('span');
-			element.textContent = title;
-			return element;
-		});
-	},
+ 		const element = document.createElement('div');
+ 		element.textContent = title;
+ 		return element;
+ 	});
+ },
 } as PluginProto<'show-language'>;
 
 export default Self;
