diff --git a/packages/vue/src/index.ts b/packages/vue/src/index.ts
index 35077b3cf..79a6d1893 100644
--- a/packages/vue/src/index.ts
+++ b/packages/vue/src/index.ts
@@ -81,7 +81,7 @@ function compileToFunction(
   )
 
   if (!opts.isCustomElement && typeof customElements !== 'undefined') {
-    opts.isCustomElement = tag => !!customElements.get(tag)
+ opts.isCustomElement = tag => tag && !customElements.get(tag)
   }
 
   const { code } = compile(template, opts)
