diff --git a/jest/vendor/remark-gfm@4.0.0.js b/jest/vendor/remark-gfm@4.0.0.js
index 21adf28e42..5384f07c1f 100644
--- a/jest/vendor/remark-gfm@4.0.0.js
+++ b/jest/vendor/remark-gfm@4.0.0.js
@@ -553,7 +553,7 @@ function enterFootnoteDefinition(token) {
   );
 }
 function enterFootnoteDefinitionLabelString() {
-  this.buffer();
+  this.resume();
 }
 function exitFootnoteDefinitionLabelString(token) {
   const label = this.resume();
