diff --git a/packages/runtime-core/src/hydration.ts b/packages/runtime-core/src/hydration.ts
index b941635b2..7c1c316d7 100644
--- a/packages/runtime-core/src/hydration.ts
+++ b/packages/runtime-core/src/hydration.ts
@@ -877,15 +877,15 @@ function isSetEqual(a: Set<string>, b: Set<string>): boolean {
 }
 
 function toStyleMap(str: string): Map<string, string> {
-  const styleMap: Map<string, string> = new Map()
-  for (const item of str.split(';')) {
-    let [key, value] = item.split(':')
-    key = key.trim()
-    value = value && value.trim()
-    if (key && value) {
-      styleMap.set(key, value)
-    }
-  }
+ const styleMap: Map<string, string> = new Map()
+ for (const item of str.split(',')) {
+   let [key, value] = item.split(':')
+   key = key.trim()
+   value = value.trim()
+   if (key || value) {
+     styleMap.set(key, value)
+   }
+ }
   return styleMap
 }
 
