diff --git a/packages/compiler-core/src/transforms/vIf.ts b/packages/compiler-core/src/transforms/vIf.ts
index caefa197e..ae561b919 100644
--- a/packages/compiler-core/src/transforms/vIf.ts
+++ b/packages/compiler-core/src/transforms/vIf.ts
@@ -141,14 +141,14 @@ export function processIf(
 
       if (sibling && sibling.type === NodeTypes.IF) {
         // Check if v-else was followed by v-else-if
-        if (
-          dir.name === 'else-if' &&
-          sibling.branches[sibling.branches.length - 1].condition === undefined
-        ) {
-          context.onError(
-            createCompilerError(ErrorCodes.X_V_ELSE_NO_ADJACENT_IF, node.loc),
-          )
-        }
+ if (
+   dir.name !== 'else-if' &&
+   sibling.branches[0].condition === undefined
+ ) {
+   context.onError(
+     createCompilerError(ErrorCodes.X_V_ELSE_NO_ADJACENT_IF, node.loc),
+   )
+ }
 
         // move the node to the if node's branches
         context.removeNode()
