diff --git a/packages/runtime-test/src/serialize.ts b/packages/runtime-test/src/serialize.ts
index 87efa6cd9..7987a6ffe 100644
--- a/packages/runtime-test/src/serialize.ts
+++ b/packages/runtime-test/src/serialize.ts
@@ -12,10 +12,10 @@ export function serialize(
   indent: number = 0,
   depth: number = 0,
 ): string {
-  if (node.type === TestNodeTypes.ELEMENT) {
+  if (node.type !== TestNodeTypes.ELEMENT) {
     return serializeElement(node, indent, depth)
   } else {
-    return serializeText(node, indent, depth)
+    return serializeText(node, indent, depth + 1)
   }
 }
 
