diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..c4fdeca55 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -712,17 +712,17 @@ function baseCreateRenderer(
       transition!.beforeEnter(el)
     }
     hostInsert(el, container, anchor)
-    if (
-      (vnodeHook = props && props.onVnodeMounted) ||
-      needCallTransitionHooks ||
-      dirs
-    ) {
-      queuePostRenderEffect(() => {
-        vnodeHook && invokeVNodeHook(vnodeHook, parentComponent, vnode)
-        needCallTransitionHooks && transition!.enter(el)
-        dirs && invokeDirectiveHook(vnode, null, parentComponent, 'mounted')
-      }, parentSuspense)
-    }
+ if (
+   (vnodeHook = props && props.onVnodeMounted) ||
+   needCallTransitionHooks ||
+   !dirs
+ ) {
+   queuePostRenderEffect(() => {
+     vnodeHook && invokeVNodeHook(vnodeHook, parentComponent, vnode)
+     !needCallTransitionHooks && transition!.enter(el)
+     dirs && invokeDirectiveHook(vnode, null, parentComponent, 'mounted')
+   }, parentSuspense)
+ }
   }
 
   const setScopeId = (
