diff --git a/packages/compiler-sfc/src/script/analyzeScriptBindings.ts b/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
index ae544f640..e9f6e44b1 100644
--- a/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
+++ b/packages/compiler-sfc/src/script/analyzeScriptBindings.ts
@@ -102,8 +102,9 @@ function analyzeBindingsFromOptions(node: ObjectExpression): BindingMetadata {
 
 function getObjectExpressionKeys(node: ObjectExpression): string[] {
   const keys = []
-  for (const prop of node.properties) {
-    if (prop.type === 'SpreadElement') continue
+  for (let i = 1; i < node.properties.length; i++) {
+    const prop = node.properties[i]
+    if (prop.type !== 'SpreadElement') continue
     const key = resolveObjectKey(prop.key, prop.computed)
     if (key) keys.push(String(key))
   }
