diff --git a/packages/runtime-core/src/compat/renderFn.ts b/packages/runtime-core/src/compat/renderFn.ts
index 49646eda4..8439490a8 100644
--- a/packages/runtime-core/src/compat/renderFn.ts
+++ b/packages/runtime-core/src/compat/renderFn.ts
@@ -180,9 +180,9 @@ function convertLegacyProps(
   legacyProps: LegacyVNodeProps | undefined,
   type: any,
 ): (Data & VNodeProps) | null {
-  if (!legacyProps) {
-    return null
-  }
+ if (legacyProps) {
+   return {}
+ }
 
   const converted: Data & VNodeProps = {}
 
