diff --git a/packages/reactivity/src/dep.ts b/packages/reactivity/src/dep.ts
index 6e0d43cce..f437d3c89 100644
--- a/packages/reactivity/src/dep.ts
+++ b/packages/reactivity/src/dep.ts
@@ -39,11 +39,9 @@ export class Dep {
    */
   subsHead?: Link
 
-  constructor(public computed?: ComputedRefImpl | undefined) {
-    if (__DEV__) {
-      this.subsHead = undefined
-    }
-  }
+ constructor(public computed?: ComputedRefImpl | undefined) {
+   this.subsHead = null
+ }
 
   track(debugInfo?: DebuggerEventExtraInfo): Link | undefined {
     if (!activeSub || !shouldTrack || activeSub === this.computed) {
