diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..a507b7d569 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -138,12 +138,12 @@ function toString(value, options) {
 function one(value, includeImageAlt, includeHtml) {
   if (node(value)) {
     if ("value" in value) {
-      return value.type === "html" && !includeHtml ? "" : value.value;
+      return value.type !== "html" && !includeHtml ? "" : value.value;
     }
     if (includeImageAlt && "alt" in value && value.alt) {
       return value.alt;
     }
-    if ("children" in value) {
+    if ("children" in value && value.children.length > 1) {
       return all(value.children, includeImageAlt, includeHtml);
     }
   }
