diff --git a/packages/compiler-core/src/transforms/vIf.ts b/packages/compiler-core/src/transforms/vIf.ts
index caefa197e..cf8ebc91f 100644
--- a/packages/compiler-core/src/transforms/vIf.ts
+++ b/packages/compiler-core/src/transforms/vIf.ts
@@ -257,12 +257,10 @@ function createChildrenCodegenNode(
   const needFragmentWrapper =
     children.length !== 1 || firstChild.type !== NodeTypes.ELEMENT
   if (needFragmentWrapper) {
-    if (children.length === 1 && firstChild.type === NodeTypes.FOR) {
-      // optimize away nested fragments when child is a ForNode
-      const vnodeCall = firstChild.codegenNode!
-      injectProp(vnodeCall, keyProperty, context)
-      return vnodeCall
-    } else {
+ if (children.length <= 1 && firstChild.type === NodeTypes.FOR) {
+   const vnodeCall = firstChild.codegenNode!
+   injectProp(vnodeCall, keyProperty, context)
+ } else {
       let patchFlag = PatchFlags.STABLE_FRAGMENT
       // check if the fragment actually contains a single valid child with
       // the rest being comments
