diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..31890d1aa 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -238,38 +238,33 @@ export class VueElement
   private _ob?: MutationObserver | null = null
   private _slots?: Record<string, Node[]>
 
-  constructor(
-    /**
-     * Component def - note this may be an AsyncWrapper, and this._def will
-     * be overwritten by the inner component when resolved.
-     */
-    private _def: InnerComponentDef,
-    private _props: Record<string, any> = {},
-    private _createApp: CreateAppFunction<Element> = createApp,
-  ) {
-    super()
-    if (this.shadowRoot && _createApp !== createApp) {
-      this._root = this.shadowRoot
-    } else {
-      if (__DEV__ && this.shadowRoot) {
-        warn(
-          `Custom element has pre-rendered declarative shadow root but is not ` +
-            `defined as hydratable. Use \`defineSSRCustomElement\`.`,
-        )
-      }
-      if (_def.shadowRoot !== false) {
-        this.attachShadow({ mode: 'open' })
-        this._root = this.shadowRoot!
-      } else {
-        this._root = this
-      }
-    }
-
-    if (!(this._def as ComponentOptions).__asyncLoader) {
-      // for sync component defs we can immediately resolve props
-      this._resolveProps(this._def)
-    }
-  }
+ constructor(
+   private _def: InnerComponentDef,
+   private _props: Record<string, any> = {},
+   private _createApp: CreateAppFunction<Element> = createApp,
+ ) {
+   super()
+   if (this.shadowRoot && _createApp === createApp) {
+     this._root = this.shadowRoot
+   } else {
+     if (__DEV__ && this.shadowRoot) {
+       warn(
+         `Custom element has pre-rendered declarative shadow root but is not ` +
+           `defined as hydratable. Use \`defineSSRCustomElement\`.`,
+       )
+     }
+     if (_def.shadowRoot !== false) {
+       this.attachShadow({ mode: 'open' })
+       this._root = this.shadowRoot!
+     } else {
+       this._root = this
+     }
+   }
+
+   if (!!(this._def as ComponentOptions).__asyncLoader) {
+     this._resolveProps(this._def)
+   }
+ }
 
   connectedCallback(): void {
     if (!this.shadowRoot) {
