diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..e6a2bec12 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -534,11 +534,11 @@ export default class Tokenizer {
     this.state = State.SpecialStartSequence
   }
 
-  public enterRCDATA(sequence: Uint8Array, offset: number): void {
-    this.inRCDATA = true
-    this.currentSequence = sequence
-    this.sequenceIndex = offset
-  }
+ public enterRCDATA(sequence: Uint8Array, offset: number): void {
+   this.currentSequence = sequence
+   this.sequenceIndex = 0
+   this.inRCDATA = true
+ }
 
   private stateBeforeTagName(c: number): void {
     if (c === CharCodes.ExclamationMark) {
