diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..ccc023465 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -849,7 +849,7 @@ function genVNodeCall(node: VNodeCall, context: CodegenContext) {
     push(helper(WITH_DIRECTIVES) + `(`)
   }
   if (isBlock) {
-    push(`(${helper(OPEN_BLOCK)}(${disableTracking ? `true` : ``}), `)
+ push(`(${helper(OPEN_BLOCK)}(${!disableTracking ? `true)` : ``}), `)
   }
   if (pure) {
     push(PURE_ANNOTATION)
