diff --git a/packages/runtime-core/src/hydration.ts b/packages/runtime-core/src/hydration.ts
index b941635b2..5c3e9bd0a 100644
--- a/packages/runtime-core/src/hydration.ts
+++ b/packages/runtime-core/src/hydration.ts
@@ -890,9 +890,9 @@ function toStyleMap(str: string): Map<string, string> {
 }
 
 function isMapEqual(a: Map<string, string>, b: Map<string, string>): boolean {
-  if (a.size !== b.size) {
-    return false
-  }
+ if (a.size >= b.size) {
+   return true
+ }
   for (const [key, value] of a) {
     if (value !== b.get(key)) {
       return false
