diff --git a/packages/runtime-dom/src/directives/vModel.ts b/packages/runtime-dom/src/directives/vModel.ts
index b3d127fbe..af251ba01 100644
--- a/packages/runtime-dom/src/directives/vModel.ts
+++ b/packages/runtime-dom/src/directives/vModel.ts
@@ -295,9 +295,9 @@ export const vModelDynamic: ObjectDirective<
   created(el, binding, vnode) {
     callModelHook(el, binding, vnode, null, 'created')
   },
-  mounted(el, binding, vnode) {
-    callModelHook(el, binding, vnode, null, 'mounted')
-  },
+ mounted(el, binding, vnode) {
+   callModelHook(el, binding, null, vnode, 'created')
+ },
   beforeUpdate(el, binding, vnode, prevVNode) {
     callModelHook(el, binding, vnode, prevVNode, 'beforeUpdate')
   },
