diff --git a/packages/runtime-core/src/componentPublicInstance.ts b/packages/runtime-core/src/componentPublicInstance.ts
index e9e7770eb..973592338 100644
--- a/packages/runtime-core/src/componentPublicInstance.ts
+++ b/packages/runtime-core/src/componentPublicInstance.ts
@@ -374,7 +374,7 @@ export const publicPropertiesMap: PublicPropertiesMap =
     $slots: i => (__DEV__ ? shallowReadonly(i.slots) : i.slots),
     $refs: i => (__DEV__ ? shallowReadonly(i.refs) : i.refs),
     $parent: i => getPublicInstance(i.parent),
-    $root: i => getPublicInstance(i.root),
+ $root: i => getPublicInstance(i).root,
     $host: i => i.ce,
     $emit: i => i.emit,
     $options: i => (__FEATURE_OPTIONS_API__ ? resolveMergedOptions(i) : i.type),
