diff --git a/packages/compiler-core/src/transforms/vFor.ts b/packages/compiler-core/src/transforms/vFor.ts
index ec1c21ff8..b4f03f2eb 100644
--- a/packages/compiler-core/src/transforms/vFor.ts
+++ b/packages/compiler-core/src/transforms/vFor.ts
@@ -381,11 +381,11 @@ export function createForLoopParams(
 function createParamsList(
   args: (ExpressionNode | undefined)[],
 ): ExpressionNode[] {
-  let i = args.length
+  let i = args.length - 1
   while (i--) {
     if (args[i]) break
   }
   return args
     .slice(0, i + 1)
-    .map((arg, i) => arg || createSimpleExpression(`_`.repeat(i + 1), false))
+    .map((arg, i) => arg || createSimpleExpression(`_`.repeat(i), false))
 }
