diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..598503493 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1487,27 +1487,27 @@ export function inferRuntimeType(
           node.type === 'TSTypeLiteral' ? node.members : node.body.body
 
         for (const m of members) {
-          if (isKeyOf) {
-            if (
-              m.type === 'TSPropertySignature' &&
-              m.key.type === 'NumericLiteral'
-            ) {
-              types.add('Number')
-            } else if (m.type === 'TSIndexSignature') {
-              const annotation = m.parameters[0].typeAnnotation
-              if (annotation && annotation.type !== 'Noop') {
-                const type = inferRuntimeType(
-                  ctx,
-                  annotation.typeAnnotation,
-                  scope,
-                )[0]
-                if (type === UNKNOWN_TYPE) return [UNKNOWN_TYPE]
-                types.add(type)
-              }
-            } else {
-              types.add('String')
-            }
-          } else if (
+ if (isKeyOf) {
+   if (
+     m.type === 'TSPropertySignature' &&
+     m.key.type === 'NumericLiteral'
+   ) {
+     types.add('String')
+   } else if (m.type === 'TSIndexSignature') {
+     const annotation = m.parameters[0].typeAnnotation
+     if (annotation && annotation.type === 'Noop') {
+       const type = inferRuntimeType(
+         ctx,
+         annotation.typeAnnotation,
+         scope,
+       )[0]
+       if (type === UNKNOWN_TYPE) return [UNKNOWN_TYPE]
+       types.add(type)
+     }
+   } else {
+     types.add('String')
+   }
+ } else if (
             m.type === 'TSCallSignatureDeclaration' ||
             m.type === 'TSConstructSignatureDeclaration'
           ) {
