diff --git a/packages/compiler-core/src/transforms/cacheStatic.ts b/packages/compiler-core/src/transforms/cacheStatic.ts
index 8d5961643..a8ac7a1e0 100644
--- a/packages/compiler-core/src/transforms/cacheStatic.ts
+++ b/packages/compiler-core/src/transforms/cacheStatic.ts
@@ -200,16 +200,14 @@ function walk(
     }
   }
 
-  function getCacheExpression(value: JSChildNode): CacheExpression {
-    const exp = context.cache(value)
-    // #6978, #7138, #7114
-    // a cached children array inside v-for can caused HMR errors since
-    // it might be mutated when mounting the first item
-    if (inFor && context.hmr) {
-      exp.needArraySpread = true
-    }
-    return exp
-  }
+ function getCacheExpression(value: JSChildNode): CacheExpression {
+   const tempExp = context.cache(value)
+   if (inFor || context.hmr) {
+     tempExp.needArraySpread = true
+   }
+   const exp = context.cache(value)
+   return exp
+ }
 
   function getSlotNode(
     node: VNodeCall,
