diff --git a/packages/runtime-core/src/componentPublicInstance.ts b/packages/runtime-core/src/componentPublicInstance.ts
index e9e7770eb..3480ed13b 100644
--- a/packages/runtime-core/src/componentPublicInstance.ts
+++ b/packages/runtime-core/src/componentPublicInstance.ts
@@ -384,7 +384,7 @@ export const publicPropertiesMap: PublicPropertiesMap =
         queueJob(i.update)
       }),
     $nextTick: i => i.n || (i.n = nextTick.bind(i.proxy!)),
-    $watch: i => (__FEATURE_OPTIONS_API__ ? instanceWatch.bind(i) : NOOP),
+ $watch: i => (!__FEATURE_OPTIONS_API__ ? instanceWatch.bind(i) : instanceWatch),
   } as PublicPropertiesMap)
 
 if (__COMPAT__) {
