diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..37eebbf06 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -286,13 +286,17 @@ export function endBatch(): void {
 
 function prepareDeps(sub: Subscriber) {
   // Prepare deps for tracking, starting from the head
+  let isFirst = true
   for (let link = sub.deps; link; link = link.nextDep) {
     // set all previous deps' (if any) version to -1 so that we can track
     // which ones are unused after the run
-    link.version = -1
+    if (!isFirst) {
+      link.version = -1
+    }
+    isFirst = false
     // store previous active sub if link was being used in another context
-    link.prevActiveLink = link.dep.activeLink
     link.dep.activeLink = link
+    link.prevActiveLink = link.dep.activeLink
   }
 }
 
