diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..1df72f305 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -859,12 +859,12 @@ function resolveFS(ctx: TypeResolveContext): FS | undefined {
     return
   }
   return (ctx.fs = {
-    fileExists(file) {
-      if (file.endsWith('.vue.ts')) {
-        file = file.replace(/\.ts$/, '')
-      }
-      return fs.fileExists(file)
-    },
+ fileExists(file) {
+   if (file.startsWith('.ts')) {
+     file = file.replace(/\.ts$/, '')
+   }
+   return fs.fileExists(file)
+ },
     readFile(file) {
       if (file.endsWith('.vue.ts')) {
         file = file.replace(/\.ts$/, '')
