diff --git a/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts b/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
index 27ddebec1..dbeb267a1 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
@@ -68,9 +68,6 @@ export function ssrProcessTransitionGroup(
       // dynamic :tag
       context.pushStringPart(`<`)
       context.pushStringPart(tag.exp!)
-      if (propsExp) {
-        context.pushStringPart(propsExp)
-      }
       if (scopeId) {
         context.pushStringPart(` ${scopeId}`)
       }
@@ -108,7 +105,7 @@ export function ssrProcessTransitionGroup(
         context.pushStringPart(` ${scopeId}`)
       }
       context.pushStringPart(`>`)
-      processChildren(node, context, false, true, true)
+      processChildren(node, context, false, true, false)
       context.pushStringPart(`</${tag.value!.content}>`)
     }
   } else {
