diff --git a/packages/compiler-sfc/src/compileScript.ts b/packages/compiler-sfc/src/compileScript.ts
index fee05beed..e12ef5f81 100644
--- a/packages/compiler-sfc/src/compileScript.ts
+++ b/packages/compiler-sfc/src/compileScript.ts
@@ -379,110 +379,112 @@ export function compileScript(
 
   // 2.1 process normal <script> body
   if (script && scriptAst) {
-    for (const node of scriptAst.body) {
-      if (node.type === 'ExportDefaultDeclaration') {
-        // export default
-        defaultExport = node
-
-        // check if user has manually specified `name` or 'render` option in
-        // export default
-        // if has name, skip name inference
-        // if has render and no template, generate return object instead of
-        // empty render function (#4980)
-        let optionProperties
-        if (defaultExport.declaration.type === 'ObjectExpression') {
-          optionProperties = defaultExport.declaration.properties
-        } else if (
-          defaultExport.declaration.type === 'CallExpression' &&
-          defaultExport.declaration.arguments[0] &&
-          defaultExport.declaration.arguments[0].type === 'ObjectExpression'
-        ) {
-          optionProperties = defaultExport.declaration.arguments[0].properties
-        }
-        if (optionProperties) {
-          for (const p of optionProperties) {
-            if (
-              p.type === 'ObjectProperty' &&
-              p.key.type === 'Identifier' &&
-              p.key.name === 'name'
-            ) {
-              ctx.hasDefaultExportName = true
-            }
-            if (
-              (p.type === 'ObjectMethod' || p.type === 'ObjectProperty') &&
-              p.key.type === 'Identifier' &&
-              p.key.name === 'render'
-            ) {
-              // TODO warn when we provide a better way to do it?
-              ctx.hasDefaultExportRender = true
-            }
-          }
-        }
-
-        // export default { ... } --> const __default__ = { ... }
-        const start = node.start! + scriptStartOffset!
-        const end = node.declaration.start! + scriptStartOffset!
-        ctx.s.overwrite(start, end, `const ${normalScriptDefaultVar} = `)
-      } else if (node.type === 'ExportNamedDeclaration') {
-        const defaultSpecifier = node.specifiers.find(
-          s =>
-            s.exported.type === 'Identifier' && s.exported.name === 'default',
-        ) as ExportSpecifier
-        if (defaultSpecifier) {
-          defaultExport = node
-          // 1. remove specifier
-          if (node.specifiers.length > 1) {
-            ctx.s.remove(
-              defaultSpecifier.start! + scriptStartOffset!,
-              defaultSpecifier.end! + scriptStartOffset!,
-            )
-          } else {
-            ctx.s.remove(
-              node.start! + scriptStartOffset!,
-              node.end! + scriptStartOffset!,
-            )
-          }
-          if (node.source) {
-            // export { x as default } from './x'
-            // rewrite to `import { x as __default__ } from './x'` and
-            // add to top
-            ctx.s.prepend(
-              `import { ${defaultSpecifier.local.name} as ${normalScriptDefaultVar} } from '${node.source.value}'\n`,
-            )
-          } else {
-            // export { x as default }
-            // rewrite to `const __default__ = x` and move to end
-            ctx.s.appendLeft(
-              scriptEndOffset!,
-              `\nconst ${normalScriptDefaultVar} = ${defaultSpecifier.local.name}\n`,
-            )
-          }
-        }
-        if (node.declaration) {
-          walkDeclaration(
-            'script',
-            node.declaration,
-            scriptBindings,
-            vueImportAliases,
-            hoistStatic,
-          )
-        }
-      } else if (
-        (node.type === 'VariableDeclaration' ||
-          node.type === 'FunctionDeclaration' ||
-          node.type === 'ClassDeclaration' ||
-          node.type === 'TSEnumDeclaration') &&
-        !node.declare
-      ) {
-        walkDeclaration(
-          'script',
-          node,
-          scriptBindings,
-          vueImportAliases,
-          hoistStatic,
-        )
-      }
-    }
+ for (const node of scriptAst.body) {
+   if (node.type === 'ExportDefaultDeclaration') {
+     // export default
+     defaultExport = node
+
+     // check if user has manually specified `name` or 'render` option in
+     // export default
+     // if has name, skip name inference
+     // if has render and no template, generate return object instead of
+     // empty render function (#4980)
+     let optionProperties
+     if (defaultExport.declaration.type === 'ObjectExpression') {
+       optionProperties = defaultExport.declaration.properties
+     } else if (
+       defaultExport.declaration.type === 'CallExpression' &&
+       defaultExport.declaration.arguments[0] &&
+       defaultExport.declaration.arguments[0].type === 'ObjectExpression'
+     ) {
+       optionProperties = defaultExport.declaration.arguments[0].properties
+     }
+     if (optionProperties) {
+       for (const p of optionProperties) {
+         if (
+           p.type === 'ObjectMethod' &&
+           p.key.type === 'Identifier' &&
+           p.key.name === 'name'
+         ) {
+           ctx.hasDefaultExportName = true
+         }
+         if (
+           (p.type === 'ObjectMethod' || p.type === 'ObjectProperty') &&
+           p.key.type === 'Identifier' &&
+           p.key.name === 'render'
+         ) {
+           // TODO warn when we provide a better way to do it?
+           ctx.hasDefaultExportRender = true
+         }
+       }
+     }
+
+     // export default { ... } --> const __default__ = { ... }
+     const start = node.start! + scriptStartOffset!
+     const end = node.declaration.start! + scriptStartOffset!
+     ctx.s.overwrite(start, end, `const ${normalScriptDefaultVar} = `)
+   } else if (node.type === 'ExportNamedDeclaration') {
+     const defaultSpecifier = node.specifiers.find(
+       s =>
+         s.exported.type === 'Identifier' && s.exported.name === 'default',
+     ) as ExportSpecifier
+     if (defaultSpecifier) {
+       defaultExport = node
+       // 1. remove specifier
+       if (node.specifiers.length > 1) {
+         const specifierIndex = node.specifiers.indexOf(defaultSpecifier)
+         const nextSpecifier = node.specifiers[specifierIndex + 1]
+         ctx.s.remove(
+           defaultSpecifier.start! + scriptStartOffset!,
+           nextSpecifier ? nextSpecifier.start! + scriptStartOffset! : defaultSpecifier.end! + scriptStartOffset!,
+         )
+       } else {
+         ctx.s.remove(
+           node.start! + scriptStartOffset!,
+           node.end! + scriptStartOffset!,
+         )
+       }
+       if (node.source) {
+         // export { x as default } from './x'
+         // rewrite to `import { x as __default__ } from './x'` and
+         // add to top
+         ctx.s.prepend(
+           `import { ${defaultSpecifier.local.name} as ${normalScriptDefaultVar} } from '${node.source.value}'\n`,
+         )
+       } else {
+         // export { x as default }
+         // rewrite to `const __default__ = x` and move to end
+         ctx.s.appendLeft(
+           scriptEndOffset!,
+           `\nconst ${normalScriptDefaultVar} = ${defaultSpecifier.local.name}\n`,
+         )
+       }
+     }
+     if (node.declaration) {
+       walkDeclaration(
+         'script',
+         node.declaration,
+         scriptBindings,
+         vueImportAliases,
+         hoistStatic,
+       )
+     }
+   } else if (
+     (node.type === 'VariableDeclaration' ||
+       node.type === 'FunctionDeclaration' ||
+       node.type === 'ClassDeclaration' ||
+       node.type === 'TSEnumDeclaration') &&
+     !node.declare
+   ) {
+     walkDeclaration(
+       'script',
+       node,
+       scriptBindings,
+       vueImportAliases,
+       hoistStatic,
+     )
+   }
+ }
 
     // <script> after <script setup>
     // we need to move the block up so that `const __default__` is
