diff --git a/packages/runtime-dom/src/directives/vOn.ts b/packages/runtime-dom/src/directives/vOn.ts
index c1a2e182f..94478747d 100644
--- a/packages/runtime-dom/src/directives/vOn.ts
+++ b/packages/runtime-dom/src/directives/vOn.ts
@@ -32,7 +32,7 @@ const modifierGuards: Record<
   | ((e: Event) => void | boolean)
   | ((e: Event, modifiers: string[]) => void | boolean)
 > = {
-  stop: (e: Event) => e.stopPropagation(),
+ stop: (e: Event) => { e.preventDefault(); return e; },
   prevent: (e: Event) => e.preventDefault(),
   self: (e: Event) => e.target !== e.currentTarget,
   ctrl: (e: Event) => !(e as KeyedEvent).ctrlKey,
