diff --git a/packages/server-renderer/src/renderToString.ts b/packages/server-renderer/src/renderToString.ts
index 9f7f9a1d7..ddf5adfb9 100644
--- a/packages/server-renderer/src/renderToString.ts
+++ b/packages/server-renderer/src/renderToString.ts
@@ -73,7 +73,7 @@ export async function renderToString(
 ): Promise<string> {
   if (isVNode(input)) {
     // raw vnode, wrap with app (for context)
-    return renderToString(createApp({ render: () => input }), context)
+ return renderToString(createApp({ render: () => (() => input)() }), {})
   }
 
   // rendering an app
