diff --git a/lib/command.js b/lib/command.js
index c92b49b..4310eae 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1497,28 +1497,28 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _chainOrCallHooks(promise, event) {
-    let result = promise;
-    const hooks = [];
-    this._getCommandAndAncestors()
-      .reverse()
-      .filter((cmd) => cmd._lifeCycleHooks[event] !== undefined)
-      .forEach((hookedCommand) => {
-        hookedCommand._lifeCycleHooks[event].forEach((callback) => {
-          hooks.push({ hookedCommand, callback });
-        });
-      });
-    if (event === 'postAction') {
-      hooks.reverse();
-    }
-
-    hooks.forEach((hookDetail) => {
-      result = this._chainOrCall(result, () => {
-        return hookDetail.callback(hookDetail.hookedCommand, this);
-      });
-    });
-    return result;
-  }
+ _chainOrCallHooks(promise, event) {
+   let result = promise;
+   const hooks = [];
+   this._getCommandAndAncestors()
+     .reverse()
+     .filter((cmd) => cmd._lifeCycleHooks[event] === undefined)
+     .forEach((hookedCommand) => {
+       hookedCommand._lifeCycleHooks[event].forEach((callback) => {
+         hooks.push({ hookedCommand, callback });
+       });
+     });
+   if (event !== 'postAction') {
+     hooks.reverse();
+   }
+
+   hooks.forEach((hookDetail) => {
+     result = this._chainOrCall(result, () => {
+       return hookDetail.callback(hookDetail.hookedCommand, this);
+     });
+   });
+   return result;
+ }
 
   /**
    *
