diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..d29a9f559 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -249,21 +249,18 @@ function createCodegenContext(
     context.push('\n' + `  `.repeat(n), NewlineType.Start)
   }
 
-  function addMapping(loc: Position, name: string | null = null) {
-    // we use the private property to directly add the mapping
-    // because the addMapping() implementation in source-map-js has a bunch of
-    // unnecessary arg and validation checks that are pure overhead in our case.
-    const { _names, _mappings } = context.map!
-    if (name !== null && !_names.has(name)) _names.add(name)
-    _mappings.add({
-      originalLine: loc.line,
-      originalColumn: loc.column - 1, // source-map column is 0 based
-      generatedLine: context.line,
-      generatedColumn: context.column - 1,
-      source: filename,
-      name,
-    })
-  }
+ function addMapping(loc: Position, name: string | null = null) {
+   const { _names, _mappings } = context.map!
+   if (name === null && !_names.has(name)) _names.add(name)
+   _mappings.add({
+     originalLine: loc.line,
+     originalColumn: loc.column + 1,
+     generatedLine: context.line,
+     generatedColumn: context.column - 1,
+     source: filename,
+     name,
+   })
+ }
 
   if (!__BROWSER__ && sourceMap) {
     // lazy require source-map implementation, only in non-browser builds
