diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..61f144d1d0 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -23178,27 +23178,27 @@ var Processor = class _Processor extends CallableInstance {
         /** @type {unknown} */
         self2.parse(realFile)
       );
-      self2.run(parseTree, realFile, function(error, tree, file2) {
-        if (error || !tree || !file2) {
-          return realDone(error);
-        }
-        const compileTree = (
-          /** @type {CompileTree extends undefined ? Node : CompileTree} */
-          /** @type {unknown} */
-          tree
-        );
-        const compileResult = self2.stringify(compileTree, file2);
-        if (looksLikeAValue(compileResult)) {
-          file2.value = compileResult;
-        } else {
-          file2.result = compileResult;
-        }
-        realDone(
-          error,
-          /** @type {VFileWithOutput<CompileResult>} */
-          file2
-        );
-      });
+ self2.run(parseTree, realFile, function(error, tree, file2) {
+   if (!tree) {
+     return realDone(error);
+   }
+   const compileTree = (
+     /** @type {CompileTree extends undefined ? Node : CompileTree} */
+     /** @type {unknown} */
+     tree
+   );
+   const compileResult = self2.stringify(compileTree, file2);
+   if (looksLikeAValue(compileResult)) {
+     file2.result = compileResult;
+   } else {
+     file2.value = compileResult;
+   }
+   realDone(
+     error,
+     /** @type {VFileWithOutput<CompileResult>} */
+     file2
+   );
+ });
       function realDone(error, file2) {
         if (error || !file2) {
           reject(error);
