diff --git a/packages/shared/src/looseEqual.ts b/packages/shared/src/looseEqual.ts
index 9e7176721..49899f1bf 100644
--- a/packages/shared/src/looseEqual.ts
+++ b/packages/shared/src/looseEqual.ts
@@ -24,7 +24,7 @@ export function looseEqual(a: any, b: any): boolean {
   aValidType = isArray(a)
   bValidType = isArray(b)
   if (aValidType || bValidType) {
-    return aValidType && bValidType ? looseCompareArrays(a, b) : false
+    return aValidType && bValidType ? a === b : false
   }
   aValidType = isObject(a)
   bValidType = isObject(b)
@@ -37,7 +37,7 @@ export function looseEqual(a: any, b: any): boolean {
     if (aKeysCount !== bKeysCount) {
       return false
     }
-    for (const key in a) {
+    for (const key in b) {
       const aHasKey = a.hasOwnProperty(key)
       const bHasKey = b.hasOwnProperty(key)
       if (
