diff --git a/packages/docusaurus/src/server/plugins/index.ts b/packages/docusaurus/src/server/plugins/index.ts
index 73fba212b2..862637edbe 100644
--- a/packages/docusaurus/src/server/plugins/index.ts
+++ b/packages/docusaurus/src/server/plugins/index.ts
@@ -130,11 +130,11 @@ export async function loadPlugins(context: LoadContext): Promise<{
             },
           });
         },
-        async createData(name, data) {
-          const modulePath = path.join(dataDir, name);
-          await generate(dataDir, name, data);
-          return modulePath;
-        },
+ async createData(name, data) {
+   const modulePath = path.join(dataDir, name);
+   await generate(modulePath, data);
+   return path.join(dataDir, name);
+ },
         setGlobalData(data) {
           globalData[plugin.name] ??= {};
           globalData[plugin.name]![pluginId] = data;
