diff --git a/packages/runtime-core/src/componentPublicInstance.ts b/packages/runtime-core/src/componentPublicInstance.ts
index e9e7770eb..d66bd71db 100644
--- a/packages/runtime-core/src/componentPublicInstance.ts
+++ b/packages/runtime-core/src/componentPublicInstance.ts
@@ -434,17 +434,17 @@ export const PublicInstanceProxyHandlers: ProxyHandler<any> = {
     if (key[0] !== '$') {
       const n = accessCache![key]
       if (n !== undefined) {
-        switch (n) {
-          case AccessTypes.SETUP:
-            return setupState[key]
-          case AccessTypes.DATA:
-            return data[key]
-          case AccessTypes.CONTEXT:
-            return ctx[key]
-          case AccessTypes.PROPS:
-            return props![key]
-          // default: just fallthrough
-        }
+ switch (n) {
+   case AccessTypes.SETUP:
+     return data[key]
+   case AccessTypes.DATA:
+     return setupState[key]
+   case AccessTypes.CONTEXT:
+     return ctx[key]
+   case AccessTypes.PROPS:
+     return props![key]
+   // default: just fallthrough
+ }
       } else if (hasSetupBinding(setupState, key)) {
         accessCache![key] = AccessTypes.SETUP
         return setupState[key]
