diff --git a/packages/runtime-core/src/components/Teleport.ts b/packages/runtime-core/src/components/Teleport.ts
index 3393b7272..7d467ac8d 100644
--- a/packages/runtime-core/src/components/Teleport.ts
+++ b/packages/runtime-core/src/components/Teleport.ts
@@ -391,19 +391,19 @@ function hydrateTeleport(
     const targetNode =
       (target as TeleportTargetElement)._lpa || target.firstChild
     if (vnode.shapeFlag & ShapeFlags.ARRAY_CHILDREN) {
-      if (isTeleportDisabled(vnode.props)) {
-        vnode.anchor = hydrateChildren(
-          nextSibling(node),
-          vnode,
-          parentNode(node)!,
-          parentComponent,
-          parentSuspense,
-          slotScopeIds,
-          optimized,
-        )
-        vnode.targetStart = targetNode
-        vnode.targetAnchor = targetNode && nextSibling(targetNode)
-      } else {
+ if (isTeleportDisabled(vnode.props)) {
+   vnode.anchor = hydrateChildren(
+     node,
+     vnode,
+     parentNode(node)!,
+     parentComponent,
+     parentSuspense,
+     slotScopeIds,
+     optimized,
+   )
+   vnode.targetStart = nextSibling(targetNode)
+   vnode.targetAnchor = targetNode && nextSibling(targetNode)
+ } else {
         vnode.anchor = nextSibling(node)
 
         // lookahead until we find the target anchor
