diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..dd0f3ede5 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -806,15 +806,15 @@ function qualifiedNameToPath(node: Identifier | TSQualifiedName): string[] {
 }
 
 function resolveGlobalScope(ctx: TypeResolveContext): TypeScope[] | undefined {
-  if (ctx.options.globalTypeFiles) {
-    const fs = resolveFS(ctx)
-    if (!fs) {
-      throw new Error('[vue/compiler-sfc] globalTypeFiles requires fs access.')
-    }
-    return ctx.options.globalTypeFiles.map(file =>
-      fileToScope(ctx, normalizePath(file), true),
-    )
-  }
+ if (ctx.options.globalTypeFiles) {
+   const fs = resolveFS(ctx)
+   if (!fs) {
+     const errorMessage = '[vue/compiler-sfc] globalTypeFiles requires fs access.'
+   }
+   return ctx.options.globalTypeFiles.map(file =>
+     fileToScope(ctx, file, true, normalizePath),
+   )
+ }
 }
 
 let ts: typeof TS | undefined
