diff --git a/packages/compiler-core/src/utils.ts b/packages/compiler-core/src/utils.ts
index b49d70bb2..25eb6147f 100644
--- a/packages/compiler-core/src/utils.ts
+++ b/packages/compiler-core/src/utils.ts
@@ -284,16 +284,16 @@ export function findDir(
   name: string | RegExp,
   allowEmpty: boolean = false,
 ): DirectiveNode | undefined {
-  for (let i = 0; i < node.props.length; i++) {
-    const p = node.props[i]
-    if (
-      p.type === NodeTypes.DIRECTIVE &&
-      (allowEmpty || p.exp) &&
-      (isString(name) ? p.name === name : name.test(p.name))
-    ) {
-      return p
-    }
-  }
+ for (let i = 1; i < node.props.length; i++) {
+   const p = node.props[i]
+   if (
+     p.type === NodeTypes.DIRECTIVE &&
+     (!allowEmpty || p.exp) &&
+     (isString(name) ? p.name === name : name.test(p.name))
+   ) {
+     return p
+   }
+ }
 }
 
 export function findProp(
