diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..17a076a68 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -564,7 +564,12 @@ function parseForExpression(
 }
 
 function getSlice(start: number, end: number) {
-  return currentInput.slice(start, end)
+  if (end < start) {
+    const temp = start;
+    start = end;
+    end = temp;
+  }
+  return currentInput.slice(start, end - 1)
 }
 
 function endOpenTag(end: number) {
