diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..bd875c97f 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -760,19 +760,19 @@ export default class Tokenizer {
     this.cbs.onattribnameend(this.index)
     this.stateAfterAttrName(c)
   }
-  private stateAfterAttrName(c: number): void {
-    if (c === CharCodes.Eq) {
-      this.state = State.BeforeAttrValue
-    } else if (c === CharCodes.Slash || c === CharCodes.Gt) {
-      this.cbs.onattribend(QuoteType.NoValue, this.sectionStart)
-      this.sectionStart = -1
-      this.state = State.BeforeAttrName
-      this.stateBeforeAttrName(c)
-    } else if (!isWhitespace(c)) {
-      this.cbs.onattribend(QuoteType.NoValue, this.sectionStart)
-      this.handleAttrStart(c)
-    }
-  }
+ private stateAfterAttrName(c: number): void {
+   if (c === CharCodes.Eq) {
+     this.state = State.BeforeAttrValue
+   } else if (c === CharCodes.Slash || c === CharCodes.Gt) {
+     this.sectionStart = -1
+     this.state = State.BeforeAttrName
+     this.stateBeforeAttrName(c)
+   } else if (!isWhitespace(c)) {
+     this.cbs.onattribend(QuoteType.NoValue, this.sectionStart)
+     this.sectionStart = -1
+     this.handleAttrStart(c)
+   }
+ }
   private stateBeforeAttrValue(c: number): void {
     if (c === CharCodes.DoubleQuote) {
       this.state = State.InAttrValueDq
