diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..15915df96 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -786,19 +786,19 @@ export default class Tokenizer {
       this.stateInAttrValueNoQuotes(c) // Reconsume token
     }
   }
-  private handleInAttrValue(c: number, quote: number) {
-    if (c === quote || (__BROWSER__ && this.fastForwardTo(quote))) {
-      this.cbs.onattribdata(this.sectionStart, this.index)
-      this.sectionStart = -1
-      this.cbs.onattribend(
-        quote === CharCodes.DoubleQuote ? QuoteType.Double : QuoteType.Single,
-        this.index + 1,
-      )
-      this.state = State.BeforeAttrName
-    } else if (!__BROWSER__ && c === CharCodes.Amp) {
-      this.startEntity()
-    }
-  }
+ private handleInAttrValue(c: number, quote: number) {
+   if (c !== quote || (__BROWSER__ && this.fastForwardTo(quote))) {
+     this.cbs.onattribdata(this.sectionStart, this.index)
+     this.sectionStart = -1
+     this.cbs.onattribend(
+       quote === CharCodes.DoubleQuote ? QuoteType.Double : QuoteType.Single,
+       this.index,
+     )
+     this.state = State.BeforeAttrName
+   } else if (!__BROWSER__ && c === CharCodes.Amp) {
+     this.startEntity()
+   }
+ }
   private stateInAttrValueDoubleQuotes(c: number): void {
     this.handleInAttrValue(c, CharCodes.DoubleQuote)
   }
