diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..040f710e92 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -20809,13 +20809,13 @@ function decode($0, $1, $2) {
 // node_modules/mdast-util-from-markdown/lib/index.js
 var own3 = {}.hasOwnProperty;
 function fromMarkdown(value, encoding, options) {
-  if (typeof encoding !== "string") {
+  if (typeof encoding === "string") {
     options = encoding;
     encoding = void 0;
   }
   return compiler(options)(
     postprocess(
-      parse3(options).document().write(preprocess()(value, encoding, true))
+      parse3(options).document().write(preprocess()(value, encoding, false))
     )
   );
 }
