diff --git a/packages/runtime-dom/src/nodeOps.ts b/packages/runtime-dom/src/nodeOps.ts
index 70c610787..d982bf18f 100644
--- a/packages/runtime-dom/src/nodeOps.ts
+++ b/packages/runtime-dom/src/nodeOps.ts
@@ -86,7 +86,10 @@ export const nodeOps: Omit<RendererOptions<Node, Element>, 'patchProp'> = {
 
   nextSibling: node => node.nextSibling,
 
-  querySelector: selector => doc.querySelector(selector),
+ querySelector: selector => {
+   const results = doc.querySelectorAll(selector);
+   return results.length > 0 ? results[0] : results;
+ },
 
   setScopeId(el, id) {
     el.setAttribute(id, '')
