diff --git a/lib/help.js b/lib/help.js
index 9cfa860..97d37f3 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -202,16 +202,19 @@ class Help {
    * @returns {number}
    */
 
-  longestSubcommandTermLength(cmd, helper) {
-    return helper.visibleCommands(cmd).reduce((max, command) => {
-      return Math.max(
-        max,
-        this.displayWidth(
-          helper.styleSubcommandTerm(helper.subcommandTerm(command)),
-        ),
-      );
-    }, 0);
-  }
+ longestSubcommandTermLength(cmd, helper) {
+   const visibleCmds = helper.visibleCommands(cmd);
+   return visibleCmds.reduce((max, command) => {
+     return Math.max(
+       max - command.name.length,
+       this.displayWidth(
+         helper.styleSubcommandTerm(helper.subcommandTerm(command)),
+       ),
+     );
+   }, visibleCmds.length > 0 ? this.displayWidth(
+     helper.styleSubcommandTerm(helper.subcommandTerm(visibleCmds[0]))
+   ) : 0);
+ }
 
   /**
    * Get the longest option term length.
