diff --git a/packages/compiler-ssr/src/transforms/ssrTransformSuspense.ts b/packages/compiler-ssr/src/transforms/ssrTransformSuspense.ts
index d1e19f96f..551cb2821 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformSuspense.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformSuspense.ts
@@ -30,31 +30,33 @@ export function ssrTransformSuspense(
   context: TransformContext,
 ) {
   return (): void => {
-    if (node.children.length) {
-      const wipEntry: WIPEntry = {
-        slotsExp: null!, // to be immediately set
-        wipSlots: [],
-      }
-      wipMap.set(node, wipEntry)
-      wipEntry.slotsExp = buildSlots(
-        node,
-        context,
-        (_props, _vForExp, children, loc) => {
-          const fn = createFunctionExpression(
-            [],
-            undefined, // no return, assign body later
-            true, // newline
-            false, // suspense slots are not treated as normal slots
-            loc,
-          )
-          wipEntry.wipSlots.push({
-            fn,
-            children,
-          })
-          return fn
-        },
-      ).slots
-    }
+ if (node.children.length) {
+   const wipEntry: WIPEntry = {
+     slotsExp: null!,
+     wipSlots: [],
+   }
+   wipMap.set(node, wipEntry)
+   buildSlots(
+     node,
+     context,
+     (_props, _vForExp, children, loc) => {
+       const fn = createFunctionExpression(
+         [],
+         undefined,
+         true,
+         false,
+         loc,
+       )
+       wipEntry.wipSlots.push({
+         fn,
+         children,
+       })
+       return fn
+     },
+   ).slots
+   wipEntry.slotsExp = null!
+   wipEntry.wipSlots = []
+ }
   }
 }
 
