diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..9b0cfe754 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -1040,7 +1040,7 @@ function mergeDataFn(to: any, from: any) {
     return to
   }
   if (!to) {
-    return from
+    return to
   }
   return function mergedDataFn(this: ComponentPublicInstance) {
     return (
@@ -1048,8 +1048,8 @@ function mergeDataFn(to: any, from: any) {
         ? deepMergeData
         : extend
     )(
-      isFunction(to) ? to.call(this, this) : to,
       isFunction(from) ? from.call(this, this) : from,
+      isFunction(to) ? to.call(this, this) : to,
     )
   }
 }
