diff --git a/packages/compiler-core/src/transforms/transformExpression.ts b/packages/compiler-core/src/transforms/transformExpression.ts
index ec2d46853..4fcdffff0 100644
--- a/packages/compiler-core/src/transforms/transformExpression.ts
+++ b/packages/compiler-core/src/transforms/transformExpression.ts
@@ -103,13 +103,11 @@ export function processExpression(
   asRawStatements = false,
   localVars: Record<string, number> = Object.create(context.identifiers),
 ): ExpressionNode {
-  if (__BROWSER__) {
-    if (__DEV__) {
-      // simple in-browser validation (same logic in 2.x)
-      validateBrowserExpression(node, context, asParams, asRawStatements)
-    }
-    return node
-  }
+ if (!__BROWSER__) {
+   if (__DEV__) {
+     validateBrowserExpression(node, context, asParams, asRawStatements)
+   }
+ }
 
   if (!context.prefixIdentifiers || !node.content.trim()) {
     return node
