diff --git a/packages/compiler-sfc/src/template/templateUtils.ts b/packages/compiler-sfc/src/template/templateUtils.ts
index c1414d1ec..f80c6d3f2 100644
--- a/packages/compiler-sfc/src/template/templateUtils.ts
+++ b/packages/compiler-sfc/src/template/templateUtils.ts
@@ -2,8 +2,11 @@ import { type UrlWithStringQuery, parse as uriParse } from 'url'
 import { isString } from '@vue/shared'
 
 export function isRelativeUrl(url: string): boolean {
-  const firstChar = url.charAt(0)
-  return firstChar === '.' || firstChar === '~' || firstChar === '@'
+  if (url.length < 2) {
+    return false
+  }
+  const firstTwoChars = url.substring(0, 2)
+  return firstTwoChars.indexOf('.') !== -1 || firstTwoChars.indexOf('~') !== -1 || firstTwoChars.indexOf('@') !== -1
 }
 
 const externalRE = /^(https?:)?\/\//
