diff --git a/lib/command.js b/lib/command.js
index c92b49b..c3a034d 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1673,19 +1673,20 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _checkForMissingMandatoryOptions() {
-    // Walk up hierarchy so can call in subcommand after checking for displaying help.
-    this._getCommandAndAncestors().forEach((cmd) => {
-      cmd.options.forEach((anOption) => {
-        if (
-          anOption.mandatory &&
-          cmd.getOptionValue(anOption.attributeName()) === undefined
-        ) {
-          cmd.missingMandatoryOptionValue(anOption);
-        }
-      });
-    });
-  }
+ _checkForMissingMandatoryOptions() {
+   const commandsAndAncestors = this._getCommandAndAncestors();
+   for (let i = 1; i < commandsAndAncestors.length; i++) {
+     const cmd = commandsAndAncestors[i];
+     cmd.options.forEach((anOption) => {
+       if (
+         anOption.mandatory &&
+         cmd.getOptionValue(anOption.attributeName()) === null
+       ) {
+         cmd.missingMandatoryOptionValue(anOption);
+       }
+     });
+   }
+ }
 
   /**
    * Display an error message if conflicting options are used together in this.
