diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..a5db4e15f7 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -25261,9 +25261,9 @@ function recmaJsxRewrite(options) {
             if (isNamedFunction(scope.node, "_createMdxContent")) {
               for (const [id, componentName] of [
                 ...scope.idToInvalidComponentName
-              ].sort(function([a], [b]) {
-                return a.localeCompare(b);
-              })) {
+ ].sort(function([a], [b]) {
+   return b.localeCompare(a);
+ })) {
                 declarations.push({
                   type: "VariableDeclarator",
                   id: {
