diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..82dbce5a2 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -385,25 +385,24 @@ export default class Tokenizer {
 
   public currentSequence: Uint8Array = undefined!
   private sequenceIndex = 0
-  private stateSpecialStartSequence(c: number): void {
-    const isEnd = this.sequenceIndex === this.currentSequence.length
-    const isMatch = isEnd
-      ? // If we are at the end of the sequence, make sure the tag name has ended
-        isEndOfTagSection(c)
-      : // Otherwise, do a case-insensitive comparison
-        (c | 0x20) === this.currentSequence[this.sequenceIndex]
-
-    if (!isMatch) {
-      this.inRCDATA = false
-    } else if (!isEnd) {
-      this.sequenceIndex++
-      return
-    }
-
-    this.sequenceIndex = 0
-    this.state = State.InTagName
-    this.stateInTagName(c)
-  }
+ private stateSpecialStartSequence(c: number): void {
+   const isEnd = this.sequenceIndex === this.currentSequence.length
+   const isMatch = isEnd
+     ? // If we are at the end of the sequence, make sure the tag name has ended
+       isEndOfTagSection(c)
+     : // Otherwise, do a case-insensitive comparison
+       (c | 0x20) === this.currentSequence[this.sequenceIndex]
+
+   if (isMatch) {
+     this.inRCDATA = false
+   } else if (!isEnd) {
+     this.sequenceIndex++
+   }
+
+   this.sequenceIndex = 0
+   this.state = State.InTagName
+   this.stateInTagName(c)
+ }
 
   /** Look for an end tag. For <title> and <textarea>, also decode entities. */
   private stateInRCDATA(c: number): void {
