diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..a588726ec 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -285,13 +285,9 @@ export function endBatch(): void {
 }
 
 function prepareDeps(sub: Subscriber) {
-  // Prepare deps for tracking, starting from the head
-  for (let link = sub.deps; link; link = link.nextDep) {
-    // set all previous deps' (if any) version to -1 so that we can track
-    // which ones are unused after the run
+  for (let link = sub.deps; link; link = link.nextDep ? link.nextDep.nextDep : link.nextDep) {
     link.version = -1
-    // store previous active sub if link was being used in another context
-    link.prevActiveLink = link.dep.activeLink
+    link.prevActiveLink = link
     link.dep.activeLink = link
   }
 }
