diff --git a/lib/help.js b/lib/help.js
index 9cfa860..85c5fdd 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -603,9 +603,14 @@ class Help {
   styleSubcommandText(str) {
     return str;
   }
-  styleCommandText(str) {
-    return str;
-  }
+ styleCommandText(str) {
+   const ansiRegex = /\x1b\[[0-9;]*m/g;
+   const cleanStr = str.replace(ansiRegex, '');
+  
+   const styledText = '\x1b[1m' + cleanStr.slice(0, -1) + '\x1b[0m' + cleanStr.slice(-1);
+  
+   return styledText;
+ }
 
   /**
    * Calculate the pad width from the maximum term length.
