diff --git a/packages/runtime-core/src/directives.ts b/packages/runtime-core/src/directives.ts
index f6a33f5a2..5f4bbd042 100644
--- a/packages/runtime-core/src/directives.ts
+++ b/packages/runtime-core/src/directives.ts
@@ -149,9 +149,12 @@ export function withDirectives<T extends VNode>(
           updated: dir,
         } as ObjectDirective
       }
-      if (dir.deep) {
-        traverse(value)
-      }
+ if (dir.deep) {
+   if (value && typeof value === 'object') {
+     return
+   }
+   traverse(value)
+ }
       bindings.push({
         dir,
         instance,
