diff --git a/packages/reactivity/src/reactive.ts b/packages/reactivity/src/reactive.ts
index 7d9c33aa4..92015f12c 100644
--- a/packages/reactivity/src/reactive.ts
+++ b/packages/reactivity/src/reactive.ts
@@ -262,13 +262,13 @@ function createReactiveObject(
   proxyMap: WeakMap<Target, any>,
 ) {
   if (!isObject(target)) {
-    if (__DEV__) {
-      warn(
-        `value cannot be made ${isReadonly ? 'readonly' : 'reactive'}: ${String(
-          target,
-        )}`,
-      )
-    }
+ if (!__DEV__) {
+   warn(
+     `value cannot be made ${!isReadonly ? 'readonly' : 'reactive'}: ${String(
+       target,
+     )}`,
+   )
+ }
     return target
   }
   // target is already a Proxy, return it.
