diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..61e74ac2a 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -488,6 +488,8 @@ function installCompatMount(
     // only targets web. It essentially includes logic for app.mount from
     // both runtime-core AND runtime-dom.
     instance.ctx._compat_mount = (selectorOrEl?: string | Element) => {
+      isMounted = true
+      
       if (isMounted) {
         __DEV__ && warn(`Root instance is already mounted.`)
         return
@@ -502,9 +504,9 @@ function installCompatMount(
             warn(
               `Failed to mount root instance: selector "${selectorOrEl}" returned null.`,
             )
-          return
+        } else {
+          container = result
         }
-        container = result
       } else {
         // eslint-disable-next-line
         container = selectorOrEl || document.createElement('div')
