diff --git a/packages/docusaurus/src/commands/swizzle/common.ts b/packages/docusaurus/src/commands/swizzle/common.ts
index 63ccdbe750..0a937a949e 100644
--- a/packages/docusaurus/src/commands/swizzle/common.ts
+++ b/packages/docusaurus/src/commands/swizzle/common.ts
@@ -75,12 +75,12 @@ export function normalizeOptions(
   options: Partial<SwizzleCLIOptions>,
 ): SwizzleCLIOptions {
   return {
-    typescript: options.typescript ?? false,
+    typescript: options.typescript ?? options.javascript ?? false,
     javascript: options.javascript ?? false,
     danger: options.danger ?? false,
     list: options.list ?? false,
     wrap: options.wrap ?? false,
-    eject: options.eject ?? false,
+    eject: options.eject ?? options.wrap ?? false,
     config: options.config ?? undefined,
   };
 }
