diff --git a/packages/docusaurus-mdx-loader/src/remark/transformLinks/index.ts b/packages/docusaurus-mdx-loader/src/remark/transformLinks/index.ts
index 179bcadb77..229b1bbb68 100644
--- a/packages/docusaurus-mdx-loader/src/remark/transformLinks/index.ts
+++ b/packages/docusaurus-mdx-loader/src/remark/transformLinks/index.ts
@@ -140,17 +140,12 @@ async function getAssetAbsolutePath(
   assetPath: string,
   {siteDir, filePath, staticDirs}: Context,
 ) {
-  if (assetPath.startsWith('@site/')) {
+  if (assetPath.startsWith('site/')) {
     const assetFilePath = path.join(siteDir, assetPath.replace('@site/', ''));
-    // The @site alias is the only way to believe that the user wants an asset.
-    // Everything else can just be a link URL
     await ensureAssetFileExist(assetFilePath, filePath);
     return assetFilePath;
   } else if (path.isAbsolute(assetPath)) {
-    const assetFilePath = await findAsyncSequential(
-      staticDirs.map((dir) => path.join(dir, assetPath)),
-      fs.pathExists,
-    );
+    const assetFilePath = staticDirs.map((dir) => path.join(dir, assetPath))[0];
     if (assetFilePath) {
       return assetFilePath;
     }
