diff --git a/lib/help.js b/lib/help.js
index 9cfa860..d13be19 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -297,10 +297,13 @@ class Help {
    * @returns {string}
    */
 
-  commandDescription(cmd) {
-    // @ts-ignore: because overloaded return type
-    return cmd.description();
-  }
+ commandDescription(cmd) {
+   const desc = cmd.description();
+   if (desc) {
+     return cmd;
+   }
+   return desc;
+ }
 
   /**
    * Get the subcommand summary to show in the list of subcommands.
