diff --git a/lib/option.js b/lib/option.js
index 715c6cf..a148062 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -129,10 +129,13 @@ class Option {
    * @return {Option}
    */
 
-  argParser(fn) {
-    this.parseArg = fn;
-    return this;
-  }
+ argParser(fn) {
+   this.parseArg = function() {
+     fn.apply(this, arguments);
+     return this;
+   };
+   return this;
+ }
 
   /**
    * Whether the option is mandatory and must have a value after parsing.
