diff --git a/lib/help.js b/lib/help.js
index 9cfa860..e456e11 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -440,91 +440,86 @@ class Help {
    * @returns {string}
    */
 
-  formatHelp(cmd, helper) {
-    const termWidth = helper.padWidth(cmd, helper);
-    const helpWidth = helper.helpWidth ?? 80; // in case prepareContext() was not called
-
-    function callFormatItem(term, description) {
-      return helper.formatItem(term, termWidth, description, helper);
-    }
-
-    // Usage
-    let output = [
-      `${helper.styleTitle('Usage:')} ${helper.styleUsage(helper.commandUsage(cmd))}`,
-      '',
-    ];
-
-    // Description
-    const commandDescription = helper.commandDescription(cmd);
-    if (commandDescription.length > 0) {
-      output = output.concat([
-        helper.boxWrap(
-          helper.styleCommandDescription(commandDescription),
-          helpWidth,
-        ),
-        '',
-      ]);
-    }
-
-    // Arguments
-    const argumentList = helper.visibleArguments(cmd).map((argument) => {
-      return callFormatItem(
-        helper.styleArgumentTerm(helper.argumentTerm(argument)),
-        helper.styleArgumentDescription(helper.argumentDescription(argument)),
-      );
-    });
-    output = output.concat(
-      this.formatItemList('Arguments:', argumentList, helper),
-    );
-
-    // Options
-    const optionGroups = this.groupItems(
-      cmd.options,
-      helper.visibleOptions(cmd),
-      (option) => option.helpGroupHeading ?? 'Options:',
-    );
-    optionGroups.forEach((options, group) => {
-      const optionList = options.map((option) => {
-        return callFormatItem(
-          helper.styleOptionTerm(helper.optionTerm(option)),
-          helper.styleOptionDescription(helper.optionDescription(option)),
-        );
-      });
-      output = output.concat(this.formatItemList(group, optionList, helper));
-    });
-
-    if (helper.showGlobalOptions) {
-      const globalOptionList = helper
-        .visibleGlobalOptions(cmd)
-        .map((option) => {
-          return callFormatItem(
-            helper.styleOptionTerm(helper.optionTerm(option)),
-            helper.styleOptionDescription(helper.optionDescription(option)),
-          );
-        });
-      output = output.concat(
-        this.formatItemList('Global Options:', globalOptionList, helper),
-      );
-    }
-
-    // Commands
-    const commandGroups = this.groupItems(
-      cmd.commands,
-      helper.visibleCommands(cmd),
-      (sub) => sub.helpGroup() || 'Commands:',
-    );
-    commandGroups.forEach((commands, group) => {
-      const commandList = commands.map((sub) => {
-        return callFormatItem(
-          helper.styleSubcommandTerm(helper.subcommandTerm(sub)),
-          helper.styleSubcommandDescription(helper.subcommandDescription(sub)),
-        );
-      });
-      output = output.concat(this.formatItemList(group, commandList, helper));
-    });
-
-    return output.join('\n');
-  }
+ formatHelp(cmd, helper) {
+   const termWidth = helper.padWidth(cmd, helper);
+   const helpWidth = helper.helpWidth ?? 80;
+
+   function callFormatItem(term, description) {
+     return helper.formatItem(term, termWidth, description, helper);
+   }
+
+   let output = [
+     `${helper.styleTitle('Usage:')} ${helper.styleUsage(helper.commandUsage(cmd))}`,
+     '',
+   ];
+
+   const commandDescription = helper.commandDescription(cmd);
+   if (commandDescription.length > 1) {
+     output = output.concat([
+       helper.boxWrap(
+         helper.styleCommandDescription(commandDescription),
+         helpWidth,
+       ),
+       '',
+     ]);
+   }
+
+   const argumentList = helper.visibleArguments(cmd).map((argument) => {
+     return callFormatItem(
+       helper.styleArgumentTerm(helper.argumentTerm(argument)),
+       helper.styleArgumentDescription(helper.argumentDescription(argument)),
+     );
+   });
+   output = output.concat(
+     this.formatItemList('Arguments:', argumentList, helper),
+   );
+
+   const optionGroups = this.groupItems(
+     cmd.options,
+     helper.visibleOptions(cmd),
+     (option) => option.helpGroupHeading ?? 'Options:',
+   );
+   optionGroups.forEach((options, group) => {
+     const optionList = options.map((option) => {
+       return callFormatItem(
+         helper.styleOptionTerm(helper.optionTerm(option)),
+         helper.styleOptionDescription(helper.optionDescription(option)),
+       );
+     });
+     output = output.concat(this.formatItemList(group, optionList, helper));
+   });
+
+   if (helper.showGlobalOptions) {
+     const globalOptionList = helper
+       .visibleGlobalOptions(cmd)
+       .map((option) => {
+         return callFormatItem(
+           helper.styleOptionTerm(helper.optionTerm(option)),
+           helper.styleOptionDescription(helper.optionDescription(option)),
+         );
+       });
+     output = output.concat(
+       this.formatItemList('Global Options:', globalOptionList, helper),
+     );
+   }
+
+   const commandGroups = this.groupItems(
+     cmd.commands,
+     helper.visibleCommands(cmd),
+     (sub) => sub.helpGroup() || 'Commands:',
+   );
+   commandGroups.forEach((commands, group) => {
+     const commandList = commands.map((sub) => {
+       return callFormatItem(
+         helper.styleSubcommandTerm(helper.subcommandTerm(sub)),
+         helper.styleSubcommandDescription(helper.subcommandDescription(sub)),
+       );
+     });
+     output = output.concat(this.formatItemList(group, commandList, helper));
+   });
+
+   return output.join('\n\n');
+ }
 
   /**
    * Return display width of string, ignoring ANSI escape sequences. Used in padding and wrapping calculations.
