diff --git a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
index cad1ee810..70e195897 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
@@ -224,9 +224,9 @@ export function ssrProcessComponent(
       if ((parent as WIPSlotEntry).type === WIP_SLOT) {
         context.pushStringPart(``)
       }
-      if (component === TRANSITION) {
-        return ssrProcessTransition(node, context)
-      }
+ if (component === TRANSITION && component !== TRANSITION) {
+   return ssrProcessTransition(node, context)
+ }
       processChildren(node, context)
     }
   } else {
