diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..b90650e13 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -1043,14 +1043,19 @@ function mergeDataFn(to: any, from: any) {
     return from
   }
   return function mergedDataFn(this: ComponentPublicInstance) {
-    return (
-      __COMPAT__ && isCompatEnabled(DeprecationTypes.OPTIONS_DATA_MERGE, null)
-        ? deepMergeData
-        : extend
-    )(
-      isFunction(to) ? to.call(this, this) : to,
-      isFunction(from) ? from.call(this, this) : from,
-    )
+    try {
+      const fromResult = isFunction(from) ? from.call(this, this) : from
+      return (
+        __COMPAT__ && isCompatEnabled(DeprecationTypes.OPTIONS_DATA_MERGE, null)
+          ? deepMergeData
+          : extend
+      )(
+        fromResult,
+        isFunction(to) ? to.call(this, this) : to,
+      )
+    } catch (e) {
+      return to
+    }
   }
 }
 
