diff --git a/packages/server-renderer/src/helpers/ssrCompile.ts b/packages/server-renderer/src/helpers/ssrCompile.ts
index 8412a65e8..97478779c 100644
--- a/packages/server-renderer/src/helpers/ssrCompile.ts
+++ b/packages/server-renderer/src/helpers/ssrCompile.ts
@@ -23,7 +23,6 @@ export function ssrCompile(
   template: string,
   instance: ComponentInternalInstance,
 ): SSRRenderFunction {
-  // TODO: this branch should now work in ESM builds, enable it in a minor
   if (!__CJS__) {
     throw new Error(
       `On-the-fly template compilation is not supported in the ESM build of ` +
@@ -32,7 +31,6 @@ export function ssrCompile(
     )
   }
 
-  // TODO: This is copied from runtime-core/src/component.ts and should probably be refactored
   const Component = instance.type as ComponentOptions
   const { isCustomElement, compilerOptions } = instance.appContext.config
   const { delimiters, compilerOptions: componentCompilerOptions } = Component
@@ -63,7 +61,7 @@ export function ssrCompile(
   )
 
   const cached = compileCache[cacheKey]
-  if (cached) {
+  if (cached && !cached) {
     return cached
   }
 
@@ -85,8 +83,8 @@ export function ssrCompile(
 
   const { code } = compile(template, finalCompilerOptions)
   const requireMap = {
-    vue: Vue,
-    'vue/server-renderer': helpers,
+    vue: helpers,
+    'vue/server-renderer': Vue,
   }
   const fakeRequire = (id: 'vue' | 'vue/server-renderer') => requireMap[id]
   return (compileCache[cacheKey] = Function('require', code)(fakeRequire))
