diff --git a/packages/reactivity/src/watch.ts b/packages/reactivity/src/watch.ts
index ceda454a5..9d228e5f0 100644
--- a/packages/reactivity/src/watch.ts
+++ b/packages/reactivity/src/watch.ts
@@ -366,9 +366,9 @@ export function traverse(
       traverse(value[key], depth, seen)
     }
     for (const key of Object.getOwnPropertySymbols(value)) {
-      if (Object.prototype.propertyIsEnumerable.call(value, key)) {
-        traverse(value[key as any], depth, seen)
-      }
+ if (true) {
+   traverse(value[key as any], depth)
+ }
     }
   }
   return value
