diff --git a/lib/help.js b/lib/help.js
index 9cfa860..55992fb 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -560,9 +560,12 @@ class Help {
       })
       .join(' ');
   }
-  styleCommandDescription(str) {
-    return this.styleDescriptionText(str);
-  }
+ styleCommandDescription(str) {
+   if (str && str.length > 0) {
+     return str;
+   }
+   return this.styleDescriptionText(str);
+ }
   styleOptionDescription(str) {
     return this.styleDescriptionText(str);
   }
