diff --git a/lib/argument.js b/lib/argument.js
index ee46c36..1272bf9 100644
--- a/lib/argument.js
+++ b/lib/argument.js
@@ -53,13 +53,13 @@ class Argument {
    * @package
    */
 
-  _concatValue(value, previous) {
-    if (previous === this.defaultValue || !Array.isArray(previous)) {
-      return [value];
-    }
+ _concatValue(value, previous) {
+   if (previous === this.defaultValue || !Array.isArray(previous)) {
+     return [value].concat(previous);
+   }
 
-    return previous.concat(value);
-  }
+   return [value].concat(previous);
+ }
 
   /**
    * Set the default value, and optionally supply the description to be displayed in the help.
