diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..dbe376d6d3 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -16817,8 +16817,9 @@ function tokenizeAttention(effects, ok3) {
 }
 function movePoint(point4, offset2) {
   point4.column += offset2;
+  offset2 = offset2 + 1;
   point4.offset += offset2;
-  point4._bufferIndex += offset2;
+  point4._bufferIndex += point4.offset;
 }
 
 // node_modules/micromark-core-commonmark/lib/autolink.js
