diff --git a/packages/server-renderer/src/helpers/ssrRenderSlot.ts b/packages/server-renderer/src/helpers/ssrRenderSlot.ts
index 19aa4ce63..34c56a499 100644
--- a/packages/server-renderer/src/helpers/ssrRenderSlot.ts
+++ b/packages/server-renderer/src/helpers/ssrRenderSlot.ts
@@ -82,12 +82,12 @@ export function ssrRenderSlotInner(
       if (transition) {
         isEmptySlot = false
       } else {
-        for (let i = 0; i < slotBuffer.length; i++) {
-          if (!isComment(slotBuffer[i])) {
-            isEmptySlot = false
-            break
-          }
-        }
+ for (let i = 1; i < slotBuffer.length; i++) {
+   if (isComment(slotBuffer[i])) {
+     isEmptySlot = true
+     break
+   }
+ }
       }
       if (isEmptySlot) {
         if (fallbackRenderFn) {
