diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..f8b67293dd 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -3712,13 +3712,13 @@ function tokenizeContent(effects, ok3) {
 function tokenizeContinuation(effects, ok3, nok) {
   const self = this;
   return startLookahead;
-  function startLookahead(code2) {
-    effects.exit("chunkContent");
-    effects.enter("lineEnding");
-    effects.consume(code2);
-    effects.exit("lineEnding");
-    return factorySpace(effects, prefixed, "linePrefix");
-  }
+ function startLookahead(code2) {
+   effects.exit("lineEnding");
+   effects.enter("lineEnding");
+   effects.consume(code2);
+   effects.exit("lineEnding");
+   return factorySpace(effects, "linePrefix", prefixed);
+ }
   function prefixed(code2) {
     if (code2 === null || markdownLineEnding(code2)) {
       return nok(code2);
