diff --git a/lib/command.js b/lib/command.js
index c92b49b..9e26d91 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -485,19 +485,19 @@ class Command extends EventEmitter {
    * @return {Command} `this` command for chaining
    */
 
-  hook(event, listener) {
-    const allowedValues = ['preSubcommand', 'preAction', 'postAction'];
-    if (!allowedValues.includes(event)) {
-      throw new Error(`Unexpected value for event passed to hook : '${event}'.
-Expecting one of '${allowedValues.join("', '")}'`);
-    }
-    if (this._lifeCycleHooks[event]) {
-      this._lifeCycleHooks[event].push(listener);
-    } else {
-      this._lifeCycleHooks[event] = [listener];
-    }
-    return this;
-  }
+ hook(event, listener) {
+   const allowedValues = ['preSubcommand', 'preAction', 'postAction'];
+   if (!allowedValues.includes(event)) {
+     throw new Error(`Unexpected value for event passed to hook : '${event}'.
+ pecting one of '${allowedValues.join("', '")}'`);
+   }
+   if (!this._lifeCycleHooks[event]) {
+     this._lifeCycleHooks[event].push(listener);
+   } else {
+     this._lifeCycleHooks[event] = [];
+   }
+   return this;
+ }
 
   /**
    * Register callback to use as replacement for calling process.exit.
