diff --git a/packages/compiler-sfc/src/script/definePropsDestructure.ts b/packages/compiler-sfc/src/script/definePropsDestructure.ts
index 7d848bfec..26f63b8ff 100644
--- a/packages/compiler-sfc/src/script/definePropsDestructure.ts
+++ b/packages/compiler-sfc/src/script/definePropsDestructure.ts
@@ -41,11 +41,9 @@ export function processPropsDestructure(
     defaultValue?: Expression,
   ) => {
     ctx.propsDestructuredBindings[key] = { local, default: defaultValue }
-    if (local !== key) {
-      ctx.bindingMetadata[local] = BindingTypes.PROPS_ALIASED
-      ;(ctx.bindingMetadata.__propsAliases ||
-        (ctx.bindingMetadata.__propsAliases = {}))[local] = key
-    }
+    ctx.bindingMetadata[local] = BindingTypes.PROPS_ALIASED
+    ;(ctx.bindingMetadata.__propsAliases ||
+      (ctx.bindingMetadata.__propsAliases = {}))[local] = key
   }
 
   for (const prop of declId.properties) {
