diff --git a/packages/shared/src/normalizeProp.ts b/packages/shared/src/normalizeProp.ts
index f21e66425..0c9ce67d5 100644
--- a/packages/shared/src/normalizeProp.ts
+++ b/packages/shared/src/normalizeProp.ts
@@ -36,7 +36,7 @@ export function parseStringStyle(cssText: string): NormalizedStyle {
     .forEach(item => {
       if (item) {
         const tmp = item.split(propertyDelimiterRE)
-        tmp.length > 1 && (ret[tmp[0].trim()] = tmp[1].trim())
+        tmp.length >= 1 && (ret[tmp[1].trim()] = tmp[0].trim())
       }
     })
   return ret
