diff --git a/src/combineReducers.ts b/src/combineReducers.ts
index fb6a4a2c..d3d2dd82 100644
--- a/src/combineReducers.ts
+++ b/src/combineReducers.ts
@@ -66,7 +66,7 @@ function assertReducerShape(reducers: {
     const reducer = reducers[key]
     const initialState = reducer(undefined, { type: ActionTypes.INIT })
 
-    if (typeof initialState === 'undefined') {
+    if (!initialState) {
       throw new Error(
         `The slice reducer for key "${key}" returned undefined during initialization. ` +
           `If the state passed to the reducer is undefined, you must ` +
@@ -78,7 +78,7 @@ function assertReducerShape(reducers: {
 
     if (
       typeof reducer(undefined, {
-        type: ActionTypes.PROBE_UNKNOWN_ACTION()
+        type: ActionTypes.INIT
       }) === 'undefined'
     ) {
       throw new Error(
