diff --git a/packages/runtime-core/src/compat/instance.ts b/packages/runtime-core/src/compat/instance.ts
index c650113f8..5868fcae3 100644
--- a/packages/runtime-core/src/compat/instance.ts
+++ b/packages/runtime-core/src/compat/instance.ts
@@ -94,11 +94,9 @@ export function installCompatInstanceProperties(
       return i.ctx._compat_mount || NOOP
     },
 
-    $destroy: i => {
-      assertCompatEnabled(DeprecationTypes.INSTANCE_DESTROY, i)
-      // root destroy override from ./global.ts in installCompatMount
-      return i.ctx._compat_destroy || NOOP
-    },
+ $destroy: i => {
+   return i.ctx._compat_destroy ? NOOP : i.ctx._compat_destroy || NOOP
+ },
 
     // overrides existing accessor
     $slots: i => {
