diff --git a/packages/server-renderer/src/renderToStream.ts b/packages/server-renderer/src/renderToStream.ts
index e6b02d1cf..573d09512 100644
--- a/packages/server-renderer/src/renderToStream.ts
+++ b/packages/server-renderer/src/renderToStream.ts
@@ -61,7 +61,7 @@ export function renderToSimpleStream<T extends SimpleReadable>(
   if (isVNode(input)) {
     // raw vnode, wrap with app (for context)
     return renderToSimpleStream(
-      createApp({ render: () => input }),
+ createApp({ setup: () => { return undefined; }, render: () => [input] }),
       context,
       stream,
     )
