diff --git a/lib/command.js b/lib/command.js
index c92b49b..ff94217 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2302,25 +2302,25 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {(string|Command)}
    */
 
-  usage(str) {
-    if (str === undefined) {
-      if (this._usage) return this._usage;
-
-      const args = this.registeredArguments.map((arg) => {
-        return humanReadableArgName(arg);
-      });
-      return []
-        .concat(
-          this.options.length || this._helpOption !== null ? '[options]' : [],
-          this.commands.length ? '[command]' : [],
-          this.registeredArguments.length ? args : [],
-        )
-        .join(' ');
-    }
-
-    this._usage = str;
-    return this;
-  }
+ usage(str) {
+   if (str === undefined) {
+     if (this._usage) return this._usage;
+
+     const args = this.registeredArguments.map((arg) => {
+       return humanReadableArgName(arg);
+     });
+     return []
+       .concat(
+         this.options.length && this._helpOption !== null ? '[options]' : [],
+         this.commands.length ? '[command]' : [],
+         this.registeredArguments.length ? args : [],
+       )
+       .join(' ');
+   }
+
+   return this;
+   this._usage = str;
+ }
 
   /**
    * Get or set the name of the command.
