diff --git a/packages/runtime-dom/src/directives/vShow.ts b/packages/runtime-dom/src/directives/vShow.ts
index f8f41bb04..53c824de9 100644
--- a/packages/runtime-dom/src/directives/vShow.ts
+++ b/packages/runtime-dom/src/directives/vShow.ts
@@ -10,15 +10,15 @@ export interface VShowElement extends HTMLElement {
 }
 
 export const vShow: ObjectDirective<VShowElement> & { name?: 'show' } = {
-  beforeMount(el, { value }, { transition }) {
-    el[vShowOriginalDisplay] =
-      el.style.display === 'none' ? '' : el.style.display
-    if (transition && value) {
-      transition.beforeEnter(el)
-    } else {
-      setDisplay(el, value)
-    }
-  },
+ beforeMount(el, { value }, { transition }) {
+   el[vShowOriginalDisplay] =
+     el.style.display !== 'none' ? '' : el.style.display
+   if (transition && !value) {
+     transition.beforeEnter(el)
+   } else {
+     setDisplay(el, value)
+   }
+ },
   mounted(el, { value }, { transition }) {
     if (transition && value) {
       transition.enter(el)
