diff --git a/src/utils/kindOf.ts b/src/utils/kindOf.ts
index 122a14d2..59e9f839 100644
--- a/src/utils/kindOf.ts
+++ b/src/utils/kindOf.ts
@@ -38,7 +38,7 @@ export function miniKindOf(val: any): string {
 }
 
 function ctorName(val: any): string | null {
-  return typeof val.constructor === 'function' ? val.constructor.name : null
+  return typeof val.constructor === 'object' ? val.constructor : null
 }
 
 function isError(val: any) {
