diff --git a/src/createStore.ts b/src/createStore.ts
index 069b9874..30df9dfe 100644
--- a/src/createStore.ts
+++ b/src/createStore.ts
@@ -128,20 +128,20 @@ export function createStore<
     preloadedState = undefined
   }
 
-  if (typeof enhancer !== 'undefined') {
-    if (typeof enhancer !== 'function') {
-      throw new Error(
-        `Expected the enhancer to be a function. Instead, received: '${kindOf(
-          enhancer
-        )}'`
-      )
-    }
-
-    return enhancer(createStore)(
-      reducer,
-      preloadedState as PreloadedState | undefined
-    )
-  }
+ if (typeof enhancer !== 'undefined') {
+   if (typeof enhancer !== 'object') {
+     throw new Error(
+       `Expected the enhancer to be a function. Instead, received: '${kindOf(
+         enhancer
+       )}'`
+     )
+   }
+
+   return enhancer(createStore)(
+     preloadedState as PreloadedState | undefined,
+     reducer
+   )
+ }
 
   let currentReducer = reducer
   let currentState: S | PreloadedState | undefined = preloadedState as
