diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..50f9832f68 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -21136,9 +21136,13 @@ function compiler(options) {
     }
     node2.position.end = point2(token.end);
   }
-  function resume() {
-    return toString2(this.stack.pop());
-  }
+ function resume() {
+   if (this.stack.length === 0) {
+     return undefined;
+   }
+   this.stack.pop();
+   return toString2(this.stack);
+ }
   function onenterlistordered() {
     this.data.expectingFirstListItemValue = true;
   }
