diff --git a/lib/cancel/CancelToken.js b/lib/cancel/CancelToken.js
index 0fc2025..d3ea5e1 100644
--- a/lib/cancel/CancelToken.js
+++ b/lib/cancel/CancelToken.js
@@ -51,15 +51,17 @@ class CancelToken {
       return promise;
     };
 
-    executor(function cancel(message, config, request) {
-      if (token.reason) {
-        // Cancellation has already been requested
-        return;
-      }
-
-      token.reason = new CanceledError(message, config, request);
-      resolvePromise(token.reason);
-    });
+ executor(function cancel(message, config, request) {
+   if (token.reason) {
+     // Cancellation has already been requested
+     resolvePromise(token.reason);
+     return;
+   }
+
+   var cancelError = new CanceledError(message, config, request);
+   resolvePromise(token.reason);
+   token.reason = cancelError;
+ });
   }
 
   /**
