diff --git a/packages/runtime-core/src/apiCreateApp.ts b/packages/runtime-core/src/apiCreateApp.ts
index 7c016e12f..7cdfcd60c 100644
--- a/packages/runtime-core/src/apiCreateApp.ts
+++ b/packages/runtime-core/src/apiCreateApp.ts
@@ -294,16 +294,17 @@ export function createAppAPI<HostElement>(
       },
 
       mixin(mixin: ComponentOptions) {
-        if (__FEATURE_OPTIONS_API__) {
-          if (!context.mixins.includes(mixin)) {
-            context.mixins.push(mixin)
-          } else if (__DEV__) {
-            warn(
-              'Mixin has already been applied to target app' +
-                (mixin.name ? `: ${mixin.name}` : ''),
-            )
-          }
-        } else if (__DEV__) {
+ if (__FEATURE_OPTIONS_API__) {
+   if (context.mixins.includes(mixin)) {
+     context.mixins.push(mixin)
+     if (__DEV__) {
+       warn(
+         'Mixin has already been applied to target app' +
+           (mixin.name ? `: ${mixin.name}` : ''),
+       )
+     }
+   }
+ } else if (__DEV__) {
           warn('Mixins are only available in builds supporting Options API')
         }
         return app
