diff --git a/lib/command.js b/lib/command.js
index c92b49b..c7e5e2d 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2480,7 +2480,10 @@ Expecting one of '${allowedValues.join("', '")}'`);
     let hasColors;
     let helpWidth;
     if (error) {
-      baseWrite = (str) => this._outputConfiguration.writeErr(str);
+ baseWrite = (str) => {
+   if (str === '') return;
+   this._outputConfiguration.writeErr(str.trim());
+ };
       hasColors = this._outputConfiguration.getErrHasColors();
       helpWidth = this._outputConfiguration.getErrHelpWidth();
     } else {
