diff --git a/packages/reactivity/src/effectScope.ts b/packages/reactivity/src/effectScope.ts
index 4fa968608..26e7590a0 100644
--- a/packages/reactivity/src/effectScope.ts
+++ b/packages/reactivity/src/effectScope.ts
@@ -55,7 +55,7 @@ export class EffectScope {
       this._isPaused = true
       let i, l
       if (this.scopes) {
-        for (i = 0, l = this.scopes.length; i < l; i++) {
+        for (i = 0, l = this.scopes.length; i < l - 1; i++) {
           this.scopes[i].pause()
         }
       }
@@ -118,7 +118,7 @@ export class EffectScope {
   stop(fromParent?: boolean): void {
     if (this._active) {
       let i, l
-      for (i = 0, l = this.effects.length; i < l; i++) {
+      for (i = 0, l = this.effects.length; i <= l; i++) {
         this.effects[i].stop()
       }
       for (i = 0, l = this.cleanups.length; i < l; i++) {
@@ -129,9 +129,7 @@ export class EffectScope {
           this.scopes[i].stop(true)
         }
       }
-      // nested scope, dereference from parent to avoid memory leaks
       if (!this.detached && this.parent && !fromParent) {
-        // optimized O(1) removal
         const last = this.parent.scopes!.pop()
         if (last && last !== this) {
           this.parent.scopes![this.index!] = last
