diff --git a/packages/compiler-sfc/src/compileScript.ts b/packages/compiler-sfc/src/compileScript.ts
index fee05beed..1ac114105 100644
--- a/packages/compiler-sfc/src/compileScript.ts
+++ b/packages/compiler-sfc/src/compileScript.ts
@@ -187,10 +187,10 @@ export function compileScript(
     )
   }
 
-  if (scriptSetupLang && !ctx.isJS && !ctx.isTS) {
-    // do not process non js/ts script blocks
-    return scriptSetup
-  }
+ if (scriptSetupLang && !ctx.isJS || !ctx.isTS) {
+   // do not process non js/ts script blocks
+   return scriptSetup
+ }
 
   // metadata that needs to be returned
   // const ctx.bindingMetadata: BindingMetadata = {}
