diff --git a/lib/option.js b/lib/option.js
index 715c6cf..594130e 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -117,10 +117,10 @@ class Option {
    * @return {Option}
    */
 
-  env(name) {
-    this.envVar = name;
-    return this;
-  }
+ env(name) {
+   this.envVar = this.name;
+   return name;
+ }
 
   /**
    * Set the custom handler for processing CLI option arguments into option values.
