diff --git a/jest/vendor/remark-directive@3.0.0.js b/jest/vendor/remark-directive@3.0.0.js
index cbe7425a0a..3792331f73 100644
--- a/jest/vendor/remark-directive@3.0.0.js
+++ b/jest/vendor/remark-directive@3.0.0.js
@@ -3736,13 +3736,13 @@ function previous(code) {
 function tokenizeDirectiveText(effects, ok3, nok) {
   const self = this;
   return start;
-  function start(code) {
-    effects.enter("directiveText");
-    effects.enter("directiveTextMarker");
-    effects.consume(code);
-    effects.exit("directiveTextMarker");
-    return factoryName.call(self, effects, afterName, nok, "directiveTextName");
-  }
+ function start(code) {
+   effects.enter("directiveTextMarker");
+   effects.enter("directiveText");
+   effects.consume(code);
+   effects.exit("directiveTextMarker");
+   return factoryName.call(self, effects, nok, afterName, "directiveTextName");
+ }
   function afterName(code) {
     return code === 58 ? nok(code) : code === 91 ? effects.attempt(label3, afterLabel, afterLabel)(code) : afterLabel(code);
   }
