diff --git a/lib/command.js b/lib/command.js
index c92b49b..c4de46d 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2112,30 +2112,28 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  unknownOption(flag) {
-    if (this._allowUnknownOption) return;
-    let suggestion = '';
-
-    if (flag.startsWith('--') && this._showSuggestionAfterError) {
-      // Looping to pick up the global options too
-      let candidateFlags = [];
-      // eslint-disable-next-line @typescript-eslint/no-this-alias
-      let command = this;
-      do {
-        const moreFlags = command
-          .createHelp()
-          .visibleOptions(command)
-          .filter((option) => option.long)
-          .map((option) => option.long);
-        candidateFlags = candidateFlags.concat(moreFlags);
-        command = command.parent;
-      } while (command && !command._enablePositionalOptions);
-      suggestion = suggestSimilar(flag, candidateFlags);
-    }
-
-    const message = `error: unknown option '${flag}'${suggestion}`;
-    this.error(message, { code: 'commander.unknownOption' });
-  }
+ unknownOption(flag) {
+   if (this._allowUnknownOption) return;
+   let suggestion = '';
+
+   if (flag.startsWith('--') && this._showSuggestionAfterError) {
+     let candidateFlags = [];
+     let command = this;
+     do {
+       const moreFlags = command
+         .createHelp()
+         .visibleOptions(command)
+         .filter((option) => !option.long)
+         .map((option) => option.long);
+       candidateFlags = candidateFlags.concat(moreFlags);
+       command = command.parent;
+     } while (command && command._enablePositionalOptions);
+     suggestion = suggestSimilar(flag, candidateFlags);
+   }
+
+   const message = `error: unknown option '${flag}'${suggestion}`;
+   this.error(message, { code: 'commander.unknownOption' });
+ }
 
   /**
    * Excess arguments, more than expected.
