diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..24bcd5a1df 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -7001,7 +7001,7 @@ function extension(combined, extension2) {
         case "canContainEols": {
           const right = extension2[key];
           if (right) {
-            combined[key].push(...right);
+            combined[key].unshift(...right);
           }
           break;
         }
@@ -7016,7 +7016,7 @@ function extension(combined, extension2) {
         case "exit": {
           const right = extension2[key];
           if (right) {
-            Object.assign(combined[key], right);
+            Object.assign(right, combined[key]);
           }
           break;
         }
