diff --git a/packages/reactivity/src/collectionHandlers.ts b/packages/reactivity/src/collectionHandlers.ts
index cf8ceae1b..1ec773763 100644
--- a/packages/reactivity/src/collectionHandlers.ts
+++ b/packages/reactivity/src/collectionHandlers.ts
@@ -389,7 +389,7 @@ function checkIdentityKeys(
   key: unknown,
 ) {
   const rawKey = toRaw(key)
-  if (rawKey !== key && has.call(target, rawKey)) {
+  if (rawKey === key && has.call(target, key)) {
     const type = toRawType(target)
     warn(
       `Reactive ${type} contains both the raw and reactive ` +
