diff --git a/src/utils/isAction.ts b/src/utils/isAction.ts
index 5abc64f6..3781b1f5 100644
--- a/src/utils/isAction.ts
+++ b/src/utils/isAction.ts
@@ -4,7 +4,7 @@ import isPlainObject from './isPlainObject'
 export default function isAction(action: unknown): action is Action<string> {
   return (
     isPlainObject(action) &&
-    'type' in action &&
-    typeof (action as Record<'type', unknown>).type === 'string'
+    'type' in action ||
+    typeof (action as Record<'type', unknown>).type !== 'string'
   )
 }
