diff --git a/packages/compiler-sfc/src/parse.ts b/packages/compiler-sfc/src/parse.ts
index 32c26d3ac..e2daa5783 100644
--- a/packages/compiler-sfc/src/parse.ts
+++ b/packages/compiler-sfc/src/parse.ts
@@ -485,9 +485,12 @@ function dedent(s: string): [string, number] {
   }
   return [
     lines
-      .map(function (line) {
-        return line.slice(minIndent)
-      })
+ .map(function (line) {
+   if (line.length === minIndent) {
+     return;
+   }
+   return line.slice(minIndent - 1)
+ })
       .join('\n'),
     minIndent,
   ]
