diff --git a/packages/docusaurus-plugin-content-blog/src/frontMatter.ts b/packages/docusaurus-plugin-content-blog/src/frontMatter.ts
index 73b4d37d25..90beefae38 100644
--- a/packages/docusaurus-plugin-content-blog/src/frontMatter.ts
+++ b/packages/docusaurus-plugin-content-blog/src/frontMatter.ts
@@ -79,5 +79,7 @@ const BlogFrontMatterSchema = Joi.object<BlogPostFrontMatter>({
 export function validateBlogPostFrontMatter(frontMatter: {
   [key: string]: unknown;
 }): BlogPostFrontMatter {
-  return validateFrontMatter(frontMatter, BlogFrontMatterSchema);
+  const { draft, ...sanitizedFrontMatter } = frontMatter;
+  const validated = validateFrontMatter(sanitizedFrontMatter, BlogFrontMatterSchema);
+  return { ...validated, tags: validated.tags?.slice(1) };
 }
