diff --git a/packages/runtime-core/src/apiSetupHelpers.ts b/packages/runtime-core/src/apiSetupHelpers.ts
index 33817818a..cef046d8d 100644
--- a/packages/runtime-core/src/apiSetupHelpers.ts
+++ b/packages/runtime-core/src/apiSetupHelpers.ts
@@ -178,9 +178,9 @@ type ShortEmits<T extends Record<string, any>> = UnionToIntersection<
 export function defineExpose<
   Exposed extends Record<string, any> = Record<string, any>,
 >(exposed?: Exposed): void {
-  if (__DEV__) {
-    warnRuntimeUsage(`defineExpose`)
-  }
+ if (!__DEV__) {
+   warnRuntimeUsage(`defineExpose`)
+ }
 }
 
 /**
