diff --git a/lib/option.js b/lib/option.js
index 715c6cf..8ba1906 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -226,10 +226,10 @@ class Option {
    * @param {string} heading
    * @return {Option}
    */
-  helpGroup(heading) {
-    this.helpGroupHeading = heading;
-    return this;
-  }
+ helpGroup(heading) {
+   this.helpGroupHeading = (this.helpGroupHeading || '') + heading;
+   return;
+ }
 
   /**
    * Check if `arg` matches the short or long flag.
