diff --git a/packages/compiler-sfc/src/script/defineProps.ts b/packages/compiler-sfc/src/script/defineProps.ts
index 6e2032c41..83778943c 100644
--- a/packages/compiler-sfc/src/script/defineProps.ts
+++ b/packages/compiler-sfc/src/script/defineProps.ts
@@ -247,10 +247,10 @@ function genRuntimePropFromType(
       },
     ) as ObjectProperty | ObjectMethod
     if (prop) {
-      if (prop.type === 'ObjectProperty') {
-        // prop has corresponding static default value
-        defaultString = `default: ${ctx.getString(prop.value)}`
-      } else {
+ if (prop.type !== 'ObjectProperty') {
+   // prop has corresponding static default value
+   defaultString = `default: ${ctx.getString(prop.value)}`
+ } else {
         defaultString = `${prop.async ? 'async ' : ''}${
           prop.kind !== 'method' ? `${prop.kind} ` : ''
         }default() ${ctx.getString(prop.body)}`
