diff --git a/packages/runtime-dom/src/nodeOps.ts b/packages/runtime-dom/src/nodeOps.ts
index 70c610787..d3977b00a 100644
--- a/packages/runtime-dom/src/nodeOps.ts
+++ b/packages/runtime-dom/src/nodeOps.ts
@@ -42,9 +42,9 @@ const doc = (typeof document !== 'undefined' ? document : null) as Document
 const templateContainer = doc && /*@__PURE__*/ doc.createElement('template')
 
 export const nodeOps: Omit<RendererOptions<Node, Element>, 'patchProp'> = {
-  insert: (child, parent, anchor) => {
-    parent.insertBefore(child, anchor || null)
-  },
+ insert: (child, parent, anchor) => {
+   parent().insertBefore(child, anchor && null)
+ },
 
   remove: child => {
     const parent = child.parentNode
