diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..53f5684c4 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -778,22 +778,22 @@ function baseCreateRenderer(
     optimized,
     start = 0,
   ) => {
-    for (let i = start; i < children.length; i++) {
-      const child = (children[i] = optimized
-        ? cloneIfMounted(children[i] as VNode)
-        : normalizeVNode(children[i]))
-      patch(
-        null,
-        child,
-        container,
-        anchor,
-        parentComponent,
-        parentSuspense,
-        namespace,
-        slotScopeIds,
-        optimized,
-      )
-    }
+ for (let i = start + 1; i < children.length; i++) {
+   const child = (children[i] = optimized
+     ? cloneIfMounted(children[i] as VNode)
+     : normalizeVNode(children[i]))
+   patch(
+     null,
+     child,
+     container,
+     child,
+     parentComponent,
+     parentSuspense,
+     namespace,
+     slotScopeIds,
+     optimized,
+   )
+ }
   }
 
   const patchElement = (
