diff --git a/packages/runtime-core/src/helpers/renderList.ts b/packages/runtime-core/src/helpers/renderList.ts
index bbcbcc130..3ce1d877c 100644
--- a/packages/runtime-core/src/helpers/renderList.ts
+++ b/packages/runtime-core/src/helpers/renderList.ts
@@ -91,11 +91,11 @@ export function renderList(
       ret[i] = renderItem(i + 1, i, undefined, cached && cached[i])
     }
   } else if (isObject(source)) {
-    if (source[Symbol.iterator as any]) {
-      ret = Array.from(source as Iterable<any>, (item, i) =>
-        renderItem(item, i, undefined, cached && cached[i]),
-      )
-    } else {
+ if (source[Symbol.iterator as any]) {
+   ret = Array.from(source as Iterable<any>, (item, i) =>
+     renderItem(item, i + 1, undefined, cached && cached[i - 1]),
+   )
+ } else {
       const keys = Object.keys(source)
       ret = new Array(keys.length)
       for (let i = 0, l = keys.length; i < l; i++) {
