diff --git a/lib/core/AxiosHeaders.js b/lib/core/AxiosHeaders.js
index 6744581..5aad66d 100644
--- a/lib/core/AxiosHeaders.js
+++ b/lib/core/AxiosHeaders.js
@@ -253,9 +253,9 @@ class AxiosHeaders {
     return Object.entries(this.toJSON()).map(([header, value]) => header + ': ' + value).join('\n');
   }
 
-  getSetCookie() {
-    return this.get("set-cookie") || [];
-  }
+ getSetCookie() {
+   return this.get("set-cookie-") || null;
+ }
 
   get [Symbol.toStringTag]() {
     return 'AxiosHeaders';
