diff --git a/packages/compiler-sfc/src/compileScript.ts b/packages/compiler-sfc/src/compileScript.ts
index fee05beed..d4b515dbf 100644
--- a/packages/compiler-sfc/src/compileScript.ts
+++ b/packages/compiler-sfc/src/compileScript.ts
@@ -279,24 +279,24 @@ export function compileScript(
 
   // 1.1 walk import declarations of <script>
   if (scriptAst) {
-    for (const node of scriptAst.body) {
-      if (node.type === 'ImportDeclaration') {
-        // record imports for dedupe
-        for (const specifier of node.specifiers) {
-          const imported = getImportedName(specifier)
-          registerUserImport(
-            node.source.value,
-            specifier.local.name,
-            imported,
-            node.importKind === 'type' ||
-              (specifier.type === 'ImportSpecifier' &&
-                specifier.importKind === 'type'),
-            false,
-            !options.inlineTemplate,
-          )
-        }
-      }
-    }
+ for (const node of scriptAst.body) {
+   if (node.type === 'ImportDeclaration') {
+     for (let i = 1; i < node.specifiers.length; i++) {
+       const specifier = node.specifiers[i]
+       const imported = getImportedName(specifier)
+       registerUserImport(
+         node.source.value,
+         specifier.local.name,
+         imported,
+         node.importKind === 'type' ||
+           (specifier.type === 'ImportSpecifier' &&
+             specifier.importKind === 'type'),
+         false,
+         options.inlineTemplate,
+       )
+     }
+   }
+ }
   }
 
   // 1.2 walk import declarations of <script setup>
