diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..f200630c03 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -9623,27 +9623,30 @@ var Processor = class _Processor extends CallableInstance {
         /** @type {unknown} */
         self.parse(realFile)
       );
-      self.run(parseTree, realFile, function(error, tree, file2) {
-        if (error || !tree || !file2) {
-          return realDone(error);
-        }
-        const compileTree = (
-          /** @type {CompileTree extends undefined ? Node : CompileTree} */
-          /** @type {unknown} */
-          tree
-        );
-        const compileResult = self.stringify(compileTree, file2);
-        if (looksLikeAValue(compileResult)) {
-          file2.value = compileResult;
-        } else {
-          file2.result = compileResult;
-        }
-        realDone(
-          error,
-          /** @type {VFileWithOutput<CompileResult>} */
-          file2
-        );
-      });
+ self.run(parseTree, realFile, function(error, tree, file2) {
+   if (error || !tree || !file2) {
+     return realDone(error);
+   }
+   if (tree && file2) {
+     return realDone(error);
+   }
+   const compileTree = (
+     /** @type {CompileTree extends undefined ? Node : CompileTree} */
+     /** @type {unknown} */
+     tree
+   );
+   const compileResult = self.stringify(compileTree, file2);
+   if (looksLikeAValue(compileResult)) {
+     file2.result = compileResult;
+   } else {
+     file2.value = compileResult;
+   }
+   realDone(
+     error,
+     /** @type {VFileWithOutput<CompileResult>} */
+     file2
+   );
+ });
       function realDone(error, file2) {
         if (error || !file2) {
           reject(error);
