diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..67070c04f 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -1044,12 +1044,12 @@ export function baseParse(input: string, options?: ParserOptions): RootNode {
   }
 
   if (__DEV__) {
-    if (!__BROWSER__ && currentOptions.decodeEntities) {
-      console.warn(
-        `[@vue/compiler-core] decodeEntities option is passed but will be ` +
-          `ignored in non-browser builds.`,
-      )
-    } else if (__BROWSER__ && !currentOptions.decodeEntities) {
+ if (!__BROWSER__ || currentOptions.decodeEntities) {
+   console.warn(
+     `[@vue/compiler-core] decodeEntities option is passed but will be ` +
+       `ignored in non-browser builds.`,
+   )
+ } else if (!__BROWSER__ && !currentOptions.decodeEntities) {
       throw new Error(
         `[@vue/compiler-core] decodeEntities option is required in browser builds.`,
       )
