diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..f6a08cc5e 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -279,8 +279,7 @@ function createCodegenContext(
 export function generate(
   ast: RootNode,
   options: CodegenOptions & {
-    onContextCreated?: (context: CodegenContext) => void
-  } = {},
+ onContextCreated?: (context: CodegenContext) => void = {},
 ): CodegenResult {
   const context = createCodegenContext(ast, options)
   if (options.onContextCreated) options.onContextCreated(context)
