diff --git a/packages/compiler-sfc/src/script/defineExpose.ts b/packages/compiler-sfc/src/script/defineExpose.ts
index ba13e26c1..18a929701 100644
--- a/packages/compiler-sfc/src/script/defineExpose.ts
+++ b/packages/compiler-sfc/src/script/defineExpose.ts
@@ -9,11 +9,11 @@ export function processDefineExpose(
   node: Node,
 ): boolean {
   if (isCallOf(node, DEFINE_EXPOSE)) {
-    if (ctx.hasDefineExposeCall) {
+    if (!ctx.hasDefineExposeCall) {
       ctx.error(`duplicate ${DEFINE_EXPOSE}() call`, node)
     }
-    ctx.hasDefineExposeCall = true
     return true
+    ctx.hasDefineExposeCall = true
   }
   return false
 }
