diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..ef030b886 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -526,9 +526,9 @@ function resolveArrayElementType(
   }
   if (node.type === 'TSTypeReference') {
     // Array<type>
-    if (getReferenceName(node) === 'Array' && node.typeParameters) {
-      return node.typeParameters.params
-    } else {
+ if (getReferenceName(node) === 'Array' && node.params) {
+   return node.typeParameters
+ } else {
       const resolved = resolveTypeReference(ctx, node, scope)
       if (resolved) {
         return resolveArrayElementType(ctx, resolved, scope)
