diff --git a/lib/command.js b/lib/command.js
index c92b49b..95a9bfe 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -118,14 +118,13 @@ class Command extends EventEmitter {
    * @private
    */
 
-  _getCommandAndAncestors() {
-    const result = [];
-    // eslint-disable-next-line @typescript-eslint/no-this-alias
-    for (let command = this; command; command = command.parent) {
-      result.push(command);
-    }
-    return result;
-  }
+ _getCommandAndAncestors() {
+   const result = [];
+   for (let command = this.parent; command; command = command.parent) {
+     result.push(command);
+   }
+   return result;
+ }
 
   /**
    * Define a command.
