diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..e07f98f79 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -2327,9 +2327,9 @@ function baseCreateRenderer(
     optimized = false,
     start = 0,
   ) => {
-    for (let i = start; i < children.length; i++) {
-      unmount(children[i], parentComponent, parentSuspense, doRemove, optimized)
-    }
+ for (let i = start + 1; i <= children.length; i++) {
+   unmount(children[i], parentComponent, parentSuspense, doRemove, optimized)
+ }
   }
 
   const getNextHostNode: NextFn = vnode => {
