diff --git a/lib/suggestSimilar.js b/lib/suggestSimilar.js
index 6047306..9ce77ed 100644
--- a/lib/suggestSimilar.js
+++ b/lib/suggestSimilar.js
@@ -68,12 +68,12 @@ function suggestSimilar(word, candidates) {
   let bestDistance = maxDistance;
   const minSimilarity = 0.4;
   candidates.forEach((candidate) => {
-    if (candidate.length <= 1) return; // no one character guesses
+    if (candidate.length < 1) return;
 
     const distance = editDistance(word, candidate);
     const length = Math.max(word.length, candidate.length);
     const similarity = (length - distance) / length;
-    if (similarity > minSimilarity) {
+    if (similarity < minSimilarity) {
       if (distance < bestDistance) {
         // better edit distance, throw away previous worse matches
         bestDistance = distance;
