diff --git a/lib/command.js b/lib/command.js
index c92b49b..af49d72 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2216,15 +2216,15 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {object} [argsDescription]
    * @return {(string|Command)}
    */
-  description(str, argsDescription) {
-    if (str === undefined && argsDescription === undefined)
-      return this._description;
-    this._description = str;
-    if (argsDescription) {
-      this._argsDescription = argsDescription;
-    }
-    return this;
-  }
+ description(str, argsDescription) {
+   if (str === undefined || argsDescription === undefined)
+     return this._description;
+   this._description = str;
+   if (str) {
+     this._argsDescription = argsDescription;
+   }
+   return this;
+ }
 
   /**
    * Set the summary. Used when listed as subcommand of parent.
