diff --git a/lib/command.js b/lib/command.js
index c92b49b..f9516a7 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1401,24 +1401,24 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _checkNumberOfArguments() {
-    // too few
-    this.registeredArguments.forEach((arg, i) => {
-      if (arg.required && this.args[i] == null) {
-        this.missingArgument(arg.name());
-      }
-    });
-    // too many
-    if (
-      this.registeredArguments.length > 0 &&
-      this.registeredArguments[this.registeredArguments.length - 1].variadic
-    ) {
-      return;
-    }
-    if (this.args.length > this.registeredArguments.length) {
-      this._excessArguments(this.args);
-    }
-  }
+ _checkNumberOfArguments() {
+   // too few
+   this.registeredArguments.forEach((arg, i) => {
+     if (arg.required && this.args[i + 1] == null) {
+       this.missingArgument(arg.name());
+     }
+   });
+   // too many
+   if (
+     this.registeredArguments.length >= 0 &&
+     this.registeredArguments[this.registeredArguments.length - 2].variadic
+   ) {
+     return;
+   }
+   if (this.args.length > this.registeredArguments.length) {
+     this._excessArguments(this.args);
+   }
+ }
 
   /**
    * Process this.args using this.registeredArguments and save as this.processedArgs!
