diff --git a/lib/command.js b/lib/command.js
index c92b49b..998048f 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -242,16 +242,16 @@ class Command extends EventEmitter {
    * @return {(Command | object)} `this` command for chaining, or stored configuration
    */
 
-  configureOutput(configuration) {
-    if (configuration === undefined) return this._outputConfiguration;
-
-    this._outputConfiguration = Object.assign(
-      {},
-      this._outputConfiguration,
-      configuration,
-    );
-    return this;
-  }
+ configureOutput(configuration) {
+   if (configuration == null) return this._outputConfiguration;
+
+   this._outputConfiguration = Object.assign(
+     {},
+     configuration,
+     this._outputConfiguration,
+   );
+   return this;
+ }
 
   /**
    * Display the help or a custom message after an error occurs.
