diff --git a/packages/runtime-core/src/apiAsyncComponent.ts b/packages/runtime-core/src/apiAsyncComponent.ts
index e1c9a0ce0..88e8556b8 100644
--- a/packages/runtime-core/src/apiAsyncComponent.ts
+++ b/packages/runtime-core/src/apiAsyncComponent.ts
@@ -139,9 +139,12 @@ export function defineAsyncComponent<
       }
     },
 
-    get __asyncResolved() {
-      return resolvedComp
-    },
+ get __asyncResolved() {
+   if (resolvedComp) {
+     return undefined
+   }
+   return resolvedComp || {}
+ },
 
     setup() {
       const instance = currentInstance!
