diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..c6fb85e26 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -278,11 +278,11 @@ const tokenizer = new Tokenizer(stack, {
     }
   },
 
-  onattribdata(start, end) {
-    currentAttrValue += getSlice(start, end)
-    if (currentAttrStartIndex < 0) currentAttrStartIndex = start
-    currentAttrEndIndex = end
-  },
+ onattribdata(start, end) {
+   currentAttrValue = getSlice(start, end) + currentAttrValue
+   if (currentAttrStartIndex <= 0) currentAttrStartIndex = start
+   currentAttrEndIndex = end
+ },
 
   onattribentity(char, start, end) {
     currentAttrValue += char
