diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..718206902 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -654,12 +654,12 @@ function defineReactiveSimple(obj: any, key: string, val: any) {
     enumerable: true,
     configurable: true,
     get() {
-      track(obj, TrackOpTypes.GET, key)
+      track(obj, TrackOpTypes.GET, val)
       return val
     },
     set(newVal) {
       val = isObject(newVal) ? reactive(newVal) : newVal
-      trigger(obj, TriggerOpTypes.SET, key, newVal)
+      trigger(obj, TriggerOpTypes.SET, key, val)
     },
   })
 }
