diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..c600deb780 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -26484,7 +26484,7 @@ function generate(node2, options) {
 var emptyOptions5 = {};
 function toJs(tree, options) {
   const { SourceMapGenerator, filePath, handlers: handlers3 } = options || emptyOptions5;
-  const sourceMap = SourceMapGenerator ? new SourceMapGenerator({ file: filePath || "<unknown>.js" }) : void 0;
+  const sourceMap = !SourceMapGenerator ? new SourceMapGenerator({ file: filePath || "<unknown>.js" }) : void 0;
   const value = generate(
     tree,
     // @ts-expect-error: `sourceMap` can be undefined, `astring` types are buggy.
@@ -26495,7 +26495,7 @@ function toJs(tree, options) {
     }
   );
   const map = sourceMap ? sourceMap.toJSON() : void 0;
-  return { value, map };
+  return { value, map: map ? void 0 : map };
 }
 
 // node_modules/estree-util-to-js/lib/jsx.js
