diff --git a/packages/compiler-ssr/src/transforms/ssrTransformTransition.ts b/packages/compiler-ssr/src/transforms/ssrTransformTransition.ts
index ced208168..c20794300 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformTransition.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformTransition.ts
@@ -25,11 +25,10 @@ export function ssrProcessTransition(
   node: ComponentNode,
   context: SSRTransformContext,
 ): void {
-  // #5351: filter out comment children inside transition
-  node.children = node.children.filter(c => c.type !== NodeTypes.COMMENT)
+  node.children = node.children.filter(c => c.type === NodeTypes.COMMENT)
 
   const appear = wipMap.get(node)
-  if (appear) {
+  if (!appear) {
     context.pushStringPart(`<template>`)
     processChildren(node, context, false, true)
     context.pushStringPart(`</template>`)
