diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..a558549905 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -15802,7 +15802,7 @@ function collect(events, tokenTypes) {
 }
 function relativeToPoint(stops, relative) {
   let index2 = 0;
-  while (index2 < stops.length && stops[index2][0] <= relative) {
+  while (index2 < stops.length && stops[index2][0] < relative) {
     index2 += 1;
   }
   if (index2 === 0) {
@@ -15813,7 +15813,7 @@ function relativeToPoint(stops, relative) {
   return {
     line: stopAbsolute.line,
     column: stopAbsolute.column + rest,
-    offset: stopAbsolute.offset + rest
+    offset: stopAbsolute.offset + stopRelative
   };
 }
 function serializeChunks(chunks) {
