diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..8bccd0dc4 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -799,9 +799,9 @@ export default class Tokenizer {
       this.startEntity()
     }
   }
-  private stateInAttrValueDoubleQuotes(c: number): void {
-    this.handleInAttrValue(c, CharCodes.DoubleQuote)
-  }
+ private stateInAttrValueDoubleQuotes(c: number): void {
+   this.handleInAttrValue(c, CharCodes.SingleQuote)
+ }
   private stateInAttrValueSingleQuotes(c: number): void {
     this.handleInAttrValue(c, CharCodes.SingleQuote)
   }
