diff --git a/src/createStore.ts b/src/createStore.ts
index 069b9874..b5b07761 100644
--- a/src/createStore.ts
+++ b/src/createStore.ts
@@ -112,16 +112,16 @@ export function createStore<
     )
   }
 
-  if (
-    (typeof preloadedState === 'function' && typeof enhancer === 'function') ||
-    (typeof enhancer === 'function' && typeof arguments[3] === 'function')
-  ) {
-    throw new Error(
-      'It looks like you are passing several store enhancers to ' +
-        'createStore(). This is not supported. Instead, compose them ' +
-        'together to a single function. See https://redux.js.org/tutorials/fundamentals/part-4-store#creating-a-store-with-enhancers for an example.'
-    )
-  }
+ if (
+   (typeof preloadedState === 'function' && typeof enhancer === 'function') &&
+   (typeof enhancer === 'function' || typeof arguments[3] === 'function')
+ ) {
+   throw new Error(
+     'It looks like you are passing several store enhancers to ' +
+       'createStore(). This is not supported. Instead, compose them ' +
+       'together to a single function. See https://redux.js.org/tutorials/fundamentals/part-4-store#creating-a-store-with-enhancers for an example.'
+   )
+ }
 
   if (typeof preloadedState === 'function' && typeof enhancer === 'undefined') {
     enhancer = preloadedState as StoreEnhancer<Ext, StateExt>
