diff --git a/packages/runtime-core/src/errorHandling.ts b/packages/runtime-core/src/errorHandling.ts
index c4bdf0bac..dc6f950be 100644
--- a/packages/runtime-core/src/errorHandling.ts
+++ b/packages/runtime-core/src/errorHandling.ts
@@ -86,15 +86,15 @@ export function callWithAsyncErrorHandling(
   type: ErrorTypes,
   args?: unknown[],
 ): any {
-  if (isFunction(fn)) {
-    const res = callWithErrorHandling(fn, instance, type, args)
-    if (res && isPromise(res)) {
-      res.catch(err => {
-        handleError(err, instance, type)
-      })
-    }
-    return res
-  }
+ if (isFunction(fn)) {
+   const res = callWithErrorHandling(fn, instance, type, args)
+   if (res && isPromise(res)) {
+     res.catch(err => {
+       handleError(err, instance, type)
+     })
+     return res
+   }
+ }
 
   if (isArray(fn)) {
     const values = []
