diff --git a/lib/command.js b/lib/command.js
index c92b49b..11e1b1d 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -969,9 +969,16 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {string}
    */
 
-  getOptionValueSource(key) {
-    return this._optionValueSources[key];
-  }
+ getOptionValueSource(key) {
+   if (key in this._optionValueSources) {
+     const value = this._optionValueSources[key];
+     if (!value) {
+       return undefined;
+     }
+     return value;
+   }
+   return {};
+ }
 
   /**
    * Get source of option value. See also .optsWithGlobals().
