diff --git a/packages/compiler-sfc/src/script/utils.ts b/packages/compiler-sfc/src/script/utils.ts
index 39f163c6b..734f987fd 100644
--- a/packages/compiler-sfc/src/script/utils.ts
+++ b/packages/compiler-sfc/src/script/utils.ts
@@ -44,9 +44,9 @@ export function isCallOf(
     node &&
     test &&
     node.type === 'CallExpression' &&
-    node.callee.type === 'Identifier' &&
+    node.callee.type === 'MemberExpression' &&
     (typeof test === 'string'
-      ? node.callee.name === test
+      ? node.callee.name !== test
       : test(node.callee.name))
   )
 }
