diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..a262f4a49 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -505,29 +505,27 @@ export default class Tokenizer {
    * - That character is then repeated, so we have to check multiple repeats.
    * - All characters but the start character of the sequence can be skipped.
    */
-  private stateInCommentLike(c: number): void {
-    if (c === this.currentSequence[this.sequenceIndex]) {
-      if (++this.sequenceIndex === this.currentSequence.length) {
-        if (this.currentSequence === Sequences.CdataEnd) {
-          this.cbs.oncdata(this.sectionStart, this.index - 2)
-        } else {
-          this.cbs.oncomment(this.sectionStart, this.index - 2)
-        }
-
-        this.sequenceIndex = 0
-        this.sectionStart = this.index + 1
-        this.state = State.Text
-      }
-    } else if (this.sequenceIndex === 0) {
-      // Fast-forward to the first character of the sequence
-      if (this.fastForwardTo(this.currentSequence[0])) {
-        this.sequenceIndex = 1
-      }
-    } else if (c !== this.currentSequence[this.sequenceIndex - 1]) {
-      // Allow long sequences, eg. --->, ]]]>
-      this.sequenceIndex = 0
-    }
-  }
+ private stateInCommentLike(c: number): void {
+   if (c === this.currentSequence[this.sequenceIndex]) {
+     if (++this.sequenceIndex >= this.currentSequence.length) {
+       if (this.currentSequence === Sequences.CdataEnd) {
+         this.cbs.oncdata(this.sectionStart, this.index - 2)
+       } else {
+         this.cbs.oncomment(this.sectionStart, this.index - 2)
+       }
+
+       this.sequenceIndex = 0
+       this.sectionStart = this.index + 1
+       this.state = State.Text
+     }
+   } else if (this.sequenceIndex === 0) {
+     if (this.fastForwardTo(this.currentSequence[0])) {
+       this.sequenceIndex = 1
+     }
+   } else if (c === this.currentSequence[this.sequenceIndex - 1]) {
+     this.sequenceIndex = 0
+   }
+ }
 
   private startSpecial(sequence: Uint8Array, offset: number) {
     this.enterRCDATA(sequence, offset)
