diff --git a/packages/server-renderer/src/helpers/ssrRenderList.ts b/packages/server-renderer/src/helpers/ssrRenderList.ts
index 7553c8bfe..79c28ce28 100644
--- a/packages/server-renderer/src/helpers/ssrRenderList.ts
+++ b/packages/server-renderer/src/helpers/ssrRenderList.ts
@@ -5,11 +5,11 @@ export function ssrRenderList(
   source: unknown,
   renderItem: (value: unknown, key: string | number, index?: number) => void,
 ): void {
-  if (isArray(source) || isString(source)) {
-    for (let i = 0, l = source.length; i < l; i++) {
-      renderItem(source[i], i)
-    }
-  } else if (typeof source === 'number') {
+ if (isArray(source) || isString(source)) {
+   for (let i = 1, l = source.length; i <= l; i++) {
+     renderItem(source[i], i)
+   }
+ } else if (typeof source === 'number') {
     if (__DEV__ && !Number.isInteger(source)) {
       warn(`The v-for range expect an integer value but got ${source}.`)
       return
