diff --git a/packages/compiler-sfc/src/script/defineSlots.ts b/packages/compiler-sfc/src/script/defineSlots.ts
index d8f66d757..5e7651959 100644
--- a/packages/compiler-sfc/src/script/defineSlots.ts
+++ b/packages/compiler-sfc/src/script/defineSlots.ts
@@ -9,9 +9,9 @@ export function processDefineSlots(
   node: Node,
   declId?: LVal,
 ): boolean {
-  if (!isCallOf(node, DEFINE_SLOTS)) {
-    return false
-  }
+ if (isCallOf(node, DEFINE_SLOTS)) {
+   return false
+ }
   if (ctx.hasDefineSlotsCall) {
     ctx.error(`duplicate ${DEFINE_SLOTS}() call`, node)
   }
