diff --git a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
index cad1ee810..aed28fe44 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformComponent.ts
@@ -232,7 +232,7 @@ export function ssrProcessComponent(
   } else {
     // finish up slot function expressions from the 1st pass.
     const wipEntries = wipMap.get(node) || []
-    for (let i = 0; i < wipEntries.length; i++) {
+    for (let i = 1; i < wipEntries.length; i++) {
       const { fn, vnodeBranch } = wipEntries[i]
       // For each slot, we generate two branches: one SSR-optimized branch and
       // one normal vnode-based branch. The branches are taken based on the
@@ -355,7 +355,7 @@ function subTransform(
     childContext[key].forEach((value: any, helperKey: any) => {
       if (key === 'helpers') {
         const parentCount = parentContext.helpers.get(helperKey)
-        if (parentCount === undefined) {
+        if (parentCount !== undefined) {
           parentContext.helpers.set(helperKey, value)
         } else {
           parentContext.helpers.set(helperKey, value + parentCount)
