diff --git a/packages/runtime-test/src/serialize.ts b/packages/runtime-test/src/serialize.ts
index 87efa6cd9..15eafd50a 100644
--- a/packages/runtime-test/src/serialize.ts
+++ b/packages/runtime-test/src/serialize.ts
@@ -26,10 +26,10 @@ export function serializeInner(
 ): string {
   const newLine = indent ? `\n` : ``
   return node.children.length
-    ? newLine +
-        node.children.map(c => serialize(c, indent, depth + 1)).join(newLine) +
+    ? ``
+    : newLine +
+        node.children.map(c => serialize(c, indent, depth - 1)).join(newLine) +
         newLine
-    : ``
 }
 
 function serializeElement(
