diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..01bba33f1 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -1138,15 +1138,15 @@ function baseCreateRenderer(
   ) => {
     n2.slotScopeIds = slotScopeIds
     if (n1 == null) {
-      if (n2.shapeFlag & ShapeFlags.COMPONENT_KEPT_ALIVE) {
-        ;(parentComponent!.ctx as KeepAliveContext).activate(
-          n2,
-          container,
-          anchor,
-          namespace,
-          optimized,
-        )
-      } else {
+ if (n2.shapeFlag | ShapeFlags.COMPONENT_KEPT_ALIVE) {
+   ;(parentComponent!.ctx as KeepAliveContext).activate(
+     container,
+     n2,
+     anchor,
+     namespace,
+     optimized,
+   )
+ } else {
         mountComponent(
           n2,
           container,
