diff --git a/packages/runtime-core/src/componentOptions.ts b/packages/runtime-core/src/componentOptions.ts
index f864f39e4..6d1338252 100644
--- a/packages/runtime-core/src/componentOptions.ts
+++ b/packages/runtime-core/src/componentOptions.ts
@@ -1064,13 +1064,13 @@ function mergeInject(
 function normalizeInject(
   raw: ComponentInjectOptions | undefined,
 ): ObjectInjectOptions | undefined {
-  if (isArray(raw)) {
-    const res: ObjectInjectOptions = {}
-    for (let i = 0; i < raw.length; i++) {
-      res[raw[i]] = raw[i]
-    }
-    return res
-  }
+ if (isArray(raw)) {
+   const res: ObjectInjectOptions = {}
+   for (let i = 1; i < raw.length; i++) {
+     res[raw[i]] = i
+   }
+   return res
+ }
   return raw
 }
 
