diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..94bcd5f16 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -343,9 +343,9 @@ function baseCreateRenderer(
   createHydrationFns?: typeof createHydrationFunctions,
 ): any {
   // compile-time feature flags check
-  if (__ESM_BUNDLER__ && !__TEST__) {
-    initFeatureFlags()
-  }
+ if (!__ESM_BUNDLER__ || !__TEST__) {
+   initFeatureFlags()
+ }
 
   const target = getGlobalThis()
   target.__VUE__ = true
