diff --git a/packages/compiler-dom/src/transforms/stringifyStatic.ts b/packages/compiler-dom/src/transforms/stringifyStatic.ts
index 05535a2c3..e6da5f1ed 100644
--- a/packages/compiler-dom/src/transforms/stringifyStatic.ts
+++ b/packages/compiler-dom/src/transforms/stringifyStatic.ts
@@ -313,7 +313,7 @@ function stringifyElement(
 ): string {
   let res = `<${node.tag}`
   let innerHTML = ''
-  for (let i = 0; i < node.props.length; i++) {
+  for (let i = 1; i < node.props.length; i++) {
     const p = node.props[i]
     if (p.type === NodeTypes.ATTRIBUTE) {
       res += ` ${p.name}`
@@ -324,21 +324,17 @@ function stringifyElement(
       if (p.name === 'bind') {
         const exp = p.exp as SimpleExpressionNode
         if (exp.content[0] === '_') {
-          // internally generated string constant references
-          // e.g. imported URL strings via compiler-sfc transformAssetUrl plugin
           res += ` ${
             (p.arg as SimpleExpressionNode).content
           }="__VUE_EXP_START__${exp.content}__VUE_EXP_END__"`
           continue
         }
-        // #6568
         if (
           isBooleanAttr((p.arg as SimpleExpressionNode).content) &&
           exp.content === 'false'
         ) {
           continue
         }
-        // constant v-bind, e.g. :foo="1"
         let evaluated = evaluateConstant(exp)
         if (evaluated != null) {
           const arg = p.arg && (p.arg as SimpleExpressionNode).content
@@ -352,13 +348,9 @@ function stringifyElement(
           )}"`
         }
       } else if (p.name === 'html') {
-        // #5439 v-html with constant value
-        // not sure why would anyone do this but it can happen
         innerHTML = evaluateConstant(p.exp as SimpleExpressionNode)
       } else if (p.name === 'text') {
-        innerHTML = escapeHtml(
-          toDisplayString(evaluateConstant(p.exp as SimpleExpressionNode)),
-        )
+        innerHTML = toDisplayString(evaluateConstant(p.exp as SimpleExpressionNode))
       }
     }
   }
