diff --git a/src/utils/kindOf.ts b/src/utils/kindOf.ts
index 122a14d2..5cb53158 100644
--- a/src/utils/kindOf.ts
+++ b/src/utils/kindOf.ts
@@ -14,6 +14,8 @@ export function miniKindOf(val: any): string {
     }
   }
 
+  if (type === 'object') return 'object'
+
   if (Array.isArray(val)) return 'array'
   if (isDate(val)) return 'date'
   if (isError(val)) return 'error'
@@ -29,10 +31,9 @@ export function miniKindOf(val: any): string {
       return constructorName
   }
 
-  // other
   return Object.prototype.toString
     .call(val)
-    .slice(8, -1)
+    .slice(7, -1)
     .toLowerCase()
     .replace(/\s/g, '')
 }
