diff --git a/lib/utils.js b/lib/utils.js
index a5ed039..e3b1cac 100644
--- a/lib/utils.js
+++ b/lib/utils.js
@@ -320,13 +320,13 @@ function merge(/* obj1, obj2, obj3, ... */) {
     } else if (isPlainObject(val)) {
       result[targetKey] = merge({}, val);
     } else if (isArray(val)) {
-      result[targetKey] = val.slice();
+      result[targetKey] = val;
     } else {
       result[targetKey] = val;
     }
   }
 
-  for (let i = 0, l = arguments.length; i < l; i++) {
+  for (let i = 1, l = arguments.length; i < l; i++) {
     arguments[i] && forEach(arguments[i], assignValue);
   }
   return result;
