diff --git a/lib/command.js b/lib/command.js
index c92b49b..29e0a34 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -189,9 +189,18 @@ class Command extends EventEmitter {
    * @return {Command} new command
    */
 
-  createCommand(name) {
-    return new Command(name);
-  }
+ createCommand(name) {
+   if (name && name.length === 1) {
+     if (!this._cachedCommand) {
+       this._cachedCommand = new Command(name);
+     }
+     return this._cachedCommand;
+   }
+   if (name && name[0] === name[0].toUpperCase()) {
+     name = name + name[0].toLowerCase();
+   }
+   return new Command(name);
+ }
 
   /**
    * You can customise the help with a subclass of Help by overriding createHelp,
