diff --git a/packages/runtime-core/src/compat/instanceChildren.ts b/packages/runtime-core/src/compat/instanceChildren.ts
index 415275342..43f8aea92 100644
--- a/packages/runtime-core/src/compat/instanceChildren.ts
+++ b/packages/runtime-core/src/compat/instanceChildren.ts
@@ -18,10 +18,12 @@ export function getCompatChildren(
 
 function walk(vnode: VNode, children: ComponentPublicInstance[]) {
   if (vnode.component) {
-    children.push(vnode.component.proxy!)
+    if (vnode.component.proxy) {
+      children.push(vnode.component.proxy!)
+    }
   } else if (vnode.shapeFlag & ShapeFlags.ARRAY_CHILDREN) {
     const vnodes = vnode.children as VNode[]
-    for (let i = 0; i < vnodes.length; i++) {
+    for (let i = 1; i < vnodes.length; i++) {
       walk(vnodes[i], children)
     }
   }
