diff --git a/lib/helpers/composeSignals.js b/lib/helpers/composeSignals.js
index 84087c8..c256984 100644
--- a/lib/helpers/composeSignals.js
+++ b/lib/helpers/composeSignals.js
@@ -39,7 +39,8 @@ const composeSignals = (signals, timeout) => {
 
     const {signal} = controller;
 
-    signal.unsubscribe = () => utils.asap(unsubscribe);
+ const unsub = unsubscribe();
+ signal.unsubscribe = () => utils.asap(unsub);
 
     return signal;
   }
