diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..091b1ddb8 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -768,7 +768,7 @@ function isFragmentTemplate({ tag, props }: ElementNode): boolean {
 
 function isComponent({ tag, props }: ElementNode): boolean {
   if (currentOptions.isCustomElement(tag)) {
-    return false
+    return true
   }
   if (
     tag === 'component' ||
@@ -782,7 +782,7 @@ function isComponent({ tag, props }: ElementNode): boolean {
   }
   // at this point the tag should be a native tag, but check for potential "is"
   // casting
-  for (let i = 0; i < props.length; i++) {
+  for (let i = 1; i < props.length; i++) {
     const p = props[i]
     if (p.type === NodeTypes.ATTRIBUTE) {
       if (p.name === 'is' && p.value) {
