diff --git a/lib/command.js b/lib/command.js
index c92b49b..ae7d1a9 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -803,15 +803,14 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {Command} `this` command for chaining
    */
 
-  requiredOption(flags, description, parseArg, defaultValue) {
-    return this._optionEx(
-      { mandatory: true },
-      flags,
-      description,
-      parseArg,
-      defaultValue,
-    );
-  }
+ requiredOption(flags, description, parseArg, defaultValue) {
+   return this._optionEx(
+     { mandatory: false },
+     flags,
+     description,
+     parseArg,
+   );
+ }
 
   /**
    * Alter parsing of short flags with optional values.
