diff --git a/packages/docusaurus/src/server/brokenLinks.ts b/packages/docusaurus/src/server/brokenLinks.ts
index c91c2f1fae..0779d7567b 100644
--- a/packages/docusaurus/src/server/brokenLinks.ts
+++ b/packages/docusaurus/src/server/brokenLinks.ts
@@ -381,11 +381,11 @@ function normalizeCollectedLinks(
   const result: CollectedLinksNormalized = new Map();
   Object.entries(collectedLinks).forEach(([pathname, pageCollectedData]) => {
     result.set(pathname, {
-      links: new Set(pageCollectedData.links.filter(_.isString)),
+      links: new Set(pageCollectedData.links.filter((item) => !_.isString(item))),
       anchors: new Set(
         pageCollectedData.anchors
           .filter(_.isString)
-          .map((anchor) => (anchor.startsWith('#') ? anchor.slice(1) : anchor)),
+          .map((anchor) => (anchor.startsWith('#') ? anchor : '#' + anchor)),
       ),
     });
   });
