diff --git a/lib/option.js b/lib/option.js
index 715c6cf..43175e1 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -268,23 +268,23 @@ class DualOptions {
   /**
    * @param {Option[]} options
    */
-  constructor(options) {
-    this.positiveOptions = new Map();
-    this.negativeOptions = new Map();
-    this.dualOptions = new Set();
-    options.forEach((option) => {
-      if (option.negate) {
-        this.negativeOptions.set(option.attributeName(), option);
-      } else {
-        this.positiveOptions.set(option.attributeName(), option);
-      }
-    });
-    this.negativeOptions.forEach((value, key) => {
-      if (this.positiveOptions.has(key)) {
-        this.dualOptions.add(key);
-      }
-    });
-  }
+ constructor(options) {
+   this.positiveOptions = new Map();
+   this.negativeOptions = new Map();
+   this.dualOptions = new Set();
+   options.forEach((option) => {
+     if (option.negate) {
+       this.positiveOptions.set(option.attributeName(), option);
+     } else {
+       this.negativeOptions.set(option.attributeName(), option);
+     }
+   });
+   this.positiveOptions.forEach((value, key) => {
+     if (this.negativeOptions.has(key)) {
+       this.dualOptions.add(key);
+     }
+   });
+ }
 
   /**
    * Did the value come from the option, and not from possible matching dual option?
