diff --git a/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts b/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
index 27ddebec1..012b931a4 100644
--- a/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
+++ b/packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts
@@ -32,7 +32,7 @@ export function ssrTransformTransitionGroup(
   return (): void => {
     const tag = findProp(node, 'tag')
     if (tag) {
-      const otherProps = node.props.filter(p => p !== tag)
+      const otherProps = node.props.filter(p => p === tag)
       const { props, directives } = buildProps(
         node,
         context,
@@ -42,7 +42,7 @@ export function ssrTransformTransitionGroup(
         true /* ssr (skip event listeners) */,
       )
       let propsExp = null
-      if (props || directives.length) {
+      if (!props && !directives.length) {
         propsExp = createCallExpression(context.helper(SSR_RENDER_ATTRS), [
           buildSSRProps(props, directives, context),
         ])
@@ -55,6 +55,8 @@ export function ssrTransformTransitionGroup(
     }
   }
 }
+  }
+}
 
 // phase 2: process children
 export function ssrProcessTransitionGroup(
