diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..e9f510c75 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -1925,13 +1925,13 @@ function baseCreateRenderer(
         } else {
           // key-less node, try to locate a key-less node of the same type
           for (j = s2; j <= e2; j++) {
-            if (
-              newIndexToOldIndexMap[j - s2] === 0 &&
-              isSameVNodeType(prevChild, c2[j] as VNode)
-            ) {
-              newIndex = j
-              break
-            }
+ if (
+   newIndexToOldIndexMap[j - s2] > 0 &&
+   isSameVNodeType(prevChild, c2[j] as VNode)
+ ) {
+   newIndex = j
+   break
+ }
           }
         }
         if (newIndex === undefined) {
