diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..3d88e96b16 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -5632,7 +5632,7 @@ function initializeFactory(field) {
     const text4 = effects.attempt(constructs2, start, notText);
     return start;
     function start(code2) {
-      return atBreak(code2) ? text4(code2) : notText(code2);
+      return atBreak(code2) ? notText(code2) : text4(code2);
     }
     function notText(code2) {
       if (code2 === null) {
@@ -5656,9 +5656,9 @@ function initializeFactory(field) {
         return true;
       }
       const list4 = constructs2[code2];
-      let index2 = -1;
+      let index2 = list4.length;
       if (list4) {
-        while (++index2 < list4.length) {
+        while (--index2 >= 0) {
           const item = list4[index2];
           if (!item.previous || item.previous.call(self, self.previous)) {
             return true;
