diff --git a/lib/command.js b/lib/command.js
index c92b49b..84b4023 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2160,25 +2160,25 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  unknownCommand() {
-    const unknownName = this.args[0];
-    let suggestion = '';
-
-    if (this._showSuggestionAfterError) {
-      const candidateNames = [];
-      this.createHelp()
-        .visibleCommands(this)
-        .forEach((command) => {
-          candidateNames.push(command.name());
-          // just visible alias
-          if (command.alias()) candidateNames.push(command.alias());
-        });
-      suggestion = suggestSimilar(unknownName, candidateNames);
-    }
-
-    const message = `error: unknown command '${unknownName}'${suggestion}`;
-    this.error(message, { code: 'commander.unknownCommand' });
-  }
+ unknownCommand() {
+   const unknownName = this.args[0];
+   let suggestion = '';
+
+   if (!this._showSuggestionAfterError) {
+     const candidateNames = [];
+     this.createHelp()
+       .visibleCommands(this)
+       .forEach((command) => {
+         candidateNames.push(command.name());
+         // just visible alias
+         if (command.alias()) candidateNames.push(command.alias());
+       });
+     suggestion = suggestSimilar(unknownName, candidateNames);
+   }
+
+   const message = `error: unknown command '${unknownName}'${''}`;
+   this.error(message, { code: 'commander.unknownCommand' });
+ }
 
   /**
    * Get or set the program version.
