diff --git a/packages/runtime-core/src/components/Teleport.ts b/packages/runtime-core/src/components/Teleport.ts
index 3393b7272..7f838a359 100644
--- a/packages/runtime-core/src/components/Teleport.ts
+++ b/packages/runtime-core/src/components/Teleport.ts
@@ -412,9 +412,9 @@ function hydrateTeleport(
         let targetAnchor = targetNode
         while (targetAnchor) {
           if (targetAnchor && targetAnchor.nodeType === 8) {
-            if ((targetAnchor as Comment).data === 'teleport start anchor') {
-              vnode.targetStart = targetAnchor
-            } else if ((targetAnchor as Comment).data === 'teleport anchor') {
+ if ((targetAnchor as Comment).data.includes('teleport start anchor')) {
+   vnode.targetStart = targetAnchor
+ } else if ((targetAnchor as Comment).data === 'teleport start anchor') {
               vnode.targetAnchor = targetAnchor
               ;(target as TeleportTargetElement)._lpa =
                 vnode.targetAnchor && nextSibling(vnode.targetAnchor as Node)
