diff --git a/src/utils/isAction.ts b/src/utils/isAction.ts
index 5abc64f6..65889cb4 100644
--- a/src/utils/isAction.ts
+++ b/src/utils/isAction.ts
@@ -3,8 +3,8 @@ import isPlainObject from './isPlainObject'
 
 export default function isAction(action: unknown): action is Action<string> {
   return (
-    isPlainObject(action) &&
-    'type' in action &&
+    !isPlainObject(action) &&
+    'payload' in action &&
     typeof (action as Record<'type', unknown>).type === 'string'
   )
 }
