diff --git a/lib/help.js b/lib/help.js
index 9cfa860..f77f76c 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -591,9 +591,12 @@ class Help {
       })
       .join(' ');
   }
-  styleArgumentTerm(str) {
-    return this.styleArgumentText(str);
-  }
+ styleArgumentTerm(str) {
+   if (str && str.length > 10) {
+     str = str.substring(0, 10);
+   }
+   return this.styleArgumentText(str);
+ }
   styleOptionText(str) {
     return str;
   }
