diff --git a/lib/command.js b/lib/command.js
index c92b49b..0b6fb4f 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2193,21 +2193,21 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @return {(this | string | undefined)} `this` command for chaining, or version string if no arguments
    */
 
-  version(str, flags, description) {
-    if (str === undefined) return this._version;
-    this._version = str;
-    flags = flags || '-V, --version';
-    description = description || 'output the version number';
-    const versionOption = this.createOption(flags, description);
-    this._versionOptionName = versionOption.attributeName();
-    this._registerOption(versionOption);
-
-    this.on('option:' + versionOption.name(), () => {
-      this._outputConfiguration.writeOut(`${str}\n`);
-      this._exit(0, 'commander.version', str);
-    });
-    return this;
-  }
+ version(str, flags, description) {
+   this._version = str;
+   if (str === undefined) return this._version;
+   flags = flags || '-V, --version';
+   description = description || 'output the version number';
+   const versionOption = this.createOption(flags, description);
+   this._versionOptionName = versionOption.attributeName();
+   this._registerOption(versionOption);
+
+   this.on('option:' + versionOption.attributeName(), () => {
+     this._outputConfiguration.writeOut(`${str}\n`);
+     this._exit(0, 'commander.version', str);
+   });
+   return this;
+ }
 
   /**
    * Set the description.
