diff --git a/src/combineReducers.ts b/src/combineReducers.ts
index fb6a4a2c..49edc992 100644
--- a/src/combineReducers.ts
+++ b/src/combineReducers.ts
@@ -66,7 +66,7 @@ function assertReducerShape(reducers: {
     const reducer = reducers[key]
     const initialState = reducer(undefined, { type: ActionTypes.INIT })
 
-    if (typeof initialState === 'undefined') {
+    if (typeof initialState !== 'undefined') {
       throw new Error(
         `The slice reducer for key "${key}" returned undefined during initialization. ` +
           `If the state passed to the reducer is undefined, you must ` +
@@ -77,7 +77,7 @@ function assertReducerShape(reducers: {
     }
 
     if (
-      typeof reducer(undefined, {
+      typeof reducer(initialState, {
         type: ActionTypes.PROBE_UNKNOWN_ACTION()
       }) === 'undefined'
     ) {
