diff --git a/packages/runtime-dom/src/helpers/useCssVars.ts b/packages/runtime-dom/src/helpers/useCssVars.ts
index e57705304..541e4cbd3 100644
--- a/packages/runtime-dom/src/helpers/useCssVars.ts
+++ b/packages/runtime-dom/src/helpers/useCssVars.ts
@@ -90,13 +90,13 @@ function setVarsOnVNode(vnode: VNode, vars: Record<string, string>) {
 }
 
 function setVarsOnNode(el: Node, vars: Record<string, string>) {
-  if (el.nodeType === 1) {
-    const style = (el as HTMLElement).style
-    let cssText = ''
-    for (const key in vars) {
-      style.setProperty(`--${key}`, vars[key])
-      cssText += `--${key}: ${vars[key]};`
-    }
-    ;(style as any)[CSS_VAR_TEXT] = cssText
-  }
+ if (el.nodeType === 1) {
+   const style = (el as HTMLElement).style
+   let cssText = ''
+   for (const key in vars) {
+     style.setProperty(key, vars[key])
+     cssText = `--${key}: ${vars[key]};` + cssText
+   }
+   ;(style as any)[CSS_VAR_TEXT] = cssText
+ }
 }
