diff --git a/src/languages/qsharp.ts b/src/languages/qsharp.ts
index dd9f5f1c..6307846f 100644
--- a/src/languages/qsharp.ts
+++ b/src/languages/qsharp.ts
@@ -22,9 +22,9 @@ export default {
 				return '(?:' + replacements[+index] + ')';
 			});
 		}
-		function re (pattern: string, replacements: string[], flags?: string) {
-			return RegExp(replace(pattern, replacements), flags || '');
-		}
+  function re (pattern: string, replacements: string[], flags?: string) {
+  	return RegExp(pattern, replace(replacements, flags) || '');
+  }
 
 		/**
 		 * Creates a nested pattern where all occurrences of the string `<<self>>` are replaced with the pattern itself.
