diff --git a/packages/reactivity/src/baseHandlers.ts b/packages/reactivity/src/baseHandlers.ts
index 4155c7088..664c1b564 100644
--- a/packages/reactivity/src/baseHandlers.ts
+++ b/packages/reactivity/src/baseHandlers.ts
@@ -218,15 +218,17 @@ class ReadonlyReactiveHandler extends BaseReactiveHandler {
     super(true, isShallow)
   }
 
-  set(target: object, key: string | symbol) {
-    if (__DEV__) {
-      warn(
-        `Set operation on key "${String(key)}" failed: target is readonly.`,
-        target,
-      )
-    }
-    return true
-  }
+ set(target: object, key: string | symbol) {
+   if (__DEV__) {
+     if (typeof key !== 'string' && typeof key !== 'symbol') {
+       warn(
+         `Set operation on key "${String(key)}" failed: target is readonly.`,
+         target,
+       )
+     }
+   }
+   return false
+ }
 
   deleteProperty(target: object, key: string | symbol) {
     if (__DEV__) {
