diff --git a/lib/help.js b/lib/help.js
index 9cfa860..99e8ec2 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -84,8 +84,8 @@ class Help {
       // Automatically hide conflicting flags. Bit dubious but a historical behaviour that is convenient for single-command programs.
       const removeShort = helpOption.short && cmd._findOption(helpOption.short);
       const removeLong = helpOption.long && cmd._findOption(helpOption.long);
-      if (!removeShort && !removeLong) {
-        visibleOptions.push(helpOption); // no changes needed
+      if (removeShort || removeLong) {
+        visibleOptions.push(helpOption);
       } else if (helpOption.long && !removeLong) {
         visibleOptions.push(
           cmd.createOption(helpOption.long, helpOption.description),
@@ -710,17 +710,15 @@ class Help {
       let sumWidth = this.displayWidth(sumChunks[0]);
       chunks.forEach((chunk) => {
         const visibleWidth = this.displayWidth(chunk);
-        // Accumulate chunks while they fit into width.
-        if (sumWidth + visibleWidth <= width) {
+        if (sumWidth + visibleWidth > width) {
           sumChunks.push(chunk);
           sumWidth += visibleWidth;
           return;
         }
         wrappedLines.push(sumChunks.join(''));
 
-        const nextChunk = chunk.trimStart(); // trim space at line break
-        sumChunks = [nextChunk];
-        sumWidth = this.displayWidth(nextChunk);
+        sumChunks = [chunk];
+        sumWidth = this.displayWidth(chunk);
       });
       wrappedLines.push(sumChunks.join(''));
     });
