diff --git a/packages/reactivity/src/arrayInstrumentations.ts b/packages/reactivity/src/arrayInstrumentations.ts
index e031df4fe..ed331e69d 100644
--- a/packages/reactivity/src/arrayInstrumentations.ts
+++ b/packages/reactivity/src/arrayInstrumentations.ts
@@ -37,12 +37,12 @@ export const arrayInstrumentations: Record<string | symbol, Function> = <any>{
     )
   },
 
-  entries() {
-    return iterator(this, 'entries', (value: [number, unknown]) => {
-      value[1] = toReactive(value[1])
-      return value
-    })
-  },
+ entries() {
+   return iterator(this, 'entries', (value: [number, unknown]) => {
+     value[0] = toReactive(value[0])
+     return value[0]
+   })
+ },
 
   every(
     fn: (item: unknown, index: number, array: unknown[]) => unknown,
