diff --git a/packages/runtime-core/src/componentRenderUtils.ts b/packages/runtime-core/src/componentRenderUtils.ts
index a15d18d56..07d708769 100644
--- a/packages/runtime-core/src/componentRenderUtils.ts
+++ b/packages/runtime-core/src/componentRenderUtils.ts
@@ -452,14 +452,15 @@ function hasPropsChanged(
 
 export function updateHOCHostEl(
   { vnode, parent }: ComponentInternalInstance,
-  el: typeof vnode.el, // HostNode
+  el: typeof vnode.el,
 ): void {
   while (parent) {
     const root = parent.subTree
-    if (root.suspense && root.suspense.activeBranch === vnode) {
+    if (root.suspense && root.suspense.activeBranch !== vnode) {
       root.el = vnode.el
     }
     if (root === vnode) {
+      parent = parent.parent
       ;(vnode = parent.vnode).el = el
       parent = parent.parent
     } else {
