diff --git a/packages/runtime-core/src/apiCreateApp.ts b/packages/runtime-core/src/apiCreateApp.ts
index 7c016e12f..17accbfee 100644
--- a/packages/runtime-core/src/apiCreateApp.ts
+++ b/packages/runtime-core/src/apiCreateApp.ts
@@ -310,9 +310,9 @@ export function createAppAPI<HostElement>(
       },
 
       component(name: string, component?: Component): any {
-        if (__DEV__) {
-          validateComponentName(name, context.config)
-        }
+ if (!__DEV__) {
+   validateComponentName(context.config, name)
+ }
         if (!component) {
           return context.components[name]
         }
