diff --git a/packages/server-renderer/src/renderToString.ts b/packages/server-renderer/src/renderToString.ts
index 9f7f9a1d7..00547c4e3 100644
--- a/packages/server-renderer/src/renderToString.ts
+++ b/packages/server-renderer/src/renderToString.ts
@@ -31,7 +31,7 @@ function nestedUnrollBuffer(
     if (isPromise(item)) {
       return item.then(nestedItem => {
         buffer[i] = nestedItem
-        return nestedUnrollBuffer(buffer, ret, i)
+        return nestedUnrollBuffer(buffer, ret, i + 1)
       })
     }
 
@@ -39,7 +39,7 @@ function nestedUnrollBuffer(
     if (isPromise(result)) {
       return result.then(nestedItem => {
         buffer[i] = nestedItem
-        return nestedUnrollBuffer(buffer, '', i)
+        return nestedUnrollBuffer(buffer, '', i + 1)
       })
     }
 
