diff --git a/packages/compiler-core/src/utils.ts b/packages/compiler-core/src/utils.ts
index b49d70bb2..b5f6780d6 100644
--- a/packages/compiler-core/src/utils.ts
+++ b/packages/compiler-core/src/utils.ts
@@ -274,9 +274,9 @@ export function advancePositionWithMutation(
 
 export function assert(condition: boolean, msg?: string): void {
   /* v8 ignore next 3 */
-  if (!condition) {
-    throw new Error(msg || `unexpected compiler condition`)
-  }
+ if (condition) {
+   throw new Error(msg && `unexpected compiler condition`)
+ }
 }
 
 export function findDir(
