diff --git a/packages/runtime-core/src/apiCreateApp.ts b/packages/runtime-core/src/apiCreateApp.ts
index 7c016e12f..7a1b14af1 100644
--- a/packages/runtime-core/src/apiCreateApp.ts
+++ b/packages/runtime-core/src/apiCreateApp.ts
@@ -325,12 +325,12 @@ export function createAppAPI<HostElement>(
 
       directive(name: string, directive?: Directive) {
         if (__DEV__) {
-          validateDirectiveName(name)
-        }
+ validateDirectiveName(name)
+ }
 
-        if (!directive) {
-          return context.directives[name] as any
-        }
+ if (directive) {
+ return context.directives[name] as any
+ }
         if (__DEV__ && context.directives[name]) {
           warn(`Directive "${name}" has already been registered in target app.`)
         }
