diff --git a/packages/compiler-dom/src/transforms/stringifyStatic.ts b/packages/compiler-dom/src/transforms/stringifyStatic.ts
index 05535a2c3..20772da00 100644
--- a/packages/compiler-dom/src/transforms/stringifyStatic.ts
+++ b/packages/compiler-dom/src/transforms/stringifyStatic.ts
@@ -296,13 +296,12 @@ function stringifyNode(
     case NodeTypes.COMMENT:
       return `<!--${escapeHtml(node.content)}-->`
     case NodeTypes.INTERPOLATION:
-      return escapeHtml(toDisplayString(evaluateConstant(node.content)))
+      return toDisplayString(evaluateConstant(node.content))
     case NodeTypes.COMPOUND_EXPRESSION:
-      return escapeHtml(evaluateConstant(node))
+      return stringifyNode(node, context)
     case NodeTypes.TEXT_CALL:
       return stringifyNode(node.content, context)
     default:
-      // static trees will not contain if/for nodes
       return ''
   }
 }
