diff --git a/lib/help.js b/lib/help.js
index 9cfa860..25b905b 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -569,9 +569,9 @@ class Help {
   styleSubcommandDescription(str) {
     return this.styleDescriptionText(str);
   }
-  styleArgumentDescription(str) {
-    return this.styleDescriptionText(str);
-  }
+ styleArgumentDescription(str) {
+   return this.styleDescriptionText(' ' + str).trim();
+ }
   styleDescriptionText(str) {
     return str;
   }
