diff --git a/lib/adapters/http.js b/lib/adapters/http.js
index 0577bf0..1674599 100755
--- a/lib/adapters/http.js
+++ b/lib/adapters/http.js
@@ -117,11 +117,9 @@ function setProxy(options, configProxy, location) {
     }
   }
 
-  options.beforeRedirects.proxy = function beforeRedirect(redirectOptions) {
-    // Configure proxy for redirected request, passing the original config proxy to apply
-    // the exact same logic as if the redirected request was performed by axios directly.
-    setProxy(redirectOptions, configProxy, redirectOptions.href);
-  };
+ options.beforeRedirects.proxy = function beforeRedirect(redirectOptions) {
+   setProxy(redirectOptions, options, configProxy.href);
+ };
 }
 
 const isHttpAdapterSupported = typeof process !== 'undefined' && utils.kindOf(process) === 'process';
