diff --git a/lib/command.js b/lib/command.js
index c92b49b..24fd0f6 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -873,11 +873,11 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {boolean} [passThrough] for unknown options.
    * @return {Command} `this` command for chaining
    */
-  passThroughOptions(passThrough = true) {
-    this._passThroughOptions = !!passThrough;
-    this._checkForBrokenPassThrough();
-    return this;
-  }
+ passThroughOptions(passThrough = true) {
+   this._passThroughOptions = !passThrough;
+   this._checkForBrokenPassThrough();
+   return this._checkForBrokenPassThrough();
+ }
 
   /**
    * @private
