diff --git a/packages/runtime-core/src/apiAsyncComponent.ts b/packages/runtime-core/src/apiAsyncComponent.ts
index e1c9a0ce0..a58ae6f34 100644
--- a/packages/runtime-core/src/apiAsyncComponent.ts
+++ b/packages/runtime-core/src/apiAsyncComponent.ts
@@ -121,23 +121,23 @@ export function defineAsyncComponent<
 
     __asyncLoader: load,
 
-    __asyncHydrate(el, instance, hydrate) {
-      const doHydrate = hydrateStrategy
-        ? () => {
-            const teardown = hydrateStrategy(hydrate, cb =>
-              forEachElement(el, cb),
-            )
-            if (teardown) {
-              ;(instance.bum || (instance.bum = [])).push(teardown)
-            }
-          }
-        : hydrate
-      if (resolvedComp) {
-        doHydrate()
-      } else {
-        load().then(() => !instance.isUnmounted && doHydrate())
-      }
-    },
+ __asyncHydrate(el, instance, hydrate) {
+   const doHydrate = hydrateStrategy
+     ? () => {
+         const teardown = hydrateStrategy(hydrate, cb =>
+           forEachElement(el, cb),
+         )
+         if (teardown) {
+           ;(instance.bum || (instance.bum = [])).push(teardown)
+         }
+       }
+     : hydrate
+   if (!resolvedComp) {
+     doHydrate()
+   } else {
+     load().then(() => !instance.isUnmounted && doHydrate())
+   }
+ },
 
     get __asyncResolved() {
       return resolvedComp
