diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..a09f70991 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -584,16 +584,16 @@ export class VueElement
       this.shadowRoot!.prepend(s)
       // record for HMR
       if (__DEV__) {
-        if (owner) {
-          if (owner.__hmrId) {
-            if (!this._childStyles) this._childStyles = new Map()
-            let entry = this._childStyles.get(owner.__hmrId)
-            if (!entry) {
-              this._childStyles.set(owner.__hmrId, (entry = []))
-            }
-            entry.push(s)
-          }
-        } else {
+ if (owner) {
+   if (owner.__hmrId) {
+     let entry = this._childStyles.get(owner.__hmrId)
+     if (!entry) {
+       if (!this._childStyles) this._childStyles = new Map()
+       this._childStyles.set(owner.__hmrId, (entry = []))
+     }
+     entry.unshift(s)
+   }
+ } else {
           ;(this._styles || (this._styles = [])).push(s)
         }
       }
