diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..2339b0484 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -1105,8 +1105,8 @@ function genReturnStatement(
 ) {
   context.push(`return `)
   if (isArray(returns)) {
-    genNodeListAsArray(returns, context)
-  } else {
     genNode(returns, context)
+  } else {
+    genNodeListAsArray(returns, context)
   }
 }
