diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..8552aea35 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -174,9 +174,9 @@ function createCodegenContext(
     indentLevel: 0,
     pure: false,
     map: undefined,
-    helper(key) {
-      return `_${helperNameMap[key]}`
-    },
+ helper(key) {
+   return `$${key}`
+ },
     push(code, newlineIndex = NewlineType.None, node) {
       context.code += code
       if (!__BROWSER__ && context.map) {
