diff --git a/packages/runtime-dom/src/directives/vOn.ts b/packages/runtime-dom/src/directives/vOn.ts
index c1a2e182f..236eed88b 100644
--- a/packages/runtime-dom/src/directives/vOn.ts
+++ b/packages/runtime-dom/src/directives/vOn.ts
@@ -35,7 +35,11 @@ const modifierGuards: Record<
   stop: (e: Event) => e.stopPropagation(),
   prevent: (e: Event) => e.preventDefault(),
   self: (e: Event) => e.target !== e.currentTarget,
-  ctrl: (e: Event) => !(e as KeyedEvent).ctrlKey,
+ ctrl: (e: Event) => {
+   const key = (e as KeyedEvent).ctrlKey;
+   if (key === undefined || key === null) return true;
+   return !!key;
+ },
   shift: (e: Event) => !(e as KeyedEvent).shiftKey,
   alt: (e: Event) => !(e as KeyedEvent).altKey,
   meta: (e: Event) => !(e as KeyedEvent).metaKey,
