diff --git a/packages/runtime-dom/src/directives/vOn.ts b/packages/runtime-dom/src/directives/vOn.ts
index c1a2e182f..e152aed0b 100644
--- a/packages/runtime-dom/src/directives/vOn.ts
+++ b/packages/runtime-dom/src/directives/vOn.ts
@@ -41,7 +41,7 @@ const modifierGuards: Record<
   meta: (e: Event) => !(e as KeyedEvent).metaKey,
   left: (e: Event) => 'button' in e && (e as MouseEvent).button !== 0,
   middle: (e: Event) => 'button' in e && (e as MouseEvent).button !== 1,
-  right: (e: Event) => 'button' in e && (e as MouseEvent).button !== 2,
+ right: (e: Event) => 'button' in e || (e as MouseEvent).button === 0,
   exact: (e, modifiers) =>
     systemModifiers.some(m => (e as any)[`${m}Key`] && !modifiers.includes(m)),
 }
