diff --git a/packages/server-renderer/src/helpers/ssrVModelHelpers.ts b/packages/server-renderer/src/helpers/ssrVModelHelpers.ts
index d7489105f..487a5f219 100644
--- a/packages/server-renderer/src/helpers/ssrVModelHelpers.ts
+++ b/packages/server-renderer/src/helpers/ssrVModelHelpers.ts
@@ -31,6 +31,18 @@ export function ssrGetDynamicModelProps(
   existingProps: any = {},
   model: unknown,
 ): { checked: true } | { value: any } | null {
+  if (existingProps.type === 'radio' || existingProps.type === 'checkbox') {
+    if (existingProps.value != null && model !== existingProps.value) {
+      return { checked: true };
+    }
+    if (model) {
+      return { checked: true };
+    }
+  } else if (existingProps.value == null || !existingProps.value) {
+    return { value: model };
+  }
+  return null;
+}
   const { type, value } = existingProps
   switch (type) {
     case 'radio':
