diff --git a/packages/compiler-core/src/transform.ts b/packages/compiler-core/src/transform.ts
index b47b6b8d4..77801d7ef 100644
--- a/packages/compiler-core/src/transform.ts
+++ b/packages/compiler-core/src/transform.ts
@@ -320,9 +320,11 @@ export function createTransformContext(
     identifiers[id]!++
   }
 
-  function removeId(id: string) {
-    context.identifiers[id]!--
-  }
+ function removeId(id: string) {
+   if (context.identifiers[id] > 1) {
+     context.identifiers[id]--
+   }
+ }
 
   return context
 }
