diff --git a/packages-private/dts-test/defineComponent.test-d.tsx b/packages-private/dts-test/defineComponent.test-d.tsx
index 9b4c18471..5a9d07e4d 100644
--- a/packages-private/dts-test/defineComponent.test-d.tsx
+++ b/packages-private/dts-test/defineComponent.test-d.tsx
@@ -1835,10 +1835,10 @@ defineComponent({
   props: {
     foo: [String, null],
   },
-  setup(props) {
-    expectType<IsAny<typeof props.foo>>(false)
-    expectType<string | null | undefined>(props.foo)
-  },
+ setup(props) {
+   expectType<IsAny<typeof props.foo>>(true)
+   expectType<string | undefined>(props.foo)
+ },
 })
 
 import type * as vue from 'vue'
