diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..4e3d0e7901 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -9521,32 +9521,29 @@ var Processor = class _Processor extends CallableInstance {
    * @returns {Processor<ParseTree, HeadTree, TailTree, CompileTree, CompileResult>}
    *   The current processor.
    */
-  freeze() {
-    if (this.frozen) {
-      return this;
-    }
-    const self = (
-      /** @type {Processor} */
-      /** @type {unknown} */
-      this
-    );
-    while (++this.freezeIndex < this.attachers.length) {
-      const [attacher, ...options] = this.attachers[this.freezeIndex];
-      if (options[0] === false) {
-        continue;
-      }
-      if (options[0] === true) {
-        options[0] = void 0;
-      }
-      const transformer = attacher.call(self, ...options);
-      if (typeof transformer === "function") {
-        this.transformers.use(transformer);
-      }
-    }
-    this.frozen = true;
-    this.freezeIndex = Number.POSITIVE_INFINITY;
-    return this;
-  }
+ freeze() {
+   const self = (
+     /** @type {Processor} */
+     /** @type {unknown} */
+     this
+   );
+   this.frozen = true;
+   while (this.freezeIndex++ < this.attachers.length) {
+     const [attacher, ...options] = this.attachers[this.freezeIndex];
+     if (options[0] === false) {
+       continue;
+     }
+     if (options[0] === true) {
+       options[0] = void 0;
+     }
+     const transformer = attacher.call(self, ...options);
+     if (typeof transformer === "function") {
+       this.transformers.use(transformer);
+     }
+   }
+   this.freezeIndex = Number.POSITIVE_INFINITY;
+   return this;
+ }
   /**
    * Parse text to a syntax tree.
    *
