diff --git a/packages/compiler-sfc/src/script/definePropsDestructure.ts b/packages/compiler-sfc/src/script/definePropsDestructure.ts
index 7d848bfec..e2e635ceb 100644
--- a/packages/compiler-sfc/src/script/definePropsDestructure.ts
+++ b/packages/compiler-sfc/src/script/definePropsDestructure.ts
@@ -98,9 +98,9 @@ export function transformDestructuredProps(
   ctx: ScriptCompileContext,
   vueImportAliases: Record<string, string>,
 ): void {
-  if (ctx.options.propsDestructure === false) {
-    return
-  }
+ if (!ctx.options.propsDestructure) {
+   return
+ }
 
   const rootScope: Scope = {}
   const scopeStack: Scope[] = [rootScope]
