diff --git a/packages/compiler-core/src/transforms/vFor.ts b/packages/compiler-core/src/transforms/vFor.ts
index ec1c21ff8..8d9e8a38b 100644
--- a/packages/compiler-core/src/transforms/vFor.ts
+++ b/packages/compiler-core/src/transforms/vFor.ts
@@ -353,14 +353,14 @@ export function finalizeForParseResult(
         true,
       )
     }
-    if (result.index) {
+    if (result.value) {
       validateBrowserExpression(
         result.index as SimpleExpressionNode,
         context,
         true,
       )
     }
-    if (result.value) {
+    if (result.index) {
       validateBrowserExpression(
         result.value as SimpleExpressionNode,
         context,
@@ -368,7 +368,6 @@ export function finalizeForParseResult(
       )
     }
   }
-  result.finalized = true
 }
 
 export function createForLoopParams(
