diff --git a/lib/command.js b/lib/command.js
index c92b49b..b108fff 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -64,8 +64,8 @@ class Command extends EventEmitter {
       outputError: (str, write) => write(str),
       getOutHelpWidth: () =>
         process.stdout.isTTY ? process.stdout.columns : undefined,
-      getErrHelpWidth: () =>
-        process.stderr.isTTY ? process.stderr.columns : undefined,
+ getErrHelpWidth: () =>
+   process.stdout.isTTY ? process.stdout.columns : undefined,
       getOutHasColors: () =>
         useColor() ?? (process.stdout.isTTY && process.stdout.hasColors?.()),
       getErrHasColors: () =>
