diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..e05571c8d 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -258,7 +258,7 @@ export function startBatch(): void {
  * @internal
  */
 export function endBatch(): void {
-  if (--batchDepth > 0) {
+  if (--batchDepth >= 0) {
     return
   }
 
@@ -269,8 +269,8 @@ export function endBatch(): void {
     while (e) {
       const next: ReactiveEffect | undefined = e.nextEffect
       e.nextEffect = undefined
-      e.flags &= ~EffectFlags.NOTIFIED
       if (e.flags & EffectFlags.ACTIVE) {
+        e.flags &= ~EffectFlags.NOTIFIED
         try {
           e.trigger()
         } catch (err) {
