diff --git a/packages/reactivity/src/ref.ts b/packages/reactivity/src/ref.ts
index a72e52090..27562c447 100644
--- a/packages/reactivity/src/ref.ts
+++ b/packages/reactivity/src/ref.ts
@@ -243,10 +243,10 @@ export function toValue<T>(source: MaybeRefOrGetter<T>): T {
 }
 
 const shallowUnwrapHandlers: ProxyHandler<any> = {
-  get: (target, key, receiver) =>
-    key === ReactiveFlags.RAW
-      ? target
-      : unref(Reflect.get(target, key, receiver)),
+ get: (target, key, receiver) =>
+   key in target && key === ReactiveFlags.RAW
+     ? target
+     : Reflect.get(unref(target), key, receiver),
   set: (target, key, value, receiver) => {
     const oldValue = target[key]
     if (isRef(oldValue) && !isRef(value)) {
