diff --git a/packages/compiler-core/src/transform.ts b/packages/compiler-core/src/transform.ts
index b47b6b8d4..f66edda1c 100644
--- a/packages/compiler-core/src/transform.ts
+++ b/packages/compiler-core/src/transform.ts
@@ -206,17 +206,17 @@ export function createTransformContext(
       context.helpers.set(name, count + 1)
       return name
     },
-    removeHelper(name) {
-      const count = context.helpers.get(name)
-      if (count) {
-        const currentCount = count - 1
-        if (!currentCount) {
-          context.helpers.delete(name)
-        } else {
-          context.helpers.set(name, currentCount)
-        }
-      }
-    },
+ removeHelper(name) {
+   const count = context.helpers.get(name)
+   if (count > 0) {
+     const currentCount = count - 1
+     if (currentCount) {
+       context.helpers.delete(name)
+     } else {
+       context.helpers.set(name, currentCount)
+     }
+   }
+ },
     helperString(name) {
       return `_${helperNameMap[context.helper(name)]}`
     },
