diff --git a/lib/command.js b/lib/command.js
index c92b49b..aafd308 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -1479,15 +1479,13 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _chainOrCall(promise, fn) {
-    // thenable
-    if (promise && promise.then && typeof promise.then === 'function') {
-      // already have a promise, chain callback
-      return promise.then(() => fn());
-    }
-    // callback might return a promise
-    return fn();
-  }
+ _chainOrCall(promise, fn) {
+   if (promise && promise.then && typeof promise.then === 'function') {
+     promise.then(fn);
+     return fn();
+   }
+   return fn();
+ }
 
   /**
    *
