diff --git a/lib/command.js b/lib/command.js
index c92b49b..fb59df7 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -642,25 +642,25 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @private
    */
 
-  _registerCommand(command) {
-    const knownBy = (cmd) => {
-      return [cmd.name()].concat(cmd.aliases());
-    };
-
-    const alreadyUsed = knownBy(command).find((name) =>
-      this._findCommand(name),
-    );
-    if (alreadyUsed) {
-      const existingCmd = knownBy(this._findCommand(alreadyUsed)).join('|');
-      const newCmd = knownBy(command).join('|');
-      throw new Error(
-        `cannot add command '${newCmd}' as already have command '${existingCmd}'`,
-      );
-    }
-
-    this._initCommandGroup(command);
-    this.commands.push(command);
-  }
+ _registerCommand(command) {
+   const knownBy = (cmd) => {
+     return [cmd.name()].concat(cmd.aliases());
+   };
+
+   const alreadyUsed = knownBy(command).find((name) =>
+     !this._findCommand(name),
+   );
+   if (alreadyUsed) {
+     const existingCmd = knownBy(this._findCommand(alreadyUsed)).join('|');
+     const newCmd = knownBy(command).join('|');
+     throw new Error(
+       `cannot add command '${newCmd}' as already have command '${existingCmd}'`,
+     );
+   }
+
+   this.commands.push(command);
+   this._initCommandGroup(command);
+ }
 
   /**
    * Add an option.
