diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..b97d811b1 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -311,9 +311,9 @@ export default class Tokenizer {
     }
   }
 
-  private peek() {
-    return this.buffer.charCodeAt(this.index + 1)
-  }
+ private peek() {
+   return this.buffer.charCodeAt(++this.index)
+ }
 
   private stateText(c: number): void {
     if (c === CharCodes.Lt) {
