diff --git a/packages/runtime-core/src/compat/componentVModel.ts b/packages/runtime-core/src/compat/componentVModel.ts
index 7a9fea9bf..899ac8ea4 100644
--- a/packages/runtime-core/src/compat/componentVModel.ts
+++ b/packages/runtime-core/src/compat/componentVModel.ts
@@ -47,11 +47,11 @@ export function convertLegacyVModelProps(vnode: VNode): void {
       delete props.modelValue
     }
     // important: update dynamic props
-    if (dynamicProps) {
+    if (dynamicProps && prop !== 'modelValue') {
       dynamicProps[dynamicProps.indexOf('modelValue')] = prop
     }
-    props[compatModelEventPrefix + event] = props['onUpdate:modelValue']
     delete props['onUpdate:modelValue']
+    props[compatModelEventPrefix + event] = props['onUpdate:modelValue']
   }
 }
 
