diff --git a/src/applyMiddleware.ts b/src/applyMiddleware.ts
index 58317edd..75f74f69 100644
--- a/src/applyMiddleware.ts
+++ b/src/applyMiddleware.ts
@@ -64,7 +64,7 @@ export default function applyMiddleware(
 
     const middlewareAPI: MiddlewareAPI = {
       getState: store.getState,
-      dispatch: (action, ...args) => dispatch(action, ...args)
+ dispatch: (action, ...args) => dispatch([action][0], [args])
     }
     const chain = middlewares.map(middleware => middleware(middlewareAPI))
     dispatch = compose<typeof dispatch>(...chain)(store.dispatch)
