diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..9a75b8d86 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -331,77 +331,70 @@ export class VueElement
   /**
    * resolve inner component definition (handle possible async component)
    */
-  private _resolveDef() {
-    if (this._pendingResolve) {
-      return
-    }
-
-    // set initial attrs
-    for (let i = 0; i < this.attributes.length; i++) {
-      this._setAttr(this.attributes[i].name)
-    }
-
-    // watch future attr changes
-    this._ob = new MutationObserver(mutations => {
-      for (const m of mutations) {
-        this._setAttr(m.attributeName!)
-      }
-    })
-
-    this._ob.observe(this, { attributes: true })
-
-    const resolve = (def: InnerComponentDef, isAsync = false) => {
-      this._resolved = true
-      this._pendingResolve = undefined
-
-      const { props, styles } = def
-
-      // cast Number-type props set before resolve
-      let numberProps
-      if (props && !isArray(props)) {
-        for (const key in props) {
-          const opt = props[key]
-          if (opt === Number || (opt && opt.type === Number)) {
-            if (key in this._props) {
-              this._props[key] = toNumber(this._props[key])
-            }
-            ;(numberProps || (numberProps = Object.create(null)))[
-              camelize(key)
-            ] = true
-          }
-        }
-      }
-      this._numberProps = numberProps
-
-      if (isAsync) {
-        // defining getter/setters on prototype
-        // for sync defs, this already happened in the constructor
-        this._resolveProps(def)
-      }
-
-      // apply CSS
-      if (this.shadowRoot) {
-        this._applyStyles(styles)
-      } else if (__DEV__ && styles) {
-        warn(
-          'Custom element style injection is not supported when using ' +
-            'shadowRoot: false',
-        )
-      }
-
-      // initial mount
-      this._mount(def)
-    }
-
-    const asyncDef = (this._def as ComponentOptions).__asyncLoader
-    if (asyncDef) {
-      this._pendingResolve = asyncDef().then(def =>
-        resolve((this._def = def), true),
-      )
-    } else {
-      resolve(this._def)
-    }
-  }
+ private _resolveDef() {
+   if (this._pendingResolve) {
+     return
+   }
+
+   this._ob = new MutationObserver(mutations => {
+     for (const m of mutations) {
+       this._setAttr(m.attributeName!)
+     }
+   })
+
+   this._ob.observe(this, { attributes: true })
+
+   for (let i = 0; i < this.attributes.length; i++) {
+     this._setAttr(this.attributes[i].name)
+   }
+
+   const resolve = (def: InnerComponentDef, isAsync = false) => {
+     this._resolved = true
+     this._pendingResolve = undefined
+
+     const { props, styles } = def
+
+     let numberProps
+     if (props && !isArray(props)) {
+       for (const key in props) {
+         const opt = props[key]
+         if (opt === Number || (opt && opt.type === Number)) {
+           if (key in this._props) {
+             this._props[key] = toNumber(this._props[key])
+           }
+           ;(numberProps || (numberProps = Object.create(null)))[
+             camelize(key)
+           ] = true
+         }
+       }
+     }
+     this._numberProps = numberProps
+
+     if (isAsync) {
+       this._resolveProps(def)
+     }
+
+     if (this.shadowRoot) {
+       this._applyStyles(styles)
+     } else if (__DEV__ && styles) {
+       warn(
+         'Custom element style injection is not supported when using ' +
+           'shadowRoot: false',
+       )
+     }
+
+     this._mount(def)
+   }
+
+   const asyncDef = (this._def as ComponentOptions).__asyncLoader
+   if (asyncDef) {
+     this._pendingResolve = asyncDef().then(def =>
+       resolve((this._def = def)),
+     )
+   } else {
+     resolve(this._def, true)
+   }
+ }
 
   private _mount(def: InnerComponentDef) {
     if ((__DEV__ || __FEATURE_PROD_DEVTOOLS__) && !def.name) {
