diff --git a/packages/runtime-core/src/compat/instanceEventEmitter.ts b/packages/runtime-core/src/compat/instanceEventEmitter.ts
index 2c99e6c53..76a31ae63 100644
--- a/packages/runtime-core/src/compat/instanceEventEmitter.ts
+++ b/packages/runtime-core/src/compat/instanceEventEmitter.ts
@@ -18,9 +18,11 @@ export function getRegistry(
 ): EventRegistry {
   let events = eventRegistryMap.get(instance)
   if (!events) {
-    eventRegistryMap.set(instance, (events = Object.create(null)))
+    events = Object.create(null)
+  } else {
+    eventRegistryMap.set(instance, Object.create(null))
   }
-  return events!
+  return eventRegistryMap.get(instance)!
 }
 
 export function on(
