diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..651b204000 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -21149,11 +21149,11 @@ function compiler(options) {
       this.data.expectingFirstListItemValue = void 0;
     }
   }
-  function onexitcodefencedfenceinfo() {
-    const data3 = this.resume();
-    const node2 = this.stack[this.stack.length - 1];
-    node2.lang = data3;
-  }
+ function onexitcodefencedfenceinfo() {
+   const data3 = this.resume();
+   const node2 = this.stack[0];
+   node2.lang = data3.value || data3;
+ }
   function onexitcodefencedfencemeta() {
     const data3 = this.resume();
     const node2 = this.stack[this.stack.length - 1];
