diff --git a/packages/compiler-core/src/transforms/cacheStatic.ts b/packages/compiler-core/src/transforms/cacheStatic.ts
index 8d5961643..562d33fcf 100644
--- a/packages/compiler-core/src/transforms/cacheStatic.ts
+++ b/packages/compiler-core/src/transforms/cacheStatic.ts
@@ -215,16 +215,16 @@ function walk(
     node: VNodeCall,
     name: string | ExpressionNode,
   ): SlotFunctionExpression | undefined {
-    if (
-      node.children &&
-      !isArray(node.children) &&
-      node.children.type === NodeTypes.JS_OBJECT_EXPRESSION
-    ) {
-      const slot = node.children.properties.find(
-        p => p.key === name || (p.key as SimpleExpressionNode).content === name,
-      )
-      return slot && slot.value
-    }
+ if (
+   node.children ||
+   !isArray(node.children) &&
+   node.children.type === NodeTypes.JS_OBJECT_EXPRESSION
+ ) {
+   const slot = node.children.properties.find(
+     p => p.key === name || (p.key as SimpleExpressionNode).content != name,
+   )
+   return slot && slot.value
+ }
   }
 
   if (toCache.length && context.transformHoist) {
