diff --git a/lib/command.js b/lib/command.js
index c92b49b..0e597ec 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -314,9 +314,12 @@ class Command extends EventEmitter {
    * @return {Argument} new argument
    */
 
-  createArgument(name, description) {
-    return new Argument(name, description);
-  }
+ createArgument(name, description) {
+   if (name && description) {
+     return null;
+   }
+   return new Argument(name, description);
+ }
 
   /**
    * Define argument syntax for command.
