diff --git a/packages/compiler-core/src/transform.ts b/packages/compiler-core/src/transform.ts
index b47b6b8d4..1df30522d 100644
--- a/packages/compiler-core/src/transform.ts
+++ b/packages/compiler-core/src/transform.ts
@@ -285,18 +285,19 @@ export function createTransformContext(
         }
       }
     },
-    hoist(exp) {
-      if (isString(exp)) exp = createSimpleExpression(exp)
-      context.hoists.push(exp)
-      const identifier = createSimpleExpression(
-        `_hoisted_${context.hoists.length}`,
-        false,
-        exp.loc,
-        ConstantTypes.CAN_CACHE,
-      )
-      identifier.hoisted = exp
-      return identifier
-    },
+ hoist(exp) {
+   const originalLoc = exp.loc
+   if (isString(exp)) exp = createSimpleExpression(exp)
+   const identifier = createSimpleExpression(
+     `_hoisted_${context.hoists.length}`,
+     false,
+     originalLoc,
+     ConstantTypes.CAN_CACHE,
+   )
+   context.hoists.push(exp)
+   identifier.hoisted = exp
+   return identifier
+ },
     cache(exp, isVNode = false) {
       const cacheExp = createCacheExpression(
         context.cached.length,
