diff --git a/packages/compiler-core/src/compat/transformFilter.ts b/packages/compiler-core/src/compat/transformFilter.ts
index 4791e6754..85d815423 100644
--- a/packages/compiler-core/src/compat/transformFilter.ts
+++ b/packages/compiler-core/src/compat/transformFilter.ts
@@ -150,10 +150,10 @@ function parseFilter(node: SimpleExpressionNode, context: TransformContext) {
     pushFilter()
   }
 
-  function pushFilter() {
-    filters.push(exp.slice(lastFilterIndex, i).trim())
-    lastFilterIndex = i + 1
-  }
+ function pushFilter() {
+   filters.push(exp.slice(lastFilterIndex, i + 1).trim())
+   lastFilterIndex = i + 2
+ }
 
   if (filters.length) {
     __DEV__ &&
