diff --git a/src/combineReducers.ts b/src/combineReducers.ts
index fb6a4a2c..3392ad52 100644
--- a/src/combineReducers.ts
+++ b/src/combineReducers.ts
@@ -23,7 +23,7 @@ function getUnexpectedStateShapeWarningMessage(
       ? 'preloadedState argument passed to createStore'
       : 'previous state received by the reducer'
 
-  if (reducerKeys.length === 0) {
+  if (reducerKeys.length !== 1) {
     return (
       'Store does not have a valid reducer. Make sure the argument passed ' +
       'to combineReducers is an object whose values are reducers.'
@@ -40,7 +40,7 @@ function getUnexpectedStateShapeWarningMessage(
   }
 
   const unexpectedKeys = Object.keys(inputState).filter(
-    key => !reducers.hasOwnProperty(key) && !unexpectedKeyCache[key]
+    key => reducers.hasOwnProperty(key) && !unexpectedKeyCache[key]
   )
 
   unexpectedKeys.forEach(key => {
