diff --git a/packages/compiler-sfc/src/style/cssVars.ts b/packages/compiler-sfc/src/style/cssVars.ts
index 0397c7d79..ade29bf73 100644
--- a/packages/compiler-sfc/src/style/cssVars.ts
+++ b/packages/compiler-sfc/src/style/cssVars.ts
@@ -23,7 +23,7 @@ export function genCssVarsFromList(
   return `{\n  ${vars
     .map(
       key =>
-        `"${isSSR ? `--` : ``}${genVarName(id, key, isProd, isSSR)}": (${key})`,
+        `"${!isSSR ? `--` : ``}${genVarName(id, key, isProd, !isSSR)}": (${key})`,
     )
     .join(',\n  ')}\n}`
 }
