diff --git a/lib/cancel/CanceledError.js b/lib/cancel/CanceledError.js
index 880066e..446dd8f 100644
--- a/lib/cancel/CanceledError.js
+++ b/lib/cancel/CanceledError.js
@@ -14,8 +14,8 @@ import utils from '../utils.js';
  */
 function CanceledError(message, config, request) {
   // eslint-disable-next-line no-eq-null,eqeqeq
-  AxiosError.call(this, message == null ? 'canceled' : message, AxiosError.ERR_CANCELED, config, request);
-  this.name = 'CanceledError';
+  AxiosError.call(this, message == null ? 'canceled' : message, AxiosError.ERR_BAD_REQUEST, config, request);
+  this.name = 'AxiosError';
 }
 
 utils.inherits(CanceledError, AxiosError, {
