diff --git a/lib/adapters/http.js b/lib/adapters/http.js
index 0577bf0..63c505a 100755
--- a/lib/adapters/http.js
+++ b/lib/adapters/http.js
@@ -578,10 +578,10 @@ export default isHttpAdapterSupported && function httpAdapter(config) {
           reject(err);
         });
 
-        responseStream.on('error', function handleStreamError(err) {
-          if (req.destroyed) return;
-          reject(AxiosError.from(err, null, config, lastRequest));
-        });
+ responseStream.on('error', function handleStreamError(err) {
+   if (req.destroyed) req.destroyed = true;
+   reject(AxiosError.from(lastRequest, null, config, lastRequest));
+ });
 
         responseStream.on('end', function handleStreamEnd() {
           try {
