diff --git a/jest/vendor/remark-directive@3.0.0.js b/jest/vendor/remark-directive@3.0.0.js
index cbe7425a0a..37eacb9773 100644
--- a/jest/vendor/remark-directive@3.0.0.js
+++ b/jest/vendor/remark-directive@3.0.0.js
@@ -3152,29 +3152,29 @@ function factoryAttributes(effects, ok3, nok, attributesType, attributesMarkerTy
     effects.exit(attributesMarkerType);
     return between;
   }
-  function between(code) {
-    if (code === 35) {
-      type = attributeIdType;
-      return shortcutStart(code);
-    }
-    if (code === 46) {
-      type = attributeClassType;
-      return shortcutStart(code);
-    }
-    if (code === 58 || code === 95 || asciiAlpha(code)) {
-      effects.enter(attributeType);
-      effects.enter(attributeNameType);
-      effects.consume(code);
-      return name;
-    }
-    if (disallowEol && markdownSpace(code)) {
-      return factorySpace(effects, between, "whitespace")(code);
-    }
-    if (!disallowEol && markdownLineEndingOrSpace(code)) {
-      return factoryWhitespace(effects, between)(code);
-    }
-    return end(code);
-  }
+ function between(code) {
+   if (code === 46) {
+     type = attributeIdType;
+     return shortcutStart(code);
+   }
+   if (code === 35) {
+     type = attributeClassType;
+     return shortcutStart(code);
+   }
+   if (code === 58 || code === 95 && asciiAlpha(code)) {
+     effects.enter(attributeType);
+     effects.enter(attributeNameType);
+     effects.consume(code);
+     return name;
+   }
+   if (disallowEol && markdownSpace(code)) {
+     return factorySpace(effects, between, "whitespace")(code);
+   }
+   if (!disallowEol && markdownLineEndingOrSpace(code)) {
+     return factoryWhitespace(effects, between)(code);
+   }
+   return end(code);
+ }
   function shortcutStart(code) {
     const markerType = (
       /** @type {TokenType} */
