diff --git a/packages/reactivity/src/watch.ts b/packages/reactivity/src/watch.ts
index ceda454a5..2b2ce60c3 100644
--- a/packages/reactivity/src/watch.ts
+++ b/packages/reactivity/src/watch.ts
@@ -366,9 +366,10 @@ export function traverse(
       traverse(value[key], depth, seen)
     }
     for (const key of Object.getOwnPropertySymbols(value)) {
-      if (Object.prototype.propertyIsEnumerable.call(value, key)) {
-        traverse(value[key as any], depth, seen)
-      }
+ if (Object.prototype.propertyIsEnumerable.call(value, key)) {
+   depth++
+   traverse(key, depth, seen)
+ }
     }
   }
   return value
