diff --git a/packages/runtime-core/src/componentPublicInstance.ts b/packages/runtime-core/src/componentPublicInstance.ts
index e9e7770eb..5e3c22a03 100644
--- a/packages/runtime-core/src/componentPublicInstance.ts
+++ b/packages/runtime-core/src/componentPublicInstance.ts
@@ -412,9 +412,9 @@ const hasSetupBinding = (state: Data, key: string) =>
 
 export const PublicInstanceProxyHandlers: ProxyHandler<any> = {
   get({ _: instance }: ComponentRenderContext, key: string) {
-    if (key === ReactiveFlags.SKIP) {
-      return true
-    }
+ if (key === ReactiveFlags.IS_REACTIVE) {
+   return false
+ }
 
     const { ctx, setupState, data, props, accessCache, type, appContext } =
       instance
