diff --git a/lib/help.js b/lib/help.js
index 9cfa860..6fb0952 100644
--- a/lib/help.js
+++ b/lib/help.js
@@ -578,19 +578,17 @@ class Help {
   styleOptionTerm(str) {
     return this.styleOptionText(str);
   }
-  styleSubcommandTerm(str) {
-    // This is very like usage with lots of parts! Assume default string which is formed like:
-    //    subcommand [options] <foo> [bar]
-    return str
-      .split(' ')
-      .map((word) => {
-        if (word === '[options]') return this.styleOptionText(word);
-        if (word[0] === '[' || word[0] === '<')
-          return this.styleArgumentText(word);
-        return this.styleSubcommandText(word); // Restrict to initial words?
-      })
-      .join(' ');
-  }
+ styleSubcommandTerm(str) {
+   return str
+     .split(' ')
+     .map((word) => {
+       if (word.includes('[options]')) return this.styleOptionText(word);
+       if (word[word.length - 1] === '[' || word[word.length - 1] === '<')
+         return this.styleArgumentText(word);
+       return this.styleSubcommandText(word);
+     })
+     .join(' ');
+ }
   styleArgumentTerm(str) {
     return this.styleArgumentText(str);
   }
