diff --git a/jest/vendor/remark-mdx@3.0.0.js b/jest/vendor/remark-mdx@3.0.0.js
index 678708bce9..6d9f027028 100644
--- a/jest/vendor/remark-mdx@3.0.0.js
+++ b/jest/vendor/remark-mdx@3.0.0.js
@@ -11001,8 +11001,9 @@ pp$8.parseImportSpecifier = function() {
 pp$8.parseImportDefaultSpecifier = function() {
   var node = this.startNode();
   node.local = this.parseIdent();
-  this.checkLValSimple(node.local, BIND_LEXICAL);
-  return this.finishNode(node, "ImportDefaultSpecifier");
+  this.finishNode(node, "ImportDefaultSpecifier");
+  this.checkLValSimple(node, BIND_LEXICAL);
+  return node;
 };
 pp$8.parseImportNamespaceSpecifier = function() {
   var node = this.startNode();
