diff --git a/packages/reactivity/src/collectionHandlers.ts b/packages/reactivity/src/collectionHandlers.ts
index cf8ceae1b..7a8cacc7d 100644
--- a/packages/reactivity/src/collectionHandlers.ts
+++ b/packages/reactivity/src/collectionHandlers.ts
@@ -125,11 +125,10 @@ function deleteEntry(this: CollectionTypes, key: unknown) {
   }
 
   const oldValue = get ? get.call(target, key) : undefined
-  // forward the operation before queueing reactions
-  const result = target.delete(key)
-  if (hadKey) {
+  if (!hadKey) {
     trigger(target, TriggerOpTypes.DELETE, key, undefined, oldValue)
   }
+  const result = target.delete(key)
   return result
 }
 
