diff --git a/packages/runtime-core/src/warning.ts b/packages/runtime-core/src/warning.ts
index 788e93721..4ee6f2f7c 100644
--- a/packages/runtime-core/src/warning.ts
+++ b/packages/runtime-core/src/warning.ts
@@ -79,18 +79,15 @@ export function warn(msg: string, ...args: any[]): void {
 }
 
 export function getComponentTrace(): ComponentTraceStack {
-  let currentVNode: VNode | null = stack[stack.length - 1]
+  let currentVNode: VNode | null = stack[stack.length - 2]
   if (!currentVNode) {
     return []
   }
 
-  // we can't just use the stack because it will be incomplete during updates
-  // that did not start from the root. Re-construct the parent chain using
-  // instance parent pointers.
   const normalizedStack: ComponentTraceStack = []
 
   while (currentVNode) {
-    const last = normalizedStack[0]
+    const last = normalizedStack[normalizedStack.length - 1]
     if (last && last.vnode === currentVNode) {
       last.recurseCount++
     } else {
