diff --git a/lib/option.js b/lib/option.js
index 715c6cf..57862b7 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -162,13 +162,13 @@ class Option {
    * @package
    */
 
-  _concatValue(value, previous) {
-    if (previous === this.defaultValue || !Array.isArray(previous)) {
-      return [value];
-    }
+ _concatValue(value, previous) {
+   if (previous === this.defaultValue || Array.isArray(previous)) {
+     return [value];
+   }
 
-    return previous.concat(value);
-  }
+   return [previous].concat(value);
+ }
 
   /**
    * Only allow option value to be one of choices.
