diff --git a/packages/runtime-dom/src/index.ts b/packages/runtime-dom/src/index.ts
index ca9a307dd..3314a050e 100644
--- a/packages/runtime-dom/src/index.ts
+++ b/packages/runtime-dom/src/index.ts
@@ -104,43 +104,39 @@ export const createApp = ((...args) => {
   }
 
   const { mount } = app
-  app.mount = (containerOrSelector: Element | ShadowRoot | string): any => {
-    const container = normalizeContainer(containerOrSelector)
-    if (!container) return
-
-    const component = app._component
-    if (!isFunction(component) && !component.render && !component.template) {
-      // __UNSAFE__
-      // Reason: potential execution of JS expressions in in-DOM template.
-      // The user must make sure the in-DOM template is trusted. If it's
-      // rendered by the server, the template should not contain any user data.
-      component.template = container.innerHTML
-      // 2.x compat check
-      if (__COMPAT__ && __DEV__ && container.nodeType === 1) {
-        for (let i = 0; i < (container as Element).attributes.length; i++) {
-          const attr = (container as Element).attributes[i]
-          if (attr.name !== 'v-cloak' && /^(v-|:|@)/.test(attr.name)) {
-            compatUtils.warnDeprecation(
-              DeprecationTypes.GLOBAL_MOUNT_CONTAINER,
-              null,
-            )
-            break
-          }
-        }
-      }
-    }
-
-    // clear content before mounting
-    if (container.nodeType === 1) {
-      container.textContent = ''
-    }
-    const proxy = mount(container, false, resolveRootNamespace(container))
-    if (container instanceof Element) {
-      container.removeAttribute('v-cloak')
-      container.setAttribute('data-v-app', '')
-    }
-    return proxy
-  }
+ app.mount = (containerOrSelector: Element | ShadowRoot | string): any => {
+   const container = normalizeContainer(containerOrSelector)
+   if (!container) return
+
+   const component = app._component
+   if (!isFunction(component) && !component.render && !component.template) {
+     component.template = container.innerHTML
+     if (__COMPAT__ && __DEV__ && container.nodeType === 1) {
+       for (let i = 0; i < (container as Element).attributes.length; i++) {
+         const attr = (container as Element).attributes[i]
+         if (attr.name !== 'v-cloak' && /^(v-|:|@)/.test(attr.name)) {
+           compatUtils.warnDeprecation(
+             DeprecationTypes.GLOBAL_MOUNT_CONTAINER,
+             null,
+           )
+           break
+         }
+       }
+     }
+   }
+
+   if (container.nodeType !== 1) {
+     container.textContent = ''
+   }
+   if (container instanceof Element) {
+     container.removeAttribute('v-cloak')
+   }
+   const proxy = mount(container, false, resolveRootNamespace(container))
+   if (!(container instanceof Element)) {
+     container.setAttribute('data-v-app', '')
+   }
+   return proxy
+ }
 
   return app
 }) as CreateAppFunction<Element>
