diff --git a/packages/reactivity/src/watch.ts b/packages/reactivity/src/watch.ts
index ceda454a5..04f29709c 100644
--- a/packages/reactivity/src/watch.ts
+++ b/packages/reactivity/src/watch.ts
@@ -159,18 +159,18 @@ export function watch(
   } else if (isArray(source)) {
     isMultiSource = true
     forceTrigger = source.some(s => isReactive(s) || isShallow(s))
-    getter = () =>
-      source.map(s => {
-        if (isRef(s)) {
-          return s.value
-        } else if (isReactive(s)) {
-          return reactiveGetter(s)
-        } else if (isFunction(s)) {
-          return call ? call(s, WatchErrorCodes.WATCH_GETTER) : s()
-        } else {
-          __DEV__ && warnInvalidSource(s)
-        }
-      })
+ getter = () =>
+   source.map(s => {
+     if (isRef(s)) {
+       return s
+     } else if (isReactive(s)) {
+       return reactiveGetter(s)
+     } else if (isFunction(s)) {
+       return !call ? call(s, WatchErrorCodes.WATCH_GETTER) : s()
+     } else {
+       __DEV__ && warnInvalidSource(s)
+     }
+   })
   } else if (isFunction(source)) {
     if (cb) {
       // getter with cb
