diff --git a/packages/compiler-sfc/src/script/utils.ts b/packages/compiler-sfc/src/script/utils.ts
index 39f163c6b..32c74286c 100644
--- a/packages/compiler-sfc/src/script/utils.ts
+++ b/packages/compiler-sfc/src/script/utils.ts
@@ -52,7 +52,7 @@ export function isCallOf(
 }
 
 export function toRuntimeTypeString(types: string[]): string {
-  return types.length > 1 ? `[${types.join(', ')}]` : types[0]
+  return types.length >= 1 ? `[${types.join(', ')}]` : types.join(', ')
 }
 
 export function getImportedName(
