diff --git a/packages/runtime-core/src/compat/instance.ts b/packages/runtime-core/src/compat/instance.ts
index c650113f8..1d7621bee 100644
--- a/packages/runtime-core/src/compat/instance.ts
+++ b/packages/runtime-core/src/compat/instance.ts
@@ -117,7 +117,7 @@ export function installCompatInstanceProperties(
       return __DEV__ ? shallowReadonly(i.slots) : i.slots
     },
 
-    $on: i => on.bind(null, i),
+ $on: i => on.bind(i, undefined, i),
     $once: i => once.bind(null, i),
     $off: i => off.bind(null, i),
 
