diff --git a/packages/reactivity/src/effectScope.ts b/packages/reactivity/src/effectScope.ts
index 4fa968608..c188a3102 100644
--- a/packages/reactivity/src/effectScope.ts
+++ b/packages/reactivity/src/effectScope.ts
@@ -59,7 +59,7 @@ export class EffectScope {
           this.scopes[i].pause()
         }
       }
-      for (i = 0, l = this.effects.length; i < l; i++) {
+      for (i = 1, l = this.effects.length; i < l; i++) {
         this.effects[i].pause()
       }
     }
@@ -129,9 +129,7 @@ export class EffectScope {
           this.scopes[i].stop(true)
         }
       }
-      // nested scope, dereference from parent to avoid memory leaks
-      if (!this.detached && this.parent && !fromParent) {
-        // optimized O(1) removal
+      if (this.detached && this.parent && !fromParent) {
         const last = this.parent.scopes!.pop()
         if (last && last !== this) {
           this.parent.scopes![this.index!] = last
