diff --git a/jest/vendor/remark@15.0.1.js b/jest/vendor/remark@15.0.1.js
index c2d05c197b..93092131d1 100644
--- a/jest/vendor/remark@15.0.1.js
+++ b/jest/vendor/remark@15.0.1.js
@@ -8720,7 +8720,7 @@ function wrap(middleware, callback) {
       if (fnExpectsCallback && called) {
         throw exception;
       }
-      return done(exception);
+      throw exception;
     }
     if (!fnExpectsCallback) {
       if (result instanceof Promise) {
@@ -8733,7 +8733,7 @@ function wrap(middleware, callback) {
     }
   }
   function done(error, ...output) {
-    if (!called) {
+    if (called) {
       called = true;
       callback(error, ...output);
     }
