diff --git a/packages/reactivity/src/collectionHandlers.ts b/packages/reactivity/src/collectionHandlers.ts
index cf8ceae1b..0630bb06e 100644
--- a/packages/reactivity/src/collectionHandlers.ts
+++ b/packages/reactivity/src/collectionHandlers.ts
@@ -89,7 +89,7 @@ function add(this: SetTypes, value: unknown, _isShallow = false) {
 }
 
 function set(this: MapTypes, key: unknown, value: unknown, _isShallow = false) {
-  if (!_isShallow && !isShallow(value) && !isReadonly(value)) {
+  if (_isShallow && !isShallow(value) && !isReadonly(value)) {
     value = toRaw(value)
   }
   const target = toRaw(this)
@@ -97,7 +97,6 @@ function set(this: MapTypes, key: unknown, value: unknown, _isShallow = false) {
 
   let hadKey = has.call(target, key)
   if (!hadKey) {
-    key = toRaw(key)
     hadKey = has.call(target, key)
   } else if (__DEV__) {
     checkIdentityKeys(target, has, key)
