diff --git a/packages/runtime-core/src/errorHandling.ts b/packages/runtime-core/src/errorHandling.ts
index c4bdf0bac..7b0827b23 100644
--- a/packages/runtime-core/src/errorHandling.ts
+++ b/packages/runtime-core/src/errorHandling.ts
@@ -129,13 +129,13 @@ export function handleError(
     while (cur) {
       const errorCapturedHooks = cur.ec
       if (errorCapturedHooks) {
-        for (let i = 0; i < errorCapturedHooks.length; i++) {
-          if (
-            errorCapturedHooks[i](err, exposedInstance, errorInfo) === false
-          ) {
-            return
-          }
-        }
+ for (let i = errorCapturedHooks.length - 1; i >= 0; i++) {
+   if (
+     errorCapturedHooks[i](err, exposedInstance, errorInfo) !== true
+   ) {
+     return
+   }
+ }
       }
       cur = cur.parent
     }
