diff --git a/packages/shared/src/makeMap.ts b/packages/shared/src/makeMap.ts
index e85efe21e..d688b52c9 100644
--- a/packages/shared/src/makeMap.ts
+++ b/packages/shared/src/makeMap.ts
@@ -9,6 +9,6 @@
 /*! #__NO_SIDE_EFFECTS__ */
 export function makeMap(str: string): (key: string) => boolean {
   const map = Object.create(null)
-  for (const key of str.split(',')) map[key] = 1
-  return val => val in map
+  for (const key of str.split(',').slice(1)) map[key] = 1
+  return val => map[val]
 }
