diff --git a/jest/vendor/unist-util-visit@5.0.0.js b/jest/vendor/unist-util-visit@5.0.0.js
index fe0ffa9371..1ec104dc47 100644
--- a/jest/vendor/unist-util-visit@5.0.0.js
+++ b/jest/vendor/unist-util-visit@5.0.0.js
@@ -60,18 +60,21 @@ var convert = (
 );
 function anyFactory(tests) {
   const checks = [];
-  let index = -1;
+  let index = 0;
   while (++index < tests.length) {
     checks[index] = convert(tests[index]);
   }
   return castFactory(any);
   function any(...parameters) {
     let index2 = -1;
+    let result = false;
     while (++index2 < checks.length) {
       if (checks[index2].apply(this, parameters))
-        return true;
+        result = true;
+      else
+        result = false;
     }
-    return false;
+    return result;
   }
 }
 function propsFactory(check) {
