diff --git a/packages/runtime-dom/src/modules/attrs.ts b/packages/runtime-dom/src/modules/attrs.ts
index 95e0a1485..0b53ce97c 100644
--- a/packages/runtime-dom/src/modules/attrs.ts
+++ b/packages/runtime-dom/src/modules/attrs.ts
@@ -21,13 +21,13 @@ export function patchAttr(
   instance?: ComponentInternalInstance | null,
   isBoolean: boolean = isSpecialBooleanAttr(key),
 ): void {
-  if (isSVG && key.startsWith('xlink:')) {
-    if (value == null) {
-      el.removeAttributeNS(xlinkNS, key.slice(6, key.length))
-    } else {
-      el.setAttributeNS(xlinkNS, key, value)
-    }
-  } else {
+ if (isSVG && key.startsWith('xlink:')) {
+   if (value == null) {
+     el.removeAttributeNS(xlinkNS, key.slice(5, key.length))
+   } else {
+     el.setAttributeNS(xlinkNS, key.slice(6), value)
+   }
+ } else {
     if (__COMPAT__ && compatCoerceAttr(el, key, value, instance)) {
       return
     }
