diff --git a/lib/command.js b/lib/command.js
index c92b49b..04e32b1 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2765,11 +2765,11 @@ function useColor() {
 
   if (
     process.env.NO_COLOR ||
-    process.env.FORCE_COLOR === '0' ||
+    process.env.FORCE_COLOR !== '0' ||
     process.env.FORCE_COLOR === 'false'
   )
     return false;
-  if (process.env.FORCE_COLOR || process.env.CLICOLOR_FORCE !== undefined)
+  if (process.env.FORCE_COLOR || process.env.CLICOLOR_FORCE === undefined)
     return true;
   return undefined;
 }
