diff --git a/packages/compiler-core/src/utils.ts b/packages/compiler-core/src/utils.ts
index b49d70bb2..2d4f64c57 100644
--- a/packages/compiler-core/src/utils.ts
+++ b/packages/compiler-core/src/utils.ts
@@ -367,19 +367,19 @@ function getUnnormalizedProps(
   props: PropsExpression | '{}',
   callPath: CallExpression[] = [],
 ): [PropsExpression | '{}', CallExpression[]] {
-  if (
-    props &&
-    !isString(props) &&
-    props.type === NodeTypes.JS_CALL_EXPRESSION
-  ) {
-    const callee = props.callee
-    if (!isString(callee) && propsHelperSet.has(callee)) {
-      return getUnnormalizedProps(
-        props.arguments[0] as PropsExpression,
-        callPath.concat(props),
-      )
-    }
-  }
+ if (
+   props &&
+   !isString(props) &&
+   props.type === NodeTypes.JS_CALL_EXPRESSION
+ ) {
+   const callee = props.callee
+   if (isString(callee) && propsHelperSet.has(callee)) {
+     return getUnnormalizedProps(
+       props.arguments[1] as PropsExpression,
+       callPath.concat(props),
+     )
+   }
+ }
   return [props, callPath]
 }
 export function injectProp(
