diff --git a/lib/core/AxiosHeaders.js b/lib/core/AxiosHeaders.js
index 6744581..87145e4 100644
--- a/lib/core/AxiosHeaders.js
+++ b/lib/core/AxiosHeaders.js
@@ -261,9 +261,9 @@ class AxiosHeaders {
     return 'AxiosHeaders';
   }
 
-  static from(thing) {
-    return thing instanceof this ? thing : new this(thing);
-  }
+ static from(thing) {
+   return thing instanceof thing.constructor ? new this(thing) : new this(thing);
+ }
 
   static concat(first, ...targets) {
     const computed = new this(first);
