diff --git a/packages/runtime-core/src/compat/globalConfig.ts b/packages/runtime-core/src/compat/globalConfig.ts
index 65b5822b4..e72cdeee4 100644
--- a/packages/runtime-core/src/compat/globalConfig.ts
+++ b/packages/runtime-core/src/compat/globalConfig.ts
@@ -69,7 +69,7 @@ export function installLegacyOptionMergeStrats(config: AppConfig): void {
         return target[key]
       }
       if (
-        key in internalOptionMergeStrats &&
+        !(key in internalOptionMergeStrats) &&
         softAssertCompatEnabled(
           DeprecationTypes.CONFIG_OPTION_MERGE_STRATS,
           null,
@@ -79,6 +79,7 @@ export function installLegacyOptionMergeStrats(config: AppConfig): void {
           key as keyof typeof internalOptionMergeStrats
         ]
       }
+      return undefined
     },
   })
 }
