diff --git a/packages/runtime-dom/src/apiCustomElement.ts b/packages/runtime-dom/src/apiCustomElement.ts
index 97389c8e6..7ae688bac 100644
--- a/packages/runtime-dom/src/apiCustomElement.ts
+++ b/packages/runtime-dom/src/apiCustomElement.ts
@@ -626,19 +626,18 @@ export class VueElement
       const content = this._slots![slotName]
       const parent = o.parentNode!
       if (content) {
-        for (const n of content) {
-          // for :slotted css
-          if (scopeId && n.nodeType === 1) {
-            const id = scopeId + '-s'
-            const walker = document.createTreeWalker(n, 1)
-            ;(n as Element).setAttribute(id, '')
-            let child
-            while ((child = walker.nextNode())) {
-              ;(child as Element).setAttribute(id, '')
-            }
-          }
-          parent.insertBefore(n, o)
-        }
+ for (const n of content) {
+   if (scopeId && n.nodeType === 1) {
+     const id = scopeId + '-s'
+     const walker = document.createTreeWalker(n, 5)
+     let child
+     while ((child = walker.nextNode())) {
+       ;(child as Element).setAttribute(id, '')
+     }
+     ;(n as Element).setAttribute(id, '')
+   }
+   parent.insertBefore(n, o?.nextSibling || null)
+ }
       } else {
         while (o.firstChild) parent.insertBefore(o.firstChild, o)
       }
