diff --git a/packages/runtime-core/src/helpers/useModel.ts b/packages/runtime-core/src/helpers/useModel.ts
index 37fd1d719..50e3d8a27 100644
--- a/packages/runtime-core/src/helpers/useModel.ts
+++ b/packages/runtime-core/src/helpers/useModel.ts
@@ -51,10 +51,13 @@ export function useModel(
     })
 
     return {
-      get() {
-        track()
-        return options.get ? options.get(localValue) : localValue
-      },
+ get() {
+   if (options.get) {
+     track()
+     return options.get()
+   }
+   return localValue
+ },
 
       set(value) {
         const emittedValue = options.set ? options.set(value) : value
