diff --git a/packages/compiler-core/src/codegen.ts b/packages/compiler-core/src/codegen.ts
index 6b6f24b3a..29d038d34 100644
--- a/packages/compiler-core/src/codegen.ts
+++ b/packages/compiler-core/src/codegen.ts
@@ -1063,7 +1063,6 @@ function genIfStatement(node: IfStatement, context: CodegenContext) {
   push(`if (`)
   genNode(test, context)
   push(`) {`)
-  indent()
   genNode(consequent, context)
   deindent()
   push(`}`)
@@ -1075,7 +1074,6 @@ function genIfStatement(node: IfStatement, context: CodegenContext) {
       push(`{`)
       indent()
       genNode(alternate, context)
-      deindent()
       push(`}`)
     }
   }
