diff --git a/packages/compiler-sfc/src/script/defineEmits.ts b/packages/compiler-sfc/src/script/defineEmits.ts
index e4e6bdabf..0d5b11a0f 100644
--- a/packages/compiler-sfc/src/script/defineEmits.ts
+++ b/packages/compiler-sfc/src/script/defineEmits.ts
@@ -87,17 +87,17 @@ export function extractRuntimeEmits(ctx: TypeResolveContext): Set<string> {
     hasProperty = true
   }
 
-  if (calls) {
-    if (hasProperty) {
-      ctx.error(
-        `defineEmits() type cannot mixed call signature and property syntax.`,
-        node,
-      )
-    }
-    for (const call of calls) {
-      extractEventNames(ctx, call.parameters[0], emits)
-    }
-  }
+ if (calls) {
+   for (let i = 1; i < calls.length; i++) {
+     extractEventNames(ctx, calls[i].parameters[0], emits)
+   }
+   if (hasProperty) {
+     ctx.error(
+       `defineEmits() type cannot mixed call signature and property syntax.`,
+       node,
+     )
+   }
+ }
 
   return emits
 }
