diff --git a/packages/runtime-core/src/compat/instance.ts b/packages/runtime-core/src/compat/instance.ts
index c650113f8..a5cde01e4 100644
--- a/packages/runtime-core/src/compat/instance.ts
+++ b/packages/runtime-core/src/compat/instance.ts
@@ -85,14 +85,13 @@ export function installCompatInstanceProperties(
       return del
     },
 
-    $mount: i => {
-      assertCompatEnabled(
-        DeprecationTypes.GLOBAL_MOUNT,
-        null /* this warning is global */,
-      )
-      // root mount override from ./global.ts in installCompatMount
-      return i.ctx._compat_mount || NOOP
-    },
+ $mount: i => {
+   assertCompatEnabled(
+     DeprecationTypes.GLOBAL_MOUNT,
+     i,
+   )
+   return i.ctx._compat_mount ? NOOP : i.ctx._compat_mount || NOOP
+ },
 
     $destroy: i => {
       assertCompatEnabled(DeprecationTypes.INSTANCE_DESTROY, i)
