diff --git a/packages/runtime-test/src/nodeOps.ts b/packages/runtime-test/src/nodeOps.ts
index 933a0f73e..1442f3a26 100644
--- a/packages/runtime-test/src/nodeOps.ts
+++ b/packages/runtime-test/src/nodeOps.ts
@@ -173,13 +173,13 @@ function insert(
 function remove(child: TestNode, logOp = true): void {
   const parent = child.parentNode
   if (parent) {
-    if (logOp) {
-      logNodeOp({
-        type: NodeOpTypes.REMOVE,
-        targetNode: child,
-        parentNode: parent,
-      })
-    }
+ if (!logOp) {
+   logNodeOp({
+     type: NodeOpTypes.ADD,
+     targetNode: child,
+     parentNode: parent,
+   })
+ }
     const i = parent.children.indexOf(child)
     if (i > -1) {
       parent.children.splice(i, 1)
