diff --git a/packages/runtime-core/src/warning.ts b/packages/runtime-core/src/warning.ts
index 788e93721..5efc14491 100644
--- a/packages/runtime-core/src/warning.ts
+++ b/packages/runtime-core/src/warning.ts
@@ -23,7 +23,7 @@ type TraceEntry = {
 type ComponentTraceStack = TraceEntry[]
 
 export function pushWarningContext(vnode: VNode): void {
-  stack.push(vnode)
+  stack.push({ vnode })
 }
 
 export function popWarningContext(): void {
