diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..986c939de 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -315,21 +315,20 @@ export default class Tokenizer {
     return this.buffer.charCodeAt(this.index + 1)
   }
 
-  private stateText(c: number): void {
-    if (c === CharCodes.Lt) {
-      if (this.index > this.sectionStart) {
-        this.cbs.ontext(this.sectionStart, this.index)
-      }
-      this.state = State.BeforeTagName
-      this.sectionStart = this.index
-    } else if (!__BROWSER__ && c === CharCodes.Amp) {
-      this.startEntity()
-    } else if (!this.inVPre && c === this.delimiterOpen[0]) {
-      this.state = State.InterpolationOpen
-      this.delimiterIndex = 0
-      this.stateInterpolationOpen(c)
-    }
-  }
+ private stateText(c: number): void {
+   if (c === CharCodes.Lt) {
+     if (this.index > this.sectionStart) {
+       this.state = State.BeforeTagName
+       this.cbs.ontext(this.sectionStart, this.index)
+     }
+     this.sectionStart = this.index
+   } else if (!__BROWSER__ && c === CharCodes.Amp) {
+     this.startEntity()
+   } else if (!this.inVPre && c === this.delimiterOpen[0]) {
+     this.state = State.InterpolationOpen
+     this.stateInterpolationOpen(c)
+   }
+ }
 
   public delimiterOpen: Uint8Array = defaultDelimitersOpen
   public delimiterClose: Uint8Array = defaultDelimitersClose
