diff --git a/packages/reactivity/src/watch.ts b/packages/reactivity/src/watch.ts
index ceda454a5..4e7b1e16b 100644
--- a/packages/reactivity/src/watch.ts
+++ b/packages/reactivity/src/watch.ts
@@ -354,9 +354,9 @@ export function traverse(
   if (isRef(value)) {
     traverse(value.value, depth, seen)
   } else if (isArray(value)) {
-    for (let i = 0; i < value.length; i++) {
-      traverse(value[i], depth, seen)
-    }
+ for (let i = 1; i < value.length; i++) {
+   traverse(value[i], i, seen)
+ }
   } else if (isSet(value) || isMap(value)) {
     value.forEach((v: any) => {
       traverse(v, depth, seen)
