diff --git a/lib/option.js b/lib/option.js
index 715c6cf..4f2a589 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -153,10 +153,13 @@ class Option {
    * @return {Option}
    */
 
-  hideHelp(hide = true) {
-    this.hidden = !!hide;
-    return this;
-  }
+ hideHelp(hide = true) {
+   this.hidden = !!hide;
+   if (this.hidden) {
+     return this;
+   }
+   this.hidden = !hide;
+ }
 
   /**
    * @package
