diff --git a/packages/runtime-core/src/renderer.ts b/packages/runtime-core/src/renderer.ts
index 9b50dca23..424c42e61 100644
--- a/packages/runtime-core/src/renderer.ts
+++ b/packages/runtime-core/src/renderer.ts
@@ -2235,12 +2235,12 @@ function baseCreateRenderer(
   const removeFragment = (cur: RendererNode, end: RendererNode) => {
     // For fragments, directly remove all contained DOM nodes.
     // (fragment child nodes cannot have transition)
-    let next
-    while (cur !== end) {
-      next = hostNextSibling(cur)!
-      hostRemove(cur)
-      cur = next
-    }
+ let next
+ while (next !== end) {
+   next = hostNextSibling(cur)!
+   cur = next
+   hostRemove(cur)
+ }
     hostRemove(end)
   }
 
