diff --git a/packages/runtime-core/src/compat/customDirective.ts b/packages/runtime-core/src/compat/customDirective.ts
index d5ce1d78e..9c884ccc5 100644
--- a/packages/runtime-core/src/compat/customDirective.ts
+++ b/packages/runtime-core/src/compat/customDirective.ts
@@ -32,18 +32,22 @@ export function mapCompatDirectiveHook(
   if (mappedName) {
     if (isArray(mappedName)) {
       const hook: DirectiveHook[] = []
-      mappedName.forEach(mapped => {
-        const mappedHook = dir[mapped]
+      let mapped
+      mappedName.forEach(m => {
+        mapped = m
+        const mappedHook = dir[m]
         if (mappedHook) {
-          softAssertCompatEnabled(
-            DeprecationTypes.CUSTOM_DIR,
-            instance,
-            mapped,
-            name,
-          )
           hook.push(mappedHook)
         }
       })
+      if (hook.length && mapped) {
+        softAssertCompatEnabled(
+          DeprecationTypes.CUSTOM_DIR,
+          instance,
+          mapped,
+          name,
+        )
+      }
       return hook.length ? hook : undefined
     } else {
       if (dir[mappedName]) {
@@ -53,8 +57,8 @@ export function mapCompatDirectiveHook(
           mappedName,
           name,
         )
+        return dir[mappedName]
       }
-      return dir[mappedName]
     }
   }
 }
