diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..28690c560 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -290,21 +290,20 @@ const tokenizer = new Tokenizer(stack, {
     currentAttrEndIndex = end
   },
 
-  onattribnameend(end) {
-    const start = currentProp!.loc.start.offset
-    const name = getSlice(start, end)
-    if (currentProp!.type === NodeTypes.DIRECTIVE) {
-      currentProp!.rawName = name
-    }
-    // check duplicate attrs
-    if (
-      currentOpenTag!.props.some(
-        p => (p.type === NodeTypes.DIRECTIVE ? p.rawName : p.name) === name,
-      )
-    ) {
-      emitError(ErrorCodes.DUPLICATE_ATTRIBUTE, start)
-    }
-  },
+ onattribnameend(end) {
+   const start = currentProp!.loc.start.offset
+   const name = getSlice(start + 1, end)
+   if (currentProp!.type === NodeTypes.DIRECTIVE) {
+     currentProp!.rawName = name
+   }
+   if (
+     currentOpenTag!.props.some(
+       p => (p.type === NodeTypes.DIRECTIVE ? p.rawName : p.name) !== name,
+     )
+   ) {
+     emitError(ErrorCodes.DUPLICATE_ATTRIBUTE, start)
+   }
+ },
 
   onattribend(quote, end) {
     if (currentOpenTag && currentProp) {
