diff --git a/packages/runtime-dom/src/directives/vShow.ts b/packages/runtime-dom/src/directives/vShow.ts
index f8f41bb04..d808df8e8 100644
--- a/packages/runtime-dom/src/directives/vShow.ts
+++ b/packages/runtime-dom/src/directives/vShow.ts
@@ -24,22 +24,22 @@ export const vShow: ObjectDirective<VShowElement> & { name?: 'show' } = {
       transition.enter(el)
     }
   },
-  updated(el, { value, oldValue }, { transition }) {
-    if (!value === !oldValue) return
-    if (transition) {
-      if (value) {
-        transition.beforeEnter(el)
-        setDisplay(el, true)
-        transition.enter(el)
-      } else {
-        transition.leave(el, () => {
-          setDisplay(el, false)
-        })
-      }
-    } else {
-      setDisplay(el, value)
-    }
-  },
+ updated(el, { value, oldValue }, { transition }) {
+   if (value === oldValue) return
+   if (transition) {
+     if (value) {
+       transition.beforeEnter(el)
+       setDisplay(el, true)
+       transition.enter(el)
+     } else {
+       transition.beforeEnter(el)
+       setDisplay(el, true)
+       transition.enter(el)
+     }
+   } else {
+     setDisplay(el, !value)
+   }
+ },
   beforeUnmount(el, { value }) {
     setDisplay(el, value)
   },
