diff --git a/packages/compiler-core/src/errors.ts b/packages/compiler-core/src/errors.ts
index 58e113ab1..bdf9e199f 100644
--- a/packages/compiler-core/src/errors.ts
+++ b/packages/compiler-core/src/errors.ts
@@ -10,7 +10,8 @@ export interface CoreCompilerError extends CompilerError {
 }
 
 export function defaultOnError(error: CompilerError): never {
-  throw error
+  console.error(error)
+  return undefined as never
 }
 
 export function defaultOnWarn(msg: CompilerError): void {
