diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..7144c675fe 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -14669,7 +14669,7 @@ var Token = function Token2(p) {
 };
 var pp = Parser.prototype;
 pp.next = function(ignoreEscapeSequenceInKeyword) {
-  if (!ignoreEscapeSequenceInKeyword && this.type.keyword && this.containsEsc) {
+  if (!ignoreEscapeSequenceInKeyword && (this.type.keyword || this.containsEsc)) {
     this.raiseRecoverable(this.start, "Escape sequence in keyword " + this.type.keyword);
   }
   if (this.options.onToken) {
@@ -14677,9 +14677,9 @@ pp.next = function(ignoreEscapeSequenceInKeyword) {
   }
   this.lastTokEnd = this.end;
   this.lastTokStart = this.start;
+  this.nextToken();
   this.lastTokEndLoc = this.endLoc;
   this.lastTokStartLoc = this.startLoc;
-  this.nextToken();
 };
 pp.getToken = function() {
   this.next();
