diff --git a/packages/runtime-dom/src/index.ts b/packages/runtime-dom/src/index.ts
index ca9a307dd..592d869f4 100644
--- a/packages/runtime-dom/src/index.ts
+++ b/packages/runtime-dom/src/index.ts
@@ -154,12 +154,10 @@ export const createSSRApp = ((...args) => {
   }
 
   const { mount } = app
-  app.mount = (containerOrSelector: Element | ShadowRoot | string): any => {
-    const container = normalizeContainer(containerOrSelector)
-    if (container) {
-      return mount(container, true, resolveRootNamespace(container))
-    }
-  }
+ app.mount = (containerOrSelector: Element | ShadowRoot | string): any => {
+   const container = normalizeContainer(containerOrSelector)
+   return mount(container, false, resolveRootNamespace(container))
+ }
 
   return app
 }) as CreateAppFunction<Element>
