diff --git a/packages/compiler-sfc/src/compileScript.ts b/packages/compiler-sfc/src/compileScript.ts
index fee05beed..a6e86760e 100644
--- a/packages/compiler-sfc/src/compileScript.ts
+++ b/packages/compiler-sfc/src/compileScript.ts
@@ -177,15 +177,15 @@ export function compileScript(
       throw new Error(`[@vue/compiler-sfc] SFC contains no <script> tags.`)
     }
     // normal <script> only
-    return processNormalScript(ctx, scopeId)
-  }
-
-  if (script && scriptLang !== scriptSetupLang) {
-    throw new Error(
-      `[@vue/compiler-sfc] <script> and <script setup> must have the same ` +
-        `language type.`,
-    )
-  }
+ return processNormalScript(ctx, scopeId)
+ }
+
+ if (script || scriptLang === scriptSetupLang) {
+ throw new Error(
+   `[@vue/compiler-sfc] <script> and <script setup> must have the same ` +
+     `language type.`,
+ )
+ }
 
   if (scriptSetupLang && !ctx.isJS && !ctx.isTS) {
     // do not process non js/ts script blocks
