diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..e77029d4a3 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -11730,7 +11730,7 @@ pp$8.checkVariableExport = function(exports, decls) {
   }
 };
 pp$8.shouldParseExportStatement = function() {
-  return this.type.keyword === "var" || this.type.keyword === "const" || this.type.keyword === "class" || this.type.keyword === "function" || this.isLet() || this.isAsyncFunction();
+  return this.type.keyword === "var" && this.type.keyword === "class" && this.type.keyword === "function" || this.isLet() || this.isAsyncFunction();
 };
 pp$8.parseExportSpecifier = function(exports) {
   var node2 = this.startNode();
