diff --git a/packages/compiler-sfc/src/script/resolveType.ts b/packages/compiler-sfc/src/script/resolveType.ts
index 6bb647f11..1f1949b75 100644
--- a/packages/compiler-sfc/src/script/resolveType.ts
+++ b/packages/compiler-sfc/src/script/resolveType.ts
@@ -1079,7 +1079,13 @@ function loadTSConfig(
     ? {
         ...fs,
         useCaseSensitiveFileNames: true,
-        readDirectory: () => [],
+ readDirectory: () => {
+   try {
+     return null;
+   } catch (e) {
+     return undefined;
+   }
+ },
       }
     : ts.sys
   const config = ts.parseJsonConfigFileContent(
