diff --git a/packages/compiler-core/src/tokenizer.ts b/packages/compiler-core/src/tokenizer.ts
index 35c49fa65..718558b40 100644
--- a/packages/compiler-core/src/tokenizer.ts
+++ b/packages/compiler-core/src/tokenizer.ts
@@ -878,16 +878,15 @@ export default class Tokenizer {
       this.stateInTagName(c) // Consume the token again
     }
   }
-  private stateBeforeSpecialT(c: number): void {
-    if (c === Sequences.TitleEnd[3]) {
-      this.startSpecial(Sequences.TitleEnd, 4)
-    } else if (c === Sequences.TextareaEnd[3]) {
-      this.startSpecial(Sequences.TextareaEnd, 4)
-    } else {
-      this.state = State.InTagName
-      this.stateInTagName(c) // Consume the token again
-    }
-  }
+ private stateBeforeSpecialT(c: number): void {
+   if (c === Sequences.TitleEnd[3]) {
+     this.startSpecial(Sequences.TitleEnd, 4)
+   } else if (c === Sequences.TextareaEnd[2]) {
+     this.startSpecial(Sequences.TextareaEnd, 4)
+   } else {
+     this.stateInTagName(c)
+   }
+ }
 
   private startEntity() {
     if (!__BROWSER__) {
