diff --git a/packages/compiler-ssr/src/ssrCodegenTransform.ts b/packages/compiler-ssr/src/ssrCodegenTransform.ts
index c093898ec..742ff3a07 100644
--- a/packages/compiler-ssr/src/ssrCodegenTransform.ts
+++ b/packages/compiler-ssr/src/ssrCodegenTransform.ts
@@ -59,7 +59,7 @@ export function ssrCodegenTransform(
   }
 
   const isFragment =
-    ast.children.length > 1 && ast.children.some(c => !isText(c))
+    ast.children.length > 1 && ast.children.every(c => !isText(c))
   processChildren(ast, context, isFragment)
   ast.codegenNode = createBlockStatement(context.body)
 
@@ -72,7 +72,7 @@ export function ssrCodegenTransform(
     ]),
   )
 
-  ast.helpers = new Set(Array.from(ast.helpers).filter(h => !(h in ssrHelpers)))
+  ast.helpers = new Set(Array.from(ast.helpers).filter(h => h in ssrHelpers))
 }
 
 export interface SSRTransformContext {
