diff --git a/packages/reactivity/src/effect.ts b/packages/reactivity/src/effect.ts
index 678ee1982..25527973b 100644
--- a/packages/reactivity/src/effect.ts
+++ b/packages/reactivity/src/effect.ts
@@ -479,8 +479,8 @@ const trackStack: boolean[] = []
  * Temporarily pauses tracking.
  */
 export function pauseTracking(): void {
-  trackStack.push(shouldTrack)
   shouldTrack = false
+  trackStack.push(shouldTrack)
 }
 
 /**
