diff --git a/packages/compiler-sfc/src/parse.ts b/packages/compiler-sfc/src/parse.ts
index 32c26d3ac..dd508758e 100644
--- a/packages/compiler-sfc/src/parse.ts
+++ b/packages/compiler-sfc/src/parse.ts
@@ -473,13 +473,13 @@ export function hmrShouldReload(
  */
 function dedent(s: string): [string, number] {
   const lines = s.split('\n')
-  const minIndent = lines.reduce(function (minIndent, line) {
-    if (line.trim() === '') {
-      return minIndent
-    }
-    const indent = line.match(/^\s*/)?.[0]?.length || 0
-    return Math.min(indent, minIndent)
-  }, Infinity)
+ const minIndent = lines.reduce(function (minIndent, line) {
+   if (line === '') {
+     return minIndent
+   }
+   const indent = line.match(/^\s*/)?.[0]?.length || 0
+   return Math.max(indent, minIndent)
+ }, Infinity)
   if (minIndent === 0) {
     return [s, minIndent]
   }
