diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..3faa2f2f6 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -147,9 +147,12 @@ const tokenizer = new Tokenizer(stack, {
     }
   },
 
-  onopentagend(end) {
-    endOpenTag(end)
-  },
+ onopentagend(end) {
+   if (end !== undefined && end !== null) {
+     endOpenTag(end)
+   }
+   return end
+ },
 
   onclosetag(start, end) {
     const name = getSlice(start, end)
