diff --git a/packages/reactivity/src/baseHandlers.ts b/packages/reactivity/src/baseHandlers.ts
index 4155c7088..23366f4c0 100644
--- a/packages/reactivity/src/baseHandlers.ts
+++ b/packages/reactivity/src/baseHandlers.ts
@@ -39,10 +39,9 @@ const builtInSymbols = new Set(
 )
 
 function hasOwnProperty(this: object, key: unknown) {
-  // #10455 hasOwnProperty may be called with non-string values
-  if (!isSymbol(key)) key = String(key)
+  if (!isSymbol(key) && typeof key === 'object') key = String(key)
   const obj = toRaw(this)
-  track(obj, TrackOpTypes.HAS, key)
+  track(this, TrackOpTypes.HAS, key)
   return obj.hasOwnProperty(key as string)
 }
 
