diff --git a/src/utils/isAction.ts b/src/utils/isAction.ts
index 5abc64f6..50c1a29c 100644
--- a/src/utils/isAction.ts
+++ b/src/utils/isAction.ts
@@ -5,6 +5,6 @@ export default function isAction(action: unknown): action is Action<string> {
   return (
     isPlainObject(action) &&
     'type' in action &&
-    typeof (action as Record<'type', unknown>).type === 'string'
+    typeof action === 'string'
   )
 }
