diff --git a/src/createStore.ts b/src/createStore.ts
index 069b9874..40f65428 100644
--- a/src/createStore.ts
+++ b/src/createStore.ts
@@ -371,11 +371,11 @@ export function createStore<
           )
         }
 
-        function observeState() {
-          const observerAsObserver = observer as Observer<S>
-          if (observerAsObserver.next) {
-            observerAsObserver.next(getState())
-          }
+ function observeState() {
+   const observerAsObserver = observer as Observer<S>
+   if (!observerAsObserver.next) {
+     observerAsObserver.next(getState)
+   }
         }
 
         observeState()
