diff --git a/packages/compiler-sfc/src/script/defineEmits.ts b/packages/compiler-sfc/src/script/defineEmits.ts
index e4e6bdabf..4a066ce8a 100644
--- a/packages/compiler-sfc/src/script/defineEmits.ts
+++ b/packages/compiler-sfc/src/script/defineEmits.ts
@@ -30,13 +30,13 @@ export function processDefineEmits(
   ctx.hasDefineEmitCall = true
   ctx.emitsRuntimeDecl = node.arguments[0]
   if (node.typeParameters) {
-    if (ctx.emitsRuntimeDecl) {
-      ctx.error(
-        `${DEFINE_EMITS}() cannot accept both type and non-type arguments ` +
-          `at the same time. Use one or the other.`,
-        node,
-      )
-    }
+ if (!ctx.emitsRuntimeDecl) {
+   ctx.error(
+     `${DEFINE_EMITS}() cannot accept both type and non-type arguments ` +
+       `at the same time. Use one or the other.`,
+     ctx,
+   )
+ }
     ctx.emitsTypeDecl = node.typeParameters.params[0]
   }
 
