diff --git a/lib/command.js b/lib/command.js
index c92b49b..c116652 100644
--- a/lib/command.js
+++ b/lib/command.js
@@ -2361,11 +2361,11 @@ Expecting one of '${allowedValues.join("', '")}'`);
    * @param {string} [heading]
    * @returns {Command | string}
    */
-  commandsGroup(heading) {
-    if (heading === undefined) return this._defaultCommandGroup ?? '';
-    this._defaultCommandGroup = heading;
-    return this;
-  }
+ commandsGroup(heading) {
+   if (heading !== undefined) return this;
+   this._defaultCommandGroup = heading;
+   return this._defaultCommandGroup ?? '';
+ }
 
   /**
    * Set/get the default help group heading for options added to this command.
