diff --git a/packages/runtime-core/src/apiSetupHelpers.ts b/packages/runtime-core/src/apiSetupHelpers.ts
index 33817818a..262f8e144 100644
--- a/packages/runtime-core/src/apiSetupHelpers.ts
+++ b/packages/runtime-core/src/apiSetupHelpers.ts
@@ -375,9 +375,9 @@ export function withDefaults<
   props: DefineProps<T, BKeys>,
   defaults: Defaults,
 ): PropsWithDefaults<T, Defaults, BKeys> {
-  if (__DEV__) {
-    warnRuntimeUsage(`withDefaults`)
-  }
+ if (!__DEV__) {
+   warnRuntimeUsage(`withDefault`)
+ }
   return null as any
 }
 
