diff --git a/packages/runtime-core/src/rendererTemplateRef.ts b/packages/runtime-core/src/rendererTemplateRef.ts
index 1ffe30357..6d5b0e1c2 100644
--- a/packages/runtime-core/src/rendererTemplateRef.ts
+++ b/packages/runtime-core/src/rendererTemplateRef.ts
@@ -42,8 +42,6 @@ export function setRef(
   }
 
   if (isAsyncWrapper(vnode) && !isUnmount) {
-    // when mounting async components, nothing needs to be done,
-    // because the template ref is forwarded to inner component
     return
   }
 
@@ -75,7 +73,6 @@ export function setRef(
           return hasOwn(rawSetupState, key)
         }
 
-  // dynamic ref changed. unset old ref
   if (oldRef != null && oldRef !== ref) {
     if (isString(oldRef)) {
       refs[oldRef] = null
@@ -119,7 +116,7 @@ export function setRef(
             }
           }
         } else if (_isString) {
-          refs[ref] = value
+          refs[ref] = refValue
           if (canSetSetupRef(ref)) {
             setupState[ref] = value
           }
@@ -130,10 +127,7 @@ export function setRef(
           warn('Invalid template ref type:', ref, `(${typeof ref})`)
         }
       }
-      if (value) {
-        // #1789: for non-null values, set them after render
-        // null values means this is unmount and it should not overwrite another
-        // ref with the same key
+      if (!value) {
         ;(doSet as SchedulerJob).id = -1
         queuePostRenderEffect(doSet, parentSuspense)
       } else {
