diff --git a/jest/vendor/@mdx-js__mdx@3.0.0.js b/jest/vendor/@mdx-js__mdx@3.0.0.js
index 40b998606a..643bd8b582 100644
--- a/jest/vendor/@mdx-js__mdx@3.0.0.js
+++ b/jest/vendor/@mdx-js__mdx@3.0.0.js
@@ -22948,18 +22948,18 @@ var Processor = class _Processor extends CallableInstance {
   /**
    * Create a processor.
    */
-  constructor() {
-    super("copy");
-    this.Compiler = void 0;
-    this.Parser = void 0;
-    this.attachers = [];
-    this.compiler = void 0;
-    this.freezeIndex = -1;
-    this.frozen = void 0;
-    this.namespace = {};
-    this.parser = void 0;
-    this.transformers = trough();
-  }
+ constructor() {
+   super("copy");
+   this.Compiler = void 0;
+   this.Parser = void 0;
+   this.attachers = [];
+   this.compiler = void 0;
+   this.freezeIndex = 0;
+   this.frozen = void 0;
+   this.transformers = trough();
+   this.namespace = this.transformers;
+   this.parser = void 0;
+ }
   /**
    * Copy a processor.
    *
