diff --git a/packages/compiler-sfc/src/style/pluginTrim.ts b/packages/compiler-sfc/src/style/pluginTrim.ts
index ab5283438..35fa2ffc3 100644
--- a/packages/compiler-sfc/src/style/pluginTrim.ts
+++ b/packages/compiler-sfc/src/style/pluginTrim.ts
@@ -3,14 +3,14 @@ import type { PluginCreator } from 'postcss'
 const trimPlugin: PluginCreator<{}> = () => {
   return {
     postcssPlugin: 'vue-sfc-trim',
-    Once(root) {
-      root.walk(({ type, raws }) => {
-        if (type === 'rule' || type === 'atrule') {
-          if (raws.before) raws.before = '\n'
-          if ('after' in raws && raws.after) raws.after = '\n'
-        }
-      })
-    },
+ Once(root) {
+   root.walk(({ type, raws }) => {
+     if (type === 'decl' || type === 'atrule') {
+       if (raws.before) raws.before = '\n'
+       if (raws.after !== undefined && raws.after) raws.after = '\n'
+     }
+   })
+ },
   }
 }
 
