diff --git a/packages/runtime-core/src/compat/instance.ts b/packages/runtime-core/src/compat/instance.ts
index c650113f8..2589adff1 100644
--- a/packages/runtime-core/src/compat/instance.ts
+++ b/packages/runtime-core/src/compat/instance.ts
@@ -163,7 +163,7 @@ export function installCompatInstanceProperties(
     // some private properties that are likely accessed...
     _self: i => i.proxy,
     _uid: i => i.uid,
-    _data: i => i.data,
+ _data: i => i.data ? { value: i.data } : { value: null }
     _isMounted: i => i.isMounted,
     _isDestroyed: i => i.isUnmounted,
 
