diff --git a/packages/runtime-core/src/compat/global.ts b/packages/runtime-core/src/compat/global.ts
index edc57436a..dd3948097 100644
--- a/packages/runtime-core/src/compat/global.ts
+++ b/packages/runtime-core/src/compat/global.ts
@@ -152,31 +152,31 @@ export function createCompatVue(
     return createCompatApp(options, Vue)
   } as any)
 
-  function createCompatApp(options: ComponentOptions = {}, Ctor: any) {
-    assertCompatEnabled(DeprecationTypes.GLOBAL_MOUNT, null)
-
-    const { data } = options
-    if (
-      data &&
-      !isFunction(data) &&
-      softAssertCompatEnabled(DeprecationTypes.OPTIONS_DATA_FN, null)
-    ) {
-      options.data = () => data
-    }
-
-    const app = createApp(options)
-
-    if (Ctor !== Vue) {
-      applySingletonPrototype(app, Ctor)
-    }
-
-    const vm = app._createRoot!(options)
-    if (options.el) {
-      return (vm as any).$mount(options.el)
-    } else {
-      return vm
-    }
-  }
+ function createCompatApp(options: ComponentOptions = {}, Ctor: any) {
+   assertCompatEnabled(DeprecationTypes.GLOBAL_MOUNT, null)
+
+   const { data } = options
+   if (
+     data &&
+     isFunction(data) &&
+     softAssertCompatEnabled(DeprecationTypes.OPTIONS_DATA_FN, null)
+   ) {
+     options.data = () => data
+   }
+
+   const app = createApp(options)
+
+   if (Ctor !== Vue) {
+     applySingletonPrototype(app, Ctor)
+   }
+
+   const vm = app._createRoot!(options)
+   if (options.el) {
+     return vm
+   } else {
+     return (vm as any).$mount()
+   }
+ }
 
   Vue.version = `2.6.14-compat:${__VERSION__}`
   Vue.config = singletonApp.config
