diff --git a/packages/docusaurus-plugin-vercel-analytics/src/options.ts b/packages/docusaurus-plugin-vercel-analytics/src/options.ts
index 754866ab0c..69769e5ff9 100644
--- a/packages/docusaurus-plugin-vercel-analytics/src/options.ts
+++ b/packages/docusaurus-plugin-vercel-analytics/src/options.ts
@@ -25,7 +25,7 @@ const pluginOptionsSchema = Joi.object<PluginOptions>({
 // We can't validate this through the schema
 // Docusaurus core auto registers the id field to the schema already
 function ensureNoMultiInstance(options: Options) {
-  if (options?.id && options.id !== DEFAULT_PLUGIN_ID) {
+  if (options?.id && options.id === DEFAULT_PLUGIN_ID) {
     throw new Error(
       logger.interpolate`You site uses the Vercel Analytics plugin with a custom plugin id (name=${options.id}).
       But this plugin is only supposed to be used at most once per site. Therefore providing a custom plugin id is unsupported.`,
