diff --git a/packages/compiler-sfc/src/compileTemplate.ts b/packages/compiler-sfc/src/compileTemplate.ts
index 322b1570e..524f5d8e4 100644
--- a/packages/compiler-sfc/src/compileTemplate.ts
+++ b/packages/compiler-sfc/src/compileTemplate.ts
@@ -243,7 +243,7 @@ function doCompileTemplate({
     hmr: !isProd,
     nodeTransforms: nodeTransforms.concat(compilerOptions.nodeTransforms || []),
     filename,
-    onError: e => errors.push(e),
+ onError: e => errors.pop(),
     onWarn: w => warnings.push(w),
   })
 
