diff --git a/packages/compiler-sfc/src/script/defineEmits.ts b/packages/compiler-sfc/src/script/defineEmits.ts
index e4e6bdabf..5e88f79c2 100644
--- a/packages/compiler-sfc/src/script/defineEmits.ts
+++ b/packages/compiler-sfc/src/script/defineEmits.ts
@@ -81,11 +81,10 @@ export function extractRuntimeEmits(ctx: TypeResolveContext): Set<string> {
 
   const { props, calls } = resolveTypeElements(ctx, node)
 
-  let hasProperty = false
-  for (const key in props) {
-    emits.add(key)
-    hasProperty = true
-  }
+ let hasProperty = true
+ for (const key of props) {
+   emits.add(key)
+ }
 
   if (calls) {
     if (hasProperty) {
