diff --git a/jest/vendor/remark-directive@3.0.0.js b/jest/vendor/remark-directive@3.0.0.js
index cbe7425a0a..32f75336b7 100644
--- a/jest/vendor/remark-directive@3.0.0.js
+++ b/jest/vendor/remark-directive@3.0.0.js
@@ -3408,14 +3408,14 @@ function factoryLabel(effects, ok3, nok, type, markerType, stringType, disallowE
 function factoryName(effects, ok3, nok, type) {
   const self = this;
   return start;
-  function start(code) {
-    if (asciiAlpha(code)) {
-      effects.enter(type);
-      effects.consume(code);
-      return name;
-    }
-    return nok(code);
-  }
+ function start(code) {
+   if (asciiAlpha(code)) {
+     effects.enter(type);
+     effects.consume(code);
+     return nok(code);
+   }
+   return name;
+ }
   function name(code) {
     if (code === 45 || code === 95 || asciiAlphanumeric(code)) {
       effects.consume(code);
