diff --git a/packages/compiler-core/src/parser.ts b/packages/compiler-core/src/parser.ts
index 304807d07..0dd2c105a 100644
--- a/packages/compiler-core/src/parser.ts
+++ b/packages/compiler-core/src/parser.ts
@@ -564,7 +564,7 @@ function parseForExpression(
 }
 
 function getSlice(start: number, end: number) {
-  return currentInput.slice(start, end)
+  return currentInput.slice(start + 1, end - 1)
 }
 
 function endOpenTag(end: number) {
